--- a/httpd/keystone-uwsgi-admin.ini
+++ b/httpd/keystone-uwsgi-admin.ini
@@ -1,5 +1,5 @@
 [uwsgi]
-wsgi-file = /usr/local/bin/keystone-wsgi-admin
+wsgi-file = /usr/bin/keystone-wsgi-admin
 
 # Versions of mod_proxy_uwsgi>=2.0.6 should use a UNIX socket, see
 # http://uwsgi-docs.readthedocs.org/en/latest/Apache.html#mod-proxy-uwsgi
--- a/httpd/keystone-uwsgi-public.ini
+++ b/httpd/keystone-uwsgi-public.ini
@@ -1,5 +1,5 @@
 [uwsgi]
-wsgi-file = /usr/local/bin/keystone-wsgi-public
+wsgi-file = /usr/bin/keystone-wsgi-public
 
 # Versions of mod_proxy_uwsgi>=2.0.6 should use a UNIX socket, see
 # http://uwsgi-docs.readthedocs.org/en/latest/Apache.html#mod-proxy-uwsgi
--- a/httpd/wsgi-keystone.conf
+++ b/httpd/wsgi-keystone.conf
@@ -3,7 +3,7 @@
 <VirtualHost *:5000>
     WSGIDaemonProcess keystone-public processes=5 threads=1 user=keystone group=keystone display-name=%{GROUP}
     WSGIProcessGroup keystone-public
-    WSGIScriptAlias / /usr/local/bin/keystone-wsgi-public
+    WSGIScriptAlias / /usr/bin/keystone-wsgi-public
     WSGIApplicationGroup %{GLOBAL}
     WSGIPassAuthorization On
     LimitRequestBody 114688
@@ -13,7 +13,7 @@
     ErrorLog /var/log/apache2/keystone.log
     CustomLog /var/log/apache2/keystone_access.log combined
 
-    <Directory /usr/local/bin>
+    <Directory /usr/bin>
         <IfVersion >= 2.4>
             Require all granted
         </IfVersion>
@@ -24,7 +24,7 @@
     </Directory>
 </VirtualHost>
 
-Alias /identity /usr/local/bin/keystone-wsgi-public
+Alias /identity /usr/bin/keystone-wsgi-public
 <Location /identity>
     SetHandler wsgi-script
     Options +ExecCGI
