--- /os_dep/linux/recv_linux.c	2020-12-24 16:55:47.000000000 +0600
+++ /os_dep/linux/recv_linux.c	2021-05-07 15:06:13.043463838 +0600
@@ -353,7 +353,7 @@
 
 		rx_ok = _FALSE;
 
-#ifdef CONFIG_RTW_GRO
+#if defined (CONFIG_RTW_GRO) && LINUX_VERSION_CODE < KERNEL_VERSION(5, 12, 0)
 		if (pregistrypriv->en_gro) {
 			if (rtw_napi_gro_receive(&padapter->napi, pskb) != GRO_DROP)
 				rx_ok = _TRUE;
@@ -400,7 +400,7 @@
 	if (adapter->registrypriv.en_napi) {
 		struct dvobj_priv *dvobj;
 		struct registry_priv *registry;
-	
+
 		dvobj = adapter_to_dvobj(adapter);
 		registry = &adapter->registrypriv;
 		if (dvobj->traffic_stat.cur_rx_tp > registry->napi_threshold)
@@ -511,7 +511,7 @@
 #ifdef CONFIG_RTW_NAPI
 #ifdef CONFIG_RTW_NAPI_DYNAMIC
 		if (!skb_queue_empty(&precvpriv->rx_napi_skb_queue)
-			&& !adapter_to_dvobj(padapter)->en_napi_dynamic			
+			&& !adapter_to_dvobj(padapter)->en_napi_dynamic
 			)
 			napi_recv(padapter, RTL_NAPI_WEIGHT);
 #endif
