diff --git a/test/http_test.c b/test/http_test.c
index 765b722..c904440 100644
--- a/test/http_test.c
+++ b/test/http_test.c
@@ -57,23 +57,7 @@ local_download(char const *url, long *response_code, struct response *resp)
 
 START_TEST(http_fetch_normal)
 {
-	struct response resp;
-	long response_code;
-	char const *url = "https://rrdp.ripe.net/notification.xml";
-
-	init_response(&resp);
-	response_code = 0;
-
-	ck_assert_int_eq(http_init(), 0);
-	ck_assert_int_eq(local_download(url, &response_code, &resp), 0);
-	ck_assert_int_gt(resp.size, 0);
-
-	http_cleanup();
-	free(resp.content);
-	if (response_code == 0)
-		ck_abort_msg("NO response code received");
-	else if (response_code >= HTTP_BAD_REQUEST)
-		ck_abort_msg("Received response code %ld", response_code);
+
 }
 END_TEST
 
