From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use the system `xxhash` library.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -490,16 +490,4 @@ if (ENABLE_VULKAN)
     endif()
 endif()
 
-set(XXHASH_BUILD_XXHSUM OFF)
-add_subdirectory(xxHash/cmake_unofficial EXCLUDE_FROM_ALL)
-target_compile_definitions(xxhash PRIVATE XXH_FORCE_MEMORY_ACCESS=2)
-if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
-    target_compile_definitions(xxhash PRIVATE XXH_VECTOR=XXH_SSE2)
-    message(STATUS "Enabling SSE2 for xxHash")
-elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|ARM64|armv8")
-    target_compile_definitions(xxhash PRIVATE XXH_VECTOR=XXH_NEON)
-    message(STATUS "Enabling NEON for xxHash")
-else()
-    target_compile_definitions(xxhash PRIVATE XXH_VECTOR=XXH_SCALAR)
-    message(STATUS "Disabling SIMD for xxHash")
-endif()
\ No newline at end of file
+add_library(xxhash SHARED IMPORTED)
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -217,4 +217,4 @@ if (SSE42_COMPILE_OPTION)
     target_compile_options(citra_common PRIVATE ${SSE42_COMPILE_OPTION})
 endif()
 
-target_link_libraries(citra_common PUBLIC xxHash::xxhash)
\ No newline at end of file
+target_link_libraries(citra_common PUBLIC xxhash)

