From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Link to Catch2 only when tests are enabled.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -41,6 +41,7 @@ else()
 endif()
 
 # Catch2
+if (ENABLE_TESTS)
 add_library(catch2 INTERFACE)
 if(USE_SYSTEM_CATCH2)
     find_package(Catch2 3.0.0 REQUIRED)
@@ -50,6 +51,7 @@ else()
     add_subdirectory(catch2)
 endif()
 target_link_libraries(catch2 INTERFACE Catch2::Catch2WithMain)
+endif()
 
 # Crypto++
 if(USE_SYSTEM_CRYPTOPP)
