--- a/src/Makevars.in
+++ b/src/Makevars.in
@@ -8,10 +8,10 @@
 all: px @TARGETS@ $(SHLIB)
 
 px: px.c
-	$(CC) $(CFLAGS) -Wall px.c -o px
+	$(CC) $(CFLAGS) $(LDFLAGS) -fPIE -Wall px.c -o px
 
 interrupt: interrupt.c
-	$(CC) $(CFLAGS) -Wall interrupt.c -o interrupt
+	$(CC) $(CFLAGS) $(LDFLAGS) -fPIE -Wall interrupt.c -o interrupt
 
 clean:
 	rm -rf $(SHLIB) $(OBJECTS) px.exe px interrupt.exe interrupt
