--- a/utils/Makefile.am	2021-09-05 14:28:23.674273509 +0200
+++ b/utils/Makefile.am	2021-09-05 14:28:56.501524237 +0200
@@ -336,7 +336,7 @@
 	@printf "Octave package file \"%s\" created\n" $(PKG_FILE)
 
 install-exec-local-oct: $(PKG_FILE)
-	@$(OCTAVE_CLI) -qfHW --eval "pkg install -global $<"
+	@$(OCTAVE_CLI) -qfHW --eval "pkg install -local $<"
 	@printf "package \"%s\" installed\n" $(PKG_NAME)
 else
 install-exec-local-oct:
