--- a/configure.ac
+++ b/configure.ac
@@ -1889,15 +1889,15 @@
 if test "$ol_with_bullet" != "no" ; then
 	AC_LANG_CPLUSPLUS
 	have_libbullet=no
-	AC_CHECK_HEADER(BulletCollision/CollisionShapes/btCollisionShape.h,[have_bullet_h=yes],[have_bullet_h=no])
+	AC_CHECK_HEADER(bullet/BulletCollision/CollisionShapes/btCollisionShape.h,[have_bullet_h=yes],[have_bullet_h=no])
 	if test "$have_bullet_h" = "yes" ; then
 		SAVE_LIBS="$LIBS"
-		try_BULLET_LIBS="-lbulletcollision -lbulletmath"
+		try_BULLET_LIBS="-lBulletCollision"
 		LIBS="$try_BULLET_LIBS $LIBS"
-		AC_CHECK_LIB(bulletcollision,btBulletCollisionProbe,[
+		AC_CHECK_LIB(BulletCollision,btBulletCollisionProbe,[
 			have_libbullet=yes
 			BULLET_LIBS="$try_BULLET_LIBS"
-		],[have_libbullet=no],[-lbulletmath])
+		],[have_libbullet=no],[])
 		LIBS="$save_LIBS"
 	fi
 
