--- a/configure.ac
+++ b/configure.ac
@@ -1653,9 +1653,9 @@
 		AC_DEFINE(HAVE_LTDL_H,1,[define if you have <ltdl.h>])
 		AC_CHECK_LIB(ltdl, lt_dlinit,[have_lt_dlinit=yes],[have_lt_dlinit=no])
 		if test $have_lt_dlinit = "yes" ; then
-			eval `libtool --tag=CC --config|grep '^wl='`
+			eval `$LIBTOOL --tag=CC --config|grep '^wl='`
 			LIBS="$LIBS -lltdl"
-			EXPORT_DYNAMIC_FLAG_SPEC=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
+			EXPORT_DYNAMIC_FLAG_SPEC=`($LIBTOOL --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
 			have_runtime_loading=yes
 			MODULE_LOADPATH=$libexecdir
 		fi
