From 5e209af787bf74327c0dcba1d75d6a5477dfb0cb Mon Sep 17 00:00:00 2001
From: William Breathitt Gray <vilhelm.gray@gmail.com>
Date: Wed, 8 Sep 2021 11:27:39 +0900
Subject: [PATCH] Pass -ltinfo to linker

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 04c5c3976..364dc9a0d 100644
--- a/makefile
+++ b/makefile
@@ -1368,7 +1368,7 @@ endif
 # Use gcc to link fbc from the bootstrap .o's
 # (assuming the rtlib was built already)
 ifneq ($(filter darwin freebsd dragonfly linux netbsd openbsd solaris,$(TARGET_OS)),)
-  BOOTSTRAP_LIBS := -lncurses -lm -pthread
+  BOOTSTRAP_LIBS := -ltinfo -lncurses -lm -pthread
 endif
 $(BOOTSTRAP_FBC): rtlib $(BOOTSTRAP_OBJ)
 	$(QUIET_LINK)$(CC) -o $@ $(libdir)/fbrt0.o bootstrap/$(FBTARGET)/*.o $(libdir)/libfb.a $(BOOTSTRAP_LIBS)
-- 
2.33.0

