From 73e4b497b3afa6530fabb1d445d0d4285fa302b5 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:16:33 +0200
Subject: [PATCH 1/4] use ounit2

---
 lib_test/dune | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib_test/dune b/lib_test/dune
index ba44355..f61938d 100644
--- a/lib_test/dune
+++ b/lib_test/dune
@@ -2,10 +2,10 @@
  (name test_runner)
  (package uri)
  (modules test_runner)
- (libraries uri uri.services oUnit))
+ (libraries uri uri.services ounit2))
 
 (test
  (name test_runner_sexp)
  (package uri-sexp)
  (modules test_runner_sexp)
- (libraries uri uri.services uri-sexp oUnit))
+ (libraries uri uri.services uri-sexp ounit2))

From e40a51aec751ff397c94cbde8faa7fc01141a281 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:17:15 +0200
Subject: [PATCH 2/4] depend on ounit2

---
 uri-sexp.opam | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uri-sexp.opam b/uri-sexp.opam
index d4b2e44..fb249a5 100644
--- a/uri-sexp.opam
+++ b/uri-sexp.opam
@@ -16,7 +16,7 @@ depends: [
   "dune" {>= "1.2.0"}
   "ppx_sexp_conv" {>= "v0.13.0"}
   "sexplib0"
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 build: [
   ["dune" "subst"] {pinned}

From 2615ace2358efc166f49b6dbc7167232086cadde Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:17:48 +0200
Subject: [PATCH 3/4] depend on ounit2

---
 uri-re.opam | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uri-re.opam b/uri-re.opam
index 3d525a0..3d34e3e 100644
--- a/uri-re.opam
+++ b/uri-re.opam
@@ -15,7 +15,7 @@ for parsing URI or URLs.
 depends: [
   "ocaml" {>= "4.04.0"}
   "dune" {>= "1.2.0"}
-  "ounit" {with-test & >= "1.0.2"}
+  "ounit2" {with-test & >= "1.0.2"}
   "ppx_sexp_conv" {with-test & >= "v0.9.0"}
   "re" {>= "1.9.0"}
   "stringext" {>= "1.4.0"}

From a3b57ec4b8582b73b49bd95c163b44b885fe582f Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:18:10 +0200
Subject: [PATCH 4/4] depend on ounit2

---
 uri.opam | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/uri.opam b/uri.opam
index ab1cbf6..78d08b0 100644
--- a/uri.opam
+++ b/uri.opam
@@ -15,7 +15,7 @@ for parsing URI or URLs.
 depends: [
   "ocaml" {>= "4.04.0"}
   "dune" {>= "1.2.0"}
-  "ounit" {with-test & >= "1.0.2"}
+  "ounit2" {with-test & >= "1.0.2"}
   "ppx_sexp_conv" {with-test & >= "v0.9.0"}
   "stringext" {>= "1.4.0"}
   "angstrom" {>= "0.14.0"}
