--- a/3rd/bee.lua/bee/crash/linux/unwind_linux.cpp
+++ b/3rd/bee.lua/bee/crash/linux/unwind_linux.cpp
@@ -6,12 +6,8 @@
 namespace bee::crash {
     void unwind(ucontext_t *ctx, uint16_t skip, unwind_callback func, void *ud) noexcept {
         unw_cursor_t cursor;
-#if defined(__aarch64__)
         unw_context_t *unw_ctx = (unw_context_t *)ctx;
         unw_init_local(&cursor, unw_ctx);
-#else
-        unw_init_local(&cursor, ctx);
-#endif
         while (unw_step(&cursor) > 0) {
             unw_word_t pc;
             unw_get_reg(&cursor, UNW_REG_IP, &pc);
--- a/3rd/luamake/bee.lua/bee/crash/unwind_linux.cpp
+++ b/3rd/luamake/bee.lua/bee/crash/unwind_linux.cpp
@@ -6,12 +6,8 @@
 namespace bee::crash {
     void unwind(ucontext_t *ctx, uint16_t skip, unwind_callback func, void *ud) noexcept {
         unw_cursor_t cursor;
-#if defined(__aarch64__)
         unw_context_t *unw_ctx = (unw_context_t *)ctx;
         unw_init_local(&cursor, unw_ctx);
-#else
-        unw_init_local(&cursor, ctx);
-#endif
         while (unw_step(&cursor) > 0) {
             unw_word_t pc;
             unw_get_reg(&cursor, UNW_REG_IP, &pc);
