diff -Naur old_env/SU2_CFD/include/output/tools/CWindowingTools.hpp new_env/SU2_CFD/include/output/tools/CWindowingTools.hpp
--- old_env/SU2_CFD/include/output/tools/CWindowingTools.hpp	2020-03-31 12:26:03.000000000 +0300
+++ new_env/SU2_CFD/include/output/tools/CWindowingTools.hpp	2020-05-10 17:04:24.000000000 +0300
@@ -28,7 +28,7 @@
 #pragma once
 
 #include <vector>
-#include "../../../Common/include/option_structure.hpp"
+#include "../../../../Common/include/option_structure.hpp"
 
 class CWindowingTools{
 public:
diff -Naur old_env/UnitTests/meson.build new_env/UnitTests/meson.build
--- old_env/UnitTests/meson.build	2020-05-10 17:03:43.000000000 +0300
+++ new_env/UnitTests/meson.build	2020-05-10 17:04:35.000000000 +0300
@@ -24,7 +24,7 @@
     test_driver = executable(
         'test_driver',
         unit_test_files,
-        install : true,
+        install : false,
         dependencies : [su2_cfd_dep, common_dep, su2_deps, catch2_dep],
         cpp_args: ['-fPIC', default_warning_flags, su2_cpp_args]
     )
