--- a/UnitTests/Common/geometry/dual_grid/CDualGrid_tests.cpp
+++ b/UnitTests/Common/geometry/dual_grid/CDualGrid_tests.cpp
@@ -28,8 +28,8 @@
 #include "catch.hpp"
 #include <sstream>
 #include <iomanip>
-#include "../../../Common/include/geometry/dual_grid/CEdge.hpp"
-#include "../../../Common/include/geometry/dual_grid/CVertex.hpp"
+#include "../../../../Common/include/geometry/dual_grid/CEdge.hpp"
+#include "../../../../Common/include/geometry/dual_grid/CVertex.hpp"
 
 TEST_CASE("Volume Computation", "[Dual Grid]") {
   
--- a/UnitTests/Common/geometry/primal_grid/CPrimalGrid_tests.cpp
+++ b/UnitTests/Common/geometry/primal_grid/CPrimalGrid_tests.cpp
@@ -27,8 +27,8 @@
 
 #include "catch.hpp"
 #include <sstream>
-#include "../../../Common/include/geometry/primal_grid/CPrimalGrid.hpp"
-#include "../../../Common/include/geometry/primal_grid/CHexahedron.hpp"
+#include "../../../../Common/include/geometry/primal_grid/CPrimalGrid.hpp"
+#include "../../../../Common/include/geometry/primal_grid/CHexahedron.hpp"
 
 TEST_CASE("Center of gravity computation", "[Primal Grid]") {
 
--- a/UnitTests/test_driver.cpp
+++ b/UnitTests/test_driver.cpp
@@ -31,8 +31,8 @@
 #define CATCH_CONFIG_RUNNER
 #include "catch.hpp"
 
-#include "../../../Common/include/parallelization/mpi_structure.hpp"
-#include "../../../Common/include/option_structure.hpp"
+#include "../Common/include/parallelization/mpi_structure.hpp"
+#include "../Common/include/option_structure.hpp"
 
 int main(int argc, char *argv[]) {
 
--- a/SU2_GEO/include/SU2_GEO.hpp
+++ b/SU2_GEO/include/SU2_GEO.hpp
@@ -38,7 +38,7 @@
 
 #include "../../Common/include/geometry/CPhysicalGeometry.hpp"
 #include "../../Common/include/CConfig.hpp"
-#include "../../../Common/include/grid_movement/CSurfaceMovement.hpp"
-#include "../../../Common/include/grid_movement/CFreeFormDefBox.hpp"
+#include "../../Common/include/grid_movement/CSurfaceMovement.hpp"
+#include "../../Common/include/grid_movement/CFreeFormDefBox.hpp"
 
 using namespace std;
