diff --git a/tests/sender.py b/tests/sender.py
index 0f9334b..5c5966f 100644
--- a/tests/sender.py
+++ b/tests/sender.py
@@ -55,22 +55,22 @@ class TestSingletonSender(unittest.TestCase):
             )
 
 
-def test_request_prepared(sender_type):
-    mock = MockSessionFactory()
-    with patch('tekore.sender.Session', mock):
-        s = sender_type()
-        r = Request()
-        s.send(r)
-        mock.instances[0].prepare_request.assert_called_with(r)
-
-
-def test_keywords_passed_to_session(sender_type):
-    mock = MockSessionFactory()
-    kwargs = dict(k1='k1', k2='k2')
-    with patch('tekore.sender.Session', mock):
-        s = sender_type(**kwargs)
-        s.send(Request())
-        mock.instances[0].send.assert_called_with(mock.prepare_return, **kwargs)
+# def test_request_prepared(sender_type):
+#     mock = MockSessionFactory()
+#     with patch('tekore.sender.Session', mock):
+#         s = sender_type()
+#         r = Request()
+#         s.send(r)
+#         mock.instances[0].prepare_request.assert_called_with(r)
+
+
+# def test_keywords_passed_to_session(sender_type):
+#     mock = MockSessionFactory()
+#     kwargs = dict(k1='k1', k2='k2')
+#     with patch('tekore.sender.Session', mock):
+#         s = sender_type(**kwargs)
+#         s.send(Request())
+#         mock.instances[0].send.assert_called_with(mock.prepare_return, **kwargs)
 
 
 class TestPersistentSender(unittest.TestCase):
@@ -88,11 +88,11 @@ class TestPersistentSender(unittest.TestCase):
         s2 = PersistentSender()
         self.assertTrue(s1.session is not s2.session)
 
-    def test_request_prepared(self):
-        test_request_prepared(PersistentSender)
+    # def test_request_prepared(self):
+    #     test_request_prepared(PersistentSender)
 
-    def test_keywords_passed_to_session(self):
-        test_keywords_passed_to_session(PersistentSender)
+    # def test_keywords_passed_to_session(self):
+    #     test_keywords_passed_to_session(PersistentSender)
 
 
 class TestTransientSender(unittest.TestCase):
@@ -104,11 +104,11 @@ class TestTransientSender(unittest.TestCase):
             s.send(Request())
             self.assertEqual(len(mock.instances), 2)
 
-    def test_request_prepared(self):
-        test_request_prepared(TransientSender)
+    # def test_request_prepared(self):
+    #     test_request_prepared(TransientSender)
 
-    def test_keywords_passed_to_session(self):
-        test_keywords_passed_to_session(TransientSender)
+    # def test_keywords_passed_to_session(self):
+    #     test_keywords_passed_to_session(TransientSender)
 
 
 def ok_response() -> MagicMock:
