--- a/code/Makefile
+++ b/code/Makefile
@@ -1,8 +1,6 @@
 DEST_DIR = 	../exec
 DEST=		${DEST_DIR}/chaco
-CC = 		gcc
 IFLAG =		-Imain
-CFLAGS =	-O2
 OFLAGS =	-O2
 
 main_file=	 main/main.c
@@ -98,11 +96,11 @@
 all: ${DEST} ${DEST_DIR}/libchaco.a
 
 ${DEST}:	${FILES.o} Makefile
-		${CC} ${OFLAGS} ${FILES.o} -lm -o ${DEST}
+		${CC} ${CFLAGS} ${LDFLAGS} -fPIE ${FILES.o} -lm -o ${DEST}
 
 ${DEST_DIR}/libchaco.a:	${libFILES.o} Makefile
-		ar r ${DEST_DIR}/libchaco.a ${libFILES.o}
-		ranlib ${DEST_DIR}/libchaco.a
+		${AR} r ${DEST_DIR}/libchaco.a ${libFILES.o}
+		${RANLIB} ${DEST_DIR}/libchaco.a
 
 lint:
 		lint ${IFLAG} ${FILES.c} -lm
@@ -114,4 +112,4 @@
 		rm -f */*.o ${DEST_DIR}/core
 
 .c.o:
-		${CC} -c ${IFLAG} ${CFLAGS} -o $*.o $*.c
+		${CC} -c ${IFLAG} ${CFLAGS} -fPIC -o $*.o $*.c
