diff --git a/CMakeLists.txt b/CMakeLists.txt
index 77d2f6b..665c35f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -206,14 +206,6 @@ if (USE_ZLIB)
   target_compile_definitions(ixwebsocket PUBLIC IXWEBSOCKET_USE_ZLIB)
 endif()
 
-# brew install libdeflate
-find_package(Deflate)
-if (DEFLATE_FOUND)
-  include_directories(${DEFLATE_INCLUDE_DIRS})
-  target_link_libraries(ixwebsocket ${DEFLATE_LIBRARIES})
-  target_compile_definitions(ixwebsocket PUBLIC IXWEBSOCKET_USE_DEFLATE)
-endif()
-
 if (WIN32)
   target_link_libraries(ixwebsocket wsock32 ws2_32 shlwapi)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
