--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -232,8 +232,10 @@
 
 enable_testing()
 
+find_package(Potassco REQUIRED)
+find_package(Clasp REQUIRED)
+
 # NOTE: assumes that submodule has been initialized
-set(CLASP_BUILD_APP ${CLINGO_BUILD_APPS} CACHE BOOL "")
 if (CLINGO_USE_LIB)
     set(clingo_library_targets)
     add_library(libclingo INTERFACE IMPORTED)
@@ -241,11 +243,6 @@
     set_property(TARGET libclingo PROPERTY INTERFACE_LINK_LIBRARIES "clingo")
 else()
     set(clingo_library_targets libgringo libpotassco libclasp libreify)
-    set(CLASP_USE_LOCAL_LIB_POTASSCO ON CACHE BOOL "" FORCE)
-    if (NOT CLINGO_BUILD_SHARED AND CLINGO_INSTALL_LIB)
-        set(CLASP_INSTALL_LIB ${CLINGO_INSTALL_LIB} CACHE BOOL "" FORCE)
-    endif()
-    add_subdirectory(clasp)
     add_subdirectory(libreify)
     add_subdirectory(libgringo)
     add_subdirectory(libclingo)
--- a/libclingo/CMakeLists.txt
+++ b/libclingo/CMakeLists.txt
@@ -37,8 +37,6 @@
     ${source-group})
 # ]]]
 
-list(APPEND header "${CLASP_SOURCE_DIR}/app/clasp_app.h")
-list(APPEND source "${CLASP_SOURCE_DIR}/app/clasp_app.cpp")
 
 set(clingo_public_scope_ PUBLIC)
 set(clingo_private_scope_ PRIVATE)
@@ -62,7 +60,7 @@
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>"
     "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
     ${clingo_private_scope_}
-    "$<BUILD_INTERFACE:${CLASP_SOURCE_DIR}/app>")
+    "/usr/include/clasp/cli")
 target_compile_definitions(libclingo ${clingo_private_scope_} CLINGO_BUILD_LIBRARY)
 
 if (NOT CLINGO_BUILD_SHARED)
