--- a/libclingo/tests/CMakeLists.txt
+++ b/libclingo/tests/CMakeLists.txt
@@ -2,7 +2,6 @@
 set(ide_source_group "Source Files")
 set(source-group
     "${CMAKE_CURRENT_SOURCE_DIR}/astv2.cc"
-    "${CMAKE_CURRENT_SOURCE_DIR}/catch.hpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/clingo.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/main.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/propagator.cc"
--- a/libgringo/tests/CMakeLists.txt
+++ b/libgringo/tests/CMakeLists.txt
@@ -1,7 +1,6 @@
 # [[[source: .
 set(ide_source_group "Source Files")
 set(source-group
-    "${CMAKE_CURRENT_SOURCE_DIR}/catch.hpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/graph.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/intervals.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/main.cc"
--- a/libreify/tests/CMakeLists.txt
+++ b/libreify/tests/CMakeLists.txt
@@ -1,7 +1,6 @@
 # [[[source: .
 set(ide_source_group "Source Files")
 set(source-group
-    "${CMAKE_CURRENT_SOURCE_DIR}/catch.hpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/main.cc"
     "${CMAKE_CURRENT_SOURCE_DIR}/program.cc")
 source_group("${ide_source_group}" FILES ${source-group})
