From 18d71003c12f241831811eff86337ddf2223d754 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Tue, 31 May 2022 10:52:28 +0200
Subject: [PATCH] remove setuptools_scm constraint

--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@
         'Programming Language :: Python :: 3'
     ],
     python_requires=">=3.7",
-    setup_requires=['setuptools_scm>=4,<6.1'],
+    setup_requires=['setuptools_scm>=4'],
     install_requires=[
         'protobuf'
     ]
