diff --git a/Makefile b/Makefile2
index 3e5933a..c8787d8 100644
--- a/Makefile
+++ b/Makefile
@@ -64,19 +64,19 @@
   KERNELNAME =$(shell uname -s)
 
   ifeq ($(KERNELNAME),Linux)
-    CXXFLAGS+= -O3 -c -fmessage-length=0
-    LDFLAGS+=-s -zrelro -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
+    CXXFLAGS+= -c
+    LDFLAGS+= -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
   else ifeq ($(KERNELNAME), FreeBSD)
     # To be done...
-    CXXFLAGS+= -O3 -c -fmessage-length=0
-    LDFLAGS+=-s -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
+    CXXFLAGS+= -c
+    LDFLAGS+= -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
   else ifeq ($(KERNELNAME), Darwin)
-    CXXFLAGS+= -O3 -c -fmessage-length=0
-    LDFLAGS+=-s -L"src/sc_Squirrel3/lib" -L"/usr/local/Cellar/glew" -L"/usr/local/Cellar/sdl" -I"/usr/local/Cellar/glew" -I"/usr/local/Cellar/sdl" -framework Cocoa -framework OpenGL -lSDLmain -lSDL -lGLEW -lsquirrel -lsqstdlib
+    CXXFLAGS+= -c
+    LDFLAGS+= -L"src/sc_Squirrel3/lib" -L"/usr/local/Cellar/glew" -L"/usr/local/Cellar/sdl" -I"/usr/local/Cellar/glew" -I"/usr/local/Cellar/sdl" -framework Cocoa -framework OpenGL -lSDLmain -lSDL -lGLEW -lsquirrel -lsqstdlib
   else
     # Unknow kernel... trying default flags
-    CXXFLAGS+= -O3 -c -fmessage-length=0
-    LDFLAGS+=-s -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
+    CXXFLAGS+= -c
+    LDFLAGS+= -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
   endif
 endif
 
@@ -94,11 +94,11 @@
 installable: all
 
 squirrel: 
-	cd src/sc_Squirrel3 ; $(MAKE) sq$(CPU_BITS)
+	$(MAKE) -C src/sc_Squirrel3  sq$(CPU_BITS)
 
 clean:
 	@rm -rf obj
-	@cd src/sc_Squirrel3 ; $(MAKE) clean
+	@$(MAKE) -C src/sc_Squirrel3  clean
 	@rm -f $(PROGNAME)
 
 mrproper: clean
diff --git a/src/sc_Squirrel3/Makefile b/src/sc_Squirrel3/Makefile
index 8bc76c8..fb396eb 100644
--- a/src/sc_Squirrel3/Makefile
+++ b/src/sc_Squirrel3/Makefile
@@ -2,33 +2,33 @@
 SQUIRREL=.
 
 sq32:
-	cd squirrel; $(MAKE) 
-	cd sqstdlib; $(MAKE) 
-	cd sq; $(MAKE) 
+	$(MAKE) -C squirrel 
+	$(MAKE) -C sqstdlib 
+	$(MAKE) -C sq 
 
 sqprof:
-	cd squirrel; $(MAKE) sqprof
-	cd sqstdlib; $(MAKE) sqprof
-	cd sq; $(MAKE) sqprof
+	$(MAKE) -C squirrel sqprof
+	$(MAKE) -C sqstdlib sqprof
+	$(MAKE) -C sq sqprof
 
 sq64:
-	cd squirrel; $(MAKE) sq64
-	cd sqstdlib; $(MAKE) sq64
-	cd sq; $(MAKE) sq64
+	$(MAKE) -C squirrel sq64
+	$(MAKE) -C sqstdlib sq64
+	$(MAKE) -C sq sq64
 	
 sq64d:
-	cd squirrel; $(MAKE) sq64d
-	cd sqstdlib; $(MAKE) sq64d
-	cd sq; $(MAKE) sq64d
+	$(MAKE) -C squirrel sq64d
+	$(MAKE) -C sqstdlib sq64d
+	$(MAKE) -C sq sq64d
 
 .PHONY: clean mrproper
 
 clean:
-	cd squirrel; $(MAKE) clean
-	cd sqstdlib; $(MAKE) clean
-	cd sq; $(MAKE) clean
+	$(MAKE) -C squirrel clean
+	$(MAKE) -C sqstdlib clean
+	$(MAKE) -C sq clean
 
 mrproper:
-	cd squirrel; $(MAKE) mrproper
-	cd sqstdlib; $(MAKE) mrproper
-	cd sq; $(MAKE) mrproper
+	$(MAKE) -C squirrel mrproper
+	$(MAKE) -C sqstdlib mrproper
+	$(MAKE) -C sq mrproper
diff --git a/src/sc_Squirrel3/sq/Makefile b/src/sc_Squirrel3/sq/Makefile
index 08fb2ef..8cc1cbc 100644
--- a/src/sc_Squirrel3/sq/Makefile
+++ b/src/sc_Squirrel3/sq/Makefile
@@ -12,13 +12,13 @@ SRCS= sq.c
 	
 	
 sq32:
-	$(CXX) -O2 -s -fno-exceptions -fno-rtti -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
+	$(CXX) -O2 -fno-exceptions -fno-rtti -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
 
 sqprof:
 	$(CXX) -O2 -pg -fno-exceptions -fno-rtti -pie -gstabs -g3 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
 	
 sq64:
-	$(CXX) -O2 -s -m64 -fno-exceptions -fno-rtti -D_SQ64 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
+	$(CXX) -O2 -m64 -fno-exceptions -fno-rtti -D_SQ64 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
 	
 sq64d:
 	$(CXX) -O0 -g3 -m64 -fno-exceptions -fno-rtti -D_SQ64 -o $(OUT) $(SRCS) $(INCZ) $(LIBZ) $(LIB)
