--- a/src/ACompileSettings.h
+++ b/src/ACompileSettings.h
@@ -129,13 +129,11 @@
 
 #ifdef __linux__
 
-  #define COMPILEOPTION_SAVEFOLDERNAME "Blackvoxel"
+  #define COMPILEOPTION_SAVEFOLDERNAME ".local/share/blackvoxel"
   #define COMPILEOPTION_USEHOMEDIRSTORAGE 1
   #define COMPILEOPTION_WINXPTIMECOMPUTING 0
 
-  #if DEVELOPPEMENT_ON == 1
-    #define COMPILEOPTION_DATAFILESPATH ""
-  #else
+  #if DEVELOPPEMENT_ON == 0
     #ifndef COMPILEOPTION_DATAFILESPATH
       #if COMPILEOPTION_ONLYSCHOOLMODE
         #define COMPILEOPTION_DATAFILESPATH "/usr/share/blackvoxel_school"
