--- a/Makefile	2024-10-21 17:33:29.782931946 -0400
+++ b/Makefile	2024-10-21 17:35:55.202026636 -0400
@@ -64,8 +64,10 @@
   KERNELNAME =$(shell uname -s)
 
   ifeq ($(KERNELNAME),Linux)
-    CXXFLAGS+= -O3 -c -fmessage-length=0
-    LDFLAGS+=-L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
+    UCXXFLAGS:= $(shell emerge --info | grep -w CXXFLAGS=".*" | cut -d '"' -f2)
+    CXXFLAGS+= $(UCXXFLAGS) -c -fmessage-length=0 
+    ULDFLAGS:= $(shell emerge --info | grep -w LDFLAGS=".*" | cut -d '"' -f2)
+    LDFLAGS+= $(ULDFLAGS) -L"src/sc_Squirrel3/lib" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib
   else ifeq ($(KERNELNAME), FreeBSD)
     # To be done...
     CXXFLAGS+= -O3 -c -fmessage-length=0
