diff --git a/setup.py b/setup.py
index be115a3..ad34a47 100755
--- a/setup.py
+++ b/setup.py
@@ -13,7 +13,7 @@ here = os.path.abspath(os.path.dirname(__file__))
 with open(os.path.join(here, 'README.md'), encoding='utf-8') as f:
     long_description = f.read()
 
-define_macros = [('HB_NO_MT', '1'), ('HB_EXPERIMENTAL_API', '1')]
+define_macros = [('HB_EXPERIMENTAL_API', '1')]
 linetrace = False
 if int(os.environ.get('CYTHON_LINETRACE', '0')):
     linetrace = True
@@ -21,7 +21,7 @@ if int(os.environ.get('CYTHON_LINETRACE', '0')):
 
 extra_compile_args = []
 extra_link_args = []
-libraries = []
+libraries = ['harfbuzz', 'harfbuzz-subset']
 if platform.system() != 'Windows':
     extra_compile_args.append('-std=c++11')
     define_macros.append(('HAVE_MMAP', '1'))
@@ -39,12 +39,9 @@ if platform.system() == 'Darwin':
 extension = Extension(
     'uharfbuzz._harfbuzz',
     define_macros=define_macros,
-    include_dirs=['harfbuzz/src'],
+    include_dirs=['/usr/include/harfbuzz'],
     sources=[
         'src/uharfbuzz/_harfbuzz.pyx',
-        'harfbuzz/src/harfbuzz.cc',
-        'harfbuzz/src/hb-subset-repacker.cc',
-        'harfbuzz/src/graph/gsubgpos-context.cc',
         'src/uharfbuzz/_draw_test_funcs.cc',
     ],
     language='c++',
