--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,7 +126,7 @@
 # Library: SIONLib (Optional)
 if(ENABLE_SIONLIB)
     set(SIONLIBBASE "" CACHE FILEPATH "base path to SIONlib installation")
-    set(SIONLIB_INCLUDE_DIR "${SIONLIBBASE}/include/")
+    set(SIONLIB_INCLUDE_DIR "/usr/include/sionlib")
 
     list(APPEND INC_PRIV ${SIONLIB_INCLUDE_DIR})
     find_library(SIONLIB_MPI NAMES "sionmpi_64" 
