--- a/makefile	2022-02-09 17:44:15.642864568 +0100
+++ b/makefile	2022-02-09 17:46:30.128907145 +0100
@@ -1,7 +1,6 @@
-CC := cc
 
 ifeq ($(PREFIX),)
-	PREFIX := /usr/local
+	PREFIX := /usr
 endif
 
 BUILD_FILES_COMMON := \
@@ -19,10 +18,10 @@
 	src/zenmonitor-cli.c
 
 build:
-	$(CC) -Isrc/include `pkg-config --cflags gtk+-3.0` $(BUILD_FILES_GUI) -o zenmonitor `pkg-config --libs gtk+-3.0` -lm -no-pie -Wall $(CFLAGS)
+	$(CC) -Isrc/include `pkg-config --cflags gtk+-3.0` $(BUILD_FILES_GUI) -o zenmonitor `pkg-config --libs gtk+-3.0` -lm -Wall $(CFLAGS) ${LDFLAGS}
 
 build-cli:
-	$(CC) -Isrc/include `pkg-config --cflags glib-2.0` $(BUILD_FILES_CLI) -o zenmonitor-cli `pkg-config --libs glib-2.0` -lm -lncurses -no-pie -Wall $(CFLAGS)
+	$(CC) -Isrc/include `pkg-config --cflags glib-2.0` $(BUILD_FILES_CLI) -o zenmonitor-cli `pkg-config --libs glib-2.0` -lm -ltinfo -lncurses -Wall $(CFLAGS) ${LDFLAGS}
 
 install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
