--- a/pcsd/pcsd-ruby.service
+++ b/pcsd/pcsd-ruby.service
@@ -5,13 +5,12 @@
 Requires=network-online.target
 After=network-online.target
 # Stop the service automatically if nothing that depends on it is running
-StopWhenUnneeded=true
-# When stopping or restarting pcsd, stop or restart pcsd-ruby as well
+StopWhenUnneeded=false
+# When stopping or restarting pcsd, stop or restart pcsd-daemon as well
 PartOf=pcsd.service
 
 [Service]
-EnvironmentFile=/etc/sysconfig/pcsd
-Environment=GEM_HOME=/usr/lib/pcsd/vendor/bundle/ruby
+EnvironmentFile=/etc/default/pcsd
 # This file holds the selinux context
 ExecStart=/usr/lib/pcsd/pcsd
 Type=notify
