From 73e90ae7f31600ff9c0edf0aed3cace9cd8c8a35 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Thu, 7 Apr 2022 17:02:55 +0200
Subject: [PATCH] remove bashisms

`==` is not POSIX
--- a/m4/ac_compare_versions.m4
+++ b/m4/ac_compare_versions.m4
@@ -6,7 +6,7 @@ dnl op can be:
 dnl
 dnl lt or <
 dnl le or <=
-dnl eq or ==
+dnl eq or =
 dnl ge or >=
 dnl gt or >
 dnl
@@ -20,7 +20,7 @@ AC_DEFUN([AC_COMPARE_VERSIONS],[
 	verA="$1"
 	op="$2"
 	verB="$3"
-	if test "x$verA" == "x" || test "x$verB" == "x" || test "x$op" == x; then
+	if test "x$verA" = "x" || test "x$verB" = "x" || test "x$op" = x; then
 		AC_MSG_ERROR([ac_compare_versions: Missing parameters])
 	fi
 	case "$op" in
@@ -36,7 +36,7 @@ AC_DEFUN([AC_COMPARE_VERSIONS],[
 				result=true
 			fi
 			;;
-		"eq"|"==")
+		"eq"|"=")
 			if test "$verB" = "$verA"; then
 				result=true
 			fi
