--- a/test/Makefile.mk
+++ b/test/Makefile.mk
@@ -422,8 +422,8 @@
                           # end
 
 test_geopm_test_LDADD = libgeopmd.la \
-                        libgmock.a \
-                        libgtest.a \
+                        -lgmock \
+                        -lgtest \
                         # end
 
 test_geopm_test_CPPFLAGS = $(AM_CPPFLAGS) -Iplugin
@@ -453,4 +453,3 @@
 
 CLEAN_LOCAL_TARGETS += clean-local-gtest-script-links
 
-include test/googletest.mk
