--- a/Makefile.am
+++ b/Makefile.am
@@ -30,7 +30,7 @@
 
 geopmincludedir = $(includedir)/geopm
 
-geopminclude_HEADERS = contrib/json11/json11.hpp \
+geopminclude_HEADERS =  \
                        src/geopm/Agg.hpp \
                        src/geopm/CircularBuffer.hpp \
                        src/geopm/Exception.hpp \
@@ -60,8 +60,6 @@
              Makefile.am \
              autogen.sh \
              configure.ac \
-             contrib/json11/LICENSE.txt \
-             contrib/json11/README.md \
              dox/blurb.md \
              dox/Doxyfile.in \
              geopm-service.spec.in \
@@ -91,7 +89,6 @@
              geopmsession \
              requirements.txt \
              setup.py \
-             src/geopm/json11.hpp \
              geopm.service \
              AUTHORS \
              CODE_OF_CONDUCT.md \
@@ -107,11 +104,9 @@
 
 # Add ABI version
 libgeopmd_la_LDFLAGS = $(AM_LDFLAGS) -version-info $(geopm_abi_version)
-
+libgeopmd_la_LIBADD = -ljson11
 libgeopmd_la_SOURCES = $(include_HEADERS) \
                        $(geopminclude_HEADERS) \
-                       contrib/json11/json11.cpp \
-                       contrib/json11/json11.hpp \
                        src/GPUTopo.cpp \
                        src/GPUTopo.hpp \
                        src/GPUTopoNull.cpp \
--- a/src/geopm/MSRIOGroup.hpp
+++ b/src/geopm/MSRIOGroup.hpp
@@ -13,7 +13,7 @@
 #include <string>
 #include <set>
 
-#include "geopm/json11.hpp"
+#include "json11.hpp"
 
 #include "IOGroup.hpp"
 #include "geopm_time.h"
--- a/src/MSRIOGroup.cpp
+++ b/src/MSRIOGroup.cpp
@@ -17,7 +17,7 @@
 #include <iostream>
 #include <set>
 
-#include "geopm/json11.hpp"
+#include "json11.hpp"
 
 #include "geopm_sched.h"
 #include "geopm/Exception.hpp"
--- a/src/SaveControl.cpp
+++ b/src/SaveControl.cpp
@@ -7,7 +7,7 @@
 
 #include "SaveControl.hpp"
 
-#include "geopm/json11.hpp"
+#include "json11.hpp"
 #include "geopm/Helper.hpp"
 #include "geopm/Exception.hpp"
 #include "geopm/IOGroup.hpp"
--- a/test/MSRIOGroupTest.cpp
+++ b/test/MSRIOGroupTest.cpp
@@ -21,7 +21,7 @@
 #include <algorithm>
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
-#include "geopm/json11.hpp"
+#include "json11.hpp"
 
 #include "geopm_sched.h"
 #include "geopm_hash.h"
