diff --git a/tests/test_encoding.py b/tests/test_encoding.py
index b77ca13..7ef65d9 100644
--- a/tests/test_encoding.py
+++ b/tests/test_encoding.py
@@ -152,7 +152,7 @@ def test_basic_roundtrip(make_enc):
 def test_hyp_roundtrip(make_enc: Callable[[], tiktoken.Encoding], text):
     enc = make_enc()
 
-    assert text == enc.decode(enc.encode(text))
+    assert text == enc.decode(enc.encode(text, allowed_special="all"))
 
 
 @pytest.mark.parametrize("make_enc", ENCODING_FACTORIES)
