--- a/setup.cfg	2018-12-19 11:00:58.000000000 -0000
+++ b/setup.cfg	2021-05-08 01:39:02.631499047 -0000
@@ -1,6 +1,6 @@
 [tool:pytest]
 python_files = test*.py
-addopts = --tb=native -p no:doctest -p no:logging --cov=raven
+addopts = --tb=native -p no:doctest -p no:logging
 norecursedirs = raven build bin dist htmlcov hooks node_modules .* {args}
 django_settings_module = tests.contrib.django.settings
 python_paths = tests
