--- fs2_open-21.4.1/cmake/toolchain-gcc.cmake	2021-11-01 18:00:53.000000000 +0100
+++ fs2_open-21.4.1/cmake/toolchain-gcc.cmake	2022-03-27 17:52:26.738107292 +0200
@@ -27,6 +27,11 @@
 set(COMPILER_FLAGS "")
 set(LINKER_FLAGS "")
 
+# For C and C++, the values can be overwritten independently
+if(DEFINED ENV{LDFLAGS})
+    set(LINKER_FLAGS $ENV{LDFLAGS})
+endif()
+
 if (GCC_USE_GOLD)
 	OPTION(GCC_INCREMENTAL_LINKING "Use incremental linking" OFF)
 	set(LINKER_FLAGS "${LINKER_FLAGS} -fuse-ld=gold")
