--- fs2_open/code/CMakeLists.txt	2022-03-27 21:03:27.290788070 +0200
+++ fs2_open/code/CMakeLists.txt	2022-03-27 21:06:50.271782416 +0200
@@ -41,8 +41,6 @@
 
 target_link_libraries(code PUBLIC md5)
 
-target_link_libraries(code PUBLIC discord-rpc)
-
 target_link_libraries(code PUBLIC libRocket)
 
 target_link_libraries(code PUBLIC pcp)
--- fs2_open/code/source_groups.cmake	2021-11-01 18:00:53.000000000 +0100
+++ fs2_open/code/source_groups.cmake	2022-03-27 21:35:47.595734025 +0200
@@ -705,11 +705,6 @@
 	libs/antlr/ErrorListener.h
 	)
 
-add_file_folder("Libs\\\\Discord"
-	libs/discord/discord.cpp
-	libs/discord/discord.h
-)
-
 if (FSO_BUILD_WITH_FFMPEG)
 	add_file_folder("Libs\\\\FFmpeg"
 		libs/ffmpeg/FFmpeg.cpp
--- fs2_open/lib/CMakeLists.txt	2022-03-27 21:03:27.287788070 +0200
+++ fs2_open/lib/CMakeLists.txt	2022-03-27 21:07:12.849781787 +0200
@@ -33,8 +33,6 @@
 	include(FFmpeg.cmake)
 endif()
 
-add_subdirectory(discord)
-
 include(libRocket.cmake)
 
 add_subdirectory(libpcp)
--- fs2_open/freespace2/freespace.cpp	2021-11-01 18:00:53.000000000 +0100
+++ fs2_open/freespace2/freespace.cpp	2022-03-27 21:07:50.111780749 +0200
@@ -83,7 +83,6 @@
 #include "jumpnode/jumpnode.h"
 #include "lab/labv2.h"
 #include "lab/wmcgui.h" //So that GUI_System can be initialized
-#include "libs/discord/discord.h"
 #include "libs/ffmpeg/FFmpeg.h"
 #include "lighting/lighting.h"
 #include "localization/localize.h"
@@ -1940,8 +1939,6 @@
 	libs::ffmpeg::initialize();
 #endif
 
-	libs::discord::init();
-
 	nprintf(("General", "Ships.tbl is : %s\n", Game_ships_tbl_valid ? "VALID" : "INVALID!!!!"));
 	nprintf(("General", "Weapons.tbl is : %s\n", Game_weapons_tbl_valid ? "VALID" : "INVALID!!!!"));
 
