diff --git a/lib.rs b/lib.bk.rs
index 7fd23b9..894ba25 100644
--- a/lib.rs
+++ b/lib.bk.rs
@@ -100,6 +100,8 @@ pub fn runtime_dir() -> std::path::PathBuf {
         return std::path::PathBuf::from(dir).parent().unwrap().join(RT_DIR);
     }

+	return std::path::PathBuf::from("%%DATADIR%%").join(RT_DIR);
+
     // fallback to location of the executable being run
     std::env::current_exe()
         .ok()

