index cd69a3d..cd8b54c 100644
--- a/libcprime.pro
+++ b/libcprime.pro
@@ -42,8 +42,6 @@ unix {
         contains(DEFINES, LIB64): target.path = $$INSTALL_PREFIX/lib64
         else: target.path = $$INSTALL_PREFIX/lib
 
-        target.path	  = $$PREFIX/lib/
-
         includes.files    = cprime/*.h libcprime_global.h cprime/application/capplication.h
         includes.path     = $$PREFIX/include/cprime/
 
