diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2dfd6c..96f34b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,6 @@ if(CMAKE_BUILD_TYPE MATCHES Debug OR CMAKE_BUILD_TYPE MATCHES DEBUG)
   message(STATUS "Configuring hyprlock in Debug with CMake")
   add_compile_definitions(HYPRLAND_DEBUG)
 else()
-  add_compile_options(-O3)
   message(STATUS "Configuring hyprlock in Release with CMake")
 endif()
 
