--- /var/tmp/portage/dev-R/fs-1.5.2/work/fs-1.5.2/src/Makevars
+++ fs-1.5.2/src/Makevars
@@ -1,8 +1,6 @@
 UNAME := $(shell uname)
 
-LIBUV := libuv-1.38.1
-
-PKG_LIBS = ./$(LIBUV)/.libs/libuv.a -pthread
+PKG_LIBS = -pthread -luv
 
 SOURCES = $(wildcard *.cc unix/*.cc)
 OBJECTS = $(SOURCES:.cc=.o)
@@ -21,24 +19,13 @@
 OBJECTS +=  bsd/setmode.o bsd/strmode.o bsd/reallocarray.o
 endif
 
-PKG_CPPFLAGS = -I./$(LIBUV)/include -I. -pthread
-
-all: $(SHLIB)
-
-$(SHLIB): $(LIBUV)/.libs/libuv.a
-
-$(LIBUV)/Makefile:
-	(cd $(LIBUV) \
-	&& CC="$(CC)" CFLAGS="$(CFLAGS) $(CPPFLAGS) $(CPICFLAGS) $(C_VISIBILITY)" AR="$(AR)" RANLIB="$(RANLIB)" LDFLAGS="$(LDFLAGS)" ./configure --quiet)
+PKG_CPPFLAGS = -I./usr/include/uv -I. -pthread
 
-$(LIBUV)/.libs/libuv.a: $(LIBUV)/Makefile
-	$(MAKE) --directory=$(LIBUV) \
-		HAVE_DTRACE=0
+all:
 
-.PHONY: shlib-clean clean
+.PHONY: clean
 
 # shlib-clean: clean
 
 clean:
-	$(MAKE) --directory=$(LIBUV) distclean
 	rm -f $(OBJECTS)
