--- a/meson.build
+++ b/meson.build
@@ -303,6 +303,7 @@ if cxx.has_header('sys/time.h')
 install_man('docs/ots-sanitize.1')
 
 # https://mesonbuild.com/Dependencies.html#gtest-and-gmock
+if get_option('tests')
 gtest = dependency('gtest', main : true, fallback : ['gtest', 'gtest_main_dep'])
 
 cff_charstring = executable('cff_charstring',
@@ -342,3 +343,4 @@ test('test_fuzzed_fonts',
     'top_builddir=@0@'.format(meson.current_build_dir()),
   ]
 )
+endif
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,2 +1,3 @@
+option('tests', type : 'boolean', value : false, description : 'Enable tests')
 option('graphite', type : 'boolean', value : true, description : 'Sanitize Graphite tables')
 option('fuzzer_ldflags', type: 'string', description : 'Extra LDFLAGS used during linking of fuzzing binaries')
