From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use system `quirc`.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/3rdparty/QrDecoder.cpp
+++ b/3rdparty/QrDecoder.cpp
@@ -1,10 +1,8 @@
 #include "QrDecoder.h"
 
-#include "quirc/quirc.h"
+#include <quirc.h>
 #include <qdebug.h>
 
-#include "quirc/quirc_internal.h"
-
 QrDecoder::QrDecoder()
     : m_qr(quirc_new())
 {
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,11 +82,6 @@ set(PROJECT_SOURCES
         3rdparty/qv2ray/v2/ui/widgets/speedchart/SpeedWidget.hpp
         3rdparty/qv2ray/v2/proxy/QvProxyConfigurator.cpp
 
-        3rdparty/quirc/decode.c
-        3rdparty/quirc/identify.c
-        3rdparty/quirc/quirc.c
-        3rdparty/quirc/version_db.c
-
         3rdparty/protorpc/rpc_client.cc
         3rdparty/protorpc/rpc_conn.cc
         3rdparty/protorpc/rpc_wire.cc
@@ -283,6 +278,7 @@ target_link_libraries(Throne PRIVATE
         Threads::Threads
         ${NKR_EXTERNAL_TARGETS}
         ${PLATFORM_LIBRARIES}
+        quirc
 )
 
 qt_finalize_executable(Throne)
