From 7b171c0da2b9289468696424fe51f76e1c365bb5 Mon Sep 17 00:00:00 2001
From: rathann <dominik@greysector.net>
Date: Tue, 7 Oct 2025 00:27:19 +0200
Subject: [PATCH] [vumeter] Add missing header include for QElapsedTimer (#725)

Fixes #724
---
 src/plugins/vumeter/vumeterwidget.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/plugins/vumeter/vumeterwidget.cpp b/src/plugins/vumeter/vumeterwidget.cpp
index 786f24b5b..2834a3e19 100644
--- a/src/plugins/vumeter/vumeterwidget.cpp
+++ b/src/plugins/vumeter/vumeterwidget.cpp
@@ -33,6 +33,7 @@
 #include <QActionGroup>
 #include <QBasicTimer>
 #include <QContextMenuEvent>
+#include <QElapsedTimer>
 #include <QJsonObject>
 #include <QMenu>
 #include <QPainter>
