strips forced unsecure flags

--- a/setup.py
+++ b/setup.py
@@ -3,18 +3,8 @@
 from Cython.Build import cythonize
 from setuptools import Extension, setup

-extra_compile_args = [
-    "-flto",
-    "-O3",
-    "-ffast-math",
-    "-fomit-frame-pointer",
-    "-funroll-loops",
-]
-extra_link_args = [
-    "-flto",
-    "-O3",
-    "-s",
-]
+extra_compile_args = []
+extra_link_args = []

 if shutil.which("lld"):
     extra_compile_args.append("-fuse-ld=lld")
