strips forced compilation flags

--- a/setup.py
+++ b/setup.py
@@ -4,27 +4,8 @@
 from Cython.Build import cythonize
 from setuptools import Extension, setup

-extra_compile_args = [
-    "-g0",
-    "-O3",
-    "-ffast-math",
-    "-fomit-frame-pointer",
-    "-funroll-loops",
-    "-fvisibility=hidden",
-    "-DNDEBUG",
-]
-extra_link_args = [
-    "-Wl,-O3",
-    "-Wl,-s",
-    "-Wl,--sort-common",
-    "-Wl,-z,pack-relative-relocs",
-    "-Wl,--as-needed",
-    "-Wl,--exclude-libs,ALL",
-]
-if shutil.which("lld") and os.environ.get("CC") == "clang":
-    extra_compile_args.append("-flto=thin")
-    extra_link_args.append("-flto=thin")
-    extra_link_args.append("-fuse-ld=lld")
+extra_compile_args = []
+extra_link_args = []


 extensions = [
