--- a/src/md5.c
+++ b/src/md5.c
@@ -15,6 +15,7 @@
  * will fill a supplied 16-byte array with the digest.
  */
 #include <string.h>		/* for memcpy() */
+#include "usuals.h"
 #include "md5.h"
 
 #ifndef HIGHFIRST

--- a/src/rsaglue2.c
+++ b/src/rsaglue2.c
@@ -52,9 +52,8 @@ char signon_legalese[] = _LANG("\
 Uses the RSAREF(tm) Toolkit, which is copyright RSA Data Security, Inc.\n\
 Distributed by the Massachusetts Institute of Technology.\n");
 
-#include <global.h>
-#include <rsaref.h>
-#include <rsa.h>
+#include <rsaref/rsaref.h>
+#include <rsaref/rsa.h>
 /*
  * The functions we call in rsa.h are:
  *
