From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Copy injected code from the system.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/build/build.go
+++ b/build/build.go
@@ -42,8 +42,8 @@ func Build(cloneDir string, outputDir string) error {
 	}
 
 	additionsDir := fmt.Sprintf("%s/WebKitAdditions/WebInspectorUI", baseDir)
-	cp.Copy("injectedCode/WebInspectorUIAdditions.css", fmt.Sprintf("%s/WebInspectorUIAdditions.css", additionsDir))
-	cp.Copy("injectedCode/WebInspectorUIAdditions.js", fmt.Sprintf("%s/WebInspectorUIAdditions.js", additionsDir))
+	cp.Copy("@GENTOO_PORTAGE_EPREFIX@/usr/share/ios-safari-remote-debug/injectedCode/WebInspectorUIAdditions.css", fmt.Sprintf("%s/WebInspectorUIAdditions.css", additionsDir))
+	cp.Copy("@GENTOO_PORTAGE_EPREFIX@/usr/share/ios-safari-remote-debug/injectedCode/WebInspectorUIAdditions.js", fmt.Sprintf("%s/WebInspectorUIAdditions.js", additionsDir))
 
 	log.Debug().TimeDiff("loadFiles", time.Now(), loadFilesTime).Send()
 	getRelevantFilesTime := time.Now()
@@ -67,7 +67,7 @@ func Build(cloneDir string, outputDir string) error {
 	log.Debug().TimeDiff("getRelevantFiles", time.Now(), getRelevantFilesTime).Send()
 	bundleTime := time.Now()
 
-	cp.Copy("views", outputDir)
+	cp.Copy("@GENTOO_PORTAGE_EPREFIX@/usr/share/ios-safari-remote-debug/views", outputDir)
 	os.Mkdir(fmt.Sprintf("%s/debug", outputDir), os.ModePerm)
 	os.Mkdir(fmt.Sprintf("%s/debug/Lib", outputDir), os.ModePerm)
 	os.Mkdir(fmt.Sprintf("%s/debug/Protocols", outputDir), os.ModePerm)
