--- a/cmake/install/pre/CMakeLists.txt
+++ b/cmake/install/pre/CMakeLists.txt
@@ -1,9 +1,3 @@
-install(CODE "
-  message(\"Pre-installation cleanup of CMake files\")
-  file(REMOVE_RECURSE \"${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_CMAKEDIR}\")
-"
-  COMPONENT adios2_core-development
-)
 file(GLOB ADIOS2_MODULE_FILES 
   "${ADIOS2_SOURCE_DIR}/cmake/Find*.cmake"
   "${ADIOS2_SOURCE_DIR}/cmake/CMake*.cmake"
