--- a/config/debuginfod.m4
+++ b/config/debuginfod.m4
@@ -34,5 +34,5 @@
   fi
 fi
 AM_CONDITIONAL([HAVE_DEBUGINFOD],
-               [test x"$have_debuginfod_lib" == xyes -a x"$have_debuginfod_h" == xyes])
+               [test x"$have_debuginfod_lib" = xyes && x"$have_debuginfod_h" = xyes])
 ])
--- a/configure.ac
+++ b/configure.ac
@@ -149,13 +149,13 @@
 AM_CONDITIONAL([COND_DOCS],       [test "$with_docs" != no])
 AM_CONDITIONAL([COND_TESTS],      [test "$with_tests" != no])
 AM_CONDITIONAL([COND_ANNOCHECK],  [test "$with_annocheck" != no])
-AM_CONDITIONAL([COND_CLANG],      [test "$with_clang_plugin" == yes])
+AM_CONDITIONAL([COND_CLANG],      [test "$with_clang_plugin" = yes])
 AM_CONDITIONAL([COND_GCC_PLUGIN], [test "$with_gcc_plugin" != no])
-AM_CONDITIONAL([COND_LLVM],       [test "$with_llvm_plugin" == yes])
+AM_CONDITIONAL([COND_LLVM],       [test "$with_llvm_plugin" = yes])
 
 AC_CONFIG_FILES(Makefile)
 
-if test "$with_annocheck" != no -o "$with_gcc_plugin" != no; then
+if test "$with_annocheck" != no || "$with_gcc_plugin" != no; then
   AC_CONFIG_HEADERS(gcc-plugin/config.h)
 fi
 if test "$with_docs" != no; then
