From 002360c97f11981aee736292c372057425ef23c0 Mon Sep 17 00:00:00 2001
From: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 20:20:29 +0200
Subject: [PATCH] use ounit2

Signed-off-by: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
---
 dune-project  | 2 +-
 mustache.opam | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/dune-project b/dune-project
index 5f64694..921db30 100644
--- a/dune-project
+++ b/dune-project
@@ -27,7 +27,7 @@ Contains the `mustache` command line utility for driving logic-less templates.
 ")
  (depends
   (jsonm (>= 1.0.1))
-  (ounit :with-test)
+  (ounit2 :with-test)
   (ezjsonm :with-test)
   (menhir (>= 20180703))
   (cmdliner (>= 1.0.4))
diff --git a/mustache.opam b/mustache.opam
index d5fedca..51e5a0a 100644
--- a/mustache.opam
+++ b/mustache.opam
@@ -18,7 +18,7 @@ bug-reports: "https://github.com/rgrinberg/ocaml-mustache/issues"
 depends: [
   "dune" {>= "2.7"}
   "jsonm" {>= "1.0.1"}
-  "ounit" {with-test}
+  "ounit2" {with-test}
   "ezjsonm" {with-test}
   "menhir" {>= "20180703"}
   "cmdliner" {>= "1.0.4"}
