zig-wayland-0.2.0
--- a/deps/p/1220687c8c47a48ba285d26a05600f8700d37fc637e223ced3aa8324f3650bf52242/src/scanner.zig
+++ b/deps/p/1220687c8c47a48ba285d26a05600f8700d37fc637e223ced3aa8324f3650bf52242/src/scanner.zig
@@ -528,7 +528,7 @@
     // These interfaces are special in that their version may never be increased.
     // That is, they are pinned to version 1 forever. They also may break the
     // normally required tree object creation hierarchy.
-    const version_locked_interfaces = std.StaticStringMap(void).initComptime(.{
+    const version_locked_interfaces = std.ComptimeStringMap(void, .{
         .{"wl_display"},
         .{"wl_registry"},
         .{"wl_callback"},

river-0.3.5
--- a/river/command.zig
+++ b/river/command.zig
@@ -36,9 +36,8 @@
     vertical,
 };

-const command_impls = std.StaticStringMap(
+const command_impls = std.ComptimeStringMap(
     *const fn (*Seat, []const [:0]const u8, *?[]const u8) Error!void,
-).initComptime(
     .{
         // zig fmt: off
         .{ "attach-mode",               @import("command/attach_mode.zig").defaultAttachMode },
