--- a/src/hunter/actions.py
+++ b/src/hunter/actions.py
@@ -8,6 +8,8 @@
 from itertools import islice
 from os import getpid
 
+from colorama import AnsiToWin32
+
 from . import config
 from .util import BUILTIN_SYMBOLS
 from .util import CALL_COLORS
@@ -22,7 +20,6 @@
 from .util import frame_iterator
 from .util import iter_symbols
 from .util import safe_repr
-from .vendor.colorama import AnsiToWin32
 
 try:
     from threading import get_ident
--- a/src/hunter/util.py
+++ b/src/hunter/util.py
@@ -8,9 +8,9 @@
 from collections import defaultdict
 from collections import deque
 
+from colorama import Back
+from colorama import Fore
+from colorama import Style
-from .vendor.colorama import Back
-from .vendor.colorama import Fore
-from .vendor.colorama import Style
 
 try:
     import __builtin__ as builtins
