--- a/setup.py
+++ b/setup.py
@@ -163,10 +163,10 @@
         ':platform_system != "Windows"': ['manhole >= 1.5'],
     },
     setup_requires=[
-        'setuptools_scm>=3.3.1,!=4.0.0,<6.0',
+        'setuptools_scm>=3.3.1',
         'cython',
     ] if Cython else [
-        'setuptools_scm>=3.3.1,!=4.0.0,<6.0',
+        'setuptools_scm>=3.3.1',
     ],
     entry_points={
         'console_scripts': [
