From f34b740604cf6cdb924c056d38ec08be4c781048 Mon Sep 17 00:00:00 2001
From: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 20:24:29 +0200
Subject: [PATCH] use ounit2

Signed-off-by: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
---
 crc.opam  | 2 +-
 test/dune | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/crc.opam b/crc.opam
index 74f21a8..7923009 100644
--- a/crc.opam
+++ b/crc.opam
@@ -12,7 +12,7 @@ depends: [
   "dune"
   "ocaml" {>= "4.00.1"}
   "cstruct" {>= "1.0.1"}
-  "ounit" {with-test}
+  "ounit2" {with-test}
   "odoc" {with-doc}
   "ppx_deriving_rpc"
   "ppx_sexp_conv" {>= "v0.11.0"}
diff --git a/test/dune b/test/dune
index 769e090..bad69da 100644
--- a/test/dune
+++ b/test/dune
@@ -10,7 +10,7 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
 (executable
   (name crc_test)
   (flags (:standard))
-  (libraries oUnit crc)
+  (libraries ounit2 crc)
   (preprocess (pps ppx_deriving_rpc ppx_sexp_conv %s))
 )
 
