diff -Naur a/ford/setup.cfg b/ford/setup.cfg
--- a/setup.cfg
+++ b/setup.cfg
@@ -7,7 +7,7 @@
 author = Chris MacMackin
 author_email = cmacmackin@gmail.com
 license = GPLv3
-license_file = LICENSE
+license_files = LICENSE
 keywords = Markdown, Fortran, documentation, comments
 classifiers = 
 	Development Status :: 5 - Production/Stable
@@ -27,7 +27,7 @@
 	Tracker = https://github.com/Fortran-FOSS-Programmers/ford/issues
 
 [options]
-packages = ford
+packages = find_namespace:
 install_requires = 
 	markdown ~= 3.4.0
 	markdown-include ~= 0.7.0
@@ -41,6 +41,9 @@
 	importlib-metadata; python_version < "3.8"
 include_package_data = True
 
+[options.packages.find]
+include = ford*
+
 [options.extras_require]
 tests = pytest >= 3.3.0
 docs = 
