From 67b1355fea6904dc8ea540a6cfaa6164dc3a1a23 Mon Sep 17 00:00:00 2001
From: Peter Hill <peter.hill@york.ac.uk>
Date: Tue, 24 Jan 2023 10:54:10 +0000
Subject: [PATCH] Fix test for Python 3.11 (mutable default not allowed)

---
 test/test_sourceform.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/test_sourceform.py b/test/test_sourceform.py
index 4e6c9d3..f3a67c9 100644
--- a/test/test_sourceform.py
+++ b/test/test_sourceform.py
@@ -790,7 +790,7 @@ class FakeParent:
 class FakeVariable:
     name: str
     vartype: str
-    parent: Optional[FakeParent] = FakeParent()
+    parent: Optional[FakeParent] = field(default_factory=FakeParent)
     attribs: Optional[List[str]] = field(default_factory=list)
     intent: str = ""
     optional: bool = False
