diff --git a/os_api_ref/tests/test_microversions.py b/os_api_ref/tests/test_microversions.py
index 8281442..c338c51 100644
--- a/os_api_ref/tests/test_microversions.py
+++ b/os_api_ref/tests/test_microversions.py
@@ -44,11 +44,11 @@ class TestMicroversions(base.TestCase):
     def test_rest_method(self):
         """Test that min / max mv css class attributes are set"""
         content = self.soup.find_all(class_='rp_min_ver_2_17')
-        self.assertRegexpMatches(
+        self.assertRegex(
             str(content[0]),
             '^<div class="operation-grp rp_min_ver_2_17 rp_max_ver_2_19 ?"')
         content = self.soup.find_all(class_='rp_max_ver_2_19')
-        self.assertRegexpMatches(
+        self.assertRegex(
             str(content[0]),
             '^<div class="operation-grp rp_min_ver_2_17 rp_max_ver_2_19 ?"')
 
