--- /var/tmp/portage/sci-libs/superlu_dist-8.0.0_p20220526/work/superlu_dist-df1dbf3afd7876644ff5d0613dfa7b9d32af79de/SRC/CMakeLists.txt
+++ superlu_dist-df1dbf3afd7876644ff5d0613dfa7b9d32af79de/SRC/CMakeLists.txt
@@ -7,7 +7,6 @@
     superlu_enum_consts.h
     supermatrix.h
     util_dist.h
-    colamd.h
     gpu_api_utils.h
     gpu_wrapper.h
     ${CMAKE_CURRENT_BINARY_DIR}/superlu_dist_config.h
@@ -40,7 +39,6 @@
   xerr_dist.c
   smach_dist.c
   dmach_dist.c
-  colamd.c
   superlu_dist_version.c
   comm_tree.c
   superlu_grid3d.c    ## 3D code
@@ -291,7 +289,7 @@
 endif()
 
 set(superlu_dist_libs ${MPI_C_LIBRARIES} ${MPI_CXX_LIBRARIES} ${BLAS_LIB} ${LAPACK_LIB}
-    ${PARMETIS_LIB} ${COMBBLAS_LIB} ${CUDA_LIB} ${HIP_LIB}) 
+    ${PARMETIS_LIB} ${COMBBLAS_LIB} ${CUDA_LIB} ${HIP_LIB} -lcolamd) 
 if (NOT MSVC)
   list(APPEND superlu_dist_libs m)
 endif ()
