--- a/src/keys.rs
+++ b/src/keys.rs
@@ -216,21 +216,21 @@
 	#[allow(clippy::unused_self)]
 	const fn get_key_symbol(&self, k: KeyCode) -> &str {
 		match k {
-			KeyCode::Enter => "\u{23ce}",     //⏎
+			KeyCode::Enter => "\u{f810}",     //⏎
 			KeyCode::Left => "\u{2190}",      //←
 			KeyCode::Right => "\u{2192}",     //→
 			KeyCode::Up => "\u{2191}",        //↑
 			KeyCode::Down => "\u{2193}",      //↓
-			KeyCode::Backspace => "\u{232b}", //⌫
-			KeyCode::Home => "\u{2912}",      //⤒
-			KeyCode::End => "\u{2913}",       //⤓
-			KeyCode::PageUp => "\u{21de}",    //⇞
-			KeyCode::PageDown => "\u{21df}",  //⇟
+			KeyCode::Backspace => "\u{f56d}", //⌫
+			KeyCode::Home => "\u{f015}",      //⤒
+			KeyCode::End => "\u{fb00}",       //⤓
+			KeyCode::PageUp => "\u{f077}",    //⇞
+			KeyCode::PageDown => "\u{f078}",  //⇟
 			KeyCode::Tab => "\u{21e5}",       //⇥
 			KeyCode::BackTab => "\u{21e4}",   //⇤
-			KeyCode::Delete => "\u{2326}",    //⌦
-			KeyCode::Insert => "\u{2380}",    //⎀
-			KeyCode::Esc => "\u{238b}",       //⎋
+			KeyCode::Delete => "\u{fae7}",    //⌦
+			KeyCode::Insert => "\u{0049}",    //⎀
+			KeyCode::Esc => "ESC",       //⎋
 			_ => "?",
 		}
 	}
@@ -285,7 +285,7 @@
 				"\u{21e7}".to_string() //⇧
 			}
 			KeyModifiers::ALT => {
-				"\u{2325}".to_string() //⌥
+				"M^".to_string() //⌥
 			}
 			_ => String::new(),
 		}
