diff --git a/Makefile b/Makefile
index 3a8db4c..a5ba63b 100644
--- a/Makefile
+++ b/Makefile
@@ -5,11 +5,11 @@
 # Enable compilation of Javascript decoder with Emscripten
 #USE_EMCC=y
 # Enable x265 for the encoder
-USE_X265=y
+#USE_X265=y
 # Enable the JCTVC code (best quality but slow) for the encoder
 #USE_JCTVC=y
 # Compile bpgview (SDL and SDL_image libraries needed)
-USE_BPGVIEW=y
+#USE_BPGVIEW=y
 # Enable it to use bit depths > 12 (need more tests to validate encoder)
 #USE_JCTVC_HIGH_BIT_DEPTH=y
 # Enable the cross compilation for Windows
@@ -17,7 +17,7 @@ USE_BPGVIEW=y
 # Enable for compilation on MacOS X
 #CONFIG_APPLE=y
 # Installation prefix
-prefix=/usr/local
+#prefix=/usr/local
 
 
 #################################
