diff --git a/desktop/test-appdata.sh.in b/desktop/test-appdata.sh.in
index 25be96b1c8..fb674360d9 100755
--- a/desktop/test-appdata.sh.in
+++ b/desktop/test-appdata.sh.in
@@ -2,7 +2,7 @@

 # TODO: use validate-strict when the last errors for a strict validation
 # are fixed.
-appstream-util validate-relax ${GIMP_TESTING_ABS_TOP_BUILDDIR}/desktop/org.glimpse_editor.Glimpse.appdata.xml && \
-appstream-util validate-relax ${GIMP_TESTING_ABS_TOP_BUILDDIR}/desktop/glimpse-data-extras.metainfo.xml && \
+appstream-util validate-relax --nonet ${GIMP_TESTING_ABS_TOP_BUILDDIR}/desktop/org.glimpse_editor.Glimpse.appdata.xml && \
+appstream-util validate-relax --nonet ${GIMP_TESTING_ABS_TOP_BUILDDIR}/desktop/glimpse-data-extras.metainfo.xml && \
 if [ $(expr @GIMP_MICRO_VERSION@ % 2) = 0 ]; then
   grep TODO ${GIMP_TESTING_ABS_TOP_BUILDDIR}/desktop/org.glimpse_editor.Glimpse.appdata.xml
   if [ $? = 0 ]; then
