--- a/aldor/lib/libfoam/al/Makefile.in
+++ b/aldor/lib/libfoam/al/Makefile.in
@@ -38,7 +38,7 @@
 
 all: libfoam.al
 libfoam.al: runtime.ao
-	$(AM_V_AR)ar cr $@ $<
+	$(AM_V_AR)${AR} cr $@ $<
 
 all:
 	@grep 'fiFileInitializer(' runtime.c | grep -v rtexns > runtime-gets.txt || true
--- a/aldor/tools/unix/doranlib
+++ b/aldor/tools/unix/doranlib
@@ -4,6 +4,6 @@
 hpux*) 	;;
 irix*)  ;;
 sun4os5*) ;;
-axposf*) 	ar tls $* > /dev/null ;;
+axposf*) 	${AR} tls $* > /dev/null ;;
 *) 	ranlib $*;;
 esac
--- a/aldor/tools/unix/mkasys
+++ b/aldor/tools/unix/mkasys
@@ -12,8 +12,8 @@
 
 echo "Dumping .ao files from $lib \n\tinto current directory."
 
-aos=`ar -t $lib`
-ar -x $lib
+aos=`${AR} -t $lib`
+${AR} -x $lib
 
 for i in $aos
 do
--- a/aldor/tools/unix/mklib.sh
+++ b/aldor/tools/unix/mklib.sh
@@ -17,13 +17,13 @@
 fi
 
 echo "Rebuilding $afile for $sys"
-for x in `ar t $alfile`;
+for x in `${AR} t $alfile`;
 do
 	echo $x
 	uniar x $alfile $x
 	aldor -Csys=$sys -csmax=0 -fo -O $x
 	ofile=`basename $x .ao`.o
 	if [ ! -f $ofile ] ; then echo "Compile failed. exiting." ; exit 1 ; fi
-	ar r $afile $ofile
+	${AR} r $afile $ofile
 	rm $x $ofile
 done
--- a/lib/algebra/src/test/Makefile.in
+++ b/lib/algebra/src/test/Makefile.in
@@ -26,7 +26,7 @@
 
 $(addsuffix .test,$(tests)): %.test: %.ao
 	cp $(SUBLIB_DEPEND).al lib$(libraryname)_$*.al;				\
-	ar r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
+	${AR} r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
 	$(DBG) $(aldorexedir)/aldor		\
 	  -Nfile=$(aldorsrcdir)/aldor.conf 	\
 	  -Y.					\
--- a/lib/axllib/test/ar2.sh
+++ b/lib/axllib/test/ar2.sh
@@ -20,11 +20,11 @@
 $ALDOR -R $TMPDIR -F ao -F o triv[0-3].as 2>&1 | grep -v 'warning: conflicting types for built-in function' | grep -v "GC:"
 
 echo '== Building an archive containing triv*.ao'
-ar cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
+${AR} cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
 rm -f $TMPDIR/triv*.ao
 
 echo '== Building an archive containing triv*.o'
-ar cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.$OBJ
+${AR} cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.$OBJ
 rm -f $TMPDIR/triv*.$OBJ
 
 echo '== Creating a client for the archive'
--- a/lib/axllib/test/ar3.sh
+++ b/lib/axllib/test/ar3.sh
@@ -20,7 +20,7 @@
 $ALDOR -Mno-ALDOR_W_CantUseArchive $LM -l axllib -R $TMPDIR -F ao -F o triv*.as 2>&1 | grep -v "GC:" | grep -v "warning: conflicting types for built-in function"
 
 echo '== Building an archive containing triv*.ao'
-ar cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
+${AR} cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
 rm -f $TMPDIR/triv*.ao
 
 echo '== Building an archive containing triv*.o'
@@ -28,7 +28,7 @@
 if [ "$PL" = "win32msvc - Win32 [+] " ]; then
 	lib /nologo /out:`cygpath -m $TMPDIR`/lib/libtriv.lib `cygpath -m $TMPDIR`/triv*.obj
 else
-	ar cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
+	${AR} cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
 	ranlib $TMPDIR/lib/libtriv.a
 fi
 
--- a/lib/axllib/test/ar4.sh
+++ b/lib/axllib/test/ar4.sh
@@ -20,8 +20,8 @@
 $ALDOR -Mno-ALDOR_W_CantUseArchive $LM -l axllib -R $TMPDIR -F ao -F o triv*.as 2>&1 | grep -v "GC:" | grep -v "warning: conflicting types for built-in function"
 
 echo '== Building an archive containing triv*.ao'
-ar cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
-ar d  $TMPDIR/lib/libtriv.al triv0.ao
+${AR} cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
+${AR} d  $TMPDIR/lib/libtriv.al triv0.ao
 rm -f $TMPDIR/triv*.ao
 
 echo '== Building an archive containing triv*.o'
@@ -30,8 +30,8 @@
 	lib /nologo /out:`cygpath -m $TMPDIR`/lib/libtriv.lib `cygpath -m $TMPDIR`/triv*.obj
 	lib /nologo `cygpath -m $TMPDIR`/lib/libtriv.lib /REMOVE:`cygpath -m $TMPDIR`/triv0.obj
 else
-	ar cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
-	ar d  $TMPDIR/lib/libtriv.a triv0.o
+	${AR} cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
+	${AR} d  $TMPDIR/lib/libtriv.a triv0.o
 	doranlib $TMPDIR/lib/libtriv.a
 fi
 rm -f $TMPDIR/triv*.o
--- a/lib/axllib/test/ar5.sh
+++ b/lib/axllib/test/ar5.sh
@@ -20,8 +20,8 @@
 $ALDOR -laxllib -R $TMPDIR -F ao -F o triv*.as 2>&1 | grep -v "GC:" | grep -v "warning: conflicting types for built-in function"
 
 echo '== Building an archive containing triv*.ao'
-ar cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
-ar d  $TMPDIR/lib/libtriv.al triv4.ao
+${AR} cr $TMPDIR/lib/libtriv.al $TMPDIR/triv*.ao
+${AR} d  $TMPDIR/lib/libtriv.al triv4.ao
 rm -f $TMPDIR/triv*.ao
 
 echo '== Building an archive containing triv*.o'
@@ -29,8 +29,8 @@
 	lib /nologo /out:`cygpath -m $TMPDIR`/lib/libtriv.lib `cygpath -m $TMPDIR`/triv*.obj
 	lib /nologo `cygpath -m $TMPDIR`/lib/libtriv.lib /remove:`cygpath -m $TMPDIR`/triv4.obj
 else
-	ar cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
-	ar d  $TMPDIR/lib/libtriv.a triv4.o
+	${AR} cr $TMPDIR/lib/libtriv.a $TMPDIR/triv*.o
+	${AR} d  $TMPDIR/lib/libtriv.a triv4.o
 	doranlib $TMPDIR/lib/libtriv.a
 fi
 rm -f $TMPDIR/triv*.o
--- a/lib/axllib/test/ar6.sh
+++ b/lib/axllib/test/ar6.sh
@@ -18,7 +18,7 @@
 $ALDOR -R $TMPDIR -Y $TMPDIR -F ao numeral*.as | grep -v "GC:"
 
 echo '== Building an archive containing numeral*.ao'
-ar cr $TMPDIR/lib/libnum.al $TMPDIR/numeral*.ao
+${AR} cr $TMPDIR/lib/libnum.al $TMPDIR/numeral*.ao
 rm -f $TMPDIR/numeral*.ao
 
 echo '== Creating a client for the archive'
--- a/lib/axllib/test/arrepl0.sh
+++ b/lib/axllib/test/arrepl0.sh
@@ -14,12 +14,12 @@
 
 LOBREP=${LIB}/librepl.a
 LOBREPLIB=${LIB}/librepl.lib
-ar rcv ${LOBREP} /dev/null
+${AR} rcv ${LOBREP} /dev/null
 
 # 'ar rcv' avoids the archive creation warning from ar.
-ar rcv ${LAXLREP} /dev/null 2>&1 | grep -v 'ar: writing'
+${AR} rcv ${LAXLREP} /dev/null 2>&1 | grep -v 'ar: writing'
 #if [ "$P" != "win" ]; then 
-#	ar rcv ${LOBREP} /dev/null 2>&1 | grep -v 'ar: writing'
+#	${AR} rcv ${LOBREP} /dev/null 2>&1 | grep -v 'ar: writing'
 #fi
 
 #
--- a/lib/axllib/test/arrepl1.sh
+++ b/lib/axllib/test/arrepl1.sh
@@ -14,8 +14,8 @@
 LOBREPLIB=${LIB}/librepl.lib
 
 # 'ar rcv' avoids the archive creation warning from ar.
-ar rcv ${LAXLREP} /dev/null 2>&1 | grep -v 'ar: writing'
-ar rcv ${LOBREP} /dev/null 2>&1 | grep -v 'ar: writing'
+${AR} rcv ${LAXLREP} /dev/null 2>&1 | grep -v 'ar: writing'
+${AR} rcv ${LOBREP} /dev/null 2>&1 | grep -v 'ar: writing'
 
 #
 # The perl -p -e 's/-R *[^ ]* //g' removes the use of a -R option with an 
--- a/lib/axllib/test/bug1213.sh
+++ b/lib/axllib/test/bug1213.sh
@@ -24,7 +24,7 @@
 print << (foo bar "No problems!") << newline;
 EOF
 aldor -Fao file_with_a_long_name.as another_file_with_a_long_name.as
-ar rcv libfoo.al file_with_a_long_name.ao another_file_with_a_long_name.ao 2>&1 | grep -v 'ar: writing'
+${AR} rcv libfoo.al file_with_a_long_name.ao another_file_with_a_long_name.ao 2>&1 | grep -v 'ar: writing'
 echo ""
 echo "-----------------------------------------------------------------"
 aldor -ginterp -lfoo footest.as
--- a/lib/buildlib.am
+++ b/lib/buildlib.am
@@ -10,9 +10,9 @@
 lib$(libraryname).al: $(foreach i,$(SUBDIRS),$i/_sublib_$(libraryname).al)
 	$(AM_V_AR)		\
 	for l in $+; do		\
-	   ar x $$l;		\
-	   ar cr $@ $$(ar t $$l);\
-	   rm $$(ar t $$l);	\
+	   ${AR} x $$l;		\
+	   ${AR} cr $@ $$(${AR} t $$l);\
+	   rm $$(${AR} t $$l);	\
 	done
 
 $(libraryname).jar: $(foreach i, $(JAVA_SUBDIRS),$i/$(libraryname).jar)
--- a/lib/buildlib.mk
+++ b/lib/buildlib.mk
@@ -121,21 +121,21 @@
 	$(AM_V_ALDOR)set -e;							\
 	rm -f $*.c $*.ao;							\
 	cp $(SUBLIB_DEPEND).al lib$(libraryname)_$*.al;				\
-	ar r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
+	${AR} r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
 	$(AM_DBG) $(aldorexedir)/aldor $(aldor_args);				\
 	rm lib$(libraryname)_$*.al
 
 $(SUBLIB_DEPEND).al: $(foreach l,$(library_deps),$(librarylibdir)/$l/$(SUBLIB).al) Makefile.deps
 	$(AM_V_AR)set -e;		\
-	ar cr $@;			\
+	${AR} cr $@;			\
 	for l in $(filter %.al,$+); do	\
 	   if [ ! -f $$l ]; then	\
 	      echo "missing $$l";	\
 	      exit 1;			\
 	   fi;				\
-	   ar x $$l;			\
-	   ar r $@ $$(ar t $$l);	\
-	   rm $$(ar t $$l);		\
+	   ${AR} x $$l;			\
+	   ${AR} r $@ $$(${AR} t $$l);	\
+	   rm $$(${AR} t $$l);		\
         done
 
 $(addsuffix .fm,$(alldomains)): %.fm: %.ao
@@ -149,7 +149,7 @@
 	$(AM_V_ALDOR)set -e;							\
 	rm -f $*.c $*.ao;							\
 	cp $(SUBLIB_DEPEND).al lib$(libraryname)_$*.al;				\
-	ar r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
+	${AR} r lib$(libraryname)_$*.al $(addsuffix .ao, $(shell $(UNIQ) $*.dep));	\
 	$(AM_DBG) $(aldorexedir)/aldor -gloop 	\
 	  $(aldor_common_args) 			\
 	  -Y.					\
@@ -200,7 +200,7 @@
 $(SUBLIB).al:
 	$(AM_V_AR)							\
 	rm -f $@;							\
-	ar cr $@ $(addsuffix .ao, $(shell $(UNIQ) $(@:.al=.dep)))
+	${AR} cr $@ $(addsuffix .ao, $(shell $(UNIQ) $(@:.al=.dep)))
 
 all: Makefile $(SUBLIB).al
 all: $(addsuffix .fm,$(library))
--- a/lib/debuglib/test/cmds.compile
+++ b/lib/debuglib/test/cmds.compile
@@ -6,7 +6,7 @@
 
 cd 
 cd base/linux/srclib
-ar rv ../lib/libfoam.a *.o
+${AR} rv ../lib/libfoam.a *.o
 
 cd
 cd aldor/1.0rc0/lib/libdebug
