--- a/aldor/src/Makefile.am
+++ b/aldor/src/Makefile.am
@@ -40,12 +40,12 @@
 
 javagen_LDADD = libphase.a libstruct.a libgen.a libport.a -lm
 javagen_LDFLAGS =
-javagen_CFLAGS = -g $(STRICTCFLAGS)
+javagen_CFLAGS =  $(STRICTCFLAGS)
 
 showexp_SOURCES = showexports.c
 showexp_LDADD = libphase.a libstruct.a libgen.a libport.a -lm
 showexp_LDFLAGS =
-showexp_CFLAGS = -g $(STRICTCFLAGS)
+showexp_CFLAGS =  $(STRICTCFLAGS)
 
 aldor_SOURCES =		\
 	axlcomp.c	\
@@ -53,7 +53,7 @@
 	main.c
 aldor_LDADD = libphase.a libstruct.a libgen.a libport.a -lm
 aldor_LDFLAGS = $(build_id)
-aldor_CFLAGS  = -g $(STRICTCFLAGS)
+aldor_CFLAGS  =  $(STRICTCFLAGS)
 
 #aldortest_CFLAGS = -DTEST_STAND_ALONE -DTEST_ALL -save-temps $(STRICTCFLAGS)
 aldortest_CFLAGS = -DTEST_STAND_ALONE -DTEST_ALL -save-temps $(STRICTCFLAGS)
@@ -316,7 +316,6 @@
 testall_LDFLAGS = 
 testall_CFLAGS = -save-temps $(STRICTCFLAGS)
 
-CFLAGS=-O0 -g
 
 #
 # Special rules
--- a/m4/strict_compile.m4
+++ b/m4/strict_compile.m4
@@ -12,7 +12,7 @@
    ALDOR_CC_OPTION(-Wno-sign-compare,cfg_no_sign_compare)
    AC_MSG_CHECKING(Strict options for C compiler)
 
-   cfgSTRICTCFLAGS="-pedantic -std=c99 -Wall -Wextra -Werror -Wno-empty-body -Wno-enum-compare \
+   cfgSTRICTCFLAGS="-pedantic -std=c99 -Wall -Wextra -Wno-empty-body -Wno-enum-compare \
                     -Wno-missing-field-initializers -Wno-unused -Wno-unused-parameter \
 		    -Wno-error=format -Wno-error=type-limits -Wno-error=strict-aliasing \
 		    $cfg_no_sign_compare $cfg_no_shift_negative_value"
