--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -119,16 +119,16 @@
   set(SJPEG_HAVE_GLUT TRUE)
 endif()
 
-# build the utils library
+# build the sjpeg-utils library
 include_directories(${SJPEG_DEP_IMG_INCLUDE_DIRS})
-add_library(utils
+add_library(sjpeg-utils
             ${CMAKE_CURRENT_SOURCE_DIR}/examples/utils.cc
             ${CMAKE_CURRENT_SOURCE_DIR}/examples/utils.h
 )
 
 if(WIN32)
   # quiet warnings related to fopen, sscanf
-  target_compile_definitions(utils PRIVATE _CRT_SECURE_NO_WARNINGS)
+  target_compile_definitions(sjpeg-utils PRIVATE _CRT_SECURE_NO_WARNINGS)
 endif()
 if(SJPEG_HAVE_OPENGL)
   # check pthread for GL libraries
@@ -139,30 +139,30 @@
       list(APPEND SJPEG_DEP_GL_LIBRARIES ${CMAKE_THREAD_LIBS_INIT})
     endif()
   endif()
-  target_compile_definitions(utils PUBLIC SJPEG_HAVE_OPENGL)
+  target_compile_definitions(sjpeg-utils PUBLIC SJPEG_HAVE_OPENGL)
   add_definitions(${OPENGL_DEFINITIONS})
 endif()
 if(SJPEG_HAVE_GLUT)
   add_definitions(${GLUT_DEFINITIONS})
 endif()
 if(SJPEG_HAVE_JPEG)
-  target_compile_definitions(utils PUBLIC SJPEG_HAVE_JPEG)
+  target_compile_definitions(sjpeg-utils PUBLIC SJPEG_HAVE_JPEG)
 endif()
 if(SJPEG_HAVE_PNG)
-  target_compile_definitions(utils PUBLIC SJPEG_HAVE_PNG)
+  target_compile_definitions(sjpeg-utils PUBLIC SJPEG_HAVE_PNG)
 endif()
 if(SJPEG_DEP_IMG_LIBRARIES)
   # check pthread for GL libraries
-  target_link_libraries(utils ${SJPEG_DEP_IMG_LIBRARIES}
+  target_link_libraries(sjpeg-utils ${SJPEG_DEP_IMG_LIBRARIES}
                         ${SJPEG_DEP_GL_LIBRARIES})
 endif()
-# set_target_properties(utils PROPERTIES POSITION_INDEPENDENT_CODE ON)
+# set_target_properties(sjpeg-utils PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
 # Build the executables if asked for.
 if(SJPEG_BUILD_EXAMPLES)
   # sjpeg
   add_executable(sjpeg-bin ${CMAKE_CURRENT_SOURCE_DIR}/examples/sjpeg.cc)
-  target_link_libraries(sjpeg-bin sjpeg utils)
+  target_link_libraries(sjpeg-bin sjpeg sjpeg-utils)
   set_target_properties(sjpeg-bin PROPERTIES OUTPUT_NAME sjpeg)
 
   # vjpeg
@@ -187,7 +187,7 @@
   elseif(HAVE_GL_GLUT_H)
     add_definitions(-DHAVE_GL_GLUT_H)
   endif()
-  target_link_libraries(vjpeg ${SJPEG_DEP_GL_LIBRARIES} sjpeg utils)
+  target_link_libraries(vjpeg ${SJPEG_DEP_GL_LIBRARIES} sjpeg sjpeg-utils)
 
   install(TARGETS sjpeg-bin vjpeg RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 endif()
