--- a/PERL/Makefile.PL	2021-04-19 17:30:19.964446618 +0200
+++ b/PERL/Makefile.PL	2021-04-19 17:30:47.392884824 +0200
@@ -11,7 +11,7 @@
        AUTHOR         => 'Thomas Hergenhahn <thomas.hergenhahn@web,de>') : ()),
     LIBS              => ['-lnodave'], # e.g., '-lm'
     DEFINE            => '', # e.g., '-DHAVE_SOMETHING'
-    INC               => '-I.', # e.g., '-I. -I/usr/include/other'
+    INC               => '-I. -I..', # e.g., '-I. -I/usr/include/other'
 	# Un-comment this if you add C files to link with later:
     # OBJECT            => '$(O_FILES)', # link all the C files too
 );
