From ce966c2450183dfc1c6415161463ea8d59dcccd5 Mon Sep 17 00:00:00 2001
From: zdroyer <ec2-user@ip-172-31-13-62.us-east-2.compute.internal>
Date: Fri, 12 Apr 2019 12:47:34 +0000
Subject: [PATCH] Removed unused header wchich generated tons of warnings with
 boost 1.69
---

--- a/src/admesh/stl.h
+++ b/src/admesh/stl.h
@@ -26,7 +26,6 @@
 #include <stdio.h>
 #include <stdint.h>
 #include <stddef.h>
-#include <boost/detail/endian.hpp>
 
 #if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
     #include "windows.h"

From a0f0616b43f50124164ed274c687cf98f532b3ce Mon Sep 17 00:00:00 2001
From: Chow Loong Jin <hyperair@debian.org>
Date: Tue, 6 Nov 2018 01:10:33 +0800
Subject: [PATCH] Drop #error "admesh works correctly on little endian machines
 only"

admesh works fine on little-endian machines as of b9592961. It just doesn't
compile because of this pesky #error.
---

--- a/src/admesh/stl.h
+++ b/src/admesh/stl.h
@@ -28,10 +28,6 @@
 #include <stddef.h>
 #include <boost/detail/endian.hpp>
 
-#ifndef BOOST_LITTLE_ENDIAN
-#error "admesh works correctly on little endian machines only!"
-#endif
-
 #if defined(_WIN32) || defined(__WIN32__) || defined(WIN32)
     #include "windows.h"
     #define ADMESH_CHAR wchar_t

