--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,9 +3,7 @@
 	"setuptools",
 	"wheel",
 	"setuptools_scm",
-	"scikit-build",
-	"cmake",
-	"ninja"
+	"scikit-build"
 ]
 build-backend = "setuptools.build_meta"
 
--- a/setup.py
+++ b/setup.py
@@ -196,9 +196,7 @@
           setup_requires=[
               'wheel',
               'setuptools_scm',
-              'scikit-build',
-              'cmake',
-              'ninja'
+              'scikit-build'
           ],
           tests_require=[
               'pytest',
