Drop CMake and Ninja dependencies

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -4,8 +4,6 @@ requires = [
 	"wheel",
 	"setuptools_scm",
 	"scikit-build",
-	"cmake",
-	"ninja"
 ]
 build-backend = "setuptools.build_meta"
 
--- a/setup.py
+++ b/setup.py
@@ -197,8 +197,6 @@ def main():
               'wheel',
               'setuptools_scm',
               'scikit-build',
-              'cmake',
-              'ninja'
           ],
           tests_require=[
               'pytest',
