Fixes compilation with -Werror,-Wimplicit-function-declaration.

Bug: https://bugs.gentoo.org/888149
Bug: https://github.com/bk138/gromit-mpx/pull/173

--- gromit-mpx-1.4.3/src/gromit-mpx.c	2022-09-25 14:43:43.000000000 +0200
+++ gromit-mpx-1.4.3/src/gromit-mpx.c	2022-12-24 18:32:23.629233288 +0100
@@ -696,7 +696,7 @@
 }
 
 
-
+int app_parse_args (int argc, char **argv, GromitData *data);
 
 
 void setup_main_app (GromitData *data, int argc, char ** argv)
