diff --git a/folly/folly/system/AtFork.cpp b/folly/folly/system/AtFork.cpp
index e888e52..8c9eee5 100644
--- a/folly/folly/system/AtFork.cpp
+++ b/folly/folly/system/AtFork.cpp
@@ -20,7 +20,8 @@
 #include <folly/lang/Exception.h>
 #include <folly/portability/PThread.h>
 #include <folly/synchronization/SanitizeThread.h>
-
+#include <system_error>
+#include <stdexcept>
 namespace folly {
 
 void AtForkList::prepare() noexcept {
