--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,8 +151,6 @@
     ON
     CACHE BOOL "only build thrift compiler")
 
-add_subdirectory(folly EXCLUDE_FROM_ALL)
-add_subdirectory(fbthrift EXCLUDE_FROM_ALL)
 if(NOT (ZSTD_FOUND AND PREFER_SYSTEM_ZSTD))
   add_subdirectory(zstd/build/cmake EXCLUDE_FROM_ALL)
 endif()
@@ -383,23 +381,6 @@
 
 list(
   APPEND
-  FROZEN_THRIFT_SRC
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_data.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_data.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_types.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_types.tcc
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_types.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_types_custom_protocol.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_constants.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_constants.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_metadata.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_metadata.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_visitation.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_for_each_field.h
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_visit_union.h)
-
-list(
-  APPEND
   METADATA_THRIFT_SRC
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_constants.cpp
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_constants.h
@@ -418,35 +399,20 @@
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_visitation.h)
 
 add_custom_command(
-  OUTPUT ${FROZEN_THRIFT_SRC}
-  COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift
-  COMMAND
-    cp ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/thrift/frozen.thrift
-    ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/
-  COMMAND cd ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift &&
-          ${CMAKE_CURRENT_BINARY_DIR}/bin/thrift1 --gen mstch_cpp2 frozen.thrift
-  DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/bin/thrift1
-          ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/thrift/frozen.thrift)
-
-add_custom_command(
   OUTPUT ${METADATA_THRIFT_SRC}
   COMMAND mkdir -p ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs
   COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/thrift/metadata.thrift
           thrift/dwarfs/metadata.thrift
   COMMAND
     cd ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs &&
-    ${CMAKE_CURRENT_BINARY_DIR}/bin/thrift1 --gen mstch_cpp2:frozen2
+    thrift1 --gen mstch_cpp2:frozen2
     metadata.thrift
-  DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/bin/thrift1
-          ${CMAKE_CURRENT_SOURCE_DIR}/thrift/metadata.thrift)
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/thrift/metadata.thrift)
 
 list(
   APPEND
   INCLUDE_DIRS
-  ${CMAKE_CURRENT_BINARY_DIR}/folly
   ${CMAKE_CURRENT_BINARY_DIR}/thrift
-  ${CMAKE_CURRENT_SOURCE_DIR}/folly
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift
   ${CMAKE_CURRENT_BINARY_DIR})
 
 if(NOT (ZSTD_FOUND AND PREFER_SYSTEM_ZSTD))
@@ -472,43 +438,24 @@
 endif()
 
 add_library(
-  thrift_light
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/FieldRef.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/protocol/CompactProtocol.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/protocol/BinaryProtocol.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/protocol/DebugProtocol.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/protocol/JSONProtocolCommon.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/protocol/JSONProtocol.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp/protocol/TProtocolException.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp/util/VarintUtils.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/gen/module_types_cpp.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/frozen/Frozen.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/frozen/FrozenUtil.cpp
-  ${CMAKE_CURRENT_SOURCE_DIR}/fbthrift/thrift/lib/cpp2/frozen/schema/MemorySchema.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_data.cpp
-  ${CMAKE_CURRENT_BINARY_DIR}/thrift/lib/thrift/gen-cpp2/frozen_types.cpp)
-
-set_property(TARGET thrift_light PROPERTY CXX_STANDARD 17)
-
-target_include_directories(thrift_light PRIVATE ${INCLUDE_DIRS})
-
-add_library(
   metadata_thrift
+  STATIC
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_layouts.cpp
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_types.cpp
   ${CMAKE_CURRENT_BINARY_DIR}/thrift/dwarfs/gen-cpp2/metadata_data.cpp)
 
 set_property(TARGET metadata_thrift PROPERTY CXX_STANDARD 17)
-
+set_property(TARGET metadata_thrift PROPERTY POSITION_INDEPENDENT_CODE ON)
+set_property(TARGET metadata_thrift PROPERTY CXX_VISIBILITY_PRESET hidden)
+set_property(TARGET metadata_thrift PROPERTY VISIBILITY_INLINES_HIDDEN 1)
 target_include_directories(metadata_thrift PRIVATE ${INCLUDE_DIRS})
-
-add_dependencies(metadata_thrift thrift_light)
+target_link_libraries(metadata_thrift fmt glog folly)
 
 foreach(tgt dwarfs ${BINARY_TARGETS})
   target_include_directories(
     ${tgt} SYSTEM
     PRIVATE ${Boost_INCLUDE_DIRS} ${Python3_INCLUDE_DIRS} ${INCLUDE_DIRS}
-            ${CMAKE_CURRENT_SOURCE_DIR}/parallel-hashmap)
+            )
 
   target_include_directories(${tgt} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
@@ -565,23 +512,11 @@
   endif()
 endforeach()
 
-# not sure why exactly, copied from fsst/CMakeLists.txt
-if(CMAKE_BUILD_TYPE STREQUAL Release)
-  set_source_files_properties(fsst/fsst_avx512.cpp PROPERTIES COMPILE_FLAGS -O1)
-endif()
-
-add_library(
-  fsst
-  fsst/libfsst.cpp fsst/fsst_avx512.cpp fsst/fsst_avx512_unroll1.inc
-  fsst/fsst_avx512_unroll2.inc fsst/fsst_avx512_unroll3.inc
-  fsst/fsst_avx512_unroll4.inc)
-
-target_include_directories(dwarfs PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/fsst)
-
 target_link_libraries(
   dwarfs
+  thriftprotocol
+  thriftfrozen2
   metadata_thrift
-  thrift_light
   folly
   fsst
   ${Boost_LIBRARIES}
@@ -612,11 +547,6 @@
   if(USE_JEMALLOC)
     target_link_libraries(${tgt} ${Jemalloc_LIBRARIES})
   endif()
-  if(DWARFS_USE_EXCEPTION_TRACER)
-    target_link_libraries(
-      ${tgt} -Wl,--whole-archive folly_exception_tracer_base
-      folly_exception_tracer -Wl,--no-whole-archive)
-  endif()
 endforeach()
 
 if(STATIC_BUILD_DO_NOT_USE)
