--- a/setup.py
+++ b/setup.py
@@ -82,7 +82,7 @@
                  'Programming Language :: Python :: Implementation :: Jython',
                  'Programming Language :: Python :: Implementation :: PyPy',
                  'Topic :: Software Development :: Testing'],
-    install_requires=['distribute'],
+    install_requires=['setuptools'],
     test_suite='almosttests',
     tests_require=['pytest'],
 )
