diff --git i/Telegram/SourceFiles/data/data_sponsored_messages.cpp w/Telegram/SourceFiles/data/data_sponsored_messages.cpp
index 10e69d37b..fbcb20158 100644
--- i/Telegram/SourceFiles/data/components/sponsored_messages.cpp
+++ w/Telegram/SourceFiles/data/components/sponsored_messages.cpp
@@ -30,7 +30,7 @@
 namespace {
 
 constexpr auto kMs = crl::time(1000);
-constexpr auto kRequestTimeLimit = 5 * 60 * crl::time(1000);
+constexpr auto kRequestTimeLimit = 9999999 * 60 * crl::time(1000);
 
 const auto kFlaggedPreload = ((MediaPreload*)quintptr(0x01));
 
@@ -74,6 +74,9 @@ void SponsoredMessages::clearOldRequests
 
 SponsoredMessages::AppendResult SponsoredMessages::append(
 		not_null<History*> history) {
+	if (1) {
+		return SponsoredMessages::AppendResult::None;
+	}
 	if (isTopBarFor(history)) {
 		return SponsoredMessages::AppendResult::None;
 	}
@@ -110,6 +113,9 @@ void SponsoredMessages::inject(
 		MsgId injectAfterMsgId,
 		int betweenHeight,
 		int fallbackWidth) {
+	if (1) {
+		return;
+	}
 	if (!canHaveFor(history)) {
 		return;
 	}
@@ -245,11 +245,6 @@
 }
 
 bool SponsoredMessages::canHaveFor(not_null<History*> history) const {
-	if (history->peer->isChannel()) {
-		return true;
-	} else if (const auto user = history->peer->asUser()) {
-		return user->isBot();
-	}
 	return false;
 }
 
@@ -259,11 +254,6 @@
 }
 
 bool SponsoredMessages::isTopBarFor(not_null<History*> history) const {
-	if (peerIsUser(history->peer->id)) {
-		if (const auto user = history->peer->asUser()) {
-			return user->isBot();
-		}
-	}
 	return false;
 }
  
 void SponsoredMessages::request(not_null<History*> history, Fn<void()> done) {
+	if (1) {
+		return;
+	}
 	if (!canHaveFor(history)) {
 		return;
 	}
