diff --git a/Telegram/SourceFiles/platform/linux/tray_linux.cpp b/Telegram/SourceFiles/platform/linux/tray_linux.cpp
index 2cc92f331..82e131ea9 100644
--- a/Telegram/SourceFiles/platform/linux/tray_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/tray_linux.cpp
@@ -105,7 +105,7 @@ QIcon IconGraphic::systemIcon() const {
 }
 
 bool IconGraphic::isCounterNeeded(const State &state) const {
-	return state.systemIcon.name() != PanelIconName(
+	return true || state.systemIcon.name() != PanelIconName(
 		state.counter,
 		state.muted);
 }
