From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Fix compilation errors.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/src/dynarmic/src/dynarmic/common/common_types.h
+++ b/src/dynarmic/src/dynarmic/common/common_types.h
@@ -21,3 +21,5 @@ using f64 = double; ///< 64-bit floating point
 
 using u128 = std::array<std::uint64_t, 2>;
 static_assert(sizeof(u128) == 16, "u128 must be 128 bits wide");
+
+using size_t = std::size_t;
--- a/src/dynarmic/src/dynarmic/common/memory_pool.h
+++ b/src/dynarmic/src/dynarmic/common/memory_pool.h
@@ -6,6 +6,7 @@
 #pragma once
 
 #include <cstddef>
+#include <cstdlib>
 #include <vector>
 
 namespace Dynarmic::Common {
--- a/src/yuzu/main.cpp
+++ b/src/yuzu/main.cpp
@@ -101,9 +101,10 @@ static FileSys::VirtualFile VfsDirectoryCreateFileWrapper(const FileSys::Virtual
 #include <QUrl>
 #include <QtConcurrent/QtConcurrent>
 
-#ifdef HAVE_SDL2
 #include <QCheckBox>
 #include <QStringLiteral>
+
+#ifdef HAVE_SDL2
 #include <SDL.h> // For SDL ScreenSaver functions
 #endif
