From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Make the dependency on `mcl` global.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/src/dynarmic/externals/CMakeLists.txt
+++ b/src/dynarmic/externals/CMakeLists.txt
@@ -56,6 +56,7 @@ AddPackage(
     REPO "azahar-emu/mcl"
     SHA 7b08d83418
     HASH f943bac39c1879986decad7a442ff4288eaeca4a2907684c7914e115a55ecc43c2782ded85c0835763fe04e40d5c82220ce864423e489e648e408a84f54dc4f3
+    FIND_PACKAGE_ARGUMENTS "GLOBAL"
     OPTIONS
         "MCL_INSTALL OFF"
 )
