From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Allow overriding `NX_TZDB_ROMFS_DIR`.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/nx_tzdb/CMakeLists.txt
+++ b/externals/nx_tzdb/CMakeLists.txt
@@ -30,7 +30,7 @@ endif()
 set(NX_TZDB_VERSION "250725")
 set(NX_TZDB_ARCHIVE "${CPM_SOURCE_CACHE}/nx_tzdb/${NX_TZDB_VERSION}.zip")
 
-set(NX_TZDB_ROMFS_DIR "${CPM_SOURCE_CACHE}/nx_tzdb/tz")
+set(NX_TZDB_ROMFS_DIR CACHE PATH "${CPM_SOURCE_CACHE}/nx_tzdb/tz")
 
 if ((NOT CAN_BUILD_NX_TZDB OR YUZU_DOWNLOAD_TIME_ZONE_DATA) AND NOT EXISTS ${NX_TZDB_ROMFS_DIR})
     set(NX_TZDB_DOWNLOAD_URL "https://github.com/crueter/tzdb_to_nx/releases/download/${NX_TZDB_VERSION}/${NX_TZDB_VERSION}.zip")
