From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use the system `mbedtls` library.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -54,7 +54,7 @@ endif()
 add_subdirectory(glad)
 
 # mbedtls
-AddJsonPackage(mbedtls)
+add_library(mbedtls SHARED IMPORTED)
 
 if (mbedtls_ADDED)
     target_include_directories(mbedtls PUBLIC ${mbedtls_SOURCE_DIR}/include)
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -1196,7 +1196,7 @@ else()
     target_link_libraries(core PUBLIC Boost::headers)
 endif()
 
-target_link_libraries(core PRIVATE fmt::fmt nlohmann_json::nlohmann_json mbedtls RenderDoc::API)
+target_link_libraries(core PRIVATE fmt::fmt nlohmann_json::nlohmann_json mbedtls mbedcrypto RenderDoc::API)
 if (MINGW)
     target_link_libraries(core PRIVATE ${MSWSOCK_LIBRARY})
 endif()
