From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use the system `sirit` library.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -76,7 +76,7 @@ endif()
 set(SPIRV_WERROR OFF)
 AddJsonPackage(spirv-headers)
 
-AddJsonPackage(sirit)
+add_library(sirit SHARED IMPORTED)
 
 if(MSVC AND USE_CCACHE AND sirit_ADDED)
     get_target_property(_opts sirit COMPILE_OPTIONS)
