From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Add a missing `include` for the log header.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/src/core/hle/service/nvnflinger/hos_binder_driver_server.cpp
+++ b/src/core/hle/service/nvnflinger/hos_binder_driver_server.cpp
@@ -4,6 +4,7 @@
 #include <mutex>
 
 #include "common/common_types.h"
+#include "common/logging/log.h"
 #include "core/hle/service/nvnflinger/hos_binder_driver_server.h"
 
 namespace Service::Nvnflinger {
