From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use the system `sirit` library.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -139,14 +139,7 @@ AddPackage(
     HASH f48bbe18341ed55ea0fe280dbbbc0a44bf222278de6e716e143ca1e95ca320b06d4d23d6583fbf8d03e1428f3dac8fa00e5b82ddcd6b425e6236d85af09550a4
 )
 
-AddPackage(
-    NAME sirit
-    REPO "eden-emulator/sirit"
-    SHA db1f1e8ab5
-    HASH 73eb3a042848c63a10656545797e85f40d142009dfb7827384548a385e1e28e1ac72f42b25924ce530d58275f8638554281e884d72f9c7aaf4ed08690a414b05
-    OPTIONS
-        "SIRIT_USE_SYSTEM_SPIRV_HEADERS ON"
-)
+add_library(sirit SHARED IMPORTED)
 
 if(MSVC AND USE_CCACHE AND TARGET sirit)
     get_target_property(_opts sirit COMPILE_OPTIONS)
