From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Use the system `sirit` library.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/externals/CMakeLists.txt
+++ b/externals/CMakeLists.txt
@@ -110,7 +110,7 @@ if (USE_DISCORD_PRESENCE AND NOT TARGET DiscordRPC::discord-rpc)
 endif()
 
 # Sirit
-add_subdirectory(sirit)
+add_library(sirit SHARED IMPORTED)
 
 # httplib
 if ((ENABLE_WEB_SERVICE OR ENABLE_QT_UPDATE_CHECKER) AND NOT TARGET httplib::httplib)
