From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Always include `<common/detached_tasks>` in `src/yuzu/main.cpp`.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/src/yuzu/main.cpp
+++ b/src/yuzu/main.cpp
@@ -6,10 +6,11 @@
 
 #if YUZU_ROOM
 #include "dedicated_room/yuzu_room.h"
-#include <common/detached_tasks.h>
 #include <cstring>
 #endif
 
+#include <common/detached_tasks.h>
+
 #ifdef __unix__
 #include "qt_common/gui_settings.h"
 #endif

