Add missing headers for no PCH build.
--- a/src/shell_application_spawner.h
+++ b/src/shell_application_spawner.h
@@ -19,6 +19,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define MIRACLE_WM_SHELL_APPLICATION_SPAWNER_H
 
 #include <miral/application.h>
+#include <miral/window_specification.h>
 
 namespace miracle
 {
--- a/src/window_container.h
+++ b/src/window_container.h
@@ -20,6 +20,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.

 #include "container.h"
 #include "container_effect.h"
+#include <mir/events/input_event.h>
 #include <optional>
 
 namespace miracle
--- a/src/container_effect.h
+++ b/src/container_effect.h
@@ -17,6 +17,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #ifndef MIRACLE_CONTAINER_EFFECT_H
 #define MIRACLE_CONTAINER_EFFECT_H
+#include <glm/glm.hpp>
 #include <glm/fwd.hpp>
 
 namespace miracle
--- a/src/plugin_managed_container.cpp
+++ b/src/plugin_managed_container.cpp
@@ -21,6 +21,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "compositor_state.h"
 #include "container_listener.h"
 #include "window_controller.h"
+#include <mir/scene/session.h>
 #include <mir/scene/surface.h>
 
 using namespace miracle;
--- a/src/freestyle_window_container.cpp
+++ b/src/freestyle_window_container.cpp
@@ -23,6 +23,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "container_listener.h"
 #include "direction.h"
 #include "window_controller.h"
+#include <mir/scene/session.h>
 #include <mir/scene/surface.h>
 
 using namespace miracle;
--- a/src/plugin_handle.h
+++ b/src/plugin_handle.h
@@ -15,6 +15,8 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 **/
 
+#include <cstdint>
+
 namespace miracle
 {
 typedef uint32_t PluginHandle;
--- a/src/parent_background_internal_client.cpp
+++ b/src/parent_background_internal_client.cpp
@@ -28,6 +28,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <cmath>
 #include <cstring>
 #include <fcntl.h>
+#include <mutex>
 #include <poll.h>
 #include <sys/eventfd.h>
 #include <sys/mman.h>
--- a/src/move_service.h
+++ b/src/move_service.h
@@ -19,6 +19,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define MIRACLE_WM_MOVE_SERVICE_H
 
 #include <memory>
+#include <mutex>
 #include <mir_toolkit/event.h>
 
 namespace miracle
