--- a/src/mlpack/bindings/python/setup.py.in
+++ b/src/mlpack/bindings/python/setup.py.in
@@ -137,7 +137,7 @@
       packages=['mlpack'],
       cmdclass={ 'build_ext': build_ext },
       ext_modules = modules,
-      setup_requires=['cython', 'pytest-runner'],
+      setup_requires=['cython'],
       tests_require=['pytest>3;python_version>"3.4"', 'pytest>3,<=4.6;python_version<="3.4"',
                      'more-itertools>=4.0.0,<6.0.0;python_version<="2.7"',
                      'more-itertools>=4.0.0;python_version>"2.7"'],
