From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Build tests only when requested.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,8 +102,10 @@ endif()
 
 # Project files
 
+include(CTest)
+
 add_subdirectory(src)
-if (TARGET Catch2::Catch2 AND MASTER_PROJECT)
+if (BUILD_TESTING)
     add_subdirectory(tests)
 endif()
 
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -10,7 +10,6 @@ target_include_directories(mcl-tests PUBLIC .)
 target_compile_options(mcl-tests PRIVATE ${STAMINA_CXX_FLAGS})
 target_link_libraries(mcl-tests PRIVATE Catch2::Catch2WithMain mcl)
 
-include(CTest)
 include(Catch)
 catch_discover_tests(mcl-tests)
 enable_testing()
