From https://github.com/azahar-emu/mcl/commit/7b08d83418f628b800dfac1c9a16c3f59036fbad Mon Sep 17 00:00:00 2001
From: PabloMK7 <hackyglitch2@gmail.com>
Date: Sun, 16 Mar 2025 22:27:04 +0000
Subject: [PATCH] Fix lift_sequence in msys2

--- a/include/mcl/mp/typelist/lift_sequence.hpp
+++ b/include/mcl/mp/typelist/lift_sequence.hpp
@@ -5,6 +5,7 @@
 #pragma once
 
 #include <type_traits>
+#include <utility>
 
 #include "mcl/mp/typelist/list.hpp"
 
@@ -20,6 +21,11 @@ struct lift_sequence_impl<VLT<T, values...>> {
     using type = list<std::integral_constant<T, values>...>;
 };
 
+template<class T, T... values>
+struct lift_sequence_impl<std::integer_sequence<T, values...>> {
+    using type = list<std::integral_constant<T, values>...>;
+};
+
 }  // namespace detail
 
 /// Lifts values in value list VL to create a type list.
