From 13269f4af8d3c95220f97746c968eac3ba122a32 Mon Sep 17 00:00:00 2001
From: winter091 <winter091@yandex.ru>
Date: Sat, 1 May 2021 16:48:31 +0300
Subject: [PATCH] fix typo: vec3 -> vec4

---
 include/cglm/call/vec4.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/cglm/call/vec4.h b/include/cglm/call/vec4.h
index 28a3734..f56f599 100644
--- a/include/cglm/call/vec4.h
+++ b/include/cglm/call/vec4.h
@@ -99,7 +99,7 @@ glmc_vec4_scale(vec4 v, float s, vec4 dest);
 
 CGLM_EXPORT
 void
-glmc_vec4_scale_as(vec3 v, float s, vec3 dest);
+glmc_vec4_scale_as(vec4 v, float s, vec4 dest);
 
 CGLM_EXPORT
 void
