diff --git a/include/layer.hpp b/include/layer.hpp
index 6ebe903..4ec58ea 100644
--- a/include/layer.hpp
+++ b/include/layer.hpp
@@ -219,8 +219,8 @@ namespace Layer {
 }
 
 /// Symbol definition for Vulkan instance layer.
-__attribute__((visibility("default")))
-extern "C" PFN_vkVoidFunction layer_vkGetInstanceProcAddr(VkInstance instance, const char* pName);
+extern "C" __attribute__((visibility("default")))
+PFN_vkVoidFunction layer_vkGetInstanceProcAddr(VkInstance instance, const char* pName);
 /// Symbol definition for Vulkan device layer.
-__attribute__((visibility("default")))
-extern "C" PFN_vkVoidFunction layer_vkGetDeviceProcAddr(VkDevice device, const char* pName);
+extern "C" __attribute__((visibility("default")))
+PFN_vkVoidFunction layer_vkGetDeviceProcAddr(VkDevice device, const char* pName);
