https://bugs.gentoo.org/901159
Respect user CXXFLAGS.
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,11 +132,6 @@ if(MSVC)
 else()
   ucm_add_flags(CXX "-fno-sized-deallocation -fno-omit-frame-pointer")
   ucm_add_flags(C CXX "-mmmx -msse -msse2 -msse3 -mssse3 -mpopcnt")
-  if(KLOGG_GENERIC_CPU)
-    ucm_add_flags(C CXX "-march=x86-64 -mtune=generic")
-  else()
-    ucm_add_flags(C CXX "-march=native -mtune=generic")
-  endif()
 endif(MSVC)
 
 if(WIN32)
