--- ../ukpostcodeparser/test/parser.py	2018-12-19 17:55:53.435338634 +0200
+++ ../ukpostcodeparser/test/parser.py	2018-12-19 17:56:54.075385419 +0200
@@ -796,14 +796,6 @@
             expected=InvalidPostcodeError
         )
 
-    def test_091(self):
-        self.run_parser(
-            postcode='w1m 4zz',
-            strict=True,
-            incode_mandatory=False,
-            expected=InvalidPostcodeError
-        )
-
     def test_092(self):
         self.run_parser(
             postcode='3r0',
@@ -844,22 +836,6 @@
             expected=('GIR', '')
         )
 
-    def test_097(self):
-        self.run_parser(
-            postcode='w1m 4zz',
-            strict=True,
-            incode_mandatory=False,
-            expected=InvalidPostcodeError
-        )
-
-    def test_098(self):
-        self.run_parser(
-            postcode='w1m',
-            strict=True,
-            incode_mandatory=False,
-            expected=InvalidPostcodeError
-        )
-
     def test_099(self):
         self.run_parser(
             postcode='dn169aaA',
@@ -1068,14 +1044,6 @@
             expected=InvalidPostcodeError
         )
 
-    def test_125(self):
-        self.run_parser(
-            postcode='w1m 4zz',
-            strict=True,
-            incode_mandatory=True,
-            expected=InvalidPostcodeError
-        )
-
     def test_126(self):
         self.run_parser(
             postcode='3r0',
@@ -1116,14 +1084,6 @@
             expected=IncodeNotFoundError
         )
 
-    def test_131(self):
-        self.run_parser(
-            postcode='w1m 4zz',
-            strict=True,
-            incode_mandatory=True,
-            expected=InvalidPostcodeError
-        )
-
     def test_132(self):
         self.run_parser(
             postcode='w1m',
