diff --git a/Makefile b/Makefile
index 264f1aa..ee88b24 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ ifndef HYPRLAND_HEADERS
 endif
 
 $(PLUGIN_NAME).so: $(SOURCE_FILES) $(INCLUDE_FILES)
-	g++ -shared -fPIC --no-gnu-unique $(SOURCE_FILES) -o $(PLUGIN_NAME).so -g -I "/usr/include/pixman-1" -I "/usr/include/libdrm" -I "${HYPRLAND_HEADERS}" -Iinclude -std=c++23
+	g++ -shared -fPIC --no-gnu-unique $(SOURCE_FILES) -o $(PLUGIN_NAME).so -g -I "/usr/include/pixman-1" -I "/usr/include/libdrm" -I "${HYPRLAND_HEADERS}" -Iinclude -std=c++23 -DNO_XWAYLAND
 
 clean:
 	rm -f ./$(PLUGIN_NAME).so
