--- a/configure.ac	2023-03-31 16:42:24.000000000 -0400
+++ b/configure.ac	2024-10-21 19:11:20.577093421 -0400
@@ -184,8 +184,10 @@
 
 # TODO: We currently link everything against libraries that don't need it.
 # Use the specific library CFLAGS/LIBS variables instead of setting them here.
-CFLAGS="$CFLAGS $SDL_CFLAGS ${SAMPLERATE_CFLAGS:-} ${PNG_CFLAGS:-} ${FLUIDSYNTH_CFLAGS:-} ${LIBZ_CFLAGS:-}"
-LDFLAGS="$LDFLAGS $SDL_LIBS ${SAMPLERATE_LIBS:-} ${PNG_LIBS:-} ${FLUIDSYNTH_LIBS:-} ${LIBZ_LIBS:-}"
+UCFLAGS:=$(shell emerge --info | grep CFLAGS=".*" | cut -d'"' -f2)
+ULDFLAGS:=$(shell emerge --info | grep LDFLAGS=".*" | cut -d'"' -f2)
+CFLAGS+="$UCFLAGS $SDL_CFLAGS ${SAMPLERATE_CFLAGS:-} ${PNG_CFLAGS:-} ${LIBZ_CFLAGS:-}"
+LDFLAGS+="$ULDFLAGS $SDL_LIBS ${SAMPLERATE_LIBS:-} ${PNG_LIBS:-} ${LIBZ_LIBS:-}"
 case "$host" in
   *-*-mingw* | *-*-cygwin* | *-*-msvc* )
     LDFLAGS="$LDFLAGS -lwinmm"
