--- a/configure.ac	2024-10-21 18:54:07.483980753 -0400
+++ b/configure.ac	2024-10-21 18:54:52.224330577 -0400
@@ -166,8 +166,10 @@
 
 # TODO: We currently link everything against libraries that don't need it.
 # Use the specific library CFLAGS/LIBS variables instead of setting them here.
-CFLAGS="$CFLAGS $SDL_CFLAGS ${SAMPLERATE_CFLAGS:-} ${PNG_CFLAGS:-} ${LIBZ_CFLAGS:-}"
-LDFLAGS="$LDFLAGS $SDL_LIBS ${SAMPLERATE_LIBS:-} ${PNG_LIBS:-} ${LIBZ_LIBS:-}"
+UCFLAGS:=$(shell emerge --info | grep CFLAGS=".*" | cut -d'"' -f2)
+ULDFLAGS:=$(shell emerge --info | grep LDFLAGS=".*" | cut -d'"' -f2)
+CFLAGS+="$UCFLAGS $SDL_CFLAGS ${SAMPLERATE_CFLAGS:-} ${PNG_CFLAGS:-} ${LIBZ_CFLAGS:-}"
+LDFLAGS+="$ULDFLAGS $SDL_LIBS ${SAMPLERATE_LIBS:-} ${PNG_LIBS:-} ${LIBZ_LIBS:-}"
 case "$host" in
   *-*-mingw* | *-*-cygwin* | *-*-msvc* )
     LDFLAGS="$LDFLAGS -lwinmm"
