# Packages should not have `-Werror` unconditionally enabled.

--- a/swift-corelibs-libdispatch/cmake/modules/DispatchCompilerWarnings.cmake
+++ b/swift-corelibs-libdispatch/cmake/modules/DispatchCompilerWarnings.cmake
@@ -6,7 +6,6 @@ elseif(WIN32)
   # so that we can use __popcnt64
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-fms-extensions>)
 else()
-  add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Werror>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wall>)
   add_compile_options($<$<OR:$<COMPILE_LANGUAGE:C>,$<COMPILE_LANGUAGE:CXX>>:-Wextra>)
