--- a/lib/perl/Makefile.PL
+++ b/lib/perl/Makefile.PL
@@ -42,8 +42,8 @@
 WriteMakefile(
 	'NAME'    => 'pqos',
 	'VERSION' => "0.1",
-	'CCFLAGS' => "$Config{ccflags} -Wno-unused-variable -Wno-unused-value",
-	'LIBS'    => ['-lpqos', '-L/usr/local/lib -lpqos'],
+	'CCFLAGS' => "$Config{ccflags} -Wno-unused-variable -Wno-unused-value -I..",
+	'LIBS'    => ['-lpqos', '-L.. -lpqos'],
 	'OBJECT'  => 'pqos_wrap.o',
 	'clean'   => {FILES => "pqos_wrap.c"},
 );
@@ -56,7 +56,7 @@
 
 	my $swig_target = <<"SWIG_MAKEFILE";
 pqos_wrap.c:
-	$swig -perl -outdir \$(INST_LIB) -I/usr/local/include \$(EXTRA_SWIGFLAGS) pqos.i
+	$swig -perl -outdir \$(INST_LIB) -I.. \$(EXTRA_SWIGFLAGS) pqos.i
 SWIG_MAKEFILE
 
 	my $perltidy_targets = <<'PERLTIDY_MAKEFILE';
