--- a/examples/c/CAT_MBA/Makefile
+++ b/examples/c/CAT_MBA/Makefile
@@ -35,17 +35,17 @@
 ###############################################################################
 
 LIBDIR ?= ../../../lib
-CFLAGS =-I$(LIBDIR) \
+CFLAGS += -I$(LIBDIR) \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings  \
-	-Wformat -Wformat-security -fstack-protector -fPIE -D_FORTIFY_SOURCE=2 \
-	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-g -O2
+	-Wformat -Wformat-security -fPIE \
+	-Wunreachable-code -Wsign-compare -Wno-endif-labels
+
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
 endif
-LDFLAGS=-L$(LIBDIR) -pie -z noexecstack -z relro -z now
+LDFLAGS += -L$(LIBDIR) -pie -z noexecstack -z relro -z now
 LDLIBS=-lpqos -lpthread
 
 # ICC and GCC options
--- a/examples/c/CMT_MBM/Makefile
+++ b/examples/c/CMT_MBM/Makefile
@@ -35,17 +35,17 @@
 ###############################################################################
 
 LIBDIR ?= ../../../lib
-CFLAGS =-I$(LIBDIR) -pthread \
+CFLAGS += -I$(LIBDIR) -pthread \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector -fPIE -D_FORTIFY_SOURCE=2 \
-	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-g -O2
+	-Wformat -Wformat-security -fPIE \
+	-Wunreachable-code -Wsign-compare -Wno-endif-labels
+
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
 endif
-LDFLAGS=-L$(LIBDIR) -pie -z noexecstack -z relro -z now
+LDFLAGS += -L$(LIBDIR) -pie -z noexecstack -z relro -z now
 LDLIBS=-lpqos -lpthread
 
 # ICC and GCC options
--- a/examples/c/PSEUDO_LOCK/Makefile
+++ b/examples/c/PSEUDO_LOCK/Makefile
@@ -35,15 +35,15 @@
 ###############################################################################
 
 LIBDIR ?= ../../../lib
-LDFLAGS = -L$(LIBDIR) -pie -z noexecstack -z relro -z now
+LDFLAGS += -L$(LIBDIR) -pie -z noexecstack -z relro -z now
 LDLIBS = -lpqos -lrt -lpthread
-CFLAGS = -I$(LIBDIR) \
+CFLAGS += -I$(LIBDIR) \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings  \
-	-Wformat -Wformat-security -fstack-protector -fPIE -D_FORTIFY_SOURCE=2 \
-	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-g -O2
+	-Wformat -Wformat-security -fPIE \
+	-Wunreachable-code -Wsign-compare -Wno-endif-labels
+
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
 endif
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -37,12 +37,12 @@
 VERSION = 4.2.0
 SO_VERSION = 4
 SHARED ?= y
-LDFLAGS = -L. -lpthread -z noexecstack -z relro -z now
-CFLAGS = -pthread -I./ -D_GNU_SOURCE \
+LDFLAGS += -L. -lpthread -z noexecstack -z relro -z now
+CFLAGS += -pthread -I./ -D_GNU_SOURCE \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector \
+	-Wformat -Wformat-security \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
@@ -83,9 +83,7 @@
 
 # DEBUG build
 ifeq ($(DEBUG),y)
-CFLAGS += -g -ggdb -O0 -DDEBUG
-else
-CFLAGS += -g -O2 -D_FORTIFY_SOURCE=2
+CFLAGS += -DDEBUG
 endif
 
 # RMID CUSTOM
--- a/pqos/Makefile
+++ b/pqos/Makefile
@@ -35,13 +35,13 @@
 ###############################################################################
 
 LIBDIR ?= ../lib
-LDFLAGS = -L$(LIBDIR) -pie -z noexecstack -z relro -z now
+LDFLAGS += -L$(LIBDIR) -pie -z noexecstack -z relro -z now
 LDLIBS = -lpqos -lpthread
-CFLAGS = -I$(LIBDIR) \
+CFLAGS += -I$(LIBDIR) \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector -fPIE \
+	-Wformat -Wformat-security -fPIE \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels
 ifneq ($(EXTRA_CFLAGS),)
 CFLAGS += $(EXTRA_CFLAGS)
@@ -68,9 +68,7 @@
 
 # DEBUG build
 ifeq ($(DEBUG),y)
-CFLAGS += -g -ggdb -O0 -DDEBUG
-else
-CFLAGS += -g -O2 -D_FORTIFY_SOURCE=2
+CFLAGS += -DDEBUG
 endif
 
 # RMID CUSTOM
--- a/rdtset/Makefile
+++ b/rdtset/Makefile
@@ -35,13 +35,13 @@
 ###############################################################################
 
 LIBDIR ?= ../lib
-LDFLAGS = -L$(LIBDIR) -pie -z noexecstack -z relro -z now
+LDFLAGS += -L$(LIBDIR) -pie -z noexecstack -z relro -z now
 LDLIBS = -lpqos -lpthread
-CFLAGS = -I$(LIBDIR) \
+CFLAGS += -I$(LIBDIR) \
 	-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector -fPIE \
+	-Wformat -Wformat-security -fPIE \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
 	-D_GNU_SOURCE
 ifneq ($(EXTRA_CFLAGS),)
@@ -69,9 +69,7 @@
 
 # DEBUG build
 ifeq ($(DEBUG),y)
-CFLAGS += -g -ggdb -O0 -DDEBUG
-else
-CFLAGS += -g -O2 -D_FORTIFY_SOURCE=2
+CFLAGS += -DDEBUG
 endif
 
 # RMID CUSTOM
--- a/tools/membw/Makefile
+++ b/tools/membw/Makefile
@@ -36,17 +36,15 @@
 
 APP = membw
 
-CFLAGS=-W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
+CFLAGS += -W -Wall -Wextra -Wstrict-prototypes -Wmissing-prototypes \
 	-Wmissing-declarations -Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector -fPIE \
+	-Wformat -Wformat-security -fPIE \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
-	-Winline -msse4.2
+	-Winline
 
 ifeq ($(DEBUG),y)
-CFLAGS += -O0 -g -DDEBUG
-else
-CFLAGS += -O3 -g -D_FORTIFY_SOURCE=2
+CFLAGS += -DDEBUG
 endif
 
 HAS_AVX512 := $(shell $(CC) -mavx512f -dM -E - < /dev/null 2> /dev/null | grep -c "AVX512F")
--- a/unit-test/lib/Makefile
+++ b/unit-test/lib/Makefile
@@ -15,13 +15,13 @@
 TESTS = $(TESTS_SRCS:%.c=$(BIN_DIR)/%)
 
 
-LDFLAGS = -L$(MOCK_DIR) -lpthread -lcmocka -lpqosmock -z noexecstack -z relro -z now
-CFLAGS = -pthread -I$(LIB_DIR) -I$(MOCK_DIR) \
+LDFLAGS += -L$(MOCK_DIR) -lpthread -lcmocka -lpqosmock -z noexecstack -z relro -z now
+CFLAGS += -pthread -I$(LIB_DIR) -I$(MOCK_DIR) \
 	-D_GNU_SOURCE \
 	-W -Wall -Wextra -Wstrict-prototypes \
 	-Wold-style-definition -Wpointer-arith \
 	-Wcast-qual -Wundef -Wwrite-strings \
-	-Wformat -Wformat-security -fstack-protector \
+	-Wformat -Wformat-security \
 	-Wunreachable-code -Wsign-compare -Wno-endif-labels \
 	-D'PQOS_LOCAL=' -D'PQOS_API='
 
@@ -527,4 +527,4 @@
 
 clean:
 	rm -rf $(BIN_DIR)
-	rm -rf $(OBJ_DIR)
\ Manca newline alla fine del file
+	rm -rf $(OBJ_DIR)
