--- a/pqos/Makefile
+++ b/pqos/Makefile
@@ -108,12 +108,12 @@
 ifeq ($(shell uname), FreeBSD)
 	install -d $(BIN_DIR)
 	install -d $(MAN_DIR)
-	install -s $(APP) $(BIN_DIR)
+	install $(APP) $(BIN_DIR)
 	install $(APP)-msr $(BIN_DIR)
 	install -m 0444 $(MAN) $(MAN_DIR)
 	ln -f -s $(MAN) $(MAN_DIR)/$(APP)-msr.8
 else
-	install -D -s $(APP) $(BIN_DIR)/$(APP)
+	install -D $(APP) $(BIN_DIR)/$(APP)
 	install -D $(APP)-msr $(BIN_DIR)/$(APP)-msr
 	install -D $(APP)-os $(BIN_DIR)/$(APP)-os
 	install -m 0444 $(MAN) -D $(MAN_DIR)/$(MAN)
--- a/rdtset/Makefile
+++ b/rdtset/Makefile
@@ -115,10 +115,10 @@
 ifeq ($(shell uname), FreeBSD)
 	install -d $(BIN_DIR)
 	install -d $(MAN_DIR)
-	install -s $(APP) $(BIN_DIR)
+	install $(APP) $(BIN_DIR)
 	install -m 0444 $(MAN) $(MAN_DIR)
 else
-	install -D -s $(APP) $(BIN_DIR)/$(APP)
+	install -D $(APP) $(BIN_DIR)/$(APP)
 	install -m 0444 $(MAN) -D $(MAN_DIR)/$(MAN)
 endif
 
