--- a/Cargo.toml
+++ b/Cargo.toml
@@ -16,7 +16,6 @@
 edition = "2021"
 license = "MIT"
 keywords = ["graphics", "terminal"]
-rust-version = "1.80.1"
 repository = "https://github.com/raphamorim/rio"
 homepage = "https://raphamorim.io/rio"
 documentation = "https://github.com/raphamorim/rio#readme"
--- a/frontends/rioterm/Cargo.toml
+++ b/frontends/rioterm/Cargo.toml
@@ -6,7 +6,6 @@
 edition.workspace = true
 license.workspace = true
 keywords.workspace = true
-rust-version.workspace = true
 repository.workspace = true
 homepage.workspace = true
 documentation.workspace = true
--- a/frontends/wasm/Cargo.toml
+++ b/frontends/wasm/Cargo.toml
@@ -6,7 +6,6 @@
 edition.workspace = true
 license.workspace = true
 keywords.workspace = true
-rust-version.workspace = true
 repository.workspace = true
 homepage.workspace = true
 documentation.workspace = true
