diff -ru /var/tmp/portage/sys-cluster/extrae-4.0.1/work/extrae-4.0.1/src/cmd-line/Makefile.am extrae-4.0.1/src/cmd-line/Makefile.am
--- /var/tmp/portage/sys-cluster/extrae-4.0.1/work/extrae-4.0.1/src/cmd-line/Makefile.am	2022-05-13 17:41:45.000000000 +0200
+++ extrae-4.0.1/src/cmd-line/Makefile.am	2022-06-05 08:45:43.807262310 +0200
@@ -11,7 +11,7 @@
 extrae_cmd_CFLAGS = -I$(INCLUDE_DIR) -I$(COMMON_INC) -I$(TRACER_INC) \
  -I$(TRACER_INC)/wrappers/API -I$(TRACER_INC)/clocks -I$(TRACER_INC)/hwc -O -g
 
-extrae_cmd_LDADD = $(TRACER_LIB)/libseqtrace.la $(MERGER_LIB)/libmpi2prv.la -lm
+extrae_cmd_LDADD = $(TRACER_LIB)/libseqtrace.la $(MERGER_LIB)/libmpi2prv.la -lm @PAPI_LIBS@
 if HAVE_BINUTILS
 extrae_cmd_LDADD += $(BFD_LDFLAGS) $(BFD_RPATH) $(BFD_LIBS) $(LIBERTY_LDFLAGS) $(LIBERTY_RPATH) $(LIBERTY_LIBS)
 endif
