--- a/src/tracer/Makefile.am
+++ b/src/tracer/Makefile.am
@@ -631,6 +631,10 @@
 NEW_MPI_LIBS = -L@MPI_LIBSDIR@ @MPI_LIBS@ -static
 endif
 
+if HAVE_SIONLIB
+SIONLIB_LINKER_FLAGS = @SIONLIB_LIBS@
+endif
+
 COMMON_LINKER_FLAGS = \
  -release $(PACKAGE_VERSION) \
  $(HWC_LINKER_FLAGS) \
@@ -639,7 +643,8 @@
  $(XML2_LINKER_FLAGS) \
  $(LDL)\
  $(LIBZ_LINKER_FLAGS) \
- $(MRNET_LINKER_FLAGS)
+ $(MRNET_LINKER_FLAGS) \
+ $(SIONLIB_LINKER_FLAGS)
 
 libcudaompitrace_la_LDFLAGS = $(COMMON_LINKER_FLAGS)
 libcudaompitracef_la_LDFLAGS = $(COMMON_LINKER_FLAGS)
