Bug: https://github.com/scitokens/scitokens-cpp/pull/89

---
 test/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 22f86a6..2083bff 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,6 +1,8 @@
 
 add_executable(scitokens-gtest main.cpp)
-add_dependencies(scitokens-gtest gtest)
+if( NOT SCITOKENS_EXTERNAL_GTEST )
+    add_dependencies(scitokens-gtest gtest)
+endif()
 include_directories("${PROJECT_SOURCE_DIR}/vendor/gtest/googletest/include")
 
 if(SCITOKENS_EXTERNAL_GTEST)
