Bug: https://github.com/scitokens/scitokens-cpp/issues/82

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec5c878..611afb1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,13 +81,12 @@ endif()
 enable_testing()
 add_subdirectory(test)
 
-else()
+endif()
 get_directory_property(TARGETS BUILDSYSTEM_TARGETS)
 install(
   TARGETS ${TARGETS} 
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} )
-endif()
 
 install(
   FILES src/scitokens.h
