unbundle doctest
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -43,15 +43,16 @@ set(TOML11_TEST_NAMES
 
 if(BUILD_TESTING)
     add_library(toml11_test_utility STATIC utility.cpp)
+    find_package(doctest REQUIRED)
     target_include_directories(toml11_test_utility
-        PRIVATE ${PROJECT_SOURCE_DIR}/tests/extlib/doctest/doctest/
+        PRIVATE ${DOCTEST_INCLUDE_DIR}
         )
     target_link_libraries(toml11_test_utility PUBLIC toml11)
 
     foreach(TEST_NAME ${TOML11_TEST_NAMES})
         add_executable(${TEST_NAME} ${TEST_NAME}.cpp)
         target_include_directories(${TEST_NAME}
-            PRIVATE ${PROJECT_SOURCE_DIR}/tests/extlib/doctest/doctest/
+            PRIVATE ${DOCTEST_INCLUDE_DIR}
             )
         target_link_libraries(${TEST_NAME} PUBLIC toml11 toml11_test_utility)
         if(MSVC)
--- a/tests/test_comments.cpp
+++ b/tests/test_comments.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml.hpp>
 
--- a/tests/test_datetime.cpp
+++ b/tests/test_datetime.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/datetime.hpp>
 
--- a/tests/test_error_message.cpp
+++ b/tests/test_error_message.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml.hpp>
 
--- a/tests/test_find.cpp
+++ b/tests/test_find.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_find_or.cpp
+++ b/tests/test_find_or.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_format_floating.cpp
+++ b/tests/test_format_floating.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/parser.hpp>
 #include <toml11/serializer.hpp>
--- a/tests/test_format_integer.cpp
+++ b/tests/test_format_integer.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/parser.hpp>
 #include <toml11/serializer.hpp>
--- a/tests/test_format_table.cpp
+++ b/tests/test_format_table.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/parser.hpp>
 #include <toml11/serializer.hpp>
--- a/tests/test_get.cpp
+++ b/tests/test_get.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_get_or.cpp
+++ b/tests/test_get_or.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_literal.cpp
+++ b/tests/test_literal.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <iostream>
 
--- a/tests/test_location.cpp
+++ b/tests/test_location.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/location.hpp>
 
--- a/tests/test_parse_array.cpp
+++ b/tests/test_parse_array.cpp
@@ -5,7 +5,7 @@
 #include <toml11/parser.hpp>
 #include <toml11/types.hpp>
 
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 TEST_CASE("testing an array")
 {
--- a/tests/test_parse_boolean.cpp
+++ b/tests/test_parse_boolean.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_parse_datetime.cpp
+++ b/tests/test_parse_datetime.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_parse_floating.cpp
+++ b/tests/test_parse_floating.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_parse_inline_table.cpp
+++ b/tests/test_parse_inline_table.cpp
@@ -5,7 +5,7 @@
 #include <toml11/parser.hpp>
 #include <toml11/types.hpp>
 
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 TEST_CASE("testing an inline table v1.0")
 {
--- a/tests/test_parse_integer.cpp
+++ b/tests/test_parse_integer.cpp
@@ -1,6 +1,6 @@
 
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_parse_null.cpp
+++ b/tests/test_parse_null.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_parse_string.cpp
+++ b/tests/test_parse_string.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 #include <iostream>
--- a/tests/test_parse_table.cpp
+++ b/tests/test_parse_table.cpp
@@ -5,7 +5,7 @@
 #include <toml11/parser.hpp>
 #include <toml11/types.hpp>
 
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 TEST_CASE("testing a table")
 {
--- a/tests/test_parse_table_keys.cpp
+++ b/tests/test_parse_table_keys.cpp
@@ -5,7 +5,7 @@
 #include <toml11/parser.hpp>
 #include <toml11/types.hpp>
 
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 TEST_CASE("testing table keys")
 {
--- a/tests/test_result.cpp
+++ b/tests/test_result.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/result.hpp>
 
--- a/tests/test_scanner.cpp
+++ b/tests/test_scanner.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_serialize.cpp
+++ b/tests/test_serialize.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml.hpp>
 
--- a/tests/test_spec.cpp
+++ b/tests/test_spec.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/spec.hpp>
 
--- a/tests/test_storage.cpp
+++ b/tests/test_storage.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/storage.hpp>
 
--- a/tests/test_syntax_boolean.cpp
+++ b/tests/test_syntax_boolean.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_comment.cpp
+++ b/tests/test_syntax_comment.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_datetime.cpp
+++ b/tests/test_syntax_datetime.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_floating.cpp
+++ b/tests/test_syntax_floating.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_integer.cpp
+++ b/tests/test_syntax_integer.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_key.cpp
+++ b/tests/test_syntax_key.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_syntax_string.cpp
+++ b/tests/test_syntax_string.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_traits.cpp
+++ b/tests/test_traits.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/types.hpp>
 
--- a/tests/test_types.cpp
+++ b/tests/test_types.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/types.hpp>
 
--- a/tests/test_user_defined_conversion.cpp
+++ b/tests/test_user_defined_conversion.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml.hpp>
 #include "utility.hpp"
--- a/tests/test_utility.cpp
+++ b/tests/test_utility.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/utility.hpp>
 
--- a/tests/test_value.cpp
+++ b/tests/test_value.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/test_visit.cpp
+++ b/tests/test_visit.cpp
@@ -1,5 +1,5 @@
 #define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <toml11/comments.hpp>
 #include <toml11/types.hpp>
--- a/tests/utility.cpp
+++ b/tests/utility.cpp
@@ -1,4 +1,4 @@
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include "utility.hpp"
 
--- a/tests/utility.hpp
+++ b/tests/utility.hpp
@@ -9,7 +9,7 @@
 #include <toml11/serializer.hpp>
 #include <toml11/value.hpp>
 
-#include "doctest.h"
+#include <doctest/doctest.h>
 
 #include <ostream>
 #include <string>
