--- a/GUI/src/DimemasGUI.in
+++ b/GUI/src/DimemasGUI.in
@@ -1,29 +1,9 @@
 #!/bin/sh
 
-DIMEMAS_HOME=$(dirname $(readlink -f $0))/..
-if [ -d "$DIMEMAS_HOME/lib" ]; then
-    LIBDIR="/lib"
-elif [ -d "$DIMEMAS_HOME/lib64" ]; then
-    LIBDIR="/lib64"
-else
-    echo "$DIMEMAS_HOME/{lib|lib64} not found"
-    exit 1
-fi
+DIMEMAS_HOME=/usr
 
-DIMEMAS_GUI_JAR=$DIMEMAS_HOME/$LIBDIR/GUI/@@JARFILE@@
+DIMEMAS_GUI_JAR="${DIMEMAS_HOME}/share/dimemas/lib/dimemas-gui.jar"
 
-if ! [ -e $DIMEMAS_GUI_JAR ]; then
-    echo "Dimemas GUI 'jar' file (@@JARFILE@@) not found"
-    echo "Check your Dimemas instalation."
-    exit 1
-fi
-
-if  which java >/dev/null 2>&1  ; then
-    JAVA_BIN=`which java`
-else
-    echo "No Java Runtime found on your path."
-    echo "Please check the JRE installation"
-    exit 1
-fi
+JAVA_BIN=`java-config -J`
 
 $JAVA_BIN @@JVMFLAGS@@ -DDIMEMAS_HOME=$DIMEMAS_HOME -jar $DIMEMAS_GUI_JAR $@ 
