diff --git a/SConstruct b/SConstruct
index c30f533..388be69 100644
--- a/SConstruct
+++ b/SConstruct
@@ -174,6 +174,8 @@ opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", True))
 # Compilation environment setup
 opts.Add("CXX", "C++ compiler")
 opts.Add("CC", "C compiler")
+opts.Add("AR", "Archiver")
+opts.Add("RANLIB", "Random Access Library Indexer")
 opts.Add("LINK", "Linker")
 opts.Add("CCFLAGS", "Custom flags for both the C and C++ compilers")
 opts.Add("CFLAGS", "Custom flags for the C compiler")
diff --git a/platform/x11/detect.py b/platform/x11/detect.py
index ba5fb30..d907701 100644
--- a/platform/x11/detect.py
+++ b/platform/x11/detect.py
@@ -129,7 +129,6 @@ def configure(env):
         if "clang++" not in os.path.basename(env["CXX"]):
             env["CC"] = "clang"
             env["CXX"] = "clang++"
-        env.extra_suffix = ".llvm" + env.extra_suffix
 
     if env["use_lld"]:
         if env["use_llvm"]:
@@ -189,10 +188,6 @@ def configure(env):
                 env.Append(CCFLAGS=["-flto"])
                 env.Append(LINKFLAGS=["-flto"])
 
-        if not env["use_llvm"]:
-            env["RANLIB"] = "gcc-ranlib"
-            env["AR"] = "gcc-ar"
-
     env.Append(CCFLAGS=["-pipe"])
     env.Append(LINKFLAGS=["-pipe"])
 
