--- a/Lib/fontmake/__main__.py
+++ b/Lib/fontmake/__main__.py
@@ -238,7 +238,7 @@ def main(args=None):
         dest="overlaps_backend",
         metavar="BACKEND",
         choices=("booleanOperations", "pathops"),
-        default="booleanOperations",
+        default="pathops",
         help="Select library to remove overlaps. Choose between: %(choices)s "
         "(default: %(default)s)",
     )
@@ -291,7 +291,7 @@ def main(args=None):
     )
     contourGroup.add_argument(
         "--subroutinizer",
-        default=None,
+        default="cffsubr",
         choices=["compreffor", "cffsubr"],
         help="name of the library to use for compressing CFF charstrings. "
         "Choose between: %(choices)s. By default compreffor is used for CFF 1, "
