--- a/dart/external/CMakeLists.txt
+++ b/dart/external/CMakeLists.txt
@@ -1,5 +1,3 @@
 add_subdirectory(convhull_3d)
-add_subdirectory(imgui)
 add_subdirectory(ikfast)
-add_subdirectory(lodepng)
 add_subdirectory(odelcpsolver)
--- a/dart/gui/CMakeLists.txt
+++ b/dart/gui/CMakeLists.txt
@@ -42,14 +42,15 @@
 dart_add_library(${target_name}
   ${hdrs} ${srcs} ${dart_gui_headers} ${dart_gui_sources}
 )
+target_include_directories(${target_name} PRIVATE /usr/include/imgui)
 target_link_libraries(${target_name}
   PUBLIC
     dart-utils
     OpenGL::GL
     OpenGL::GLU
     GLUT::GLUT
-    ${PROJECT_NAME}-external-lodepng
-    ${PROJECT_NAME}-external-imgui
+    -limgui
+    lodepng
 )
 
 # Component
@@ -59,8 +60,6 @@
   ${PROJECT_NAME}
   ${component_name}
   utils
-  external-lodepng
-  external-imgui
 )
 add_component_dependency_packages(${PROJECT_NAME} ${component_name} OpenGL)
 if(NOT WIN32 OR CYGWIN)
--- a/dart/gui/glut/GlutWindow.cpp
+++ b/dart/gui/glut/GlutWindow.cpp
@@ -30,7 +30,7 @@
  *   POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include "dart/external/lodepng/lodepng.h"
+#include "lodepng.h"
 #include "dart/gui/glut/Window.hpp"
 
 #ifdef _WIN32
--- a/dart/gui/osg/ImGuiHandler.cpp
+++ b/dart/gui/osg/ImGuiHandler.cpp
@@ -44,8 +44,8 @@
 #include <osg/RenderInfo>
 
 #include "dart/common/Console.hpp"
-#include "dart/external/imgui/imgui.h"
-#include "dart/external/imgui/imgui_impl_opengl2.h"
+#include "imgui/imgui.h"
+#include "imgui/imgui_impl_opengl2.h"
 #include "dart/gui/osg/ImGuiWidget.hpp"
 
 namespace dart {
--- a/dart/gui/osg/ImGuiWidget.cpp
+++ b/dart/gui/osg/ImGuiWidget.cpp
@@ -38,7 +38,7 @@
 
 #include "dart/gui/osg/ImGuiWidget.hpp"
 
-#include "dart/external/imgui/imgui.h"
+#include "imgui/imgui.h"
 
 namespace dart {
 namespace gui {
--- a/examples/atlas_simbicon/AtlasSimbiconWidget.cpp
+++ b/examples/atlas_simbicon/AtlasSimbiconWidget.cpp
@@ -38,7 +38,7 @@
 
 #include "AtlasSimbiconWidget.hpp"
 
-#include "dart/external/imgui/imgui.h"
+#include "imgui/imgui.h"
 
 #include "AtlasSimbiconWorldNode.hpp"
 
--- a/examples/box_stacking/main.cpp
+++ b/examples/box_stacking/main.cpp
@@ -33,7 +33,7 @@
 #include <iostream>
 
 #include <dart/dart.hpp>
-#include <dart/external/imgui/imgui.h>
+#include "imgui/imgui.h"
 #include <dart/gui/osg/osg.hpp>
 #include <dart/utils/utils.hpp>
 
--- a/examples/fetch/main.cpp
+++ b/examples/fetch/main.cpp
@@ -32,7 +32,7 @@
 
 #include <dart/collision/bullet/bullet.hpp>
 #include <dart/dart.hpp>
-#include <dart/external/imgui/imgui.h>
+#include "imgui/imgui.h"
 #include <dart/gui/osg/osg.hpp>
 #include <dart/utils/utils.hpp>
 
--- a/examples/heightmap/main.cpp
+++ b/examples/heightmap/main.cpp
@@ -33,7 +33,7 @@
 #include <cmath>
 
 #include <dart/dart.hpp>
-#include <dart/external/imgui/imgui.h>
+#include "imgui/imgui.h"
 #include <dart/gui/osg/osg.hpp>
 #include <dart/utils/urdf/urdf.hpp>
 #include <dart/utils/utils.hpp>
--- a/examples/imgui/main.cpp
+++ b/examples/imgui/main.cpp
@@ -31,7 +31,7 @@
  */
 
 #include <dart/dart.hpp>
-#include <dart/external/imgui/imgui.h>
+#include "imgui/imgui.h"
 #include <dart/gui/osg/osg.hpp>
 
 //==============================================================================
--- a/examples/point_cloud/main.cpp
+++ b/examples/point_cloud/main.cpp
@@ -33,7 +33,7 @@
 #include <cmath>
 
 #include <dart/dart.hpp>
-#include <dart/external/imgui/imgui.h>
+#include "imgui/imgui.h"
 #include <dart/gui/osg/osg.hpp>
 #include <dart/utils/urdf/urdf.hpp>
 #include <dart/utils/utils.hpp>
--- a/examples/tinkertoy/TinkertoyWidget.cpp
+++ b/examples/tinkertoy/TinkertoyWidget.cpp
@@ -38,7 +38,7 @@
 
 #include "TinkertoyWidget.hpp"
 
-#include "dart/external/imgui/imgui.h"
+#include "imgui/imgui.h"
 
 #include "TinkertoyWorldNode.hpp"
 
