--- a/unittests/CMakeLists.txt
+++ b/unittests/CMakeLists.txt
@@ -31,18 +31,7 @@
 #
 
 # GoogleTest setup
-include_directories(BEFORE SYSTEM ${DART_SOURCE_DIR}/unittests/gtest/include)
-include_directories(BEFORE SYSTEM ${DART_SOURCE_DIR}/unittests/gtest)
-add_library(gtest STATIC gtest/src/gtest-all.cc)
-add_library(gtest_main STATIC gtest/src/gtest_main.cc)
-target_link_libraries(gtest_main gtest)
-if(NOT WIN32)
-  target_link_libraries(gtest pthread)
-endif()
-set_target_properties(
-  gtest PROPERTIES
-  ARCHIVE_OUTPUT_DIRECTORY ${DART_BINARY_DIR}/lib
-)
+include_directories(BEFORE SYSTEM /usr/include/gtest)
 
 #===============================================================================
 # This function uses following global properties:
@@ -66,7 +55,7 @@
 
   add_executable(${target_name} ${sources})
   add_test(${target_name} ${target_name})
-  target_link_libraries(${target_name} dart gtest gtest_main)
+  target_link_libraries(${target_name} dart gtest)
 
   dart_format_add(${sources})
 
