Unbundle zstd.
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -20,7 +20,8 @@ add_library(3rdparty_dummy_lib INTERFACE)
 add_subdirectory(zlib EXCLUDE_FROM_ALL)
 
 # ZSTD
-add_subdirectory(zstd EXCLUDE_FROM_ALL)
+pkg_check_modules(zstd REQUIRED IMPORTED_TARGET GLOBAL libzstd)
+add_library(3rdparty::zstd ALIAS PkgConfig::zstd)
 
 # 7zip sdk
 add_subdirectory(7zip EXCLUDE_FROM_ALL)
@@ -350,7 +351,6 @@ else()
 	add_library(3rdparty::libusb ALIAS usb-1.0-static)
 endif()
 add_library(3rdparty::zlib ALIAS 3rdparty_zlib)
-add_library(3rdparty::zstd ALIAS 3rdparty_zstd)
 add_library(3rdparty::7zip ALIAS 3rdparty_7zip)
 add_library(3rdparty::flatbuffers ALIAS 3rdparty_flatbuffers)
 add_library(3rdparty::pugixml ALIAS pugixml)
