Use luajit to avoid lua version specific pkg-config

--- a/Makefile
+++ b/Makefile
@@ -35,8 +35,8 @@ CFLAGS += -std=c11 -g -O$(OPTIM) -I./src/ -D_GNU_SOURCE=1 -DVERSION_MAJOR=$(VERS
 LIBS = -lm
 
 ### lua
-CFLAGS += $(shell pkg-config --cflags lua)
-LIBS += $(shell pkg-config --libs lua)
+CFLAGS += $(shell pkg-config --cflags luajit)
+LIBS += $(shell pkg-config --libs luajit)
 
 ### debugger
 CFLAGS += $(shell pkg-config --cflags readline)
lines 1-15/15 (END)
