index d07eb6b..c339657 100644
--- a/sni/statusnotifierbutton.cpp
+++ b/sni/statusnotifierbutton.cpp
@@ -267,7 +267,7 @@ void StatusNotifierButton::mouseReleaseEvent(QMouseEvent *event)
 {
     if (event->button() == Qt::LeftButton)
         interface->Activate(QCursor::pos().x(), QCursor::pos().y());
-    else if (event->button() == Qt::MidButton)
+    else if (event->button() == Qt::MiddleButton)
         interface->SecondaryActivate(QCursor::pos().x(), QCursor::pos().y());
     else if (Qt::RightButton == event->button())
     {
