--- a/setup.py
+++ b/setup.py
@@ -136,7 +136,7 @@
     package_dir={"": "src"},
     packages=find_packages("src"),
     entry_points={"console_scripts": ["cffsubr = cffsubr.__main__:main"]},
-    ext_modules=[tx],
+    ext_modules=[],
     zip_safe=False,
     cmdclass=cmdclass,
     install_requires=[
@@ -144,9 +144,7 @@
         "fontTools >= 4.10.2",
     ],
     setup_requires=[
-        "setuptools_scm",
-        # finds all git tracked files including submodules when making sdist MANIFEST
-        "setuptools-git-ls-files",
+        "setuptools_scm"
     ],
     extras_require={"testing": ["pytest"]},
     python_requires=">=3.6",
