--- a/src/cffsubr/__init__.py
+++ b/src/cffsubr/__init__.py
@@ -60,8 +60,7 @@ def _run_embedded_tx(*args, **kwargs):
         subprocess.CompletedProcess object with the following attributes:
         args, returncode, stdout, stderr.
     """
-    with path(__name__, TX_EXE) as tx_cli:
-        return subprocess.run([str(tx_cli)] + list(args), **kwargs)
+    return subprocess.run(['tx'] + list(args), **kwargs)
 
 
 def _tx_subroutinize(data: bytes, output_format: str = CFFTableTag.CFF) -> bytes:
