From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Add a missing include of `cstdint`.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/onnxruntime/core/optimizer/transpose_optimization/optimizer_api.h
+++ b/onnxruntime/core/optimizer/transpose_optimization/optimizer_api.h
@@ -10,6 +10,7 @@
 #include <string_view>
 #include <unordered_map>
 #include <unordered_set>
+#include <cstdint>
 #include <vector>
 
 namespace onnx_transpose_optimization {
