--- a/CMakeLists.txt	2022-04-29 02:03:45.823385000 +0200
+++ b/CMakeLists.txt	2022-04-29 02:08:46.208369263 +0200
@@ -31,8 +31,9 @@
 #--------------------------------------------------------------------------------------------------
 # Dependencies:
 
-find_package(GMP REQUIRED)
-include_directories(${GMP_INCLUDE_DIR})
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GMP REQUIRED gmp)
+include_directories(${GMP_INCLUDE_DIRS})
 
 include_directories(${minisat_SOURCE_DIR})
 include_directories(${minisatp_SOURCE_DIR})
@@ -60,8 +61,8 @@
 add_library(minisatp-lib-static STATIC ${MINISATP_LIB_SOURCES})
 add_library(minisatp-lib-shared SHARED ${MINISATP_LIB_SOURCES})
 
-target_link_libraries(minisatp-lib-shared minisat-lib-shared ${GMP_LIBRARY})
-target_link_libraries(minisatp-lib-static minisat-lib-static ${GMP_LIBRARY})
+target_link_libraries(minisatp-lib-shared minisat gmp)
+target_link_libraries(minisatp-lib-static minisat gmp)
 
 set_target_properties(minisatp-lib-static PROPERTIES OUTPUT_NAME "minisatp")
 set_target_properties(minisatp-lib-shared 
