--- a/src/fontdiff/fontdiff.gyp
+++ b/src/fontdiff/fontdiff.gyp
@@ -20,9 +20,7 @@
                 'main.cc',
             ],
             'dependencies': [
-                'fontdiff_lib',
-                '../third_party/freetype/freetype.gyp:freetype',
-                '../third_party/icu/icu.gyp:icu_uc',
+                'fontdiff_lib'
             ],
             'conditions': [
                 ['OS == "linux"', {
@@ -67,14 +65,6 @@
                 'FONTDIFF_VERSION=' + '<!(echo $FONTDIFF_VERSION)',
             ],
             'dependencies': [
-                '../third_party/cairo/cairo.gyp:cairo',
-                '../third_party/dtl/dtl.gyp:dtl',
-                '../third_party/expat/expat.gyp:expat',
-                '../third_party/freetype/freetype.gyp:freetype',
-                '../third_party/harfbuzz/harfbuzz.gyp:harfbuzz',
-                '../third_party/icu/icu.gyp:icu_i18n',
-                '../third_party/icu/icu.gyp:icu_uc',
-                '../third_party/icu/icu.gyp:icu_data',
             ],
             'conditions': [
                 ['OS == "linux"', {
