Gentoo issue: https://bugs.gentoo.org/888009

diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -422,7 +422,7 @@
   char processor_name[MPI_MAX_PROCESSOR_NAME];
   int name_len;
   MPI_Get_processor_name(processor_name, &name_len);
-  printf('Hello world from processor %s, rank %d out of %d processors',
+  printf(\"Hello world from processor %s, rank %d out of %d processors\",
          processor_name, world_rank, world_size);
   MPI_Finalize();
 }"
