diff --git a/pyproject.toml b/pyproject.toml
index b204a4d..5633f69 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -16,14 +16,6 @@ dev = [
 testpaths = [
     "tests",
 ]
-addopts = [
-    "--doctest-modules",
-    "--cov=sphinxcontrib.zopeext",
-    "--cov-report=html",
-    "--cov-fail-under=85",
-    "--no-cov-on-fail",
-    "--pdb",
-]
 doctest_optionflags = [
     "ELLIPSIS",
     "NORMALIZE_WHITESPACE",
@@ -87,6 +79,6 @@ doc = [
 
 [build-system]
 requires = [
-    "pdm-backend",
+    "pdm-pep517",
 ]
-build-backend = "pdm.backend"
+build-backend = "pdm.pep517.api"
