diff --git a/misc/unix/grafx2.desktop b/misc/unix/grafx2.desktop
index 9718f06..892e7b2 100644
--- a/misc/unix/grafx2.desktop
+++ b/misc/unix/grafx2.desktop
@@ -1,9 +1,9 @@
 [Desktop Entry]
 Name=GrafX2
 Comment=Deluxe Paint Clone
-Exec=grafx2 %f
+Exec=grafx2-sdl %f
 Terminal=false
 Type=Application
 Icon=grafx2
-Categories=Application;Graphics;
+Categories=Graphics;
 MimeType=image/lbm;image/iff;image/pcx;image/tga;image/gif;image/bmp;image/x-icon;image/png;
