--- Makefile	2019-06-16 16:32:45.000000000 +0200
+++ Makefile	2019-12-07 17:14:01.888076319 +0100
@@ -30,12 +30,12 @@
 	   test-tal
 
 # Linux.
-#LDADD += `pkg-config --libs openssl` -lresolv
-#CFLAGS += `pkg-config --cflags openssl`
+LDADD += `pkg-config --libs openssl` -lresolv
+CFLAGS += `pkg-config --cflags openssl`
 
 # OpenBSD.
-CFLAGS += -I/usr/local/include/eopenssl
-LDADD += /usr/local/lib/eopenssl/libssl.a /usr/local/lib/eopenssl/libcrypto.a
+#CFLAGS += -I/usr/local/include/eopenssl
+#LDADD += /usr/local/lib/eopenssl/libssl.a /usr/local/lib/eopenssl/libcrypto.a
 
 all: $(BINS)
 
