diff --git a/../FreeBASIC-1.06.0-source-bootstrap/bootstrap/linux-x86_64/fbc.c b/bootstrap/linux-x86_64/fbc.c
index 650fc5b..12aee5a 100644
--- a/../FreeBASIC-1.06.0-source-bootstrap/bootstrap/linux-x86_64/fbc.c
+++ b/bootstrap/linux-x86_64/fbc.c
@@ -310,7 +310,7 @@ static int64 OPTION_TAKES_ARGUMENT$[58] = { -1ll, -1ll, -1ll, -1ll, 0ll, 0ll, -1
 
 int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 {
-	FBSTRING TMP$647$0;
+	FBSTRING TMP$650$0;
 	int32 fb$result$0;
 	__builtin_memset( &fb$result$0, 0, 4ll );
 	fb_Init( __FB_ARGC__$0, (uint8**)__FB_ARGV__$0, 0 );
@@ -353,10 +353,10 @@ int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 	}
 	label$1110:;
 	label$1109:;
-	__builtin_memset( &TMP$647$0, 0, 24ll );
-	FBSTRING* vr$6 = fb_StrAssign( (void*)&TMP$647$0, -1ll, (void*)((uint8*)&FBC$ + 3098ll), 261ll, 0 );
-	FBADDINCLUDEPATH( &TMP$647$0 );
-	fb_StrDelete( &TMP$647$0 );
+	__builtin_memset( &TMP$650$0, 0, 24ll );
+	FBSTRING* vr$6 = fb_StrAssign( (void*)&TMP$650$0, -1ll, (void*)((uint8*)&FBC$ + 3098ll), 261ll, 0 );
+	FBADDINCLUDEPATH( &TMP$650$0 );
+	fb_StrDelete( &TMP$650$0 );
 	int64 HAVE_INPUT_FILES$0;
 	void* vr$10 = LISTGETHEAD( (struct $5TLIST*)((uint8*)&FBC$ + 144ll) );
 	void* vr$13 = LISTGETHEAD( (struct $5TLIST*)((uint8*)&FBC$ + 376ll) );
@@ -366,9 +366,9 @@ int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 	if( *(int64*)((uint8*)&FBC$ + 136ll) < 0ll ) goto label$1112;
 	{
 		{
-			int64 TMP$648$2;
-			TMP$648$2 = *(int64*)((uint8*)&FBC$ + 136ll);
-			if( TMP$648$2 != 0ll ) goto label$1114;
+			int64 TMP$651$2;
+			TMP$651$2 = *(int64*)((uint8*)&FBC$ + 136ll);
+			if( TMP$651$2 != 0ll ) goto label$1114;
 			label$1115:;
 			{
 				FBSTRING* vr$24 = FBGETHOSTID(  );
@@ -376,7 +376,7 @@ int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 			}
 			goto label$1113;
 			label$1114:;
-			if( TMP$648$2 != 1ll ) goto label$1116;
+			if( TMP$651$2 != 1ll ) goto label$1116;
 			label$1117:;
 			{
 				FBSTRING* vr$25 = FBGETTARGETID(  );
@@ -384,7 +384,7 @@ int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 			}
 			goto label$1113;
 			label$1116:;
-			if( TMP$648$2 != 2ll ) goto label$1118;
+			if( TMP$651$2 != 2ll ) goto label$1118;
 			label$1119:;
 			{
 				if( HAVE_INPUT_FILES$0 == 0ll ) goto label$1121;
@@ -399,7 +399,7 @@ int32 main( int32 __FB_ARGC__$0, char** __FB_ARGV__$0 )
 			}
 			goto label$1113;
 			label$1118:;
-			if( TMP$648$2 != 3ll ) goto label$1122;
+			if( TMP$651$2 != 3ll ) goto label$1122;
 			label$1123:;
 			{
 				FBSTRING* vr$29 = fb_StrAllocTempDescZ( (uint8*)((uint8*)&FBC$ + 3359ll) );
@@ -1609,7 +1609,7 @@ static int64 HLINKFILES( void )
 					}
 					goto label$189;
 					label$190:;
-					static const void* tmp$649[7ll] = {
+					static const void* tmp$652[7ll] = {
 						&&label$192,
 						&&label$189,
 						&&label$189,
@@ -1619,7 +1619,7 @@ static int64 HLINKFILES( void )
 						&&label$202,
 					};
 					if( (TMP$126$5 - 2ull) > 6ull ) goto label$189;
-					goto *tmp$649[TMP$126$5 - 2ull];
+					goto *tmp$652[TMP$126$5 - 2ull];
 					label$189:;
 				}
 			}
@@ -1640,7 +1640,7 @@ static int64 HLINKFILES( void )
 		}
 		goto label$173;
 		label$174:;
-		static const void* tmp$650[9ll] = {
+		static const void* tmp$653[9ll] = {
 			&&label$175,
 			&&label$175,
 			&&label$184,
@@ -1652,7 +1652,7 @@ static int64 HLINKFILES( void )
 			&&label$184,
 		};
 		if( TMP$113$2 > 8ull ) goto label$173;
-		goto *tmp$650[TMP$113$2 - 0ull];
+		goto *tmp$653[TMP$113$2 - 0ull];
 		label$173:;
 	}
 	int64 vr$115 = FBGETOPTION( 3ll );
@@ -1718,7 +1718,7 @@ static int64 HLINKFILES( void )
 		}
 		goto label$211;
 		label$212:;
-		static const void* tmp$651[5ll] = {
+		static const void* tmp$654[5ll] = {
 			&&label$213,
 			&&label$213,
 			&&label$211,
@@ -1726,7 +1726,7 @@ static int64 HLINKFILES( void )
 			&&label$216,
 		};
 		if( TMP$142$2 > 4ull ) goto label$211;
-		goto *tmp$651[TMP$142$2 - 0ull];
+		goto *tmp$654[TMP$142$2 - 0ull];
 		label$211:;
 	}
 	if( *(int64*)((uint8*)&FBC$ + 2568ll) == 0ll ) goto label$218;
@@ -1885,13 +1885,13 @@ static int64 HLINKFILES( void )
 						}
 						goto label$249;
 						label$250:;
-						static const void* tmp$652[3ll] = {
+						static const void* tmp$655[3ll] = {
 							&&label$251,
 							&&label$252,
 							&&label$251,
 						};
 						if( (TMP$163$6 - 6ull) > 2ull ) goto label$252;
-						goto *tmp$652[TMP$163$6 - 6ull];
+						goto *tmp$655[TMP$163$6 - 6ull];
 						label$249:;
 					}
 				}
@@ -1916,13 +1916,13 @@ static int64 HLINKFILES( void )
 						}
 						goto label$253;
 						label$254:;
-						static const void* tmp$653[3ll] = {
+						static const void* tmp$656[3ll] = {
 							&&label$255,
 							&&label$256,
 							&&label$255,
 						};
 						if( (TMP$165$6 - 6ull) > 2ull ) goto label$256;
-						goto *tmp$653[TMP$165$6 - 6ull];
+						goto *tmp$656[TMP$165$6 - 6ull];
 						label$253:;
 					}
 				}
@@ -1966,7 +1966,7 @@ static int64 HLINKFILES( void )
 		}
 		goto label$229;
 		label$230:;
-		static const void* tmp$654[9ll] = {
+		static const void* tmp$657[9ll] = {
 			&&label$236,
 			&&label$231,
 			&&label$244,
@@ -1978,7 +1978,7 @@ static int64 HLINKFILES( void )
 			&&label$244,
 		};
 		if( TMP$156$2 > 8ull ) goto label$229;
-		goto *tmp$654[TMP$156$2 - 0ull];
+		goto *tmp$657[TMP$156$2 - 0ull];
 		label$229:;
 	}
 	if( *(int64*)((uint8*)&FBC$ + 2560ll) != 0ll ) goto label$265;
@@ -2134,7 +2134,7 @@ static int64 HLINKFILES( void )
 		}
 		goto label$281;
 		label$282:;
-		static const void* tmp$655[9ll] = {
+		static const void* tmp$658[9ll] = {
 			&&label$288,
 			&&label$281,
 			&&label$283,
@@ -2146,7 +2146,7 @@ static int64 HLINKFILES( void )
 			&&label$283,
 		};
 		if( TMP$183$2 > 8ull ) goto label$281;
-		goto *tmp$655[TMP$183$2 - 0ull];
+		goto *tmp$658[TMP$183$2 - 0ull];
 		label$281:;
 	}
 	int64 vr$352 = FBGETOPTION( 3ll );
@@ -2325,7 +2325,7 @@ static int64 HLINKFILES( void )
 		}
 		goto label$293;
 		label$294:;
-		static const void* tmp$656[5ll] = {
+		static const void* tmp$659[5ll] = {
 			&&label$300,
 			&&label$300,
 			&&label$293,
@@ -2333,7 +2333,7 @@ static int64 HLINKFILES( void )
 			&&label$305,
 		};
 		if( TMP$189$2 > 4ull ) goto label$293;
-		goto *tmp$656[TMP$189$2 - 0ull];
+		goto *tmp$659[TMP$189$2 - 0ull];
 		label$293:;
 	}
 	fb$result$1 = -1ll;
@@ -2412,7 +2412,7 @@ static void HREADOBJINFO( void )
 			}
 			goto label$323;
 			label$324:;
-			static const void* tmp$657[5ll] = {
+			static const void* tmp$660[5ll] = {
 				&&label$325,
 				&&label$326,
 				&&label$327,
@@ -2420,7 +2420,7 @@ static void HREADOBJINFO( void )
 				&&label$331,
 			};
 			if( TMP$212$3 > 4ull ) goto label$336;
-			goto *tmp$657[TMP$212$3 - 0ull];
+			goto *tmp$660[TMP$212$3 - 0ull];
 			label$323:;
 		}
 	}
@@ -3416,36 +3416,45 @@ static void HANDLEOPT( int64 OPTID$1, FBSTRING* ARG$1 )
 		{
 			FBSTRING TMP$294$3;
 			FBSTRING TMP$295$3;
+			FBSTRING TMP$296$3;
 			FBSTRING* vr$199 = HREPLACE( *(uint8**)ARG$1, (uint8*)",", (uint8*)" " );
 			__builtin_memset( &TMP$294$3, 0, 24ll );
 			FBSTRING* vr$202 = fb_StrConcat( &TMP$294$3, (void*)" ", 2ll, (void*)vr$199, -1ll );
 			__builtin_memset( &TMP$295$3, 0, 24ll );
 			FBSTRING* vr$205 = fb_StrConcat( &TMP$295$3, (void*)vr$202, -1ll, (void*)" ", 2ll );
-			fb_StrAssign( (void*)((uint8*)&FBC$ + 1782ll), 128ll, (void*)vr$205, -1ll, 0 );
+			__builtin_memset( &TMP$296$3, 0, 24ll );
+			FBSTRING* vr$209 = fb_StrConcat( &TMP$296$3, (void*)((uint8*)&FBC$ + 1782ll), 128ll, (void*)vr$205, -1ll );
+			fb_StrAssign( (void*)((uint8*)&FBC$ + 1782ll), 128ll, (void*)vr$209, -1ll, 0 );
 		}
 		goto label$391;
 		label$540:;
 		{
-			FBSTRING TMP$296$3;
 			FBSTRING TMP$297$3;
-			FBSTRING* vr$208 = HREPLACE( *(uint8**)ARG$1, (uint8*)",", (uint8*)" " );
-			__builtin_memset( &TMP$296$3, 0, 24ll );
-			FBSTRING* vr$211 = fb_StrConcat( &TMP$296$3, (void*)" ", 2ll, (void*)vr$208, -1ll );
+			FBSTRING TMP$298$3;
+			FBSTRING TMP$299$3;
+			FBSTRING* vr$212 = HREPLACE( *(uint8**)ARG$1, (uint8*)",", (uint8*)" " );
 			__builtin_memset( &TMP$297$3, 0, 24ll );
-			FBSTRING* vr$214 = fb_StrConcat( &TMP$297$3, (void*)vr$211, -1ll, (void*)" ", 2ll );
-			fb_StrAssign( (void*)((uint8*)&FBC$ + 2038ll), 128ll, (void*)vr$214, -1ll, 0 );
+			FBSTRING* vr$215 = fb_StrConcat( &TMP$297$3, (void*)" ", 2ll, (void*)vr$212, -1ll );
+			__builtin_memset( &TMP$298$3, 0, 24ll );
+			FBSTRING* vr$218 = fb_StrConcat( &TMP$298$3, (void*)vr$215, -1ll, (void*)" ", 2ll );
+			__builtin_memset( &TMP$299$3, 0, 24ll );
+			FBSTRING* vr$222 = fb_StrConcat( &TMP$299$3, (void*)((uint8*)&FBC$ + 2038ll), 128ll, (void*)vr$218, -1ll );
+			fb_StrAssign( (void*)((uint8*)&FBC$ + 2038ll), 128ll, (void*)vr$222, -1ll, 0 );
 		}
 		goto label$391;
 		label$541:;
 		{
-			FBSTRING TMP$298$3;
-			FBSTRING TMP$299$3;
-			FBSTRING* vr$217 = HREPLACE( *(uint8**)ARG$1, (uint8*)",", (uint8*)" " );
-			__builtin_memset( &TMP$298$3, 0, 24ll );
-			FBSTRING* vr$220 = fb_StrConcat( &TMP$298$3, (void*)" ", 2ll, (void*)vr$217, -1ll );
-			__builtin_memset( &TMP$299$3, 0, 24ll );
-			FBSTRING* vr$223 = fb_StrConcat( &TMP$299$3, (void*)vr$220, -1ll, (void*)" ", 2ll );
-			fb_StrAssign( (void*)((uint8*)&FBC$ + 1910ll), 128ll, (void*)vr$223, -1ll, 0 );
+			FBSTRING TMP$300$3;
+			FBSTRING TMP$301$3;
+			FBSTRING TMP$302$3;
+			FBSTRING* vr$225 = HREPLACE( *(uint8**)ARG$1, (uint8*)",", (uint8*)" " );
+			__builtin_memset( &TMP$300$3, 0, 24ll );
+			FBSTRING* vr$228 = fb_StrConcat( &TMP$300$3, (void*)" ", 2ll, (void*)vr$225, -1ll );
+			__builtin_memset( &TMP$301$3, 0, 24ll );
+			FBSTRING* vr$231 = fb_StrConcat( &TMP$301$3, (void*)vr$228, -1ll, (void*)" ", 2ll );
+			__builtin_memset( &TMP$302$3, 0, 24ll );
+			FBSTRING* vr$235 = fb_StrConcat( &TMP$302$3, (void*)((uint8*)&FBC$ + 1910ll), 128ll, (void*)vr$231, -1ll );
+			fb_StrAssign( (void*)((uint8*)&FBC$ + 1910ll), 128ll, (void*)vr$235, -1ll, 0 );
 		}
 		goto label$391;
 		label$542:;
@@ -3456,11 +3465,11 @@ static void HANDLEOPT( int64 OPTID$1, FBSTRING* ARG$1 )
 		label$543:;
 		{
 			{
-				FBSTRING TMP$300$4;
-				FBSTRING* vr$226 = fb_StrLcase2( ARG$1, 0 );
-				FBSTRING* vr$228 = fb_StrInit( (void*)&TMP$300$4, -1ll, (void*)vr$226, -1ll, 0 );
-				int32 vr$230 = fb_StrCompare( (void*)&TMP$300$4, -1ll, (void*)"gosub-setjmp", 13ll );
-				if( (int64)vr$230 != 0ll ) goto label$545;
+				FBSTRING TMP$303$4;
+				FBSTRING* vr$238 = fb_StrLcase2( ARG$1, 0 );
+				FBSTRING* vr$240 = fb_StrInit( (void*)&TMP$303$4, -1ll, (void*)vr$238, -1ll, 0 );
+				int32 vr$242 = fb_StrCompare( (void*)&TMP$303$4, -1ll, (void*)"gosub-setjmp", 13ll );
+				if( (int64)vr$242 != 0ll ) goto label$545;
 				label$546:;
 				{
 					FBSETOPTION( 22ll, -1ll );
@@ -3472,12 +3481,12 @@ static void HANDLEOPT( int64 OPTID$1, FBSTRING* ARG$1 )
 				}
 				label$547:;
 				label$544:;
-				fb_StrDelete( &TMP$300$4 );
+				fb_StrDelete( &TMP$303$4 );
 			}
 		}
 		goto label$391;
 		label$392:;
-		static const void* tmp$658[58ll] = {
+		static const void* tmp$661[58ll] = {
 			&&label$393,
 			&&label$394,
 			&&label$397,
@@ -3538,7 +3547,7 @@ static void HANDLEOPT( int64 OPTID$1, FBSTRING* ARG$1 )
 			&&label$543,
 		};
 		if( TMP$252$2 > 57ull ) goto label$391;
-		goto *tmp$658[TMP$252$2 - 0ull];
+		goto *tmp$661[TMP$252$2 - 0ull];
 		label$391:;
 	}
 	label$390:;
@@ -3550,8 +3559,8 @@ static int64 PARSEOPTION( uint8* OPT$1 )
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$548:;
 	{
-		uint64 TMP$302$2;
-		TMP$302$2 = (uint64)*(uint8*)OPT$1;
+		uint64 TMP$305$2;
+		TMP$305$2 = (uint64)*(uint8*)OPT$1;
 		goto label$551;
 		label$552:;
 		{
@@ -4117,7 +4126,7 @@ static int64 PARSEOPTION( uint8* OPT$1 )
 		}
 		goto label$550;
 		label$551:;
-		static const void* tmp$659[78ll] = {
+		static const void* tmp$662[78ll] = {
 			&&label$693,
 			&&label$550,
 			&&label$550,
@@ -4197,8 +4206,8 @@ static int64 PARSEOPTION( uint8* OPT$1 )
 			&&label$550,
 			&&label$690,
 		};
-		if( (TMP$302$2 - 45ull) > 77ull ) goto label$550;
-		goto *tmp$659[TMP$302$2 - 45ull];
+		if( (TMP$305$2 - 45ull) > 77ull ) goto label$550;
+		goto *tmp$662[TMP$305$2 - 45ull];
 		label$550:;
 	}
 	fb$result$1 = -1ll;
@@ -4233,9 +4242,9 @@ static void HANDLEARG( FBSTRING* ARG$1 )
 	label$705:;
 	label$704:;
 	{
-		uint8 TMP$364$2;
-		TMP$364$2 = *(uint8*)*(uint8**)ARG$1;
-		if( (uint64)(int64)TMP$364$2 != 45ull ) goto label$707;
+		uint8 TMP$367$2;
+		TMP$367$2 = *(uint8*)*(uint8**)ARG$1;
+		if( (uint64)(int64)TMP$367$2 != 45ull ) goto label$707;
 		label$708:;
 		{
 			uint8* OPT$3;
@@ -4268,7 +4277,7 @@ static void HANDLEARG( FBSTRING* ARG$1 )
 		}
 		goto label$706;
 		label$707:;
-		if( (uint64)(int64)TMP$364$2 != 64ull ) goto label$715;
+		if( (uint64)(int64)TMP$367$2 != 64ull ) goto label$715;
 		label$716:;
 		{
 			static int64 RECLEVEL$3 = 0ll;
@@ -4411,8 +4420,8 @@ static void PARSEARGSFROMFILE( FBSTRING* FILENAME$1 )
 				int64 CH$4;
 				CH$4 = (int64)*(uint8*)((uint8*)*(uint8**)&ARGS$1 + I$3);
 				{
-					uint64 TMP$370$5;
-					TMP$370$5 = (uint64)CH$4;
+					uint64 TMP$373$5;
+					TMP$373$5 = (uint64)CH$4;
 					goto label$751;
 					label$752:;
 					{
@@ -4441,7 +4450,7 @@ static void PARSEARGSFROMFILE( FBSTRING* FILENAME$1 )
 					}
 					goto label$750;
 					label$751:;
-					static const void* tmp$660[8ll] = {
+					static const void* tmp$663[8ll] = {
 						&&label$752,
 						&&label$750,
 						&&label$755,
@@ -4451,8 +4460,8 @@ static void PARSEARGSFROMFILE( FBSTRING* FILENAME$1 )
 						&&label$750,
 						&&label$755,
 					};
-					if( (TMP$370$5 - 32ull) > 7ull ) goto label$750;
-					goto *tmp$660[TMP$370$5 - 32ull];
+					if( (TMP$373$5 - 32ull) > 7ull ) goto label$750;
+					goto *tmp$663[TMP$373$5 - 32ull];
 					label$750:;
 				}
 				I$3 = I$3 + 1ll;
@@ -4504,9 +4513,9 @@ static int64 HTARGETNEEDSPIC( void )
 	if( vr$1 == 0ll ) goto label$765;
 	{
 		{
-			uint64 TMP$372$3;
+			uint64 TMP$375$3;
 			int64 vr$2 = FBGETOPTION( 3ll );
-			TMP$372$3 = (uint64)vr$2;
+			TMP$375$3 = (uint64)vr$2;
 			goto label$767;
 			label$768:;
 			{
@@ -4514,7 +4523,7 @@ static int64 HTARGETNEEDSPIC( void )
 			}
 			goto label$766;
 			label$767:;
-			static const void* tmp$661[7ll] = {
+			static const void* tmp$664[7ll] = {
 				&&label$768,
 				&&label$766,
 				&&label$766,
@@ -4523,8 +4532,8 @@ static int64 HTARGETNEEDSPIC( void )
 				&&label$766,
 				&&label$768,
 			};
-			if( (TMP$372$3 - 2ull) > 6ull ) goto label$766;
-			goto *tmp$661[TMP$372$3 - 2ull];
+			if( (TMP$375$3 - 2ull) > 6ull ) goto label$766;
+			goto *tmp$664[TMP$375$3 - 2ull];
 			label$766:;
 		}
 	}
@@ -4543,8 +4552,8 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 	{
 		int64 I$2;
 		I$2 = 1ll;
-		int64 TMP$373$2;
-		TMP$373$2 = ARGC$1 + -1ll;
+		int64 TMP$376$2;
+		TMP$376$2 = ARGC$1 + -1ll;
 		goto label$771;
 		label$774:;
 		{
@@ -4554,16 +4563,16 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 		label$772:;
 		I$2 = I$2 + 1ll;
 		label$771:;
-		if( I$2 <= TMP$373$2 ) goto label$774;
+		if( I$2 <= TMP$376$2 ) goto label$774;
 		label$773:;
 	}
 	if( *(int64*)&FBC$ < 0ll ) goto label$776;
 	{
-		FBSTRING TMP$374$2;
-		__builtin_memset( &TMP$374$2, 0, 24ll );
-		FBSTRING* vr$11 = fb_StrAssign( (void*)&TMP$374$2, -1ll, *(void**)((uint8*)((uint8*)ARGV$1 + (ARGC$1 << (3ll & 63ll))) + -8ll), 0ll, 0 );
-		HFATALINVALIDOPTION( &TMP$374$2 );
-		fb_StrDelete( &TMP$374$2 );
+		FBSTRING TMP$377$2;
+		__builtin_memset( &TMP$377$2, 0, 24ll );
+		FBSTRING* vr$11 = fb_StrAssign( (void*)&TMP$377$2, -1ll, *(void**)((uint8*)((uint8*)ARGV$1 + (ARGC$1 << (3ll & 63ll))) + -8ll), 0ll, 0 );
+		HFATALINVALIDOPTION( &TMP$377$2 );
+		fb_StrDelete( &TMP$377$2 );
 	}
 	label$776:;
 	label$775:;
@@ -4627,9 +4636,9 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 	label$790:;
 	label$789:;
 	{
-		uint64 TMP$375$2;
+		uint64 TMP$378$2;
 		int64 vr$33 = FBGETOPTION( 3ll );
-		TMP$375$2 = (uint64)vr$33;
+		TMP$378$2 = (uint64)vr$33;
 		goto label$792;
 		label$793:;
 		{
@@ -4650,21 +4659,21 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 		}
 		goto label$791;
 		label$792:;
-		static const void* tmp$662[5ll] = {
+		static const void* tmp$665[5ll] = {
 			&&label$793,
 			&&label$793,
 			&&label$794,
 			&&label$794,
 			&&label$793,
 		};
-		if( TMP$375$2 > 4ull ) goto label$794;
-		goto *tmp$662[TMP$375$2 - 0ull];
+		if( TMP$378$2 > 4ull ) goto label$794;
+		goto *tmp$665[TMP$378$2 - 0ull];
 		label$791:;
 	}
 	{
-		uint64 TMP$376$2;
+		uint64 TMP$379$2;
 		int64 vr$37 = FBGETOPTION( 3ll );
-		TMP$376$2 = (uint64)vr$37;
+		TMP$379$2 = (uint64)vr$37;
 		goto label$798;
 		label$799:;
 		{
@@ -4683,7 +4692,7 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 		}
 		goto label$797;
 		label$798:;
-		static const void* tmp$663[7ll] = {
+		static const void* tmp$666[7ll] = {
 			&&label$799,
 			&&label$800,
 			&&label$800,
@@ -4692,31 +4701,31 @@ static void HPARSEARGS( int64 ARGC$1, uint8** ARGV$1 )
 			&&label$799,
 			&&label$799,
 		};
-		if( (TMP$376$2 - 2ull) > 6ull ) goto label$800;
-		goto *tmp$663[TMP$376$2 - 2ull];
+		if( (TMP$379$2 - 2ull) > 6ull ) goto label$800;
+		goto *tmp$666[TMP$379$2 - 2ull];
 		label$797:;
 	}
 	if( *(int64*)((uint8*)&FBC$ + 64ll) < 0ll ) goto label$804;
 	{
 		{
-			int64 TMP$377$3;
+			int64 TMP$380$3;
 			int64 vr$40 = FBGETCPUFAMILY(  );
-			TMP$377$3 = vr$40;
-			if( TMP$377$3 == 0ll ) goto label$807;
+			TMP$380$3 = vr$40;
+			if( TMP$380$3 == 0ll ) goto label$807;
 			label$808:;
-			if( TMP$377$3 != 1ll ) goto label$806;
+			if( TMP$380$3 != 1ll ) goto label$806;
 			label$807:;
 			{
 			}
 			goto label$805;
 			label$806:;
 			{
-				FBSTRING TMP$378$4;
-				__builtin_memset( &TMP$378$4, 0, 24ll );
+				FBSTRING TMP$381$4;
+				__builtin_memset( &TMP$381$4, 0, 24ll );
 				FBSTRING* vr$42 = FBGETTARGETID(  );
-				FBSTRING* vr$44 = fb_StrAssign( (void*)&TMP$378$4, -1ll, (void*)vr$42, -1ll, 0 );
-				ERRREPORTEX( 319ll, *(uint8**)&TMP$378$4, -1ll, 1ll, (uint8*)0ull );
-				fb_StrDelete( &TMP$378$4 );
+				FBSTRING* vr$44 = fb_StrAssign( (void*)&TMP$381$4, -1ll, (void*)vr$42, -1ll, 0 );
+				ERRREPORTEX( 319ll, *(uint8**)&TMP$381$4, -1ll, 1ll, (uint8*)0ull );
+				fb_StrDelete( &TMP$381$4 );
 			}
 			label$809:;
 			label$805:;
@@ -4775,27 +4784,27 @@ static void FBCDETERMINEPREFIX( void )
 	int64 vr$1 = fb_StrLen( (void*)((uint8*)&FBC$ + 2576ll), 261ll );
 	if( vr$1 != 0ll ) goto label$824;
 	{
-		FBSTRING TMP$379$2;
-		FBSTRING TMP$380$2;
+		FBSTRING TMP$382$2;
 		FBSTRING TMP$383$2;
-		__builtin_memset( &TMP$379$2, 0, 24ll );
+		FBSTRING TMP$386$2;
+		__builtin_memset( &TMP$382$2, 0, 24ll );
 		FBSTRING* vr$3 = fb_ExePath(  );
-		FBSTRING* vr$5 = fb_StrAssign( (void*)&TMP$379$2, -1ll, (void*)vr$3, -1ll, 0 );
-		FBSTRING* vr$7 = PATHSTRIPDIV( &TMP$379$2 );
-		__builtin_memset( &TMP$380$2, 0, 24ll );
-		FBSTRING* vr$10 = fb_StrConcat( &TMP$380$2, (void*)vr$7, -1ll, (void*)"/", 2ll );
-		fb_StrAssign( (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)vr$10, -1ll, 0 );
-		fb_StrDelete( &TMP$379$2 );
+		FBSTRING* vr$5 = fb_StrAssign( (void*)&TMP$382$2, -1ll, (void*)vr$3, -1ll, 0 );
+		FBSTRING* vr$7 = PATHSTRIPDIV( &TMP$382$2 );
 		__builtin_memset( &TMP$383$2, 0, 24ll );
-		FBSTRING* vr$16 = fb_StrConcat( &TMP$383$2, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"../", 4ll );
+		FBSTRING* vr$10 = fb_StrConcat( &TMP$383$2, (void*)vr$7, -1ll, (void*)"/", 2ll );
+		fb_StrAssign( (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)vr$10, -1ll, 0 );
+		fb_StrDelete( &TMP$382$2 );
+		__builtin_memset( &TMP$386$2, 0, 24ll );
+		FBSTRING* vr$16 = fb_StrConcat( &TMP$386$2, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"../", 4ll );
 		fb_StrAssign( (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)vr$16, -1ll, 0 );
 	}
 	goto label$823;
 	label$824:;
 	{
-		FBSTRING TMP$384$2;
-		__builtin_memset( &TMP$384$2, 0, 24ll );
-		FBSTRING* vr$21 = fb_StrConcat( &TMP$384$2, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"/", 2ll );
+		FBSTRING TMP$387$2;
+		__builtin_memset( &TMP$387$2, 0, 24ll );
+		FBSTRING* vr$21 = fb_StrConcat( &TMP$387$2, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"/", 2ll );
 		fb_StrAssign( (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)vr$21, -1ll, 0 );
 	}
 	label$823:;
@@ -4804,9 +4813,6 @@ static void FBCDETERMINEPREFIX( void )
 
 static void FBCSETUPCOMPILERPATHS( void )
 {
-	FBSTRING TMP$387$1;
-	FBSTRING TMP$388$1;
-	FBSTRING TMP$389$1;
 	FBSTRING TMP$390$1;
 	FBSTRING TMP$391$1;
 	FBSTRING TMP$392$1;
@@ -4815,6 +4821,9 @@ static void FBCSETUPCOMPILERPATHS( void )
 	FBSTRING TMP$395$1;
 	FBSTRING TMP$396$1;
 	FBSTRING TMP$397$1;
+	FBSTRING TMP$398$1;
+	FBSTRING TMP$399$1;
+	FBSTRING TMP$400$1;
 	label$825:;
 	FBSTRING TARGETID$1;
 	FBSTRING* vr$0 = FBGETTARGETID(  );
@@ -4824,30 +4833,30 @@ static void FBCSETUPCOMPILERPATHS( void )
 	fb_StrAssign( (void*)&FBNAME$1, -1ll, (void*)"freebasic", 10ll, 0 );
 	FBSTRING LIBDIRNAME$1;
 	FBSTRING* vr$6 = fb_StrInit( (void*)&LIBDIRNAME$1, -1ll, (void*)"lib", 4ll, 0 );
-	__builtin_memset( &TMP$387$1, 0, 24ll );
-	FBSTRING* vr$11 = fb_StrConcat( &TMP$387$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"bin", 4ll );
-	__builtin_memset( &TMP$388$1, 0, 24ll );
-	FBSTRING* vr$14 = fb_StrConcat( &TMP$388$1, (void*)vr$11, -1ll, (void*)"/", 2ll );
-	__builtin_memset( &TMP$389$1, 0, 24ll );
-	FBSTRING* vr$17 = fb_StrConcat( &TMP$389$1, (void*)vr$14, -1ll, (void*)((uint8*)&FBC$ + 2295ll), 129ll );
-	fb_StrAssign( (void*)((uint8*)&FBC$ + 2837ll), 261ll, (void*)vr$17, -1ll, 0 );
 	__builtin_memset( &TMP$390$1, 0, 24ll );
-	FBSTRING* vr$23 = fb_StrConcat( &TMP$390$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"include", 8ll );
+	FBSTRING* vr$11 = fb_StrConcat( &TMP$390$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"bin", 4ll );
 	__builtin_memset( &TMP$391$1, 0, 24ll );
-	FBSTRING* vr$26 = fb_StrConcat( &TMP$391$1, (void*)vr$23, -1ll, (void*)"/", 2ll );
+	FBSTRING* vr$14 = fb_StrConcat( &TMP$391$1, (void*)vr$11, -1ll, (void*)"/", 2ll );
 	__builtin_memset( &TMP$392$1, 0, 24ll );
-	FBSTRING* vr$29 = fb_StrConcat( &TMP$392$1, (void*)vr$26, -1ll, (void*)&FBNAME$1, -1ll );
-	fb_StrAssign( (void*)((uint8*)&FBC$ + 3098ll), 261ll, (void*)vr$29, -1ll, 0 );
+	FBSTRING* vr$17 = fb_StrConcat( &TMP$392$1, (void*)vr$14, -1ll, (void*)((uint8*)&FBC$ + 2295ll), 129ll );
+	fb_StrAssign( (void*)((uint8*)&FBC$ + 2837ll), 261ll, (void*)vr$17, -1ll, 0 );
 	__builtin_memset( &TMP$393$1, 0, 24ll );
-	FBSTRING* vr$37 = fb_StrConcat( &TMP$393$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)&LIBDIRNAME$1, -1ll );
+	FBSTRING* vr$23 = fb_StrConcat( &TMP$393$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)"include", 8ll );
 	__builtin_memset( &TMP$394$1, 0, 24ll );
-	FBSTRING* vr$40 = fb_StrConcat( &TMP$394$1, (void*)vr$37, -1ll, (void*)"/", 2ll );
+	FBSTRING* vr$26 = fb_StrConcat( &TMP$394$1, (void*)vr$23, -1ll, (void*)"/", 2ll );
 	__builtin_memset( &TMP$395$1, 0, 24ll );
-	FBSTRING* vr$43 = fb_StrConcat( &TMP$395$1, (void*)vr$40, -1ll, (void*)&FBNAME$1, -1ll );
+	FBSTRING* vr$29 = fb_StrConcat( &TMP$395$1, (void*)vr$26, -1ll, (void*)&FBNAME$1, -1ll );
+	fb_StrAssign( (void*)((uint8*)&FBC$ + 3098ll), 261ll, (void*)vr$29, -1ll, 0 );
 	__builtin_memset( &TMP$396$1, 0, 24ll );
-	FBSTRING* vr$46 = fb_StrConcat( &TMP$396$1, (void*)vr$43, -1ll, (void*)"/", 2ll );
+	FBSTRING* vr$37 = fb_StrConcat( &TMP$396$1, (void*)((uint8*)&FBC$ + 2576ll), 261ll, (void*)&LIBDIRNAME$1, -1ll );
 	__builtin_memset( &TMP$397$1, 0, 24ll );
-	FBSTRING* vr$49 = fb_StrConcat( &TMP$397$1, (void*)vr$46, -1ll, (void*)&TARGETID$1, -1ll );
+	FBSTRING* vr$40 = fb_StrConcat( &TMP$397$1, (void*)vr$37, -1ll, (void*)"/", 2ll );
+	__builtin_memset( &TMP$398$1, 0, 24ll );
+	FBSTRING* vr$43 = fb_StrConcat( &TMP$398$1, (void*)vr$40, -1ll, (void*)&FBNAME$1, -1ll );
+	__builtin_memset( &TMP$399$1, 0, 24ll );
+	FBSTRING* vr$46 = fb_StrConcat( &TMP$399$1, (void*)vr$43, -1ll, (void*)"/", 2ll );
+	__builtin_memset( &TMP$400$1, 0, 24ll );
+	FBSTRING* vr$49 = fb_StrConcat( &TMP$400$1, (void*)vr$46, -1ll, (void*)&TARGETID$1, -1ll );
 	fb_StrAssign( (void*)((uint8*)&FBC$ + 3359ll), 261ll, (void*)vr$49, -1ll, 0 );
 	fb_StrDelete( &LIBDIRNAME$1 );
 	fb_StrDelete( &FBNAME$1 );
@@ -4857,17 +4866,17 @@ static void FBCSETUPCOMPILERPATHS( void )
 
 static void FBCPRINTTARGETINFO( void )
 {
-	FBSTRING TMP$399$1;
-	FBSTRING TMP$400$1;
+	FBSTRING TMP$402$1;
+	FBSTRING TMP$403$1;
 	label$827:;
 	FBSTRING S$1;
 	FBSTRING* vr$0 = FBGETTARGETID(  );
 	FBSTRING* vr$2 = fb_StrInit( (void*)&S$1, -1ll, (void*)vr$0, -1ll, 0 );
 	uint8* vr$3 = FBGETFBCARCH(  );
-	__builtin_memset( &TMP$399$1, 0, 24ll );
-	FBSTRING* vr$6 = fb_StrConcat( &TMP$399$1, (void*)", ", 3ll, (void*)vr$3, 0ll );
-	__builtin_memset( &TMP$400$1, 0, 24ll );
-	FBSTRING* vr$10 = fb_StrConcat( &TMP$400$1, (void*)&S$1, -1ll, (void*)vr$6, -1ll );
+	__builtin_memset( &TMP$402$1, 0, 24ll );
+	FBSTRING* vr$6 = fb_StrConcat( &TMP$402$1, (void*)", ", 3ll, (void*)vr$3, 0ll );
+	__builtin_memset( &TMP$403$1, 0, 24ll );
+	FBSTRING* vr$10 = fb_StrConcat( &TMP$403$1, (void*)&S$1, -1ll, (void*)vr$6, -1ll );
 	FBSTRING* vr$12 = fb_StrAssign( (void*)&S$1, -1ll, (void*)vr$10, -1ll, 0 );
 	FBSTRING* vr$14 = fb_StrConcatAssign( (void*)&S$1, -1ll, (void*)", ", 3ll, 0 );
 	int64 vr$15 = FBGETBITS(  );
@@ -4896,7 +4905,7 @@ static void FBCDETERMINEMAINNAME( void )
 	int64 vr$1 = fb_StrLen( (void*)((uint8*)&FBC$ + 1117ll), 261ll );
 	if( vr$1 != 0ll ) goto label$834;
 	{
-		FBSTRING TMP$406$2;
+		FBSTRING TMP$409$2;
 		struct $9FBCIOFILE* M$2;
 		void* vr$3 = LISTGETHEAD( (struct $5TLIST*)((uint8*)&FBC$ + 144ll) );
 		M$2 = (struct $9FBCIOFILE*)vr$3;
@@ -4922,11 +4931,11 @@ static void FBCDETERMINEMAINNAME( void )
 			label$837:;
 		}
 		label$835:;
-		__builtin_memset( &TMP$406$2, 0, 24ll );
-		FBSTRING* vr$14 = fb_StrAssign( (void*)&TMP$406$2, -1ll, (void*)((uint8*)&FBC$ + 1117ll), 261ll, 0 );
-		FBSTRING* vr$16 = HSTRIPEXT( &TMP$406$2 );
+		__builtin_memset( &TMP$409$2, 0, 24ll );
+		FBSTRING* vr$14 = fb_StrAssign( (void*)&TMP$409$2, -1ll, (void*)((uint8*)&FBC$ + 1117ll), 261ll, 0 );
+		FBSTRING* vr$16 = HSTRIPEXT( &TMP$409$2 );
 		fb_StrAssign( (void*)((uint8*)&FBC$ + 1117ll), 261ll, (void*)vr$16, -1ll, 0 );
-		fb_StrDelete( &TMP$406$2 );
+		fb_StrDelete( &TMP$409$2 );
 	}
 	label$834:;
 	label$833:;
@@ -4935,7 +4944,7 @@ static void FBCDETERMINEMAINNAME( void )
 
 static FBSTRING* HGETASMNAME( struct $9FBCIOFILE* MODULE$1, int64 STAGE$1 )
 {
-	FBSTRING TMP$411$1;
+	FBSTRING TMP$414$1;
 	FBSTRING fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 24ll );
 	label$839:;
@@ -4948,17 +4957,17 @@ static FBSTRING* HGETASMNAME( struct $9FBCIOFILE* MODULE$1, int64 STAGE$1 )
 	if( STAGE$1 != 1ll ) goto label$842;
 	{
 		{
-			int64 TMP$408$3;
+			int64 TMP$411$3;
 			int64 vr$6 = FBGETOPTION( 2ll );
-			TMP$408$3 = vr$6;
-			if( TMP$408$3 != 1ll ) goto label$844;
+			TMP$411$3 = vr$6;
+			if( TMP$411$3 != 1ll ) goto label$844;
 			label$845:;
 			{
 				EXT$1 = (uint8*)".c";
 			}
 			goto label$843;
 			label$844:;
-			if( TMP$408$3 != 2ll ) goto label$846;
+			if( TMP$411$3 != 2ll ) goto label$846;
 			label$847:;
 			{
 				EXT$1 = (uint8*)".ll";
@@ -4969,8 +4978,8 @@ static FBSTRING* HGETASMNAME( struct $9FBCIOFILE* MODULE$1, int64 STAGE$1 )
 	}
 	label$842:;
 	label$841:;
-	__builtin_memset( &TMP$411$1, 0, 24ll );
-	FBSTRING* vr$10 = fb_StrConcat( &TMP$411$1, (void*)&ASMFILE$1, -1ll, (void*)EXT$1, 0ll );
+	__builtin_memset( &TMP$414$1, 0, 24ll );
+	FBSTRING* vr$10 = fb_StrConcat( &TMP$414$1, (void*)&ASMFILE$1, -1ll, (void*)EXT$1, 0ll );
 	FBSTRING* vr$12 = fb_StrAssign( (void*)&ASMFILE$1, -1ll, (void*)vr$10, -1ll, 0 );
 	FBSTRING* vr$15 = fb_StrAssign( (void*)&fb$result$1, -1ll, (void*)&ASMFILE$1, -1ll, 0 );
 	fb_StrDelete( &ASMFILE$1 );
@@ -5004,10 +5013,10 @@ static void HCOMPILEBAS( struct $9FBCIOFILE* MODULE$1, int64 IS_MAIN$1, int64 IS
 		FBSTRING* vr$15 = fb_StrAssign( (void*)&PPONLYFILE$1, -1ll, *(void**)((uint8*)MODULE$1 + 24ll), -1ll, 0 );
 		if( *(int64*)((uint8*)MODULE$1 + 32ll) != 0ll ) goto label$855;
 		{
-			FBSTRING TMP$413$3;
+			FBSTRING TMP$416$3;
 			FBSTRING* vr$18 = HSTRIPEXT( &PPONLYFILE$1 );
-			__builtin_memset( &TMP$413$3, 0, 24ll );
-			FBSTRING* vr$21 = fb_StrConcat( &TMP$413$3, (void*)vr$18, -1ll, (void*)".pp.bas", 8ll );
+			__builtin_memset( &TMP$416$3, 0, 24ll );
+			FBSTRING* vr$21 = fb_StrConcat( &TMP$416$3, (void*)vr$18, -1ll, (void*)".pp.bas", 8ll );
 			FBSTRING* vr$23 = fb_StrAssign( (void*)&PPONLYFILE$1, -1ll, (void*)vr$21, -1ll, 0 );
 		}
 		label$855:;
@@ -5026,9 +5035,9 @@ static void HCOMPILEBAS( struct $9FBCIOFILE* MODULE$1, int64 IS_MAIN$1, int64 IS
 		int64 vr$29 = FBGETOPTION( 1ll );
 		if( vr$29 == 0ll ) goto label$859;
 		{
-			FBSTRING TMP$417$3;
-			__builtin_memset( &TMP$417$3, 0, 24ll );
-			FBSTRING* vr$33 = fb_StrConcat( &TMP$417$3, (void*)" -pp ", 6ll, (void*)&PPONLYFILE$1, -1ll );
+			FBSTRING TMP$420$3;
+			__builtin_memset( &TMP$420$3, 0, 24ll );
+			FBSTRING* vr$33 = fb_StrConcat( &TMP$420$3, (void*)" -pp ", 6ll, (void*)&PPONLYFILE$1, -1ll );
 			fb_PrintString( 0, vr$33, 0 );
 		}
 		label$859:;
@@ -5120,12 +5129,12 @@ static void HCOMPILEMODULES( void )
 	struct $9FBCIOFILE* MODULE$1;
 	ISMAIN$1 = 0ll;
 	{
-		int64 TMP$420$2;
+		int64 TMP$423$2;
 		int64 vr$1 = FBGETOPTION( 0ll );
-		TMP$420$2 = vr$1;
-		if( TMP$420$2 == 0ll ) goto label$880;
+		TMP$423$2 = vr$1;
+		if( TMP$423$2 == 0ll ) goto label$880;
 		label$881:;
-		if( TMP$420$2 != 2ll ) goto label$879;
+		if( TMP$423$2 != 2ll ) goto label$879;
 		label$880:;
 		{
 			CHECKMAIN$1 = -1ll;
@@ -5160,14 +5169,14 @@ static void HCOMPILEMODULES( void )
 	{
 		if( CHECKMAIN$1 == 0ll ) goto label$891;
 		{
-			FBSTRING TMP$421$3;
-			__builtin_memset( &TMP$421$3, 0, 24ll );
+			FBSTRING TMP$424$3;
+			__builtin_memset( &TMP$424$3, 0, 24ll );
 			FBSTRING* vr$16 = HSTRIPEXT( (FBSTRING*)MODULE$1 );
-			FBSTRING* vr$18 = fb_StrAssign( (void*)&TMP$421$3, -1ll, (void*)vr$16, -1ll, 0 );
-			FBSTRING* vr$19 = HSTRIPPATH( *(uint8**)&TMP$421$3 );
+			FBSTRING* vr$18 = fb_StrAssign( (void*)&TMP$424$3, -1ll, (void*)vr$16, -1ll, 0 );
+			FBSTRING* vr$19 = HSTRIPPATH( *(uint8**)&TMP$424$3 );
 			int32 vr$21 = fb_StrCompare( (void*)&MAINFILE$1, -1ll, (void*)vr$19, -1ll );
 			ISMAIN$1 = (int64)-((int64)vr$21 == 0ll);
-			fb_StrDelete( &TMP$421$3 );
+			fb_StrDelete( &TMP$424$3 );
 		}
 		label$891:;
 		label$890:;
@@ -5184,24 +5193,24 @@ static void HCOMPILEMODULES( void )
 
 static int64 HPARSEXPM( FBSTRING* XPMFILE$1, FBSTRING* CODE$1 )
 {
-	FBSTRING TMP$423$1;
-	FBSTRING TMP$425$1;
-	FBSTRING TMP$427$1;
-	FBSTRING TMP$435$1;
-	FBSTRING TMP$437$1;
-	FBSTRING TMP$439$1;
-	FBSTRING TMP$443$1;
+	FBSTRING TMP$426$1;
+	FBSTRING TMP$428$1;
+	FBSTRING TMP$430$1;
+	FBSTRING TMP$438$1;
+	FBSTRING TMP$440$1;
+	FBSTRING TMP$442$1;
+	FBSTRING TMP$446$1;
 	int64 fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$892:;
-	__builtin_memset( &TMP$423$1, 0, 24ll );
-	FBSTRING* vr$3 = fb_StrConcat( &TMP$423$1, (void*)CODE$1, -1ll, (void*)"\x0A" "dim shared as zstring ptr ", 28ll );
+	__builtin_memset( &TMP$426$1, 0, 24ll );
+	FBSTRING* vr$3 = fb_StrConcat( &TMP$426$1, (void*)CODE$1, -1ll, (void*)"\x0A" "dim shared as zstring ptr ", 28ll );
 	FBSTRING* vr$4 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$3, -1ll, 0 );
-	__builtin_memset( &TMP$425$1, 0, 24ll );
-	FBSTRING* vr$7 = fb_StrConcat( &TMP$425$1, (void*)CODE$1, -1ll, (void*)"fb_program_icon_data", 21ll );
+	__builtin_memset( &TMP$428$1, 0, 24ll );
+	FBSTRING* vr$7 = fb_StrConcat( &TMP$428$1, (void*)CODE$1, -1ll, (void*)"fb_program_icon_data", 21ll );
 	FBSTRING* vr$8 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$7, -1ll, 0 );
-	__builtin_memset( &TMP$427$1, 0, 24ll );
-	FBSTRING* vr$11 = fb_StrConcat( &TMP$427$1, (void*)CODE$1, -1ll, (void*)"(0 to ...) = _\x0A{ _\x0A", 20ll );
+	__builtin_memset( &TMP$430$1, 0, 24ll );
+	FBSTRING* vr$11 = fb_StrConcat( &TMP$430$1, (void*)CODE$1, -1ll, (void*)"(0 to ...) = _\x0A{ _\x0A", 20ll );
 	FBSTRING* vr$12 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$11, -1ll, 0 );
 	int64 F$1;
 	int32 vr$13 = fb_FileFree(  );
@@ -5249,21 +5258,21 @@ static int64 HPARSEXPM( FBSTRING* XPMFILE$1, FBSTRING* CODE$1 )
 		int64 vr$54 = fb_StrLen( (void*)&LN$1, -1ll );
 		if( vr$54 <= 0ll ) goto label$902;
 		{
-			FBSTRING TMP$432$3;
-			FBSTRING TMP$433$3;
+			FBSTRING TMP$435$3;
+			FBSTRING TMP$436$3;
 			if( SAW_ROWS$1 == 0ll ) goto label$904;
 			{
-				FBSTRING TMP$430$4;
-				__builtin_memset( &TMP$430$4, 0, 24ll );
-				FBSTRING* vr$57 = fb_StrConcat( &TMP$430$4, (void*)CODE$1, -1ll, (void*)", _\x0A", 5ll );
+				FBSTRING TMP$433$4;
+				__builtin_memset( &TMP$433$4, 0, 24ll );
+				FBSTRING* vr$57 = fb_StrConcat( &TMP$433$4, (void*)CODE$1, -1ll, (void*)", _\x0A", 5ll );
 				FBSTRING* vr$58 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$57, -1ll, 0 );
 			}
 			label$904:;
 			label$903:;
-			__builtin_memset( &TMP$432$3, 0, 24ll );
-			FBSTRING* vr$62 = fb_StrConcat( &TMP$432$3, (void*)"\x09@", 3ll, (void*)&LN$1, -1ll );
-			__builtin_memset( &TMP$433$3, 0, 24ll );
-			FBSTRING* vr$65 = fb_StrConcat( &TMP$433$3, (void*)CODE$1, -1ll, (void*)vr$62, -1ll );
+			__builtin_memset( &TMP$435$3, 0, 24ll );
+			FBSTRING* vr$62 = fb_StrConcat( &TMP$435$3, (void*)"\x09@", 3ll, (void*)&LN$1, -1ll );
+			__builtin_memset( &TMP$436$3, 0, 24ll );
+			FBSTRING* vr$65 = fb_StrConcat( &TMP$436$3, (void*)CODE$1, -1ll, (void*)vr$62, -1ll );
 			FBSTRING* vr$66 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$65, -1ll, 0 );
 			SAW_ROWS$1 = -1ll;
 		}
@@ -5284,17 +5293,17 @@ static int64 HPARSEXPM( FBSTRING* XPMFILE$1, FBSTRING* CODE$1 )
 	}
 	label$907:;
 	label$906:;
-	__builtin_memset( &TMP$435$1, 0, 24ll );
-	FBSTRING* vr$74 = fb_StrConcat( &TMP$435$1, (void*)CODE$1, -1ll, (void*)" _ \x0A", 5ll );
+	__builtin_memset( &TMP$438$1, 0, 24ll );
+	FBSTRING* vr$74 = fb_StrConcat( &TMP$438$1, (void*)CODE$1, -1ll, (void*)" _ \x0A", 5ll );
 	FBSTRING* vr$75 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$74, -1ll, 0 );
-	__builtin_memset( &TMP$437$1, 0, 24ll );
-	FBSTRING* vr$78 = fb_StrConcat( &TMP$437$1, (void*)CODE$1, -1ll, (void*)"}\x0A\x0A", 4ll );
+	__builtin_memset( &TMP$440$1, 0, 24ll );
+	FBSTRING* vr$78 = fb_StrConcat( &TMP$440$1, (void*)CODE$1, -1ll, (void*)"}\x0A\x0A", 4ll );
 	FBSTRING* vr$79 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$78, -1ll, 0 );
-	__builtin_memset( &TMP$439$1, 0, 24ll );
-	FBSTRING* vr$82 = fb_StrConcat( &TMP$439$1, (void*)CODE$1, -1ll, (void*)"extern as zstring ptr ptr fb_program_icon alias \x22" "fb_program_icon\x22\x0A", 67ll );
+	__builtin_memset( &TMP$442$1, 0, 24ll );
+	FBSTRING* vr$82 = fb_StrConcat( &TMP$442$1, (void*)CODE$1, -1ll, (void*)"extern as zstring ptr ptr fb_program_icon alias \x22" "fb_program_icon\x22\x0A", 67ll );
 	FBSTRING* vr$83 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$82, -1ll, 0 );
-	__builtin_memset( &TMP$443$1, 0, 24ll );
-	FBSTRING* vr$86 = fb_StrConcat( &TMP$443$1, (void*)CODE$1, -1ll, (void*)"dim shared as zstring ptr ptr fb_program_icon = @fb_program_icon_data(0)\x0A", 74ll );
+	__builtin_memset( &TMP$446$1, 0, 24ll );
+	FBSTRING* vr$86 = fb_StrConcat( &TMP$446$1, (void*)CODE$1, -1ll, (void*)"dim shared as zstring ptr ptr fb_program_icon = @fb_program_icon_data(0)\x0A", 74ll );
 	FBSTRING* vr$87 = fb_StrAssign( (void*)CODE$1, -1ll, (void*)vr$86, -1ll, 0 );
 	fb$result$1 = -1ll;
 	fb_StrDelete( &LN$1 );
@@ -5304,7 +5313,7 @@ static int64 HPARSEXPM( FBSTRING* XPMFILE$1, FBSTRING* CODE$1 )
 
 static int64 HCOMPILEXPM( void )
 {
-	int64 TMP$448$1;
+	int64 TMP$451$1;
 	int64 fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$908:;
@@ -5335,14 +5344,14 @@ static int64 HCOMPILEXPM( void )
 	FBSTRING* vr$15 = fb_StrConcatAssign( (void*)((uint8*)&FBC$ + 272ll), -1ll, (void*)".bas", 5ll, 0 );
 	if( *(int64*)((uint8*)&FBC$ + 112ll) == 0ll ) goto label$915;
 	{
-		FBSTRING TMP$446$2;
-		FBSTRING TMP$447$2;
+		FBSTRING TMP$449$2;
+		FBSTRING TMP$450$2;
 		FBSTRING* vr$16 = fb_StrAllocTempDescZEx( (uint8*)"parsing xpm: ", 13ll );
 		fb_PrintString( 0, vr$16, 2 );
-		__builtin_memset( &TMP$446$2, 0, 24ll );
-		FBSTRING* vr$21 = fb_StrConcat( &TMP$446$2, (void*)&XPMFILE$1, -1ll, (void*)" -o ", 5ll );
-		__builtin_memset( &TMP$447$2, 0, 24ll );
-		FBSTRING* vr$24 = fb_StrConcat( &TMP$447$2, (void*)vr$21, -1ll, (void*)((uint8*)&FBC$ + 272ll), -1ll );
+		__builtin_memset( &TMP$449$2, 0, 24ll );
+		FBSTRING* vr$21 = fb_StrConcat( &TMP$449$2, (void*)&XPMFILE$1, -1ll, (void*)" -o ", 5ll );
+		__builtin_memset( &TMP$450$2, 0, 24ll );
+		FBSTRING* vr$24 = fb_StrConcat( &TMP$450$2, (void*)vr$21, -1ll, (void*)((uint8*)&FBC$ + 272ll), -1ll );
 		fb_PrintString( 0, vr$24, 1 );
 	}
 	label$915:;
@@ -5367,8 +5376,8 @@ static int64 HCOMPILEXPM( void )
 	}
 	label$919:;
 	label$918:;
-	TMP$448$1 = FO$1;
-	fb_PrintString( (int32)TMP$448$1, &CODE$1, 0 );
+	TMP$451$1 = FO$1;
+	fb_PrintString( (int32)TMP$451$1, &CODE$1, 0 );
 	int32 vr$41 = fb_FileClose( (int32)FO$1 );
 	if( (int64)vr$41 == 0ll ) goto label$920;
 	void* vr$43 = fb_ErrorThrowAt( 2805, (uint8*)"src/compiler/fbc.bas", (void*)0ull, (void*)0ull );
@@ -5406,24 +5415,24 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 	label$926:;
 	label$925:;
 	{
-		int64 TMP$449$2;
+		int64 TMP$452$2;
 		int64 vr$7 = FBGETOPTION( 2ll );
-		TMP$449$2 = vr$7;
-		if( TMP$449$2 != 1ll ) goto label$928;
+		TMP$452$2 = vr$7;
+		if( TMP$452$2 != 1ll ) goto label$928;
 		label$929:;
 		{
 			{
-				int64 TMP$450$4;
+				int64 TMP$453$4;
 				int64 vr$8 = FBGETCPUFAMILY(  );
-				TMP$450$4 = vr$8;
-				if( TMP$450$4 != 0ll ) goto label$931;
+				TMP$453$4 = vr$8;
+				if( TMP$453$4 != 0ll ) goto label$931;
 				label$932:;
 				{
 					FBSTRING* vr$10 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-m32 ", 6ll, 0 );
 				}
 				goto label$930;
 				label$931:;
-				if( TMP$450$4 != 1ll ) goto label$933;
+				if( TMP$453$4 != 1ll ) goto label$933;
 				label$934:;
 				{
 					FBSTRING* vr$12 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-m64 ", 6ll, 0 );
@@ -5438,16 +5447,16 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 			goto label$935;
 			label$936:;
 			{
-				FBSTRING TMP$455$4;
-				FBSTRING TMP$456$4;
-				FBSTRING TMP$457$4;
+				FBSTRING TMP$458$4;
+				FBSTRING TMP$459$4;
+				FBSTRING TMP$460$4;
 				uint8* vr$15 = FBGETGCCARCH(  );
-				__builtin_memset( &TMP$455$4, 0, 24ll );
-				FBSTRING* vr$18 = fb_StrConcat( &TMP$455$4, (void*)"-march=", 8ll, (void*)vr$15, 0ll );
-				__builtin_memset( &TMP$456$4, 0, 24ll );
-				FBSTRING* vr$21 = fb_StrConcat( &TMP$456$4, (void*)vr$18, -1ll, (void*)" ", 2ll );
-				__builtin_memset( &TMP$457$4, 0, 24ll );
-				FBSTRING* vr$25 = fb_StrConcat( &TMP$457$4, (void*)&LN$1, -1ll, (void*)vr$21, -1ll );
+				__builtin_memset( &TMP$458$4, 0, 24ll );
+				FBSTRING* vr$18 = fb_StrConcat( &TMP$458$4, (void*)"-march=", 8ll, (void*)vr$15, 0ll );
+				__builtin_memset( &TMP$459$4, 0, 24ll );
+				FBSTRING* vr$21 = fb_StrConcat( &TMP$459$4, (void*)vr$18, -1ll, (void*)" ", 2ll );
+				__builtin_memset( &TMP$460$4, 0, 24ll );
+				FBSTRING* vr$25 = fb_StrConcat( &TMP$460$4, (void*)&LN$1, -1ll, (void*)vr$21, -1ll );
 				FBSTRING* vr$27 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$25, -1ll, 0 );
 			}
 			label$935:;
@@ -5486,12 +5495,12 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 			label$942:;
 			label$941:;
 			{
-				int64 TMP$474$4;
+				int64 TMP$477$4;
 				int64 vr$61 = FBGETCPUFAMILY(  );
-				TMP$474$4 = vr$61;
-				if( TMP$474$4 == 0ll ) goto label$945;
+				TMP$477$4 = vr$61;
+				if( TMP$477$4 == 0ll ) goto label$945;
 				label$946:;
-				if( TMP$474$4 != 1ll ) goto label$944;
+				if( TMP$477$4 != 1ll ) goto label$944;
 				label$945:;
 				{
 					int64 vr$62 = FBGETOPTION( 9ll );
@@ -5508,35 +5517,35 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 		}
 		goto label$927;
 		label$928:;
-		if( TMP$449$2 != 2ll ) goto label$949;
+		if( TMP$452$2 != 2ll ) goto label$949;
 		label$950:;
 		{
 			{
-				int64 TMP$476$4;
+				int64 TMP$479$4;
 				int64 vr$65 = FBGETCPUFAMILY(  );
-				TMP$476$4 = vr$65;
-				if( TMP$476$4 != 0ll ) goto label$952;
+				TMP$479$4 = vr$65;
+				if( TMP$479$4 != 0ll ) goto label$952;
 				label$953:;
 				{
 					FBSTRING* vr$67 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-march=x86 ", 12ll, 0 );
 				}
 				goto label$951;
 				label$952:;
-				if( TMP$476$4 != 1ll ) goto label$954;
+				if( TMP$479$4 != 1ll ) goto label$954;
 				label$955:;
 				{
 					FBSTRING* vr$69 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-march=x86-64 ", 15ll, 0 );
 				}
 				goto label$951;
 				label$954:;
-				if( TMP$476$4 != 2ll ) goto label$956;
+				if( TMP$479$4 != 2ll ) goto label$956;
 				label$957:;
 				{
 					FBSTRING* vr$71 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-march=arm ", 12ll, 0 );
 				}
 				goto label$951;
 				label$956:;
-				if( TMP$476$4 != 3ll ) goto label$958;
+				if( TMP$479$4 != 3ll ) goto label$958;
 				label$959:;
 				{
 					FBSTRING* vr$73 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"-march=aarch64 ", 16ll, 0 );
@@ -5557,12 +5566,12 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 			FBSTRING* vr$82 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)vr$80, -1ll, 0 );
 			FBSTRING* vr$84 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)" ", 2ll, 0 );
 			{
-				int64 TMP$482$4;
+				int64 TMP$485$4;
 				int64 vr$85 = FBGETCPUFAMILY(  );
-				TMP$482$4 = vr$85;
-				if( TMP$482$4 == 0ll ) goto label$964;
+				TMP$485$4 = vr$85;
+				if( TMP$485$4 == 0ll ) goto label$964;
 				label$965:;
-				if( TMP$482$4 != 1ll ) goto label$963;
+				if( TMP$485$4 != 1ll ) goto label$963;
 				label$964:;
 				{
 					int64 vr$86 = FBGETOPTION( 9ll );
@@ -5589,10 +5598,10 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 	FBSTRING* vr$102 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"\x22", 2ll, 0 );
 	FBSTRING* vr$105 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)((uint8*)&FBC$ + 2038ll), 128ll, 0 );
 	{
-		int64 TMP$486$2;
+		int64 TMP$489$2;
 		int64 vr$106 = FBGETOPTION( 2ll );
-		TMP$486$2 = vr$106;
-		if( TMP$486$2 != 1ll ) goto label$969;
+		TMP$489$2 = vr$106;
+		if( TMP$489$2 != 1ll ) goto label$969;
 		label$970:;
 		{
 			int64 vr$108 = FBCRUNBIN( (uint8*)"compiling C", 3ll, &LN$1 );
@@ -5600,7 +5609,7 @@ static int64 HCOMPILESTAGE2MODULE( struct $9FBCIOFILE* MODULE$1 )
 		}
 		goto label$968;
 		label$969:;
-		if( TMP$486$2 != 2ll ) goto label$971;
+		if( TMP$489$2 != 2ll ) goto label$971;
 		label$972:;
 		{
 			int64 vr$110 = FBCRUNBIN( (uint8*)"compiling LLVM IR", 4ll, &LN$1 );
@@ -5641,19 +5650,19 @@ static void HCOMPILESTAGE2MODULES( void )
 
 static int64 HASSEMBLEMODULE( struct $9FBCIOFILE* MODULE$1 )
 {
-	FBSTRING TMP$495$1;
-	FBSTRING TMP$496$1;
-	FBSTRING TMP$497$1;
+	FBSTRING TMP$498$1;
+	FBSTRING TMP$499$1;
+	FBSTRING TMP$500$1;
 	int64 fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$979:;
 	FBSTRING LN$1;
 	__builtin_memset( &LN$1, 0, 24ll );
 	{
-		int64 TMP$489$2;
+		int64 TMP$492$2;
 		int64 vr$2 = FBGETCPUFAMILY(  );
-		TMP$489$2 = vr$2;
-		if( TMP$489$2 != 0ll ) goto label$982;
+		TMP$492$2 = vr$2;
+		if( TMP$492$2 != 0ll ) goto label$982;
 		label$983:;
 		{
 			int64 vr$3 = FBGETOPTION( 3ll );
@@ -5670,7 +5679,7 @@ static int64 HASSEMBLEMODULE( struct $9FBCIOFILE* MODULE$1 )
 		}
 		goto label$981;
 		label$982:;
-		if( TMP$489$2 != 1ll ) goto label$986;
+		if( TMP$492$2 != 1ll ) goto label$986;
 		label$987:;
 		{
 			int64 vr$8 = FBGETOPTION( 3ll );
@@ -5705,12 +5714,12 @@ static int64 HASSEMBLEMODULE( struct $9FBCIOFILE* MODULE$1 )
 	FBSTRING* vr$19 = HGETASMNAME( MODULE$1, 2ll );
 	FBSTRING* vr$21 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)vr$19, -1ll, 0 );
 	FBSTRING* vr$23 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)"\x22 ", 3ll, 0 );
-	__builtin_memset( &TMP$495$1, 0, 24ll );
-	FBSTRING* vr$27 = fb_StrConcat( &TMP$495$1, (void*)"-o \x22", 5ll, *(void**)((uint8*)MODULE$1 + 24ll), -1ll );
-	__builtin_memset( &TMP$496$1, 0, 24ll );
-	FBSTRING* vr$30 = fb_StrConcat( &TMP$496$1, (void*)vr$27, -1ll, (void*)"\x22", 2ll );
-	__builtin_memset( &TMP$497$1, 0, 24ll );
-	FBSTRING* vr$34 = fb_StrConcat( &TMP$497$1, (void*)&LN$1, -1ll, (void*)vr$30, -1ll );
+	__builtin_memset( &TMP$498$1, 0, 24ll );
+	FBSTRING* vr$27 = fb_StrConcat( &TMP$498$1, (void*)"-o \x22", 5ll, *(void**)((uint8*)MODULE$1 + 24ll), -1ll );
+	__builtin_memset( &TMP$499$1, 0, 24ll );
+	FBSTRING* vr$30 = fb_StrConcat( &TMP$499$1, (void*)vr$27, -1ll, (void*)"\x22", 2ll );
+	__builtin_memset( &TMP$500$1, 0, 24ll );
+	FBSTRING* vr$34 = fb_StrConcat( &TMP$500$1, (void*)&LN$1, -1ll, (void*)vr$30, -1ll );
 	FBSTRING* vr$36 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$34, -1ll, 0 );
 	FBSTRING* vr$39 = fb_StrConcatAssign( (void*)&LN$1, -1ll, (void*)((uint8*)&FBC$ + 1782ll), 128ll, 0 );
 	int64 vr$41 = FBCRUNBIN( (uint8*)"assembling", 0ll, &LN$1 );
@@ -5759,30 +5768,30 @@ static void HASSEMBLEMODULES( void )
 
 static int64 HASSEMBLERC( struct $9FBCIOFILE* RC$1 )
 {
-	FBSTRING TMP$500$1;
-	FBSTRING TMP$501$1;
-	FBSTRING TMP$502$1;
 	FBSTRING TMP$503$1;
 	FBSTRING TMP$504$1;
 	FBSTRING TMP$505$1;
+	FBSTRING TMP$506$1;
+	FBSTRING TMP$507$1;
+	FBSTRING TMP$508$1;
 	int64 fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$1004:;
 	FBSTRING LN$1;
 	FBSTRING* vr$2 = fb_StrInit( (void*)&LN$1, -1ll, (void*)"--output-format=coff --include-dir=.", 37ll, 0 );
-	__builtin_memset( &TMP$500$1, 0, 24ll );
-	FBSTRING* vr$7 = fb_StrConcat( &TMP$500$1, (void*)" \x22", 3ll, (void*)RC$1, -1ll );
-	__builtin_memset( &TMP$501$1, 0, 24ll );
-	FBSTRING* vr$10 = fb_StrConcat( &TMP$501$1, (void*)vr$7, -1ll, (void*)"\x22", 2ll );
-	__builtin_memset( &TMP$502$1, 0, 24ll );
-	FBSTRING* vr$14 = fb_StrConcat( &TMP$502$1, (void*)&LN$1, -1ll, (void*)vr$10, -1ll );
-	FBSTRING* vr$16 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$14, -1ll, 0 );
 	__builtin_memset( &TMP$503$1, 0, 24ll );
-	FBSTRING* vr$20 = fb_StrConcat( &TMP$503$1, (void*)" \x22", 3ll, *(void**)((uint8*)RC$1 + 24ll), -1ll );
+	FBSTRING* vr$7 = fb_StrConcat( &TMP$503$1, (void*)" \x22", 3ll, (void*)RC$1, -1ll );
 	__builtin_memset( &TMP$504$1, 0, 24ll );
-	FBSTRING* vr$23 = fb_StrConcat( &TMP$504$1, (void*)vr$20, -1ll, (void*)"\x22", 2ll );
+	FBSTRING* vr$10 = fb_StrConcat( &TMP$504$1, (void*)vr$7, -1ll, (void*)"\x22", 2ll );
 	__builtin_memset( &TMP$505$1, 0, 24ll );
-	FBSTRING* vr$27 = fb_StrConcat( &TMP$505$1, (void*)&LN$1, -1ll, (void*)vr$23, -1ll );
+	FBSTRING* vr$14 = fb_StrConcat( &TMP$505$1, (void*)&LN$1, -1ll, (void*)vr$10, -1ll );
+	FBSTRING* vr$16 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$14, -1ll, 0 );
+	__builtin_memset( &TMP$506$1, 0, 24ll );
+	FBSTRING* vr$20 = fb_StrConcat( &TMP$506$1, (void*)" \x22", 3ll, *(void**)((uint8*)RC$1 + 24ll), -1ll );
+	__builtin_memset( &TMP$507$1, 0, 24ll );
+	FBSTRING* vr$23 = fb_StrConcat( &TMP$507$1, (void*)vr$20, -1ll, (void*)"\x22", 2ll );
+	__builtin_memset( &TMP$508$1, 0, 24ll );
+	FBSTRING* vr$27 = fb_StrConcat( &TMP$508$1, (void*)&LN$1, -1ll, (void*)vr$23, -1ll );
 	FBSTRING* vr$29 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$27, -1ll, 0 );
 	int64 vr$31 = FBCRUNBIN( (uint8*)"compiling rc", 7ll, &LN$1 );
 	fb$result$1 = vr$31;
@@ -5908,8 +5917,8 @@ static int64 HCOMPILEFBCTINF( void )
 
 static int64 HARCHIVEFILES( void )
 {
-	FBSTRING TMP$510$1;
-	FBSTRING TMP$511$1;
+	FBSTRING TMP$513$1;
+	FBSTRING TMP$514$1;
 	int64 fb$result$1;
 	__builtin_memset( &fb$result$1, 0, 8ll );
 	label$1033:;
@@ -5922,16 +5931,16 @@ static int64 HARCHIVEFILES( void )
 	label$1036:;
 	label$1035:;
 	FBSTRING LN$1;
-	__builtin_memset( &TMP$510$1, 0, 24ll );
-	FBSTRING* vr$8 = fb_StrConcat( &TMP$510$1, (void*)"-rsc \x22", 7ll, (void*)((uint8*)&FBC$ + 856ll), 261ll );
-	__builtin_memset( &TMP$511$1, 0, 24ll );
-	FBSTRING* vr$11 = fb_StrConcat( &TMP$511$1, (void*)vr$8, -1ll, (void*)"\x22 ", 3ll );
+	__builtin_memset( &TMP$513$1, 0, 24ll );
+	FBSTRING* vr$8 = fb_StrConcat( &TMP$513$1, (void*)"-rsc \x22", 7ll, (void*)((uint8*)&FBC$ + 856ll), 261ll );
+	__builtin_memset( &TMP$514$1, 0, 24ll );
+	FBSTRING* vr$11 = fb_StrConcat( &TMP$514$1, (void*)vr$8, -1ll, (void*)"\x22 ", 3ll );
 	FBSTRING* vr$13 = fb_StrInit( (void*)&LN$1, -1ll, (void*)vr$11, -1ll, 0 );
 	int64 vr$14 = FBGETOPTION( 29ll );
 	int64 vr$15 = FBISCROSSCOMP(  );
 	if( (vr$14 & ~vr$15) == 0ll ) goto label$1038;
 	{
-		FBSTRING TMP$515$2;
+		FBSTRING TMP$518$2;
 		int64 vr$18 = HCOMPILEFBCTINF(  );
 		if( vr$18 == 0ll ) goto label$1040;
 		{
@@ -5939,10 +5948,10 @@ static int64 HARCHIVEFILES( void )
 		}
 		label$1040:;
 		label$1039:;
-		__builtin_memset( &TMP$515$2, 0, 24ll );
-		FBSTRING* vr$23 = fb_StrAssign( (void*)&TMP$515$2, -1ll, (void*)"__fb_ct.inf", 12ll, 0 );
-		FBCADDTEMP( &TMP$515$2 );
-		fb_StrDelete( &TMP$515$2 );
+		__builtin_memset( &TMP$518$2, 0, 24ll );
+		FBSTRING* vr$23 = fb_StrAssign( (void*)&TMP$518$2, -1ll, (void*)"__fb_ct.inf", 12ll, 0 );
+		FBCADDTEMP( &TMP$518$2 );
+		fb_StrDelete( &TMP$518$2 );
 	}
 	label$1038:;
 	label$1037:;
@@ -5952,15 +5961,15 @@ static int64 HARCHIVEFILES( void )
 	label$1041:;
 	if( OBJFILE$1 == (FBSTRING*)0ull ) goto label$1042;
 	{
-		FBSTRING TMP$516$2;
-		FBSTRING TMP$517$2;
-		FBSTRING TMP$518$2;
-		__builtin_memset( &TMP$516$2, 0, 24ll );
-		FBSTRING* vr$30 = fb_StrConcat( &TMP$516$2, (void*)"\x22", 2ll, (void*)OBJFILE$1, -1ll );
-		__builtin_memset( &TMP$517$2, 0, 24ll );
-		FBSTRING* vr$33 = fb_StrConcat( &TMP$517$2, (void*)vr$30, -1ll, (void*)"\x22 ", 3ll );
-		__builtin_memset( &TMP$518$2, 0, 24ll );
-		FBSTRING* vr$37 = fb_StrConcat( &TMP$518$2, (void*)&LN$1, -1ll, (void*)vr$33, -1ll );
+		FBSTRING TMP$519$2;
+		FBSTRING TMP$520$2;
+		FBSTRING TMP$521$2;
+		__builtin_memset( &TMP$519$2, 0, 24ll );
+		FBSTRING* vr$30 = fb_StrConcat( &TMP$519$2, (void*)"\x22", 2ll, (void*)OBJFILE$1, -1ll );
+		__builtin_memset( &TMP$520$2, 0, 24ll );
+		FBSTRING* vr$33 = fb_StrConcat( &TMP$520$2, (void*)vr$30, -1ll, (void*)"\x22 ", 3ll );
+		__builtin_memset( &TMP$521$2, 0, 24ll );
+		FBSTRING* vr$37 = fb_StrConcat( &TMP$521$2, (void*)&LN$1, -1ll, (void*)vr$33, -1ll );
 		FBSTRING* vr$39 = fb_StrAssign( (void*)&LN$1, -1ll, (void*)vr$37, -1ll, 0 );
 		void* vr$40 = LISTGETNEXT( (void*)OBJFILE$1 );
 		OBJFILE$1 = (FBSTRING*)vr$40;
@@ -5976,30 +5985,30 @@ static int64 HARCHIVEFILES( void )
 
 static void HSETDEFAULTLIBPATHS( void )
 {
-	FBSTRING TMP$520$1;
-	FBSTRING TMP$521$1;
+	FBSTRING TMP$523$1;
+	FBSTRING TMP$524$1;
 	label$1043:;
-	__builtin_memset( &TMP$520$1, 0, 24ll );
-	FBSTRING* vr$3 = fb_StrAssign( (void*)&TMP$520$1, -1ll, (void*)((uint8*)&FBC$ + 3359ll), 261ll, 0 );
-	FBCADDDEFLIBPATH( &TMP$520$1 );
-	fb_StrDelete( &TMP$520$1 );
-	__builtin_memset( &TMP$521$1, 0, 24ll );
-	FBSTRING* vr$8 = fb_StrAssign( (void*)&TMP$521$1, -1ll, (void*)".", 2ll, 0 );
-	FBCADDDEFLIBPATH( &TMP$521$1 );
-	fb_StrDelete( &TMP$521$1 );
+	__builtin_memset( &TMP$523$1, 0, 24ll );
+	FBSTRING* vr$3 = fb_StrAssign( (void*)&TMP$523$1, -1ll, (void*)((uint8*)&FBC$ + 3359ll), 261ll, 0 );
+	FBCADDDEFLIBPATH( &TMP$523$1 );
+	fb_StrDelete( &TMP$523$1 );
+	__builtin_memset( &TMP$524$1, 0, 24ll );
+	FBSTRING* vr$8 = fb_StrAssign( (void*)&TMP$524$1, -1ll, (void*)".", 2ll, 0 );
+	FBCADDDEFLIBPATH( &TMP$524$1 );
+	fb_StrDelete( &TMP$524$1 );
 	FBCADDLIBPATHFOR( (uint8*)"libgcc.a" );
 	{
-		int64 TMP$523$2;
+		int64 TMP$526$2;
 		int64 vr$11 = FBGETOPTION( 3ll );
-		TMP$523$2 = vr$11;
-		if( TMP$523$2 != 3ll ) goto label$1046;
+		TMP$526$2 = vr$11;
+		if( TMP$526$2 != 3ll ) goto label$1046;
 		label$1047:;
 		{
 			FBCADDLIBPATHFOR( (uint8*)"libm.a" );
 		}
 		goto label$1045;
 		label$1046:;
-		if( TMP$523$2 != 0ll ) goto label$1048;
+		if( TMP$526$2 != 0ll ) goto label$1048;
 		label$1049:;
 		{
 			FBCADDLIBPATHFOR( (uint8*)"libmingw32.a" );
@@ -6012,12 +6021,12 @@ static void HSETDEFAULTLIBPATHS( void )
 
 static void FBCADDDEFLIB( uint8* LIBNAME$1 )
 {
-	FBSTRING TMP$526$1;
+	FBSTRING TMP$529$1;
 	label$1050:;
-	__builtin_memset( &TMP$526$1, 0, 24ll );
-	FBSTRING* vr$2 = fb_StrAssign( (void*)&TMP$526$1, -1ll, (void*)LIBNAME$1, 0ll, 0 );
-	STRSETADD( (struct $7TSTRSET*)((uint8*)&FBC$ + 680ll), &TMP$526$1, -1ll );
-	fb_StrDelete( &TMP$526$1 );
+	__builtin_memset( &TMP$529$1, 0, 24ll );
+	FBSTRING* vr$2 = fb_StrAssign( (void*)&TMP$529$1, -1ll, (void*)LIBNAME$1, 0ll, 0 );
+	STRSETADD( (struct $7TSTRSET*)((uint8*)&FBC$ + 680ll), &TMP$529$1, -1ll );
+	fb_StrDelete( &TMP$529$1 );
 	label$1051:;
 }
 
@@ -6051,32 +6060,32 @@ static FBSTRING* HGETFBLIBNAMESUFFIX( void )
 
 static void HADDDEFAULTLIBS( void )
 {
-	FBSTRING TMP$527$1;
-	FBSTRING TMP$528$1;
+	FBSTRING TMP$530$1;
+	FBSTRING TMP$531$1;
 	label$1058:;
-	__builtin_memset( &TMP$528$1, 0, 24ll );
+	__builtin_memset( &TMP$531$1, 0, 24ll );
 	FBSTRING* vr$1 = HGETFBLIBNAMESUFFIX(  );
-	__builtin_memset( &TMP$527$1, 0, 24ll );
-	FBSTRING* vr$4 = fb_StrConcat( &TMP$527$1, (void*)"fb", 3ll, (void*)vr$1, -1ll );
-	FBSTRING* vr$6 = fb_StrAssign( (void*)&TMP$528$1, -1ll, (void*)vr$4, -1ll, 0 );
-	FBCADDDEFLIB( *(uint8**)&TMP$528$1 );
-	fb_StrDelete( &TMP$528$1 );
+	__builtin_memset( &TMP$530$1, 0, 24ll );
+	FBSTRING* vr$4 = fb_StrConcat( &TMP$530$1, (void*)"fb", 3ll, (void*)vr$1, -1ll );
+	FBSTRING* vr$6 = fb_StrAssign( (void*)&TMP$531$1, -1ll, (void*)vr$4, -1ll, 0 );
+	FBCADDDEFLIB( *(uint8**)&TMP$531$1 );
+	fb_StrDelete( &TMP$531$1 );
 	int64 vr$8 = FBGETOPTION( 26ll );
 	if( vr$8 == 0ll ) goto label$1061;
 	{
-		FBSTRING TMP$530$2;
-		FBSTRING TMP$531$2;
-		__builtin_memset( &TMP$531$2, 0, 24ll );
+		FBSTRING TMP$533$2;
+		FBSTRING TMP$534$2;
+		__builtin_memset( &TMP$534$2, 0, 24ll );
 		FBSTRING* vr$10 = HGETFBLIBNAMESUFFIX(  );
-		__builtin_memset( &TMP$530$2, 0, 24ll );
-		FBSTRING* vr$13 = fb_StrConcat( &TMP$530$2, (void*)"fbgfx", 6ll, (void*)vr$10, -1ll );
-		FBSTRING* vr$15 = fb_StrAssign( (void*)&TMP$531$2, -1ll, (void*)vr$13, -1ll, 0 );
-		FBCADDDEFLIB( *(uint8**)&TMP$531$2 );
-		fb_StrDelete( &TMP$531$2 );
+		__builtin_memset( &TMP$533$2, 0, 24ll );
+		FBSTRING* vr$13 = fb_StrConcat( &TMP$533$2, (void*)"fbgfx", 6ll, (void*)vr$10, -1ll );
+		FBSTRING* vr$15 = fb_StrAssign( (void*)&TMP$534$2, -1ll, (void*)vr$13, -1ll, 0 );
+		FBCADDDEFLIB( *(uint8**)&TMP$534$2 );
+		fb_StrDelete( &TMP$534$2 );
 		{
-			uint64 TMP$532$3;
+			uint64 TMP$535$3;
 			int64 vr$17 = FBGETOPTION( 3ll );
-			TMP$532$3 = (uint64)vr$17;
+			TMP$535$3 = (uint64)vr$17;
 			goto label$1063;
 			label$1064:;
 			{
@@ -6086,11 +6095,11 @@ static void HADDDEFAULTLIBS( void )
 			goto label$1062;
 			label$1065:;
 			{
-				FBSTRING TMP$536$4;
-				__builtin_memset( &TMP$536$4, 0, 24ll );
-				FBSTRING* vr$20 = fb_StrAssign( (void*)&TMP$536$4, -1ll, (void*)"/usr/X11R6/lib", 15ll, 0 );
-				FBCADDDEFLIBPATH( &TMP$536$4 );
-				fb_StrDelete( &TMP$536$4 );
+				FBSTRING TMP$539$4;
+				__builtin_memset( &TMP$539$4, 0, 24ll );
+				FBSTRING* vr$20 = fb_StrAssign( (void*)&TMP$539$4, -1ll, (void*)"/usr/X11R6/lib", 15ll, 0 );
+				FBCADDDEFLIBPATH( &TMP$539$4 );
+				fb_StrDelete( &TMP$539$4 );
 				FBCADDDEFLIB( (uint8*)"X11" );
 				FBCADDDEFLIB( (uint8*)"Xext" );
 				FBCADDDEFLIB( (uint8*)"Xpm" );
@@ -6099,7 +6108,7 @@ static void HADDDEFAULTLIBS( void )
 			}
 			goto label$1062;
 			label$1063:;
-			static const void* tmp$664[9ll] = {
+			static const void* tmp$667[9ll] = {
 				&&label$1064,
 				&&label$1064,
 				&&label$1065,
@@ -6110,17 +6119,17 @@ static void HADDDEFAULTLIBS( void )
 				&&label$1065,
 				&&label$1065,
 			};
-			if( TMP$532$3 > 8ull ) goto label$1062;
-			goto *tmp$664[TMP$532$3 - 0ull];
+			if( TMP$535$3 > 8ull ) goto label$1062;
+			goto *tmp$667[TMP$535$3 - 0ull];
 			label$1062:;
 		}
 	}
 	label$1061:;
 	label$1060:;
 	{
-		uint64 TMP$542$2;
+		uint64 TMP$545$2;
 		int64 vr$23 = FBGETOPTION( 3ll );
-		TMP$542$2 = (uint64)vr$23;
+		TMP$545$2 = (uint64)vr$23;
 		goto label$1067;
 		label$1068:;
 		{
@@ -6263,7 +6272,7 @@ static void HADDDEFAULTLIBS( void )
 		}
 		goto label$1066;
 		label$1067:;
-		static const void* tmp$665[9ll] = {
+		static const void* tmp$668[9ll] = {
 			&&label$1083,
 			&&label$1068,
 			&&label$1076,
@@ -6274,8 +6283,8 @@ static void HADDDEFAULTLIBS( void )
 			&&label$1071,
 			&&label$1081,
 		};
-		if( TMP$542$2 > 8ull ) goto label$1066;
-		goto *tmp$665[TMP$542$2 - 0ull];
+		if( TMP$545$2 > 8ull ) goto label$1066;
+		goto *tmp$668[TMP$545$2 - 0ull];
 		label$1066:;
 	}
 	label$1059:;
@@ -6423,24 +6432,24 @@ static void HPRINTOPTIONS( void )
 
 static void HPRINTVERSION( void )
 {
-	FBSTRING TMP$642$1;
-	FBSTRING TMP$643$1;
-	FBSTRING TMP$644$1;
 	FBSTRING TMP$645$1;
+	FBSTRING TMP$646$1;
+	FBSTRING TMP$647$1;
+	FBSTRING TMP$648$1;
 	label$1097:;
 	FBSTRING CONFIG$1;
 	__builtin_memset( &CONFIG$1, 0, 24ll );
 	int64 vr$1 = FBGETHOSTBITS(  );
 	FBSTRING* vr$2 = fb_LongintToStr( vr$1 );
 	FBSTRING* vr$3 = FBGETHOSTID(  );
-	__builtin_memset( &TMP$642$1, 0, 24ll );
-	FBSTRING* vr$6 = fb_StrConcat( &TMP$642$1, (void*)"FreeBASIC Compiler - Version 1.06.0 (02-17-2019), built for ", 61ll, (void*)vr$3, -1ll );
-	__builtin_memset( &TMP$643$1, 0, 24ll );
-	FBSTRING* vr$9 = fb_StrConcat( &TMP$643$1, (void*)vr$6, -1ll, (void*)" (", 3ll );
-	__builtin_memset( &TMP$644$1, 0, 24ll );
-	FBSTRING* vr$12 = fb_StrConcat( &TMP$644$1, (void*)vr$9, -1ll, (void*)vr$2, -1ll );
 	__builtin_memset( &TMP$645$1, 0, 24ll );
-	FBSTRING* vr$15 = fb_StrConcat( &TMP$645$1, (void*)vr$12, -1ll, (void*)"bit)", 5ll );
+	FBSTRING* vr$6 = fb_StrConcat( &TMP$645$1, (void*)"FreeBASIC Compiler - Version 1.06.0 (04-21-2019), built for ", 61ll, (void*)vr$3, -1ll );
+	__builtin_memset( &TMP$646$1, 0, 24ll );
+	FBSTRING* vr$9 = fb_StrConcat( &TMP$646$1, (void*)vr$6, -1ll, (void*)" (", 3ll );
+	__builtin_memset( &TMP$647$1, 0, 24ll );
+	FBSTRING* vr$12 = fb_StrConcat( &TMP$647$1, (void*)vr$9, -1ll, (void*)vr$2, -1ll );
+	__builtin_memset( &TMP$648$1, 0, 24ll );
+	FBSTRING* vr$15 = fb_StrConcat( &TMP$648$1, (void*)vr$12, -1ll, (void*)"bit)", 5ll );
 	fb_PrintString( 0, vr$15, 1 );
 	FBSTRING* vr$16 = fb_StrAllocTempDescZEx( (uint8*)"Copyright (C) 2004-2019 The FreeBASIC development team.", 55ll );
 	fb_PrintString( 0, vr$16, 1 );
