--- a/src/c/Makefile
+++ b/src/c/Makefile
@@ -41,19 +41,16 @@
 all:	nt markup mnt finduses nwmktemp

 nt:	$(TANGLEOBJS)
-	$(CC) $(CFLAGS) -o nt $(TANGLEOBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o nt $(TANGLEOBJS)

 nwmktemp: nwmktemp.o
 	$(CC) $(CFLAGS) -o nwmktemp nwmktemp.o

 finduses:	$(FINDUSESOBJS)
-	$(CC) $(CFLAGS) -o finduses $(FINDUSESOBJS)

 markup:	$(MARKUPOBJS)
-	$(CC) $(CFLAGS) -o markup $(MARKUPOBJS)

 mnt:	$(MNTOBJS)
-	$(CC) $(CFLAGS) -o mnt $(MNTOBJS)

 source:	$(SRCS)
 touch: $(SRCS)
