diff -ur a/contrib/norman/Makefile b/contrib/norman/Makefile
--- a/contrib/norman/Makefile	2018-06-28 23:26:32.000000000 +0200
+++ b/contrib/norman/Makefile	2020-02-25 05:26:26.256610626 +0100
@@ -1,10 +1,10 @@
 LIB=/dev/null	# to be overridden
 DIRS=numarkup
 
-all: ;	for i in $(DIRS); do (cd $$i; make ICONC=$(ICONC) ICONT=$(ICONT) all); done
-install: ;	for i in $(DIRS); do (cd $$i; make LIB=$(LIB) BIN=$(BIN) install); done
-source: ;	for i in $(DIRS); do (cd $$i; make source); done
-clean: ;	for i in $(DIRS); do (cd $$i; make clean); done
+all: ;	for i in $(DIRS); do (cd $$i; $(MAKE) ICONC=$(ICONC) ICONT=$(ICONT) all); done
+install: ;	for i in $(DIRS); do (cd $$i; $(MAKE) LIB=$(LIB) BIN=$(BIN) install); done
+source: ;	for i in $(DIRS); do (cd $$i; $(MAKE) source); done
+clean: ;	for i in $(DIRS); do (cd $$i; $(MAKE) clean); done
 iconlib:	# cheap hack for slackmake
 	true
 
diff -ur a/contrib/norman/numarkup/Makefile b/contrib/norman/numarkup/Makefile
--- a/contrib/norman/numarkup/Makefile	2018-06-28 23:26:32.000000000 +0200
+++ b/contrib/norman/numarkup/Makefile	2020-02-25 05:32:02.290882144 +0100
@@ -11,11 +11,11 @@
 
 all: 
 	noweb -t numarkup.nw
-	make $(TARGET)
+	$(MAKE) $(TARGET)
 
 install:
 	noweb -t numarkup.nw
-	make $(TARGET)
+	$(MAKE) $(TARGET)
 	strip $(TARGET)
 	cp $(TARGET) $(LIB)
 
diff -ur a/Makefile b/Makefile
--- a/Makefile	2018-06-28 23:26:32.000000000 +0200
+++ b/Makefile	2020-02-25 05:24:54.607975694 +0100
@@ -12,14 +12,14 @@
 all:
 	@echo "You have no business running 'make' here; please look at the README file"
 	@exit 1
-source: ;	for i in src; do (cd $$i; make source); done
-www: ;		for i in src/xdoc examples; do (cd $$i; make www); done
+source: ;	for i in src; do (cd $$i; $(MAKE) source); done
+www: ;		for i in src/xdoc examples; do (cd $$i; $(MAKE) www); done
 
 clean:
-	for i in src examples contrib; do (cd $$i; make clean); done
+	for i in src examples contrib; do (cd $$i; $(MAKE) clean); done
 	rm -f nwsrcfilter *~ */*~
 clobber: clean
-	for i in src examples contrib; do (cd $$i; make clobber); done
+	for i in src examples contrib; do (cd $$i; $(MAKE) clobber); done
 
 DATE:
 	(./echo -n "Version $(VERSION) of "; date) > DATE
@@ -51,7 +51,7 @@
 
 ctan:	clean source nwsrcfilter DATE emacscheck
 	chmod +w src/Makefile
-	(cd src; make boot)
+	(cd src; $(MAKE) boot)
 	rm -f ../noweb-$(VERSION)-ctan.zip
 	find ./* ! -type d -not -name FAQ.old -not -name '.git*' -print | ./nwsrcfilter | sed 's@^@noweb/@' | ( ln -s . noweb; zip ../noweb-$(VERSION)-ctan.zip -@; rm -f noweb )
 	chmod -w src/Makefile
@@ -61,6 +61,6 @@
 	diff src/elisp/noweb-mode.el $(HOME)/emacs/noweb-mode.el
 
 checkin:
-	(cd src; make "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin)
+	(cd src; $(MAKE) "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin)
 
 
diff -ur a/src/Makefile b/src/Makefile
--- a/src/Makefile	2020-02-25 05:22:55.103146835 +0100
+++ b/src/Makefile	2020-02-25 05:29:26.557836177 +0100
@@ -41,9 +41,9 @@
 TEXNAME=$(TEXINPUTS)
 
 all: 
-	cd c; make "CC=$(CC)" "$(CFLAGS)" all 
-	for i in shell lib xdoc tex; do (cd $$i; make all); done
-	cd $(LIBSRC); make "ICONT=$(ICONT)" "ICONC=$(ICONC)" all
+	cd c; $(MAKE) "CC=$(CC)" "CFLAGS=$(CFLAGS)" all
+	for i in shell lib xdoc tex; do (cd $$i; $(MAKE) all); done
+	cd $(LIBSRC); $(MAKE) "ICONT=$(ICONT)" "ICONC=$(ICONC)" all
 
 FAQ: FAQ.html
 	sleep 1; html2ascii FAQ.html > FAQ
@@ -101,13 +101,13 @@
 	mkdir -p $(BIN) $(LIB)
 	true c/nt c/markup c/mnt c/finduses c/nwmktemp
 	cp c/nt c/markup c/mnt c/finduses c/nwmktemp $(LIB)
-	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) install
-	cd lib; make LIB=$(LIB) install
+	cd $(LIBSRC); $(MAKE) ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) install
+	cd lib; $(MAKE) LIB=$(LIB) install
 
 uninstall-code: uninstall-shell
 	rm -f $(LIB)/nt $(LIB)/markup $(LIB)/mnt $(LIB)/finduses 
-	cd $(LIBSRC); make ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) uninstall
-	cd lib; make LIB=$(LIB) uninstall
+	cd $(LIBSRC); $(MAKE) ICONT=$(ICONT) ICONC=$(ICONC) LIB=$(LIB) BIN=$(BIN) uninstall
+	cd lib; $(MAKE) LIB=$(LIB) uninstall
 install-man:
 	mkdir -p $(MAN) $(MANDIR) $(MAN7DIR)
 	sed -e "s@|LIBDIR|@$(LIBNAME)@" -e "s@|TEXINPUTS|@$(TEXNAME)@" xdoc/cpif.1 > $(MANDIR)/cpif.$(MANEXT)
@@ -194,26 +194,26 @@
 uninstall-elisp:
 	rm -f $(ELISP)/noweb-mode.el || true
 checkin:
-	for i in lib tex xdoc; do (cd $$i; make "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin); done
+	for i in lib tex xdoc; do (cd $$i; $(MAKE) "CINAME=$(CINAME)" "CIMSG=$(CIMSG)" checkin); done
 	for i in c icon awk; do (cd $$i; ci -l $(CINAME) $(CIMSG) *.nw Makefile); done
 	(cd elisp; ci -l $(CINAME) $(CIMSG) *.el)
 	ci -l $(CINAME) $(CIMSG) Makefile.nw INSTALL INSTALL.DOS README FAQ COPYRIGHT nwmake *.nw
 
 source: FAQ
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make CPIF=">" source); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) CPIF=">" source); done
 	sleep 1
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make touch); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) touch); done
 touch:
 	touch FAQ
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make touch); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) touch); done
 boot:
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make boot); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) boot); done
 clean:
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make clean); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) clean); done
 	rm -f nwsrcfilter *~ */*~
 
 clobber: clean
-	for i in c shell lib xdoc icon awk tex; do (cd $$i; make clobber); done
+	for i in c shell lib xdoc icon awk tex; do (cd $$i; $(MAKE) clobber); done
 
 Makefile: Makefile.nw
 	chmod +w Makefile
