diff -urN a/src/meson.build b/src/meson.build
--- a/src/meson.build	2023-11-19 01:26:19.000000000 -0600
+++ b/src/meson.build	2024-01-08 12:01:26.666552714 -0600
@@ -19,11 +19,11 @@
 # Required dependencies
 dependencies = [
     # ffmpeg libs
-    dependency('libavcodec', version: '>= 59.24.100'),
+    dependency('libavcodec', version: '>= 58.134.100'),
     dependency('libavformat', version: '>= 58.13.100'),
-    dependency('libswresample', version: '>= 4.5.100'),
+    dependency('libswresample', version: '>= 3.9.100'),
     dependency('libavfilter', version: '>= 7.16.100'),
-    dependency('libavutil', version: '>= 57.25.100'),
+    dependency('libavutil', version: '>= 56.70.100'),
 
     # other dependencies
     dependency('libcdio', version: '>= 2.0'),
