--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -99,7 +99,6 @@
 	perl -I ../lib -c $<; \
 
 test_py: $(VERSION_FILE)
-	$(foreach python,$(shell py3versions -r ../debian/control),$(python) setup.py test$(\n))
 
 debbisect.1: debbisect
 	help2man \
@@ -131,7 +130,6 @@
 	cp $< $@
 
 clean:
-	python3 setup.py clean -a
 	find -name '*.pyc' -delete
 	find -name __pycache__ -delete
 	rm -rf devscripts.egg-info $(BC_BUILD_DIR) .pylint.d
@@ -142,7 +140,6 @@
 test: test_pl test_sh test_py
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
