--- a/po4a/Makefile
+++ b/po4a/Makefile
@@ -34,8 +34,8 @@
 	pod2man --utf8 --center=" " --release="$(DESC_$(dir $@))" $< > $@
 %.1:: %.dbk translate
 	xsltproc --nonet \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
-	# /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/other.xsl
+	  /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl $<
+	# /usr/share/sgml/docbook/xsl-stylesheets/manpages/other.xsl
 	# (which is imported by the above stylesheet) insists in writing the output
 	# to where it wants to.  we can only move the file ourselves.
 	# ($* → de/deb-reversion.de)
--- a/scripts/deb-reversion.dbk
+++ b/scripts/deb-reversion.dbk
@@ -5,13 +5,13 @@
 <!--
 
 Process this file with an XSLT processor: `xsltproc \
--''-nonet /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/\
+-''-nonet /usr/share/sgml/docbook/xsl-stylesheets/\
 manpages/docbook.xsl manpage.dbk'.  A manual page
 <package>.<section> will be generated.  You may view the
 manual page with: nroff -man <package>.<section> | less'.  A
 typical entry in a Makefile or Makefile.am is:
 
-DB2MAN=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/\
+DB2MAN=/usr/share/sgml/docbook/xsl-stylesheets/\
 manpages/docbook.xsl
 XP=xsltproc -''-nonet
 
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -87,7 +87,7 @@
 	pod2man --utf8 --center=" " --release="Debian Utilities" $< > $@
 %.1: %.dbk
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+	  /usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl $<
 
 # Syntax checker
 test_sh: $(SH_CHECKS)
