diff --git a/test/dune b/test/dune
index 951a008..9d86605 100644
--- a/test/dune
+++ b/test/dune
@@ -1,6 +1,6 @@
 (executables
  (names getifaddrs_test nonblock_read open_close_test set_ipv4_test nonblock_test sendfd_test)
- (libraries lwt.unix tuntap macaddr ipaddr oUnit))
+ (libraries lwt.unix tuntap macaddr ipaddr ounit2))
 
 (alias
  (name    runtest)
diff --git a/tuntap.opam b/tuntap.opam
index 08002b2..87148c1 100644
--- a/tuntap.opam
+++ b/tuntap.opam
@@ -27,7 +27,7 @@ depends: [
   "ipaddr" {>= "5.0.0"}
   "macaddr" {>= "4.0.0"}
   "cmdliner"
-  "ounit" {with-test}
+  "ounit2" {with-test}
   "lwt" {with-test & >= "5.0.0"}
 ]
 build: [
