--- a/lib/RK/RkGetDic.man
+++ b/lib/RK/RkGetDic.man
@@ -41,7 +41,7 @@
 Dictionary names are got in the area \f2dics\f1 in the following format:
 .P
 .RS 0.2c
-dictionary-name-1 бя0 dictionary-name-2 бя0 ... dictionary-name-n бя0 бя0
+dictionary-name-1 \\0 dictionary-name-2 \\0 ... dictionary-name-n \\0 \\0
 .RE
 .P
 The size of the area \f2dics\f1 is given in bytes by \f2maxdics\f1.  The dictionary list is truncated in character strings when it exceeds this size.
