--- a/cmd/crxgram/crxgram.c
+++ b/cmd/crxgram/crxgram.c
@@ -485,7 +485,7 @@
 	    int j;
 
 	    nextS = scanToken(nextS, namevec, sizeof(namevec));
-	    name = strtok(namevec, SEP);
+	    name = strtok((char *)namevec, SEP);
 	    for (j = 0; name && j < 255;) {
 		if (r = probeIdent(Row, name))
 		    rdata[i][j++] = r->rownum;
--- a/cmd/cshost/cshost.c
+++ b/cmd/cshost/cshost.c
@@ -152,12 +152,12 @@
 	    goto protoerr;
 	assert(wp + 2 <= endp);
 	printf("HOST NAME:%s\n", wp ) ;
-	wp += strlen( wp ) + 1 ;
+	wp += strlen((char *) wp ) + 1 ;
 	if( *wp ) {
 	    printf("USER NAME:" ) ;
 	    while( *wp ) {
 		printf("%s ", wp ) ;
-		wp += strlen( wp ) + 1 ;
+		wp += strlen((char *) wp ) + 1 ;
 	    }
 	} else
 	    printf("ALL USER" ) ;
--- a/cmd/wtoc/wtoc.c
+++ b/cmd/wtoc/wtoc.c
@@ -144,8 +144,8 @@
      int	hindo;
 {
   if( !strcmp((char *)hinshi,"#kxuru") || !strcmp((char *)hinshi,"#sxuru")){
-    strcat(yomi,"る");
-    strcat(kouho,"る");
+    strcat((char *)yomi,"る");
+    strcat((char *)kouho,"る");
   }
   if( hindo == 0 )
     fprintf( fp, "%s %s %s \n", yomi, hinshi, kouho );
--- a/lib/canna/ebind.c
+++ b/lib/canna/ebind.c
@@ -364,7 +364,7 @@
     /* list_cbはKC_setListCallbackでd->elistcbに引っ越す */
     list_cb.client_data = (char *)arg;
     list_cb.callback_func = &EUCListCallback;
-    ret = XwcKanjiControl2(display, window, request, (char *)&list_cb);
+    ret = XwcKanjiControl2(display, window, request, (BYTE *)&list_cb);
     goto return_ret;
     /* FALLTHROUGH */
   default: /* ワイドでもEUCでも変わらないもの */
--- a/lib/canna/obind.c
+++ b/lib/canna/obind.c
@@ -500,7 +500,7 @@
     /* list_cbはKC_setListCallbackでd->elistcbに引っ越す */
     list_cb.client_data = (char *)arg;
     list_cb.callback_func = &owcListCallback;
-    ret = XwcKanjiControl2(display, window, request, (char *)&list_cb);
+    ret = XwcKanjiControl2(display, window, request, (BYTE *)&list_cb);
     goto return_ret;
   default: /* 新ワイドと変わらないもの */
     ret = XwcKanjiControl2(display, window, request, arg);
--- a/server/convert.c
+++ b/server/convert.c
@@ -1506,7 +1506,7 @@
 const char *username;
 const char *hostname;
 {
-    int (*ReqCallFunc) pro((char *, int)) ;
+    int (*ReqCallFunc) pro((BYTE *, int)) ;
     register Req0 *req0 = &Request.Request0 ;
     const char *username0 = username ? username : "";
     const char *hostname0 = hostname ? hostname : "";
--- a/server/wconvert.c
+++ b/server/wconvert.c
@@ -189,7 +189,7 @@
 size_t size;
 {
     ir_debug( Dmsg(10, "WriteClient:") );
-    ir_debug( DebugDump( 10, buf, size ) );
+    ir_debug( DebugDump( 10, (const char *)buf, size ) );
     return ClientBuf_store_reply(client->client_buf, buf, size);
 }
 #else
@@ -1372,8 +1372,8 @@
 	/* 以下、パーミッションのチェック */
 	stat = ACCES;
 	if (checkPermissionToRead(client, dirname, (char *)0) >= 0) {
-	  stat = RkwListDic(cxnum, (unsigned char *)dirname,
-			    (unsigned char *)dicnames, requestsize);
+	  stat = RkwListDic(cxnum, (char *)dirname,
+			    (char *)dicnames, requestsize);
 	}
 
       listdicdone:
@@ -1403,7 +1403,7 @@
 	ir_debug( Dmsg(5, "dicname = %s\n",
 		       (req->dicname)?req->dicname:null) );
 
-	stat = (char)RkwCreateDic(cxnum, (unsigned char *)req->dicname,
+	stat = (char)RkwCreateDic(cxnum, (char *)req->dicname,
 				  req->mode);
     }
 
@@ -1438,7 +1438,7 @@
 	ir_debug( Dmsg(5, "dicname = %s\n",
 		       (req->dicname)?req->dicname:null) );
 
-	stat = (char)RkwRemoveDic(cxnum, (unsigned char *)req->dicname,
+	stat = (char)RkwRemoveDic(cxnum, (char *)req->dicname,
 				  req->mode);
     }
 
@@ -1456,8 +1456,8 @@
 
     if (validcontext(cxnum, client, wRenameDictionary)) {
 	diclen = strlen(req->dicname) + 1 ;
-	stat = RkwRenameDic(cxnum, (unsigned char *)req->dicname,
-			    (unsigned char *)&(req->dicname)[diclen],
+	stat = RkwRenameDic(cxnum, (char *)req->dicname,
+			    (char *)&(req->dicname)[diclen],
 			    req->mode);
     }
     return SendType2Reply(client, wRenameDictionary, EXTPROTO, stat);
@@ -1950,8 +1950,8 @@
       ir_debug( Dmsg(5, "dicname = %s\n", Request.type15.dicname) );
       ir_debug( Dmsg(5, "username = %s\n", username) );
       
-      stat = (RkwQueryDic(cxnum, (unsigned char *)username,
-			  (unsigned char *)Request.type15.dicname,
+      stat = (RkwQueryDic(cxnum, (char *)username,
+			  (char *)Request.type15.dicname,
 			  dicinfo) < 0) ? -1 : 0;
       
       if (stat < 0) {
