diff -Naur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -73,17 +73,16 @@
 endif()
 
 #======================================================
-# Defining the static and shared gtk-fortran libraries:
+# Defining the shared gtk-fortran libraries:
 #======================================================
 add_library(gtk-fortran_object OBJECT ${sources})
 # To build position independent shared libraries:
 set_property(TARGET gtk-fortran_object PROPERTY POSITION_INDEPENDENT_CODE TRUE)
 
-add_library(gtk-fortran_static STATIC $<TARGET_OBJECTS:gtk-fortran_object>)
 add_library(gtk-fortran_shared SHARED $<TARGET_OBJECTS:gtk-fortran_object>)
 target_link_libraries(gtk-fortran_shared ${GTK_LIBRARIES})
 
-set_target_properties(gtk-fortran_static gtk-fortran_shared PROPERTIES 
+set_target_properties(gtk-fortran_shared PROPERTIES 
   OUTPUT_NAME ${gtk_V_fortran}
   VERSION "0.1"
   SOVERSION "0.1")
@@ -131,7 +130,7 @@
 #==================
 # Files to install:
 #==================
-install(TARGETS gtk-fortran_static gtk-fortran_shared
+install(TARGETS gtk-fortran_shared
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
