--- a/bootloader/wscript
+++ b/bootloader/wscript
@@ -652,11 +652,6 @@ def configure(ctx):
     if ctx.env.DEST_OS == 'linux' and ctx.check_cc(cflags='-Wl,--as-needed'):
         ctx.env.append_value('LINKFLAGS', '-Wl,--as-needed')
 
-    if ctx.env.CC_NAME != 'msvc':
-        # This tool allows reducing the size of executables.
-        ctx.find_program([assoc_programm(ctx, 'strip')], var='STRIP')
-        ctx.load('strip', tooldir='tools')
-
     def windowed(name, baseenv):
         """Setup windowed environment based on `baseenv`."""
         ctx.setenv(name, baseenv)  # Inherit from `baseenv`.
@@ -731,7 +726,7 @@ def build(ctx):
             includes='zlib')
 
     # By default strip final executables to make them smaller.
-    features = 'strip'
+    features = ''
     if ctx.env.CC_NAME == 'msvc':
         # Do not strip bootloaders when using MSVC.
         features = ''
