--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,9 +87,6 @@
 SET_SOURCE_FILES_PROPERTIES(config.h.in PROPERTIES GENERATED FALSE)
 SET_SOURCE_FILES_PROPERTIES(${PROJECT_BINARY_DIR}/config.h PROPERTIES GENERATED TRUE)
 
-# Special Install Files
-INSTALL(FILES README.md DESTINATION ${CMAKE_INSTALL_DATADIR}/kvtree)
-
 # Package
 SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "KV Storage Library for Perl style hashes")
 SET(CPACK_PACKAGE_VENDOR "Lawrence Livermore National Laboratory")
