--- a/conf/configure.in
+++ b/conf/configure.in
@@ -66,7 +66,6 @@
 AC_PROG_YACC
 AC_PROG_LN_S
 
-AC_PATH_PROG(AR, ar)
 ARFLAGS="crv"
 AC_SUBST(AR)
 AC_SUBST(ARFLAGS)
@@ -87,9 +86,6 @@
 STDCFLAGS="-D__STDC_LIMIT_MACROS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS"
 CXXFLAGS="$CXXFLAGS -fPIC $STDCFLAGS"
 
-dnl === Ignore user-requested debug/optimization levels
-CXXFLAGS=`echo $CXXFLAGS | sed 's/-g //; s/-O //; s/-O1 //; s/-O2 //; s/-O3 //'`
-
 SOFLAGS="$SOFLAGS -fPIC -shared -rdynamic"
 AC_SUBST(SOFLAGS)
 
