--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,5 @@
 # Linux
-CC=c++
-LIBS=-lcanna -ltermcap
-CFLAGS=-O2 -g
+LIBS += -lcanna
 
 # Solaris 2.6J
 #CC=c++
@@ -19,10 +17,10 @@
 all: $(TARGET)
 
 $(TARGET): $(OBJS)
-	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LIBS)
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -fPIE -o $@ $(OBJS) $(LIBS)
 
 .C.o:
-	$(CC) $(CFLAGS) -c $<
+	$(CXX) $(CXXFLAGS) -fPIC -c $<
 
 clean:
 	rm -rf $(TARGET) $(OBJS)
