unbundle KDStateMachineviewer
--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -34,7 +34,6 @@ if(GAMMARAY_WITH_KDSME)
         remove_definitions(-DQT_DISABLE_DEPRECATED_BEFORE=0x050500)
         add_subdirectory(KDStateMachineEditor)
     endfunction()
-    add_ksme_subdirectory()
 endif()
 
 add_subdirectory(kde)
--- a/plugins/statemachineviewer/CMakeLists.txt
+++ b/plugins/statemachineviewer/CMakeLists.txt
@@ -81,6 +81,7 @@ if(GAMMARAY_BUILD_UI AND GAMMARAY_WITH_KDSME)
         ${gammaray_statemachineviewer_shared_srcs}
     )
 
+    find_package(KDSME-qt6 REQUIRED)
     target_link_libraries(
         gammaray_statemachineviewer_ui_plugin
         KDSME::Core
--- a/plugins/statemachineviewer/statemachineviewerwidget.cpp
+++ b/plugins/statemachineviewer/statemachineviewerwidget.cpp
@@ -25,14 +25,14 @@
 #include <ui/contextmenuextension.h>
 #include <ui/clientdecorationidentityproxymodel.h>
 
-#include <elementmodel.h>
-#include <layoutproperties.h>
-#include <state.h>
-#include <transition.h>
-#include <runtimecontroller.h>
-#include <statemachinescene.h>
-#include <statemachinetoolbar.h>
-#include <statemachineview.h>
+#include <kdstatemachineeditor/core/elementmodel.h>
+#include <kdstatemachineeditor/core/layoutproperties.h>
+#include <kdstatemachineeditor/core/state.h>
+#include <kdstatemachineeditor/core/transition.h>
+#include <kdstatemachineeditor/core/runtimecontroller.h>
+#include <kdstatemachineeditor/view/statemachinescene.h>
+#include <kdstatemachineeditor/view/statemachinetoolbar.h>
+#include <kdstatemachineeditor/view/statemachineview.h>
 
 #include <QDebug>
 #include <QMenu>
