--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -193,10 +193,8 @@
 
 # std::format
 set(STL_FORMAT_USE_EXTERNAL_FMTLIB ${USE_EXTERNAL_FMTLIB} CACHE BOOL "" FORCE)
-if (NOT TARGET stl_polyfill::format)
-    add_subdirectory(external/std-format)
-endif()
-target_link_libraries(ananicy-cpp PRIVATE stl_polyfill::format)
+find_package(StlPolyfillFormat REQUIRED)
+target_include_directories(ananicy-cpp PRIVATE "/usr/include/polyfills/format")
 
 
 # spdlog
@@ -242,4 +240,4 @@
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/ananicy-cpp.service
             DESTINATION lib/systemd/system/
             COMPONENT Runtime)
-endif()
\ Manca newline alla fine del file
+endif()
