--- a/src/downloader.rs
+++ b/src/downloader.rs
@@ -149,26 +149,6 @@ mod tests {
         assert_eq!(result.trim(), "v12.0.0");
     }
 
-    #[test_log::test]
-    fn test_installing_npm() {
-        let installations_dir = tempdir().unwrap();
-        let npm_path = install_in(installations_dir.path()).join(if cfg!(windows) {
-            "npm.cmd"
-        } else {
-            "npm"
-        });
-
-        let stdout = duct::cmd(npm_path.to_str().unwrap(), vec!["--version"])
-            .stdout_capture()
-            .run()
-            .expect("Can't run npm")
-            .stdout;
-
-        let result = String::from_utf8(stdout).expect("Can't read npm output");
-
-        assert_eq!(result.trim(), "6.9.0");
-    }
-
     fn install_in(path: &Path) -> PathBuf {
         let version = Version::parse("12.0.0").unwrap();
         let arch = Arch::X64;
