diff --git a/Cargo.toml b/Cargo.toml
index a84c611..de78e58 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,7 +25,7 @@ log = "0.4.14"
 env_logger = "0.9.0"
 atty = "0.2.14"
 encoding_rs_io = "0.1.7"
-reqwest = { git = "https://github.com/seanmonstar/reqwest.git", rev = "d92d2aa3ce4667faa38454c8dae4fa9f72b91b71", features = ["blocking", "json", "rustls-tls", "rustls-tls-native-roots", "brotli"], default-features = false }
+reqwest = { path = @@REQWEST_PATH@@, features = ["blocking", "json", "rustls-tls", "rustls-tls-native-roots", "brotli"], default-features = false }
 url = "2.2.2"
 sysinfo = "0.23.2"
 thiserror = "1.0.30"
