diff -Npur a/data/phosh.in b/data/phosh.in
--- a/data/phosh.in	2021-03-20 13:53:32.367786000 +0100
+++ b/data/phosh.in	2021-03-20 13:55:19.327789721 +0100
@@ -56,6 +56,10 @@ elif  [ -f /etc/phosh/rootston.ini ]; th
   PHOC_INI=/etc/phosh/rootston.ini
 fi
 
+# Set proper locale according to /etc/locale.conf
+source /etc/locale.conf
+export LANG=$LANG
+
 # Run gnome-session through a login shell so it picks
 # variables from /etc/profile.d (XDG_*)
 [ -n "$WLR_BACKENDS" ] || WLR_BACKENDS=drm,libinput
