--- a/basho.ctd
+++ b/basho.ctd
@@ -1,4 +1,3 @@
-# $Id: basho.ctd,v 1.1 2001/01/26 21:02:02 lain Exp $
 あいお #CN 秋穂
 あいおい #CN 相生
 あいかわ #JCN*1 愛川
--- a/kaom.ctd
+++ b/kaom.ctd
@@ -1,4 +1,3 @@
-# $Id: kaom.ctd,v 1.1 2001/01/21 23:42:22 lain Exp $
 あうー、 #T35 (>.<)
 あげて、 #T35 (ノ^^)ノ
 あせ、 #T35*15 (^^; (;^_^A
--- a/keisan.ctd
+++ b/keisan.ctd
@@ -1,4 +1,3 @@
-# $Id: keisan.ctd,v 1.1 2001/01/26 21:15:34 lain Exp $
 あーかいば #T35*1 アーカイバ
 あーかいぶ #T30*8 アーカイブ
 あーきてくちゃ #T35*25 アーキテクチャ
--- a/pub.ctd
+++ b/pub.ctd
@@ -1,4 +1,3 @@
-# $Id: kaom.ctd,v 1.1 2001/05/30 19:57:12 lain Exp $
 ! #T35*76 ！ 感嘆符
 !! #T35*76 ！！
 != #T35*76 ≠
--- a/scien.ctd
+++ b/scien.ctd
@@ -1,4 +1,3 @@
-# $Id: suppli.ctd,v 1.1 2001/01/26 21:28:56 lain Exp $
 あーかいぶ #T30 アーカイブ
 あーく #T35 アーク
 あーくこうげん #T35 アーク光源
--- a/sup.ctd
+++ b/sup.ctd
@@ -1,4 +1,3 @@
-# $Id: suppli.ctd,v 1.1 2001/05/30 20:00:09 lain Exp $
 あーかいう゛ #T30 アーカイヴ
 あーかんそー #CN アーカンソー
 あーけーど #T35 アーケード
