--- a/src/liblsquic/CMakeLists.txt
+++ b/src/liblsquic/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Copyright (c) 2017 - 2022 LiteSpeed Technologies Inc.  See LICENSE.
 SET(lsquic_STAT_SRCS
-    ls-qpack/lsqpack.c
+#    ls-qpack/lsqpack.c
     lsquic_adaptive_cc.c
     lsquic_alarmset.c
     lsquic_arr.c
@@ -100,10 +100,10 @@ IF(PROJECT_NAME STREQUAL "openlitespeed")
     INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/src/spdy)
 ELSE()
     INCLUDE_DIRECTORIES(../lshpack)
-    SET(lsquic_STAT_SRCS ${lsquic_STAT_SRCS}
-        lsquic_xxhash.c
-        ../lshpack/lshpack.c
-    )
+#    SET(lsquic_STAT_SRCS ${lsquic_STAT_SRCS}
+#        lsquic_xxhash.c
+#        ../lshpack/lshpack.c
+#    )
 ENDIF()
 
 ADD_CUSTOM_COMMAND(
@@ -119,6 +119,8 @@ SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DXXH_HEADER_NAME=\\\"lsquic_xxhash.h\\\"")
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DLSQPACK_ENC_LOGGER_HEADER=\\\"lsquic_qpack_enc_logger.h\\\"")
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DLSQPACK_DEC_LOGGER_HEADER=\\\"lsquic_qpack_dec_logger.h\\\"")
 
+link_libraries(ls-qpack ls-hpack)
+
 if(NEED_LIBRT_FOR_clock_getres)
     target_link_libraries(lsquic PRIVATE ${RT_LIBRARY})
 endif()
