diff --git a/configure.ac b/configure.ac
index 9ee0a42..4a56e93 100644
--- a/configure.ac
+++ b/configure.ac
@@ -459,8 +459,8 @@ AC_ARG_ENABLE([static],
 )
 AM_CONDITIONAL(ENABLE_STATIC, test "$enable_static" = yes)
 
-AC_CHECK_LIB([tinfo], [initscr], tinfo=1,
-    AC_MSG_WARN([*** tinfo library (libtinfo) not found]))
+AC_CHECK_LIB([tinfow], [initscr], tinfo=1,
+    AC_MSG_WARN([*** tinfow library (libtinfo) not found]))
 AM_CONDITIONAL(ENABLE_TINFO, test "$tinfo" = 1)
 
 ##memory tracing##
diff --git a/src/Makefile.am b/src/Makefile.am
index c69cd2e..dd27e9f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -22,7 +22,7 @@ LDADD_static=-lcrypto -ldl
 endif
 
 if ENABLE_TINFO
-LIBS+=-ltinfo
+LIBS+=-ltinfow
 endif
 
 AM_CFLAGS=-D_FILE_OFFSET_BITS=64
