--- a/libsigrokdecode4DSL/configure.ac	2020-05-11 21:07:38.000000000 +0800
+++ b/libsigrokdecode4DSL/configure.ac	2021-06-05 01:54:35.937525565 +0800
@@ -90,7 +90,7 @@
 
 # Python 3 is always needed.
 SR_PKG_CHECK([python3], [SRD_PKGLIBS],
-	[python3 >= 3.2], [python-3.7 >= 3.7], [python-3.6 >= 3.6], [python-3.5 >= 3.5], [python-3.4 >= 3.4], [python-3.3 >= 3.3], [python-3.2 >= 3.2])
+	[python3 >= 3.2], [python-3.10 >= 3.10], [python-3.9 >= 3.9], [python-3.8 >= 3.8], [python-3.7 >= 3.7], [python-3.6 >= 3.6], [python-3.5 >= 3.5], [python-3.4 >= 3.4], [python-3.3 >= 3.3], [python-3.2 >= 3.2])
 AS_IF([test "x$sr_have_python3" = xno],
 	[AC_MSG_ERROR([Cannot find Python 3 development headers.])])
 
