https://bugs.gentoo.org/887913
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c40c925..41e5230 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -594,8 +594,6 @@ endif()
 set(CMAKE_CXX_FLAGS "-Wall -Wextra")
 set(CMAKE_BUILD_TYPE Release)
 set(CMAKE_CXX_FLAGS_DEBUG "-g")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3")
-add_compile_options(-O3)
 
 #===============================================================================
 #= Linker Configuration
