diff --git a/src/hin/conf.h b/src/hin/conf.h
index c4f7f4c..29b9375 100644
--- a/src/hin/conf.h
+++ b/src/hin/conf.h
@@ -2,9 +2,9 @@
 #ifndef HIN_CONF_H
 #define HIN_CONF_H
 
-#define HIN_CONF_PATH "workdir/main.lua"
-#define HIN_LOGDIR_PATH "build/"
-#define HIN_CWD_PATH "./"
+#define HIN_CONF_PATH "/etc/hinsightd/hinsightd.lua"
+#define HIN_LOGDIR_PATH "/var/log/hinsightd/"
+#define HIN_CWD_PATH "/var/www/localhost/"
 
 #define HIN_URING_QUEUE_DEPTH 256
 #define HIN_URING_DONT_FORK 0
@@ -20,8 +20,8 @@
 #define HIN_HTTPD_MAX_HEADER_LINE_SIZE 4096
 #define HIN_HTTPD_MAX_POST_SIZE 4121440
 #define HIN_HTTPD_MAX_DEFLATE_SIZE 0
-#define HIN_HTTPD_POST_DIRECTORY "/tmp"
-#define HIN_HTTPD_CACHE_DIRECTORY "/tmp"
+#define HIN_HTTPD_POST_DIRECTORY "/var/tmp/hinsightd"
+#define HIN_HTTPD_CACHE_DIRECTORY "/var/tmp/hinsightd"
 #define HIN_HTTPD_CACHE_CLEAN_ON_EXIT 1
 #define HIN_HTTPD_CACHE_TMPFILE 0
 #define HIN_HTTPD_CACHE_MAX_SIZE 64 * 1024 * 1024
