From a3bf429c886407198e7e3a292af80e6c60ee27d2 Mon Sep 17 00:00:00 2001
From: jrudess <jordan.r.rudess@gmail.com>
Date: Sat, 5 Nov 2022 16:24:04 -0700
Subject: [PATCH] Provide cmake override for shared-lib name

---
 source/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/source/CMakeLists.txt b/source/CMakeLists.txt
index 58b73678..f9ac7cd4 100644
--- a/source/CMakeLists.txt
+++ b/source/CMakeLists.txt
@@ -86,6 +86,7 @@ add_library(
 
 add_subdirectory(ast)
 
+set(SLANG_SHARED_LIB_NAME slang)
 add_library(slang::slang ALIAS slang_slang)
 set_target_properties(
   slang_slang
@@ -93,8 +94,8 @@ set_target_properties(
              VISIBILITY_INLINES_HIDDEN YES
              VERSION ${PROJECT_VERSION}
              SOVERSION ${PROJECT_VERSION_MAJOR}
-             EXPORT_NAME slang
-             OUTPUT_NAME slang)
+             EXPORT_NAME ${SLANG_SHARED_LIB_NAME}
+             OUTPUT_NAME ${SLANG_SHARED_LIB_NAME})
 
 # Compile options
 target_compile_options(slang_slang PRIVATE ${SLANG_WARN_FLAGS})
