Author: Ralf Treinen <treinen@debian.org>
Description: fix a spelling error in help text

Index: minisat+/Main.C
===================================================================
--- minisat+.orig/Main.C	2011-05-24 20:59:49.000000000 +0200
+++ minisat+/Main.C	2016-12-19 07:29:13.894553743 +0100
@@ -79,7 +79,7 @@
     "\n"
     "  -bdd-thres=   Threshold for prefering BDDs in mixed mode.        [def: %g]\n"
     "  -sort-thres=  Threshold for prefering sorters. Tried after BDDs. [def: %g]\n"
-    "  -goal-bias=   Bias goal function convertion towards sorters.     [def: %g]\n"
+    "  -goal-bias=   Bias goal function conversion towards sorters.     [def: %g]\n"
     "\n"
     "  -1 -first     Don\'t minimize, just give first solution found\n"
     "  -A -all       Don\'t minimize, give all solutions\n"
