# We provide media-libs/imgui in guru

--- a/meson.build
+++ b/meson.build
@@ -251,8 +251,7 @@ elif sizeof_ptr == 4
   pre_args += '-DMANGOHUD_ARCH="32bit"'
 endif

-dearimgui_sp = subproject('imgui', default_options: imgui_options)
-dearimgui_dep = dearimgui_sp.get_variable('imgui_dep')
+dearimgui_dep = dependency('imgui')

 spdlog_dep = cpp.find_library('spdlog', required: get_option('use_system_spdlog'))
 if not spdlog_dep.found()
