--- a/other/other.c
+++ b/other/other.c
@@ -328,11 +328,7 @@ static const char Pad64 = '=';
    */
 
 int
-b64_ntop(src, srclength, target, targsize)
-	u_char const *src;
-	size_t srclength;
-	char *target;
-	size_t targsize;
+b64_ntop(u_char const *src, size_t srclength, char *target, size_t targsize)
 {
 	size_t datalength = 0;
 	u_char input[3];
@@ -392,10 +388,7 @@ b64_ntop(src, srclength, target, targsize)
  */
 
 int
-b64_pton(src, target, targsize)
-	char const *src;
-	u_char *target;
-	size_t targsize;
+b64_pton(char const *src, u_char *target, size_t targsize)
 {
 	int tarindex, state, ch;
 	u_char nextbyte;
diff --git a/other/other.h b/other/other.h
index 08701fa..9ff1c0d 100644
--- a/other/other.h
+++ b/other/other.h
@@ -3,6 +3,9 @@
 
 #define _PATH_TTY "/dev/tty"
 
+int b64_ntop(u_char const *src, size_t srclength, char *target, size_t targsize);
+int b64_pton(char const *src, u_char *target, size_t targsize);
+
 #ifndef HAVE_STRLCAT
 size_t strlcat(char *dst, const char *src, size_t siz);
 #endif
