--- a/Makefile
+++ b/Makefile
@@ -11,10 +11,10 @@
 all: lisp lispc LISP-INI
 
 lisp : lisp1.o lisp2.o crc cri
-	gcc -ggdb -O0 -o lisp lisp1.o lisp2.o -lm
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -o lisp lisp1.o lisp2.o -lm
 
 lispc : lisp1c.o lisp2.o lispc1.o
-	gcc -o $@ $^ -lm
+	${CC} ${CFLAGS} -fPIC -o $@ $^ -lm
 	./cri comp
 
 LISP-INI : $(COMPILER_FILES) cri
@@ -34,25 +34,25 @@
 	./crc comp
 
 lisp1.o : lisp1.c
-	gcc -ggdb -O0 -Wall -pedantic -c lisp1.c
+	${CC} ${CFLAGS} -fPIC -Wall -pedantic -c lisp1.c
 
 lisp2.o : lisp2.c
-	gcc -ggdb -O0 -Wall -pedantic -Wno-parentheses -c lisp2.c
+	${CC} ${CFLAGS} -fPIC -Wall -pedantic -Wno-parentheses -c lisp2.c
 
 $(COMPILER_FILES): compiler.lsp lap.lsp lisp
 	./compile-compiler
 
 cr1 : cr1.c crfile.h
-	gcc -Wall -pedantic -o cr1 cr1.c
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -Wall -pedantic -o cr1 cr1.c
 
 cr2 : cr2.c crfile.h
-	gcc -Wall -pedantic -o cr2 cr2.c
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -Wall -pedantic -o cr2 cr2.c
 
 crc : crc.c crfile.h
-	gcc -Wall -pedantic -o crc crc.c
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -Wall -pedantic -o crc crc.c
 
 cri : cri.c
-	gcc -Wall -pedantic -o cri cri.c
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -Wall -pedantic -o cri cri.c
 
 clean:
 	rm -f lisp*.c *.o *~ *.bak
@@ -62,7 +62,7 @@
 	rm -f lisp lispc cr? LISP-INI
 
 sizes :  sizes.c flags.l
-	gcc -Wall -pedantic -o sizes sizes.c
+	${CC} ${CFLAGS} ${LDFLAGS} -fPIE -Wall -pedantic -o sizes sizes.c
 
 install: lisp
 	mv lisp /usr/local/bin
