--- a/Makefile
+++ b/Makefile
@@ -23,8 +23,8 @@ install:
 		install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-install" "$(LIB_DIR)/initcpio/install/grub-btrfs-overlayfs"; \
 		install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-hook" "$(LIB_DIR)/initcpio/hooks/grub-btrfs-overlayfs"; \
 	 fi
-	@install -Dm644 -t "$(SHARE_DIR)/doc/$(PKGNAME)/" README.md
-	@install -Dm644 "initramfs/readme.md" "$(SHARE_DIR)/doc/$(PKGNAME)/initramfs-overlayfs.md"
+#	@install -Dm644 -t "$(SHARE_DIR)/doc/$(PKGNAME)/" README.md
+#	@install -Dm644 "initramfs/readme.md" "$(SHARE_DIR)/doc/$(PKGNAME)/initramfs-overlayfs.md"
 
 uninstall:
 	@if test "$(shell id -u)" != 0; then \
