diff --git a//init.py b//init.py
index 495c15d..1670972 100755
--- a//init.py
+++ b//init.py
@@ -19,9 +19,9 @@
 import sys
 import getpass
 from os import path, remove, makedirs
-from PySide2 import QtGui, QtCore, QtWidgets
-from PySide2.QtCore import QSize
-from PySide2.QtGui import QIcon
+from qtpy import QtGui, QtCore, QtWidgets
+from qtpy.QtCore import QSize
+from qtpy.QtGui import QIcon
 from util.QtSingleApplication import QtSingleApplication
 from ui.mainwindow import Ui_MainWindow
 from ui.license import Ui_Form as License_Ui_Form
diff --git a//ui/about.py b//ui/about.py
index 248eae8..30d6f72 100644
--- a//ui/about.py
+++ b//ui/about.py
@@ -8,12 +8,12 @@
 ## WARNING! All changes made in this file will be lost when recompiling UI file!
 ################################################################################
 
-from PySide2.QtCore import (QCoreApplication, QDate, QDateTime, QMetaObject,
+from qtpy.QtCore import (QCoreApplication, QDate, QDateTime, QMetaObject,
     QObject, QPoint, QRect, QSize, QTime, QUrl, Qt)
-from PySide2.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
+from qtpy.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
     QFontDatabase, QIcon, QKeySequence, QLinearGradient, QPalette, QPainter,
     QPixmap, QRadialGradient)
-from PySide2.QtWidgets import *
+from qtpy.QtWidgets import *
 
 
 class Ui_Form(object):
diff --git a//ui/help.py b//ui/help.py
index 1eba729..a458f0e 100644
--- a//ui/help.py
+++ b//ui/help.py
@@ -8,12 +8,12 @@
 ## WARNING! All changes made in this file will be lost when recompiling UI file!
 ################################################################################
 
-from PySide2.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
+from qtpy.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
     QRect, QSize, QUrl, Qt)
-from PySide2.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
+from qtpy.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
     QFontDatabase, QIcon, QLinearGradient, QPalette, QPainter, QPixmap,
     QRadialGradient)
-from PySide2.QtWidgets import *
+from qtpy.QtWidgets import *
 
 
 class Ui_Form(object):
diff --git a//ui/license.py b//ui/license.py
index 7cfa021..370992b 100644
--- a//ui/license.py
+++ b//ui/license.py
@@ -8,12 +8,12 @@
 ## WARNING! All changes made in this file will be lost when recompiling UI file!
 ################################################################################
 
-from PySide2.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
+from qtpy.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
     QRect, QSize, QUrl, Qt)
-from PySide2.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
+from qtpy.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
     QFontDatabase, QIcon, QLinearGradient, QPalette, QPainter, QPixmap,
     QRadialGradient)
-from PySide2.QtWidgets import *
+from qtpy.QtWidgets import *
 
 
 class Ui_Form(object):
diff --git a//ui/mainwindow.py b//ui/mainwindow.py
index f8ccf11..4d50ff6 100644
--- a//ui/mainwindow.py
+++ b//ui/mainwindow.py
@@ -8,12 +8,12 @@
 ## WARNING! All changes made in this file will be lost when recompiling UI file!
 ################################################################################
 
-from PySide2.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
+from qtpy.QtCore import (QCoreApplication, QMetaObject, QObject, QPoint,
     QRect, QSize, QUrl, Qt)
-from PySide2.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
+from qtpy.QtGui import (QBrush, QColor, QConicalGradient, QCursor, QFont,
     QFontDatabase, QIcon, QLinearGradient, QPalette, QPainter, QPixmap,
     QRadialGradient)
-from PySide2.QtWidgets import *
+from qtpy.QtWidgets import *
 
 
 class Ui_MainWindow(object):
diff --git a//util/QtSingleApplication.py b//util/QtSingleApplication.py
index 4dec38f..13a01a6 100644
--- a//util/QtSingleApplication.py
+++ b//util/QtSingleApplication.py
@@ -1,11 +1,11 @@
 # -*- coding: utf-8 -*-
-from PySide2.QtCore import Signal, QTextStream, Qt
-# from PySide2.QtGui import *
-from PySide2.QtNetwork import QLocalSocket, QLocalServer
-from PySide2.QtWidgets import QApplication
+from qtpy.QtCore import Signal, QTextStream, Qt
+# from qtpy.QtGui import *
+from qtpy.QtNetwork import QLocalSocket, QLocalServer
+from qtpy.QtWidgets import QApplication
 
 """ Python 2.7 implementation of the C++ QtSingleApplication interface
-    Note: This is Python 3 implementation of the above, using Pyside2 - Archisman Panigrahi <apandada1@gmail.com>
+    Note: This is Python 3 implementation of the above, using qtpy - Archisman Panigrahi <apandada1@gmail.com>
     Copyright 2013, Johan Råde
     All rights reserved.
 
