--- a/testsuite/check.sh	2021-04-19 20:57:33.141266611 +0200
+++ b/testsuite/check.sh	2021-04-19 20:58:18.465330536 +0200
@@ -1102,6 +1102,11 @@
 	[ $? = 2 ] || test_failed $LINENO $i
 	"${TARLZ}" -q -tf "${testdir}"/t155_fv${i}.tar.lz --permissive ||
 		test_failed $LINENO $i
+	if [ "${i}" = 5 ] ; then
+		# Skip broken test case. See gentoo bug #783846 and
+		# https://lists.nongnu.org/archive/html/lzip-bug/2021-04/msg00001.html
+		continue
+	fi
 	if [ "${ln_works}" = yes ] ; then
 		mkdir dir2 || framework_failure
 		"${TARLZ}" -C dir2 -xf "${testdir}"/t155_fv${i}.tar.lz --permissive ||
