--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,15 +26,25 @@
 
 include_directories(include)
 
-add_executable(usort src/main.cpp src/parUtils.cpp src/binUtils.cpp src/sort_profiler.cpp)
-target_link_libraries(usort ${MPI_LIBRARIES} m)
+add_library(Usortlib src/parUtils.cpp src/binUtils.cpp)
+target_include_directories(Usortlib PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+target_link_libraries(Usortlib ${MPI_LIBRARIES} m)
+
+add_executable(usort src/main.cpp src/sort_profiler.cpp)
+target_include_directories(usort PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
+target_link_libraries(usort ${MPI_LIBRARIES} m Usortlib)
 
 if(MPI_COMPILE_FLAGS)
-  set_target_properties(usort PROPERTIES
-    COMPILE_FLAGS "${MPI_COMPILE_FLAGS}")
+  set_target_properties(Usortlib PROPERTIES COMPILE_FLAGS "${MPI_COMPILE_FLAGS}")
+  set_target_properties(usort PROPERTIES COMPILE_FLAGS "${MPI_COMPILE_FLAGS}")
 endif()
 
 if(MPI_LINK_FLAGS)
-  set_target_properties(usort PROPERTIES
-    LINK_FLAGS "${MPI_LINK_FLAGS}")
+  set_target_properties(Usortlib PROPERTIES LINK_FLAGS "${MPI_LINK_FLAGS}")
+  set_target_properties(usort PROPERTIES LINK_FLAGS "${MPI_LINK_FLAGS}")
 endif()
+
+include(GNUInstallDirs)
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/usort")
+install(TARGETS Usortlib LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(TARGETS usort LIBRARY DESTINATION "${CMAKE_INSTALL_BINDIR}")
