--- a/Makefile
+++ b/Makefile
@@ -8,18 +8,6 @@ LIBI2PD_CLIENT_PATH = $(I2PD_PATH)/libi2pd_client
 CXX ?= g++
 FLAGS = -Wall -std=c++11 -Wno-misleading-indentation
 
-ifeq ($(UNAME),Linux)
-	FLAGS += -g
-else ifeq ($(UNAME),Darwin)
-	FLAGS += -g
-else ifeq ($(UNAME),FreeBSD)
-	FLAGS += -g
-else
-# Win32
-	FLAGS += -Os -D_MT -DWIN32 -D_WINDOWS -DWIN32_LEAN_AND_MEAN
-	BOOST_SUFFIX = -mt
-endif
-
 INCFLAGS = -I$(LIBI2PD_PATH) -I$(LIBI2PD_CLIENT_PATH)
 CXXFLAGS = $(FLAGS)
 LDFLAGS = 
