--- a/Makefile
+++ b/Makefile
@@ -9,8 +9,7 @@ CXX ?= g++
 FLAGS = -Wall -std=c++11 -Wno-misleading-indentation
 
 INCFLAGS = -I$(LIBI2PD_PATH) -I$(LIBI2PD_CLIENT_PATH)
-CXXFLAGS = $(FLAGS)
-LDFLAGS = 
+CXXFLAGS += $(FLAGS)
 LIBS = $(I2PD_PATH)/$(I2PD_LIB) -lboost_system$(BOOST_SUFFIX) -lboost_date_time$(BOOST_SUFFIX) -lboost_filesystem$(BOOST_SUFFIX) -lboost_program_options$(BOOST_SUFFIX) -lssl -lcrypto -lz
 
 ifeq ($(UNAME),Linux)
