--- a/Makefile
+++ b/Makefile
@@ -14,19 +14,6 @@ DEFINES = -DOPENSSL_SUPPRESS_DEPRECATED
 LDFLAGS = 
 LDLIBS = $(I2PD_PATH)/$(I2PD_LIB) -lboost_system$(BOOST_SUFFIX) -lboost_date_time$(BOOST_SUFFIX) -lboost_filesystem$(BOOST_SUFFIX) -lboost_program_options$(BOOST_SUFFIX) -lssl -lcrypto -lz
 
-ifeq ($(UNAME),Linux)
-	CXXFLAGS += -g
-else ifeq ($(UNAME),Darwin)
-	CXXFLAGS += -g
-else ifeq ($(UNAME),FreeBSD)
-	CXXFLAGS += -g
-else
-# Win32
-	CXXFLAGS += -Os -fPIC -msse
-	DEFINES += -DWIN32_LEAN_AND_MEAN
-	BOOST_SUFFIX = -mt
-endif
-
 ifeq ($(UNAME),Linux)
 	LDLIBS += -lrt -lpthread
 else ifeq ($(UNAME),Darwin)
