--- a/Makefile
+++ b/Makefile
@@ -7,11 +7,10 @@ LIBI2PD_PATH = $(I2PD_PATH)/libi2pd
 LIBI2PD_CLIENT_PATH = $(I2PD_PATH)/libi2pd_client
 
 CXX ?= g++
-CXXFLAGS = -Wall -std=c++17
+CXXFLAGS += -Wall -std=c++17
 INCFLAGS = -I$(LIBI2PD_PATH) -I$(LIBI2PD_CLIENT_PATH)
 DEFINES = -DOPENSSL_SUPPRESS_DEPRECATED
 
-LDFLAGS = 
 LDLIBS = $(I2PD_PATH)/$(I2PD_LIB) -lboost_system$(BOOST_SUFFIX) -lboost_date_time$(BOOST_SUFFIX) -lboost_filesystem$(BOOST_SUFFIX) -lboost_program_options$(BOOST_SUFFIX) -lssl -lcrypto -lz
 
 ifeq ($(UNAME),Linux)
