diff --git a/meson.build b/meson.build
index 9eb076b..f16419d 100644
--- a/meson.build
+++ b/meson.build
@@ -98,5 +98,3 @@ configure_file(
   configuration: conf,
   install_dir: bindir
 )
-
-meson.add_install_script('meson_post_install.py')
diff --git a/passwordsafe.in b/passwordsafe.in
index acc8b3a..de801bb 100755
--- a/passwordsafe.in
+++ b/passwordsafe.in
@@ -1,4 +1,4 @@
-#!@PYTHON@
+#! /usr/bin/env python
 
 import sys
 import os
diff --git a/passwordsafe/keyfile_generator.py b/passwordsafe/keyfile_generator.py
index 02582fc..e29b25a 100644
--- a/passwordsafe/keyfile_generator.py
+++ b/passwordsafe/keyfile_generator.py
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-3.0-only
 import secrets
-from Cryptodome.Cipher import AES
-from Cryptodome.Random import get_random_bytes
+from Crypto.Cipher import AES
+from Crypto.Random import get_random_bytes
 from gi.repository import GLib
 
 
