--- a/Cargo.toml
+++ b/Cargo.toml
@@ -43,10 +43,6 @@ mio = { version = "1.0.4", features = ["os-poll"] }
 signal-hook = "0.3.15"
 signal-hook-mio = { version = "0.2.4", features = ["support-v1_0"] }
 
-[patch.crates-io]
-# Waiting on https://github.com/zmwangx/rust-ffmpeg-sys/pull/91
-ffmpeg-sys-next = { git = "https://github.com/russelltg/rust-ffmpeg-sys", branch = "vulkan" }
-
 # Uncomment for local dev
 # ffmpeg-next = { path = "../rust-ffmpeg" }
 # ffmpeg-sys-next = { path = "../rust-ffmpeg-sys" }
