--- a/lib/external/disassembler/lib/CMakeLists.txt
+++ b/lib/external/disassembler/lib/CMakeLists.txt
@@ -14,5 +14,5 @@ target_include_directories(libdisassembler PUBLIC include)
 target_link_libraries(libdisassembler PRIVATE wolv::types wolv::utils wolv::io wolv::math_eval ${NLOHMANN_JSON_LIBRARIES} ${FMT_LIBRARIES})
 
 if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    target_compile_options(libdisassembler PRIVATE -Wall -Wextra -Wpedantic -Werror)
-endif()
\ No newline at end of file
+    target_compile_options(libdisassembler PRIVATE -Wall -Wextra -Wpedantic)
+endif()
--- a/lib/external/pattern_language/cli/CMakeLists.txt
+++ b/lib/external/pattern_language/cli/CMakeLists.txt
@@ -32,7 +32,7 @@ else()
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    target_compile_options(plcli PRIVATE -Wall -Wextra -Werror -Wno-unknown-pragmas -Wno-array-bounds)
+    target_compile_options(plcli PRIVATE -Wall -Wextra -Wno-unknown-pragmas -Wno-array-bounds)
 endif()
 target_include_directories(plcli PUBLIC include ${CLI11_INCLUDE_DIRS})
 target_link_libraries(plcli PRIVATE ${CLI11_LIBRARIES} ${NLOHMANN_JSON_LIBRARIES} libwolv libpl_includes libpl-gen ${FMT_LIBRARIES})
--- a/lib/external/pattern_language/lib/CMakeLists.txt
+++ b/lib/external/pattern_language/lib/CMakeLists.txt
@@ -90,7 +90,7 @@ endif ()
 if (CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     target_compile_options(libpl PRIVATE /EHsc)
 elseif (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-    target_compile_options(libpl PRIVATE -Wall -Wextra -Werror -Wpedantic -Wno-unknown-pragmas -Wno-array-bounds)
+    target_compile_options(libpl PRIVATE -Wall -Wextra -Wpedantic -Wno-unknown-pragmas -Wno-array-bounds)
     if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
         target_compile_options(libpl PRIVATE -Wno-stringop-overflow)
     endif()
