Date: Thu, 19 Aug 2022 17:57:02 +0800
Subject: [PATCH 1/1] Remove LLVMDemangle

---
 lib/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plugins/builtin/CMakeLists.txt b/plugins/builtin/CMakeLists.txt
index 1822efa..1141abf 100644
--- a/plugins/builtin/CMakeLists.txt	2022-08-19 17:02:46.348471178 +0800
+++ b/plugins/builtin/CMakeLists.txt	2022-08-19 17:03:24.748001036 +0800
@@ -69,7 +69,7 @@
 target_include_directories(${PROJECT_NAME} PRIVATE include)
 
 # Add additional libraries here #
-target_link_libraries(${PROJECT_NAME} PRIVATE libimhex LLVMDemangle)
+target_link_libraries(${PROJECT_NAME} PRIVATE libimhex)
 
 # ---- No need to change anything from here downwards unless you know what you're doing ---- #
 
