--- a/plugins/script_loader/CMakeLists.txt
+++ b/plugins/script_loader/CMakeLists.txt
@@ -2,7 +2,6 @@ cmake_minimum_required(VERSION 3.16)
 
 include(ImHexPlugin)
 
-find_package(CoreClrEmbed)
 if (CoreClrEmbed_FOUND)
     set(IMHEX_DOTNET_SCRIPT_SUPPORT ON)
 
@@ -56,4 +55,4 @@ if (IMHEX_DOTNET_SCRIPT_SUPPORT)
 
     add_subdirectory(support/dotnet)
     add_dependencies(script_loader AssemblyLoader)
-endif()
\ No newline at end of file
+endif()

