From 8d86cd22a743c39a04d34d4cf1a104952cf7fbac Mon Sep 17 00:00:00 2001
From: xiaoming <xiangluoming2015@outlook.com>
Subject: [PATCH] require llvm 16

diff --git a/cmake/build_helpers.cmake b/cmake/build_helpers.cmake
index 0d4839e7..da078878 100644
--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -525,7 +525,7 @@ macro(addBundledLibraries)
         add_subdirectory(${EXTERN_LIBS_FOLDER}/llvm-demangle EXCLUDE_FROM_ALL)
         set_target_properties(LLVMDemangle PROPERTIES POSITION_INDEPENDENT_CODE ON)
     else()
-        find_package(LLVM REQUIRED Demangle)
+        find_package(LLVM 16 REQUIRED Demangle)
     endif()
 
     if (NOT USE_SYSTEM_YARA)
