diff --git a/c/Mf-a6le b/c/Mf-a6le
index 1564cce..0cf0bd4 100644
--- a/c/Mf-a6le
+++ b/c/Mf-a6le
@@ -16,7 +16,7 @@
 m = a6le
 Cpu = X86_64
 
-mdclib = -lm -ldl ${ncursesLib} -lrt -luuid
+mdclib = -lm -ldl ${ncursesLib} -lrt -luuid -ltinfo
 C = ${CC} ${CPPFLAGS} -m64 -msse2 -Wpointer-arith -Wall -Wextra -Werror -Wno-implicit-fallthrough -O2 ${CFLAGS}
 o = o
 mdsrc = i3le.c
diff --git a/c/Mf-ta6le b/c/Mf-ta6le
index 6adb83f..7f4d805 100644
--- a/c/Mf-ta6le
+++ b/c/Mf-ta6le
@@ -16,7 +16,7 @@
 m = ta6le
 Cpu = X86_64
 
-mdclib = -lm -ldl ${ncursesLib} -lpthread -lrt -luuid
+mdclib = -lm -ldl ${ncursesLib} -lpthread -lrt -luuid -ltinfo
 C = ${CC} ${CPPFLAGS} -m64 -msse2 -Wpointer-arith -Wall -Wextra -Werror -Wno-implicit-fallthrough -O2 -D_REENTRANT -pthread ${CFLAGS}
 o = o
 mdsrc = i3le.c
