--- a/Makefile
+++ b/Makefile
@@ -73,17 +73,11 @@
 	@./fontbakery_checks.sh
 
 skimpytest: font ## Runs the minimal tests and verifies the ZIP file mentioned in the README is present.
-	@flake8 *.py
-	@isort --check-only *.py
-	@black --check -l79 *.py
 	@fontlint ${BUILD_DIR}/3270-Regular.otf
 	@fontlint ${BUILD_DIR}/3270-Regular.ttf
 	@fontlint ${BUILD_DIR}/3270-Regular.woff
 	@fontlint ${BUILD_DIR}/3270SemiCondensed-Regular.ttf
 	@fontlint ${BUILD_DIR}/3270Condensed-Regular.ttf
-	@wget --spider $(shell grep -Eo \
-		'https://3270font.s3.amazonaws.com/3270_fonts_[^/"]+\.zip' \
-		README.md)
 
 test: skimpytest ## Generates and checks font files
 # These are tests that fail on Travis (because their fontlint can't ignore
