diff --git a/CMakeLists.txt b/CMakeLists.txt
index 281f47f..0b483a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,6 +10,8 @@ project(Cpp-Taskflow VERSION 2.4.0 LANGUAGES CXX)
 
 # build options
 option(TF_BUILD_BENCHMARKS "Enables build of benchmarks" OFF)
+option(TF_BUILD_EXAMPLES "Enable build of examples" OFF)
+option(TF_BUILD_TESTS "Enable build of tests" OFF)
 
 # Include additional language check
 include(CheckLanguage)
@@ -219,6 +221,7 @@ target_include_directories(${PROJECT_NAME} INTERFACE
 # Example program 
 # -----------------------------------------------------------------------------
 
+if (TF_BUILD_EXAMPLES)
 message(STATUS "Building examples ...")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${TF_EXAMPLE_DIR})
 
@@ -291,10 +294,12 @@ target_link_libraries(
   kmeans ${PROJECT_NAME} Threads::Threads tf::default_settings
 )
 endif(CMAKE_CUDA_COMPILER)
+endif(TF_BUILD_EXAMPLES)
 
 # -----------------------------------------------------------------------------
 # Unittest
 # -----------------------------------------------------------------------------
+if (TF_BUILD_TESTS)
 enable_testing()
 message(STATUS "Building unit tests ...")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${TF_UTEST_DIR})
@@ -616,6 +621,7 @@ add_test(cuda_kmeans.1000.8C8G ${TF_UTEST_CUDA_KMEANS} -tc=kmeans.1000.8C8G)
 add_test(cuda_kmeans.1000.16C16G ${TF_UTEST_CUDA_KMEANS} -tc=kmeans.1000.16C16G)
 
 endif(CMAKE_CUDA_COMPILER)
+endif(TF_BUILD_TESTS)
 
 # -----------------------------------------------------------------------------
 # Benchmarking (enabled by TF_BUILD_BENCHMARKS)
