diff --git a/CMakeLists.txt b/CMakeLists.txt
index f49c398..834b32e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ MESSAGE(STATUS "CMAKE_ROOT: " ${CMAKE_ROOT})
 set(CMAKE_VERBOSE_MAKEFILE ON)
 
 # Project name
-project(Cpp-Taskflow VERSION 2.3.1 LANGUAGES CXX)
+project(Cpp-Taskflow VERSION 2.4.0 LANGUAGES CXX)
 
 # build options
 option(TF_ENABLE_CUDA "Enables build of cuda code" OFF)
