diff --git a/CMakeLists.txt b/CMakeLists.txt
index c267f84..3f357aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,7 @@ option(TF_BUILD_BENCHMARKS "Enables build of benchmarks" OFF)
 
 # Include additional language check
 include(CheckLanguage)
+include(GNUInstallDirs)
 
 # Compiler vendors
 ## g++
@@ -157,8 +158,8 @@ target_link_libraries(
 #set(CMAKE_CXX_EXTENSIONS OFF)
 
 # installation path
-set(TF_INC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include")
-set(TF_LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/lib")
+set(TF_INC_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}")
+set(TF_LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
 set(TF_UTEST_DIR ${PROJECT_SOURCE_DIR}/unittests)
 set(TF_EXAMPLE_DIR ${PROJECT_SOURCE_DIR}/examples)
 set(TF_BENCHMARK_DIR ${PROJECT_SOURCE_DIR}/benchmarks)
@@ -782,7 +783,7 @@ install(TARGETS ${PROJECT_NAME} EXPORT ${PROJECT_NAME}Targets)
 install(
   EXPORT ${PROJECT_NAME}Targets 
   NAMESPACE ${PROJECT_NAME}:: 
-  DESTINATION ${TF_LIB_INSTALL_DIR}/cmake
+  DESTINATION ${TF_LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}
 )
 
 # set up config
@@ -791,7 +792,7 @@ include(CMakePackageConfigHelpers)
 configure_package_config_file(
   ${PROJECT_NAME}Config.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
-  INSTALL_DESTINATION ${TF_LIB_INSTALL_DIR}/cmake
+  INSTALL_DESTINATION ${TF_LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}
   PATH_VARS TF_INC_INSTALL_DIR
 )
 
@@ -803,6 +804,6 @@ write_basic_package_version_file(
 install(
   FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
-  DESTINATION ${TF_LIB_INSTALL_DIR}/cmake 
+  DESTINATION ${TF_LIB_INSTALL_DIR}/cmake/${PROJECT_NAME}
 )
 
