diff --git a/core_test/dune b/core_test/dune
index 5ad5e2a..cfc32d3 100644
--- a/core_test/dune
+++ b/core_test/dune
@@ -1,6 +1,6 @@
 (executables
  (names xs_test)
- (libraries lwt lwt.unix xenstore oUnit)
+ (libraries lwt lwt.unix xenstore ounit2)
  (flags :standard -safe-string))
 
 (alias
diff --git a/server_test/dune b/server_test/dune
index a281f54..a161646 100644
--- a/server_test/dune
+++ b/server_test/dune
@@ -1,6 +1,6 @@
 (executables
  (names server_test)
- (libraries lwt lwt.unix xenstore xenstore.server oUnit)
+ (libraries lwt lwt.unix xenstore xenstore.server ounit2)
  (flags :standard -safe-string))
 
 (alias
diff --git a/xenstore.opam b/xenstore.opam
index 4a598e0..f1606bf 100644
--- a/xenstore.opam
+++ b/xenstore.opam
@@ -17,7 +17,7 @@ depends: [
   "cstruct" {>= "3.2.0"}
   "ppx_cstruct" {>= "3.2.0"}
   "lwt"
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 build: [
   ["dune" "subst"] {pinned}
