--- a/Makefile
+++ b/Makefile
@@ -1,18 +1,16 @@
 BITS=32
 
 ifeq ($(BITS),32)
-  CC=gcc32
   NS=win-x86
   SCFLAGS=
 else
   BITS=64
-  CC=gcc64
   NS=win-x64
   SCFLAGS=-D_MSC_VER=1400
 endif
 
-CFLAGS=-m$(BITS) -shared -c -fno-rtti -fPIC $(SCFLAGS) -I/usr/lib/jvm/java-6-sun/include -I../common/platform-libs/jre-include/win32
-LFLAGS=-m$(BITS) -shared -fno-rtti -fPIC
+CFLAGS += -m$(BITS) -shared -c -fPIC $(SCFLAGS) -I/usr/lib/jvm/java-6-sun/include -I../common/platform-libs/jre-include/win32
+LFLAGS = $(LDFLAGS) -m$(BITS) -shared -fPIC
 
 SRC1=src/native/common/CPU.c
 SRC2=src/native/windows/WindowsDynamicLinkerImpl_JNI.c
