Description: Fix ftbfs with GCC-10

Author: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
Bug-Debian: https://bugs.debian.org/957276
Forwarded: no

--- a/src/junit/com/jogamp/gluegen/test/junit/generation/test1.h
+++ b/src/junit/com/jogamp/gluegen/test/junit/generation/test1.h
@@ -299,9 +299,6 @@ MYAPI MYAPIConfig *  MYAPIENTRY typeTest
 #define HUGE_VALF_3        ((int) 1e53)
 #define DOUBLE_DEFINE_BRACKETS_3 HUGE_VALF_3
 
-size_t unsigned_size_t_1;
-ptrdiff_t a_signed_pointer_t_1;
-
 MYAPI int32_t   MYAPIENTRY typeTestInt32T(const int32_t i1, int32_t i2);
 MYAPI uint32_t  MYAPIENTRY typeTestUInt32T(const uint32_t ui1, uint32_t ui2);
 MYAPI int64_t   MYAPIENTRY typeTestInt64T(const int64_t i1, int64_t i2);
