--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -14,13 +14,9 @@
 test_BE_LDFLAGS  = -L${top_srcdir}/src -lsynapse_backend -L@MRNET_LIBSDIR@ @MRNET_LIBS@
 
 install-data-hook:
-	mkdir -p ${prefix}/example
-	cp ${test_FE_SOURCES} ${test_BE_SOURCES} ${prefix}/example
-	cp Makefile-example ${prefix}/example/Makefile
-	cp run-example ${prefix}/example/run.sh
-	cp topology_1x4.txt ${prefix}/example
-	chmod u+x ${prefix}/example/run.sh
-	$(top_srcdir)/substitute $(SED) "@sub_SYNAPSE_HOME@" "${prefix}" ${prefix}/example/Makefile
-	$(top_srcdir)/substitute $(SED) "@sub_SYNAPSE_HOME@" "${prefix}" ${prefix}/example/run.sh
-
-
+	mkdir -p ${DESTDIR}/${prefix}/libexec/synapse/example
+	cp ${noinst_PROGRAMS} ${DESTDIR}/${prefix}/libexec/synapse/example
+	cp run-example ${DESTDIR}/${prefix}/libexec/synapse/example/run.sh
+	cp topology_1x4.txt ${DESTDIR}/${prefix}/libexec/synapse/example
+	chmod u+x ${DESTDIR}/${prefix}/libexec/synapse/example/run.sh
+	$(top_srcdir)/substitute $(SED) "@sub_SYNAPSE_HOME@" "${prefix}" ${DESTDIR}/${prefix}/libexec/synapse/example/run.sh
