--- a/config/macros.m4
+++ b/config/macros.m4
@@ -327,7 +327,7 @@
 		dnl Check for libraries.
 		AC_MSG_CHECKING([for libmrnet and libxplat])
 
-		if test -f ${MRNET_LIBSDIR}/libmrnet.a -a -f ${MRNET_LIBSDIR}/libxplat.a ; then
+		if test -f ${MRNET_LIBSDIR}/libmrnet.so -a -f ${MRNET_LIBSDIR}/libxplat.so ; then
 			MRNET_LIBS="-lmrnet -lxplat -lpthread -ldl"
 			MRNET_LIGHT_LIBS="-lmrnet_lightweight -lxplat_lightweight -lpthread -ldl"
 			AC_SUBST(MRNET_LIBS)
