--- a/test/test_core.h
+++ b/test/test_core.h
@@ -35,7 +35,7 @@
 #include "tiny_dnn/tiny_dnn.h"
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #endif  // defined(USE_OPENCL) || defined(USE_CUDA)
 
 namespace tiny_dnn {
--- a/tiny_dnn/core/framework/device.fwd.h
+++ b/tiny_dnn/core/framework/device.fwd.h
@@ -46,9 +46,9 @@
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #else
-#include "third_party/CLCudaAPI/cupp11.h"
+#include "CLCudaAPI/cupp11.h"
 #endif
 #endif
 
--- a/tiny_dnn/core/framework/program.h
+++ b/tiny_dnn/core/framework/program.h
@@ -49,9 +49,9 @@
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #else
-#include "third_party/CLCudaAPI/cupp11.h"
+#include "CLCudaAPI/cupp11.h"
 #endif
 #endif
 
--- a/tiny_dnn/core/framework/program_manager.h
+++ b/tiny_dnn/core/framework/program_manager.h
@@ -51,9 +51,9 @@
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #else
-#include "third_party/CLCudaAPI/cupp11.h"
+#include "CLCudaAPI/cupp11.h"
 #endif
 #endif
 
--- a/tiny_dnn/core/framework/tensor.h
+++ b/tiny_dnn/core/framework/tensor.h
@@ -53,9 +53,9 @@
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #else
-#include "third_party/CLCudaAPI/cupp11.h"
+#include "CLCudaAPI/cupp11.h"
 #endif
 #endif
 
--- a/tiny_dnn/core/kernels/tiny_quantized_matmul_kernel.h
+++ b/tiny_dnn/core/kernels/tiny_quantized_matmul_kernel.h
@@ -15,7 +15,7 @@
 
 // Implements a quantized eight-bit version of the matmul operation.
 
-#include "third_party/gemmlowp/public/gemmlowp.h"
+#include "gemmlowp/public/gemmlowp.h"
 #include "tiny_dnn/core/kernels/tiny_quantization_kernel.h"
 
 namespace tiny_dnn {
--- a/tiny_dnn/util/image.h
+++ b/tiny_dnn/util/image.h
@@ -39,15 +39,15 @@
 
 #define STB_IMAGE_IMPLEMENTATION
 #define STB_IMAGE_INLINE // We need this define to avoid multiple definition
-#include "third_party/stb/stb_image.h"
+#include "stb/stb_image.h"
 
 #define STB_IMAGE_RESIZE_IMPLEMENTATION
 #define STB_IMAGE_RESIZE_INLINE
-#include "third_party/stb/stb_image_resize.h"
+#include "stb/stb_image_resize.h"
 
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #define STB_IMAGE_WRITE_INLINE
-#include "third_party/stb/stb_image_write.h"
+#include "stb/stb_image_write.h"
 
 
 namespace tiny_dnn {
--- a/tiny_dnn/util/util.h
+++ b/tiny_dnn/util/util.h
@@ -52,9 +52,9 @@
 
 #if defined(USE_OPENCL) || defined(USE_CUDA)
 #ifdef USE_OPENCL
-#include "third_party/CLCudaAPI/clpp11.h"
+#include "CLCudaAPI/clpp11.h"
 #else
-#include "third_party/CLCudaAPI/cupp11.h"
+#include "CLCudaAPI/cupp11.h"
 #endif
 #endif
 
--- a/.travis/cpplint_test.sh
+++ b/.travis/cpplint_test.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-python third_party/cpplint.py \
+cpplint.py \
       --extensions=cpp,h \
       --filter=-build/header_guard,-runtime/references \
       tiny_dnn/*/* examples/*/* test/*/*/*/*
