--- a/setup.py
+++ b/setup.py
@@ -236,7 +236,6 @@
     platforms=["posix", "nt"],
     package_dir={"": "src/python"},
     packages=find_packages("src/python"),
-    ext_modules=[ots_sanitize],
     zip_safe=False,
     cmdclass=cmdclass,
     setup_requires=["setuptools_scm"],
--- a/src/python/ots/__init__.py
+++ b/src/python/ots/__init__.py
@@ -3,7 +3,7 @@
 import sys
 import os
 
-OTS_SANITIZE = os.path.join(os.path.dirname(__file__), "ots-sanitize")
+OTS_SANITIZE = "ots-sanitize"
 
 __all__ = ["sanitize", "OTSError", "CalledProcessError"]
 
