Disable OpenGL accumulation buffer for XWayland compatibility

diff --git a/Tcl/custom-definitions b/Tcl/custom-definitions
index 5267760..1221749 100644
--- a/Tcl/custom-definitions
+++ b/Tcl/custom-definitions
@@ -18,7 +18,7 @@
 # $XCRYSDEN_TOPDIR/Tcl/toglOpt.tcl
 # ------------------------------------------------------------------------
 
-#set toglOpt(accum)  false 
+set toglOpt(accum)  false 
 #set toglOpt(stereo) none
 
 
