diff --git a/core/chat/chatclient.go b/core/chat/chatclient.go
index fdcd77e3..4408bd48 100644
--- a/core/chat/chatclient.go
+++ b/core/chat/chatclient.go
@@ -3,6 +3,7 @@ package chat
 import (
 	"bytes"
 	"encoding/json"
+	"net/http"
 	"sync"
 	"time"
 
@@ -60,6 +61,9 @@ const (
 var upgrader = websocket.Upgrader{
 	ReadBufferSize:  1024,
 	WriteBufferSize: 1024,
+	CheckOrigin:     func(r *http.Request) bool {
+		return true
+	},
 }
 
 var (
