--- a/Makefile	2021-04-11 04:01:08.508885689 +0200
+++ b/Makefile	2021-04-11 04:02:33.680269181 +0200
@@ -1,6 +1,4 @@
-CC=gcc
-
-CFLAGS=-Wall -Wextra -Werror -pedantic -fstack-protector-all -pedantic -std=c99
+CFLAGS+=-Wall -Wextra -pedantic -std=c99
 SANITY_FLAGS=-Wfloat-equal -Wshadow -Wpointer-arith
 
 PREFIX ?= /usr
@@ -37,14 +35,12 @@
 
 all: $(OUTPUT)
 
-debug: CFLAGS += -g -O0
 debug: $(OUTPUT)
 
-release: CFLAGS += -static -O3
 release: $(OUTPUT)
 
 $(OUTPUT): Makefile $(SOURCE) $(HEADERS)
-	$(CC) $(CFLAGS) $(SANITY_FLAGS) $(SOURCE) -o $(OUTPUT)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(SANITY_FLAGS) $(SOURCE) -o $(OUTPUT)
 
 run: $(OUTPUT)
 	./$(OUTPUT)
