--- a/binding.gyp	2021-10-01 01:06:48.914850580 +0200
+++ b/binding.gyp	2021-10-01 01:24:44.147041657 +0200
@@ -7,7 +7,7 @@
   'targets': [
     {
       'target_name': 'better_sqlite3',
-      'dependencies': ['deps/sqlite3.gyp:sqlite3'],
+      'ldflags': ['-lsqlite3'],
       'sources': ['src/better_sqlite3.cpp'],
       'cflags': ['-std=c++14'],
       'xcode_settings': {
@@ -16,7 +16,7 @@
     },
     {
       'target_name': 'test_extension',
-      'dependencies': ['deps/sqlite3.gyp:sqlite3'],
+      'ldflags': ['-lsqlite3'],
       'conditions': [['sqlite3 == ""', { 'sources': ['deps/test_extension.c'] }]],
     },
   ],
