From e454e6e2f3edb9358bfd14c59cfaf88567389492 Mon Sep 17 00:00:00 2001
From: "Wolfgang E. Sanyer" <WolfgangESanyer@gmail.com>
Date: Tue, 2 Apr 2024 22:12:48 -0400
Subject: [PATCH] patch

---
 pyproject.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 72cf0e1..464a4a7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -65,9 +65,9 @@ full = [
 Homepage = "https://github.com/Backblaze/B2_Command_Line_Tool"
 
 [project.scripts]
-b2 = "b2._internal.b2v3.__main__:main"
-b2v3 = "b2._internal.b2v3.__main__:main"
-_b2v4 = "b2._internal._b2v4.__main__:main"
+backblaze2 = "b2._internal.b2v3.__main__:main"
+backblaze2v3 = "b2._internal.b2v3.__main__:main"
+_backblaze2v4 = "b2._internal._b2v4.__main__:main"
 
 [build-system]
 requires = ["pdm-backend"]
-- 
2.43.2

