https://git.frama-c.com/pub/frama-c/-/issues/2593

--- a/Makefile	2021-11-30 01:00:00.000000000 +0100
+++ b/Makefile	2022-01-14 15:19:18.855072759 +0100
@@ -671,6 +671,7 @@
 
 NON_OPAQUE_DEPS+= src/kernel_services/plugin_entry_points/dynamic
 
+src/kernel_services/abstract_interp/abstract_memory.cmi: BFLAGS := $(BFLAGS)
 src/kernel_services/abstract_interp/abstract_memory.cmo: BFLAGS += -rectypes
 src/kernel_services/abstract_interp/abstract_memory.cmx: OFLAGS += -rectypes
 
