From 334f83007cb24f076d59ee97d9d1c489d0e18437 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Tue, 31 May 2022 10:35:47 +0200
Subject: [PATCH] remove constraint on setuptools_scm

--- a/setup.py
+++ b/setup.py
@@ -55,7 +55,7 @@
         'Programming Language :: Python :: 3'
     ],
     python_requires=">=3.7",
-    setup_requires=['setuptools_scm>=4,<6.1'],
+    setup_requires=['setuptools_scm>=4'],
     install_requires=[
         'setuptools',
         'FontTools[ufo]',
