diff --git a/io-page-unix.opam b/io-page-unix.opam
index e7fa02a..42ba985 100644
--- a/io-page-unix.opam
+++ b/io-page-unix.opam
@@ -10,7 +10,7 @@ depends: [
   "dune"
   "io-page" {=version}
   "cstruct" {>= "2.0.0"}
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 build: [
   ["dune" "subst"] {pinned}
diff --git a/lib_test/dune b/lib_test/dune
index 254e49e..5531760 100644
--- a/lib_test/dune
+++ b/lib_test/dune
@@ -1,6 +1,6 @@
 (executables
  (names portable)
- (libraries io-page-unix io-page oUnit))
+ (libraries io-page-unix io-page ounit2))
 
 (alias
  (name runtest)
