diff --git a/Makefile b/Makefile
index 3a37443..2a68b2e 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ JPEG_LIB ?= $(JPEG_DIR)/lib`getconf LONG_BIT`
 CC    = -no-pie
 GTK   = `pkg-config --libs --cflags gtk+-2.0` `pkg-config --libs x11`
 LIBS  = `pkg-config --libs --cflags libswscale libavutil`
-JPEG  = -I$(JPEG_INCLUDE) $(JPEG_LIB)/libturbojpeg.a
+JPEG  = -I/usr/include /usr/lib`getconf LONG_BIT`/libturbojpeg.so
 SRC      = src/connection.c src/decoder.c
 NO_WARN  = -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
 
