diff --git a/Makefile b/Makefile
index 6995c18..41f56b6 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@
 CC    = -no-pie
 GTK   = `pkg-config --libs --cflags gtk+-2.0` `pkg-config --libs x11`
 LIBS  = `pkg-config --libs --cflags libswscale libavutil`
-JPEG  = -I/opt/libjpeg-turbo/include /opt/libjpeg-turbo/lib`getconf LONG_BIT`/libturbojpeg.a
+JPEG  = -I/usr/include /usr/lib`getconf LONG_BIT`/libturbojpeg.a
 SRC      = src/connection.c src/decoder.c
 NO_WARN  = -Wno-pointer-to-int-cast -Wno-int-to-pointer-cast
 
