Author: Ralf Treinen <treinen@debian.org>
Description: invoke minisat+ with switch for old-format input

--- a/ExternalWrapper.cc
+++ b/ExternalWrapper.cc
@@ -33,7 +33,7 @@
 :min_cost(LONG_MAX)
 ,solution_value (-1)
 ,_id_manager(id_manager)
-,solver_command("minisat+ -cs -ansi")
+,solver_command("minisat+ -cs -ansi -old-fmt")
 ,multiplication_string("*")
 ,temporary_directory("/tmp")
 ,leave_temporary_files(false)
