--- a/Makefile
+++ b/Makefile
@@ -13,18 +13,18 @@
 LIBD =
 LIBS =
 CXX?=g++
-CFLAGS+=-DEXTERNAL_SOLVER
-CFLAGS+=-DNDEBUG -DNCHECK
-CFLAGS+=-O3
-CFLAGS += -Wall -DBUILDDATE='"$(BUILDDATE)"' -DDISTDATE='"$(DISTDATE)"'
-CFLAGS += -DCHANGESET='"$(CHANGESET)"' -DRELDATE='"$(RELDATE)"'
-CFLAGS+=-D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -Wno-parentheses -Wno-deprecated
+CXXFLAGS+=-DEXTERNAL_SOLVER
+CXXFLAGS+=-DNDEBUG -DNCHECK
+CXXFLAGS+=-std=c++11
+CXXFLAGS += -Wall -DBUILDDATE='"$(BUILDDATE)"' -DDISTDATE='"$(DISTDATE)"'
+CXXFLAGS += -DCHANGESET='"$(CHANGESET)"' -DRELDATE='"$(RELDATE)"'
+CXXFLAGS+=-D __STDC_LIMIT_MACROS -D __STDC_FORMAT_MACROS -Wno-parentheses -Wno-deprecated
 LIBS+=-lz
 #LNFLAGS+=-static
 
 # comment out if arbitrary precision not needed
 LIBS+=-lgmpxx -lgmp
-CFLAGS+=-DGMPDEF
+CXXFLAGS+=-DGMPDEF
 
 .PHONY: build all objs
 
@@ -32,17 +32,17 @@
 
 $(XTRGT): objs 
 	@echo Linking: $@
-	@$(CXX) $(COBJS) $(LNFLAGS) $(LIBD) $(LIBS) -o $@ 
+	@$(CXX) $(CXXFLAGS) -fPIE $(CPPFLAGS) $(COBJS) $(LDFLAGS) $(LIBD) $(LIBS) -o $@ 
 
 objs: Lexer.o p.tab.o $(COBJS) 
 
 depend:
-	makedepend -- -I. $(CFLAGS) -- *.cc
+	makedepend -- -I. $(CXXFLAGS) -- *.cc
 
 ## Build rule
 %.o:	%.cc
 	@echo Compiling: $@
-	@$(CXX)  -I. $(CFLAGS) -c -o $@ $<
+	@$(CXX)  -I. $(CPPFLAGS) $(CXXFLAGS) -fPIC $(LDFLAGS) -c -o $@ $<
 
 ##  This  needs bison and ragel, comment in if parsing changes
 #Lexer.cc: p.tab.cc  l.rl Lexer.hh
