From f4345f9ef6365d29634e12e42cd9040a1693e600 Mon Sep 17 00:00:00 2001
From: Pieter Ghysels <pghysels@lbl.gov>
Date: Fri, 24 Jun 2022 17:17:05 -0700
Subject: [PATCH] multilib was not respected, see
 https://github.com/pghysels/STRUMPACK/issues/65

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,8 @@ include(CheckLibraryExists)
 # include(CMakePushCheckState)
 include(CheckCXXSourceCompiles)
 
+include(GNUInstallDirs)
+
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/Modules")
 
 # figure out Fortran/C name mangling
@@ -723,7 +725,8 @@ install(FILES
   DESTINATION include)
 
 install(TARGETS strumpack EXPORT STRUMPACKTargets
-  LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 
 # export targets and all
