diff --git a/usr/share/applications/easyroam_connect_desktop.desktop b/usr/share/applications/easyroam_connect_desktop.desktop
index af29159..58ff457 100644
--- a/usr/share/applications/easyroam_connect_desktop.desktop
+++ b/usr/share/applications/easyroam_connect_desktop.desktop
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Type=Application
-Version=1.4.3
+Version=1.0
 Name=Easyroam
 GenericName=wifi profile installer
 Icon=easyroam_connect_desktop
 Exec=/usr/share/easyroam_connect_desktop/easyroam_connect_desktop %U
 Categories=Utility;
 Keywords=easyroam;eduroam;Easyroam;Eduroam;
-StartupNotify=true
\ No newline at end of file
+StartupNotify=true
