diff --git a/3party/CMakeLists.txt b/3party/CMakeLists.txt
index 90f917b215..502f5b6fc5 100644
--- a/3party/CMakeLists.txt
+++ b/3party/CMakeLists.txt
@@ -86,8 +86,11 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
   endif()
 endif()
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(BLAKE3 REQUIRED IMPORTED_TARGET GLOBAL libblake3)
+add_library(BLAKE3::blake3 ALIAS PkgConfig::BLAKE3)
+
 add_subdirectory(agg)
-add_subdirectory(BLAKE3/c)
 add_subdirectory(bsdiff-courgette)
 add_subdirectory(glaze)
 add_subdirectory(minizip)
diff --git a/libs/search/search_quality/CMakeLists.txt b/libs/search/search_quality/CMakeLists.txt
index 376310e14e..6daeabd6ed 100644
--- a/libs/search/search_quality/CMakeLists.txt
+++ b/libs/search/search_quality/CMakeLists.txt
@@ -19,12 +19,12 @@ target_link_libraries(${PROJECT_NAME}
     indexer
 )
 
-if (NOT SKIP_QT_GUI)
-  omim_add_tool_subdirectory(assessment_tool)
+if (BUILD_TESTING)
+  if (NOT SKIP_QT_GUI)
+    omim_add_tool_subdirectory(assessment_tool)
+  endif()
+  omim_add_tool_subdirectory(features_collector_tool)
+  omim_add_tool_subdirectory(samples_generation_tool)
+  omim_add_tool_subdirectory(search_quality_tool)
+  omim_add_test_subdirectory(search_quality_tests)
 endif()
-
-omim_add_tool_subdirectory(features_collector_tool)
-omim_add_tool_subdirectory(samples_generation_tool)
-omim_add_tool_subdirectory(search_quality_tool)
-
-omim_add_test_subdirectory(search_quality_tests)
