diff --git a/transit/world_feed/world_feed_integration_tests/CMakeLists.txt b/transit/world_feed/world_feed_integration_tests/CMakeLists.txt
index 9f945568d4..ec9755ab09 100644
--- a/transit/world_feed/world_feed_integration_tests/CMakeLists.txt
+++ b/transit/world_feed/world_feed_integration_tests/CMakeLists.txt
@@ -10,7 +10,9 @@ target_link_libraries(${PROJECT_NAME}
 )
 
 # This test requires additional data from the other repository
-set(TEST_DATA_REPO_URL "https://github.com/organicmaps/world_feed_integration_tests_data.git")
+if(NOT TEST_DATA_REPO_URL)
+	set(TEST_DATA_REPO_URL "https://github.com/organicmaps/world_feed_integration_tests_data.git")
+endif()
 set(DESTINATION_FOLDER "${OMIM_DATA_DIR}/world_feed_integration_tests_data/")
 
 add_custom_command(
