--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,12 +114,12 @@
   find_program(LLD_FOUND ld.lld)
   if (LLD_FOUND)
     message(STATUS "Using ld.lld linker")
-    set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld")
+    string(APPEND CMAKE_EXE_LINKER_FLAGS " -fuse-ld=lld")
   else()
     find_program(GOLD_FOUND ld.gold)
     if (GOLD_FOUND)
       message(STATUS "Using ld.gold")
-      set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=gold")
+      string(APPEND CMAKE_EXE_LINKER_FLAGS " -fuse-ld=gold")
     endif()
   endif()
 endif()
