--- a/conf/Makefile.in
+++ b/conf/Makefile.in
@@ -191,8 +191,8 @@
 
 
 LIBI_TARGETS = 
-LIBLIBI = $(LIBDIR)/liblibi.a
-LIBLIBI_SO = $(LIBDIR)/liblibi.so
+LIBLIBI =
+LIBLIBI_SO =
 ifeq ($(MRNET_STARTUP_METHOD),libi)
     LIBI_TARGETS = $(LIBLIBI)
     ifeq ($(BUILD_SHARED_LIBS), yes)
@@ -388,9 +388,6 @@
 
 INSTALL_PREREQ = install-prep 
 INSTALL_PREREQ += xplat-install xplat_lightweight-install
-ifeq ($(MRNET_STARTUP_METHOD),libi)
-    INSTALL_PREREQ += libi-install
-endif
 INSTALL_PREREQ += install-local lightweight-install 
 
 install: $(INSTALL_PREREQ)
@@ -511,7 +508,7 @@
 
 $(LIBXPLAT_LIGHTWEIGHT_SO): xplat_lightweight-sharedobj
 
-$(LIBI_TARGETS): libi-all
+$(LIBI_TARGETS):
 
 $(LIBLMON):
 
