--- a/conf/configure.in
+++ b/conf/configure.in
@@ -215,10 +215,6 @@
 CXXFLAGS="$CXXFLAGS $STDCFLAGS"
 CFLAGS="$CFLAGS $STDCFLAGS"
 
-dnl === Ignore user-requested debug/optimization levels
-CXXFLAGS=`echo $CXXFLAGS | sed 's/-g //; s/-O //; s/-O1 //; s/-O2 //; s/-O3 //'`
-CFLAGS=`echo $CFLAGS | sed 's/-g //; s/-O //; s/-O1 //; s/-O2 //; s/-O3 //'`
-
 dnl === SETTING MRNET PLATFORM VARIABLES ===
 
 dnl MRNET_ARCH is host returned by config.guess with
@@ -275,7 +271,6 @@
 
     C_AS_CPLUSPLUS="-x c++"
 
-    AC_PATH_PROG(AR, ar)
     AC_SUBST(AR)
     ARFLAGS="crv"
     AC_SUBST(ARFLAGS)
@@ -292,7 +287,6 @@
 
     C_AS_CPLUSPLUS="-x c++"
 
-    AC_PATH_PROG(AR, ar)
     AC_SUBST(AR)
     ARFLAGS="crv"
     AC_SUBST(ARFLAGS)
@@ -306,7 +300,6 @@
 
     C_AS_CPLUSPLUS="-+"
 
-    AC_PATH_PROG(AR, ar)
     AC_SUBST(AR)
     ARFLAGS="crv"
     AC_SUBST(ARFLAGS)
@@ -342,7 +334,6 @@
 
     SOFLAGS="$SOFLAGS -fPIC -shared"
 
-    AC_PATH_PROG(AR, ar)
     AC_SUBST(AR)
     ARFLAGS="crv"
     AC_SUBST(ARFLAGS)
@@ -357,7 +348,6 @@
 
     SOFLAGS="$SOFLAGS -fPIC -shared"
 
-    AC_PATH_PROG(AR, ar)
     AC_SUBST(AR)
     ARFLAGS="crv"
     AC_SUBST(ARFLAGS)
