--- a/conf/Makefile.in
+++ b/conf/Makefile.in
@@ -497,7 +497,7 @@
 
 $(LIBMRNET_SO): $(LIBMRNET_OBJS) $(LIBXPLAT_SO)
 	@echo Building `basename $@` ...
-	$(CXX) -Wl,-soname,$(LIBMRNET_SONAME) $(SOFLAGS) -o $@ $(LIBMRNET_OBJS) $(LIBMRNET_SO_DEPLIBS)
+	$(CXX) -Wl,-soname,$(LIBMRNET_SONAME) $(SOFLAGS) $(LDFLAGS) -o $@ $(LIBMRNET_OBJS) $(LIBMRNET_SO_DEPLIBS)
 
 $(LIBXPLAT): xplat-all
 
@@ -556,7 +556,7 @@
 
 $(LIBDIR)/test_DynamicFilters.so: $(TSTDIR)/test_DynamicFilters.C
 	@echo "Building `basename $@` ..."
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ $(TSTDIR)/test_DynamicFilters.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ $(TSTDIR)/test_DynamicFilters.C
 
 $(BINDIR)/test_Recovery_FE: $(BINDIR)/%: $(CNDEP) $(OBJDIR)/%.o \
                                          $(OBJDIR)/test_Recovery.o \
@@ -578,7 +578,7 @@
 
 $(LIBDIR)/test_RecoveryFilter.so: $(TSTDIR)/test_Recovery/test_RecoveryFilter.C
 	@echo "Building `basename $@` ..."
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ $(TSTDIR)/test_Recovery/test_RecoveryFilter.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ $(TSTDIR)/test_Recovery/test_RecoveryFilter.C
 
 #################################
 # Cleaning rules
--- a/conf/Makefile.ltwt.in
+++ b/conf/Makefile.ltwt.in
@@ -101,11 +101,11 @@
 
 $(TARGET_SO): $(LTWT_OBJS)
 	@echo Building `basename $@` ...
-	$(CC) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) -o $@ $^ $(TARGET_SO_DEPLIBS)
+	$(CC) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) $(LDFLAGS) -o $@ $^ $(TARGET_SO_DEPLIBS)
 
 $(TARGET_SO_LOWFAT): $(LOWFAT_OBJS)
 	@echo Building `basename $@` ...
-	$(CC) -Wl,-soname,$(TARGET_SONAME_LOWFAT) $(SOFLAGS) -o $@ $^ $(TARGET_SO_LOWFAT_DEPLIBS)
+	$(CC) -Wl,-soname,$(TARGET_SONAME_LOWFAT) $(SOFLAGS) $(LDFLAGS) -o $@ $^ $(TARGET_SO_LOWFAT_DEPLIBS)
 
 install-prep:
 	for dir in $(ILIBDIR) $(install_INCDIR) $(install_libincdir) ; do \
--- a/Examples/FaultRecovery/Makefile
+++ b/Examples/FaultRecovery/Makefile
@@ -24,7 +24,7 @@
 	$(CC) $(CCFLAGS) $(LDFLAGS) -o $@ FaultRecovery_BE_lightweight.c $(LIBS_LIGHTWEIGHT)
 
 $(LIBDIR)/FaultRecoveryFilter.so: FaultRecoveryFilter.C
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ FaultRecoveryFilter.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ FaultRecoveryFilter.C
 
 clean:
 	$(RM) -f $(TARGETS) $(TARGETS_SO)
--- a/Examples/HeterogeneousFilters/Makefile
+++ b/Examples/HeterogeneousFilters/Makefile
@@ -21,7 +21,7 @@
 	$(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ HeteroFilters_BE.C $(LIBS)
 
 $(LIBDIR)/HeteroFilters.so: HeteroFilters.C
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ HeteroFilters.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ HeteroFilters.C
 
 clean:
 	$(RM) -f $(TARGETS) $(TARGETS_SO)
--- a/Examples/IntegerAddition/Makefile
+++ b/Examples/IntegerAddition/Makefile
@@ -40,7 +40,7 @@
 	$(CC) $(CCFLAGS) $(LDFLAGS) -o $@ IntegerAddition_BE_lightweight.c $(LIBS_LIGHTWEIGHT)
 
 $(LIBDIR)/IntegerAdditionFilter.so: IntegerAdditionFilter.C
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ IntegerAdditionFilter.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ IntegerAdditionFilter.C
 
 clean:
 	$(RM) -f $(TARGETS) $(TARGETS_SO)
--- a/Examples/PerformanceData/Makefile
+++ b/Examples/PerformanceData/Makefile
@@ -24,7 +24,7 @@
 	$(CC) $(CCFLAGS) $(LDFLAGS) -o $@ BE_lightweight.c $(LIBS_LIGHTWEIGHT)
 
 $(LIBDIR)/PerfDataFilter.so: PerfDataFilter.C
-	$(CXX) $(CXXFLAGS) $(SOFLAGS) -o $@ PerfDataFilter.C
+	$(CXX) $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ PerfDataFilter.C
 
 clean:
 	$(RM) -f $(TARGETS) $(TARGETS_SO)
--- a/Examples/PerThreadStreams/Makefile
+++ b/Examples/PerThreadStreams/Makefile
@@ -27,7 +27,7 @@
 	$(CC) -g $(CCFLAGS) $(LDFLAGS) -o $@ PerThreadStreams_BE_lightweight.c $(LIBS_LOWFAT)
 
 $(LIBDIR)/PerThreadStreamsFilter.so: PerThreadStreamsFilter.C
-	$(CXX) -g $(CXXFLAGS) $(SOFLAGS) -o $@ PerThreadStreamsFilter.C
+	$(CXX) -g $(CXXFLAGS) $(SOFLAGS) $(LDFLAGS) -o $@ PerThreadStreamsFilter.C
 
 clean:
 	$(RM) -f $(TARGETS) $(TARGETS_SO)
--- a/xplat/conf/Makefile.in
+++ b/xplat/conf/Makefile.in
@@ -1,4 +1,4 @@
-#/****************************************************************************
+# ****************************************************************************
 # * Copyright © 2003-2012 Dorian C. Arnold, Philip C. Roth, Barton P. Miller *
 # *                  Detailed MRNet usage rights in "LICENSE" file.          *
 # ****************************************************************************/
@@ -105,7 +105,7 @@
 
 $(TARGET_SO): $(XPLAT_OBJS)
 	@echo Building `basename $@` ...
-	$(CXX) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) -o $@ $(XPLAT_OBJS) $(TARGET_SO_DEPLIBS)
+	$(CXX) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) $(LDFLAGS) -o $@ $(XPLAT_OBJS) $(TARGET_SO_DEPLIBS)
 
 install-prep:
 	for dir in $(ILIBDIR) $(install_INCDIR) $(install_LIBINCDIR) ; do \
--- a/xplat/conf/Makefile.ltwt.in
+++ b/xplat/conf/Makefile.ltwt.in
@@ -95,11 +95,11 @@
 
 $(TARGET_SO): $(XPLAT_LTWT_OBJS)
 	@echo Building `basename $@` ...
-	$(CC) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) -o $@ $^ $(TARGET_SO_LIBS)
+	$(CC) -Wl,-soname,$(TARGET_SONAME) $(SOFLAGS) $(LDFLAGS) -o $@ $^ $(TARGET_SO_LIBS)
 
 $(TARGET_SO_LOWFAT): $(XPLAT_LOWFAT_OBJS)
 	@echo Building `basename $@` ...
-	$(CC) -Wl,-soname,$(TARGET_SONAME_LOWFAT) $(SOFLAGS) -o $@ $^ $(TARGET_SO_LOWFAT_LIBS)
+	$(CC) -Wl,-soname,$(TARGET_SONAME_LOWFAT) $(SOFLAGS) $(LDFLAGS) -o $@ $^ $(TARGET_SO_LOWFAT_LIBS)
 
 install-prep:
 	for dir in $(ILIBDIR) $(install_INCDIR) $(install_LIBINCDIR) ; do \
