diff --git i/setup.py w/setup.py
index 5cd1e35..a2618ab 100755
--- i/setup.py
+++ w/setup.py
@@ -24,22 +24,22 @@ class full_install(install):
         ),
         (
             "zsh-completion-dir=",
             None,
             "(Linux only) Set zsh completion directory (default: /usr/local/share/zsh/site-functions)",
         ),
     ]
 
     def initialize_options(self):
         install.initialize_options(self)
-        self.bash_completion_dir = "/etc/bash_completion.d"
-        self.zsh_completion_dir = "/usr/local/share/zsh/site-functions"
+        self.bash_completion_dir = "${D}"
+        self.zsh_completion_dir = "${D}"
 
     def run(self):
         if sys.platform.startswith("linux"):
             self.install_autocomplete()
         install.run(self)
 
     def install_autocomplete(self):
         def copy_autocomplete(src, dst):
             try:
                 if os.path.exists(dst):
