--- a/generator/permutation_gen.c
+++ b/generator/permutation_gen.c
@@ -251,7 +251,7 @@
     indices[0] -= temp_base;
     indices[1] -= temp_base;
     MPI_Datatype old_types[] = {INT64_T_MPI_TYPE, INT64_T_MPI_TYPE};
-    MPI_Type_struct(2, blocklens, indices, old_types, &slot_data_type);
+    MPI_Type_create_struct(2, blocklens, indices, old_types, &slot_data_type);
     MPI_Type_commit(&slot_data_type);
   }
 
