From: Pavel Sobolev <contact@paveloom.dev>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,6 @@ message("[CMAKE_FIND_ROOT_PATH] ${CMAKE_FIND_ROOT_PATH}")
 
 include("cmake/print.cmake")
 include("cmake/nkr.cmake")
-include("cmake/QHotkey.cmake")
 
 find_package(Threads)
 
@@ -65,7 +64,7 @@ list(APPEND NKR_EXTERNAL_TARGETS yaml-cpp)
 find_package(ZXing CONFIG REQUIRED)
 list(APPEND NKR_EXTERNAL_TARGETS ZXing::ZXing)
 
-set(BUILD_SHARED_LIBS OFF)
+find_package(QHotkey CONFIG REQUIRED)
 list(APPEND NKR_EXTERNAL_TARGETS qhotkey)
 
 # Sources
