--- a/config/ax_muster.m4
+++ b/config/ax_muster.m4
@@ -6,21 +6,11 @@
       [--with-muster=MUSTER_DIR],
       [sets the given directory as location of muster includes and libs (needs Boost libraries)]
     ),
-    [muster_dir="$withval"]
-    [muster_dir="/usr)"]
+    [muster_dir="$withval"],
+    [muster_dir="/usr"]
   )
 
-  ac_muster_installed="yes"
-
-  if test ! -f $muster_dir/include/partition.h; then
-    ac_muster_installed="no"
-  fi
-
-  if test ! -f $muster_dir/lib/libmuster.so; then
-    ac_muster_installed="no"
-  fi
-
-  if test "x${ac_muster_enabled}" = "xyes"; then
+    ac_muster_installed="yes"
 
     MUSTER_CPPFLAGS="-I$muster_dir/include"
     MUSTER_LDFLAGS="-L$muster_dir/lib -R$muster_dir/lib"
@@ -36,9 +26,5 @@
 
     # execute ACTION-IF-FOUND
     ifelse([$1], , :, [$1])
-  else
-    # execute ACTION-IF-NOT-FOUND
-    ifelse([$2], , :, [$2])
-  fi
 
 ])
