--- a/configure.ac.template
+++ b/configure.ac.template
@@ -533,17 +533,12 @@
 AC_SUBST(CLUSTERING_LIBS)
 AC_SUBST(CLUSTERING_LD_LIBRARY_PATH)
 
-dnl NOTE: the 'pcfparser_svn3942' is a temporary dependency until the new
-dnl 'common_files' package is defined
-
 AC_CONFIG_FILES([
 Makefile
 include/Makefile
 common_utilities/Makefile
-pcfparser_svn3942/Makefile
 src/Makefile
 src/BasicClasses/Makefile
-src/libParaverTraceParser/Makefile
 src/libSharedComponents/Makefile
 src/libSharedComponents/kalign2/Makefile
 src/libClustering/gmeans/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,9 +8,7 @@
 
 ##    -I@common_filesdir@
 
-## The 'pcfparser_svn3942' and 'common_utilities' directory will
-## be merged in the new 'common-files' package
-SUBDIRS = include pcfparser_svn3942 common_utilities src scripts doc
+SUBDIRS = include common_utilities src scripts doc
 
 instdir = $(includedir)
 
--- a/src/ClustersSequenceScore/Makefile.am
+++ b/src/ClustersSequenceScore/Makefile.am
@@ -9,8 +9,7 @@
  @CLUSTERING_CPPFLAGS@\
  -I$(top_srcdir)/src/libClustering\
  -I$(top_srcdir)/src/libTraceClustering\
- -I$(top_srcdir)/src/libSharedComponents\
- -I$(top_srcdir)/pcfparser_svn3942
+ -I$(top_srcdir)/src/libSharedComponents
 
 ClustersSequenceScore_bin_LDFLAGS  = @CLUSTERING_LDFLAGS@
 
--- a/src/libDistributedClustering/Makefile.am
+++ b/src/libDistributedClustering/Makefile.am
@@ -30,8 +30,6 @@
 libDistributedClustering_la_CPPFLAGS = @CLUSTERING_CPPFLAGS@\
  -I$(top_srcdir)/src/libClustering\
  -I$(top_srcdir)/src/libSharedComponents\
- -I$(top_srcdir)/pcfparser_svn3942\
- -I$(top_srcdir)/src/libParaverTraceParser\
  -I/usr/include/ANN
  
 libDistributedClustering_la_LDFLAGS  = @CLUSTERING_LDFLAGS@ @CLUSTERING_CPPFLAGS@
--- a/src/libSharedComponents/Makefile.am
+++ b/src/libSharedComponents/Makefile.am
@@ -75,21 +75,12 @@
 
 libSharedComponents_la_CPPFLAGS = @CLUSTERING_CPPFLAGS@\
   -I${top_srcdir}/src/libClustering\
-  -I${top_srcdir}/src/libParaverTraceParser\
   -I/usr/include/ANN
-if MAKE_OLD_PCFPARSER
-  libSharedComponents_la_CPPFLAGS += -I${top_srcdir}/pcfparser_svn3942/old
-else
-  libSharedComponents_la_CPPFLAGS += -I${top_srcdir}/pcfparser_svn3942
-endif
-  
 
 libSharedComponents_la_LDFLAGS  = @CLUSTERING_LDFLAGS@ @CLUSTERING_CPPFLAGS@
 
 libSharedComponents_la_LIBADD =  \
-	$(top_builddir)/pcfparser_svn3942/libparavertraceconfig.la \
 	$(top_builddir)/src/libClustering/libClustering.la \
-	$(top_builddir)/src/libParaverTraceParser/libParaverTraceParser.la \
 	./kalign2/libKalign2.la \
 	@CLUSTERING_LIBS@
 
--- a/src/libTraceClustering/Makefile.am
+++ b/src/libTraceClustering/Makefile.am
@@ -9,9 +9,7 @@
 	-I$(top_srcdir)/src/BasicClasses \
 	-I/usr/include/ANN \
 	-I$(top_srcdir)/src/libSharedComponents \
-	-I$(top_srcdir)/src/libParaverTraceParser \
-	-I$(top_srcdir)/common_utilities \
-	-I$(top_srcdir)/pcfparser_svn3942
+	-I$(top_srcdir)/common_utilities
 
 instdir = $(includedir)
 
@@ -27,9 +25,7 @@
 libTraceClustering_la_CPPFLAGS = @CLUSTERING_CPPFLAGS@\
   -I$(top_srcdir)/src/libClustering\
   -I$(top_srcdir)/src/libSharedComponents\
-  -I$(top_srcdir)/pcfparser_svn3942\
-  -I/usr/include/ANN\
-  -I$(top_srcdir)/src/libParaverTraceParser
+  -I/usr/include/ANN
 
 libTraceClustering_la_LDFLAGS  = @CLUSTERING_CPPFLAGS@ @CLUSTERING_LDFLAGS@
 libTraceClustering_la_LIBADD =  \
@@ -37,8 +33,5 @@
 	$(top_builddir)/src/libSharedComponents/libSharedComponents.la \
 	@CLUSTERING_LIBS@
 
-##	$(top_srcdir)/src/libParaverTraceParser/libParaverTraceParser.la
-
-
 
 
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -4,7 +4,6 @@
 
 SUBDIRS = \
 	BasicClasses \
-	libParaverTraceParser \
 	libClustering \
 	libSharedComponents \
 	libTraceClustering \
