From a1a62c6928fc19a7268f4fa627f0bb0243b5eb94 Mon Sep 17 00:00:00 2001
From: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 18:40:19 +0200
Subject: [PATCH] switch to ounit2

Signed-off-by: Alessandro-Barbieri <lssndrbarbieri@gmail.com>
---
 test/dune               | 2 +-
 xenstore_transport.opam | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/test/dune b/test/dune
index 3b5134b..f92c0df 100644
--- a/test/dune
+++ b/test/dune
@@ -1,4 +1,4 @@
 (tests
  (names client_test server_test)
  (package xenstore_transport)
- (libraries xenstore_transport xenstore xenstore.unix oUnit))
+ (libraries xenstore_transport xenstore xenstore.unix ounit2))
diff --git a/xenstore_transport.opam b/xenstore_transport.opam
index 0934220..23caac4 100644
--- a/xenstore_transport.opam
+++ b/xenstore_transport.opam
@@ -23,7 +23,7 @@ depends: [
   "dune" {build & >= "1.0"}
   "lwt"
   "xenstore" {>= "2.0.0"}
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 build: [
   ["dune" "subst"] {pinned}
