Bug: https://bugs.gentoo.org/785196

--- a/configure.in
+++ b/configure.in
@@ -9,16 +9,6 @@
 [extern int test (int i, double x);],
 AC_DEFINE(HAVE_PROTOTYPES))])dnl
 dnl
-dnl Brook Milligan's compiler check
-dnl Check for the sun ansi c compiler, acc
-define(LOCAL_PROG_ACC,
-[AC_BEFORE([$0], [AC_PROG_CPP])AC_PROVIDE([$0])dnl
-AC_PROGRAM_CHECK(CC, acc, acc, "")])dnl
-dnl David Stewart's modified compiler check
-define(LOCAL_PROG_CC,
-[AC_BEFORE([$0], [AC_PROG_CPP])AC_PROVIDE([$0])dnl
-AC_PROGRAM_CHECK(CC, acc, acc, cc)])dnl
-dnl
 dnl
 dnl
 dnl ----------------------------------------------------------------------
@@ -29,7 +19,6 @@
 PROGS=""
 AC_SUBST(PROGS)dnl
 LOCAL_PROG_ACC
-AC_PROGRAM_CHECK(CC, cc, cc, gcc)
 dnl AC_PROG_CC
 AC_PROG_CPP
 AC_AIX
