diff -ru xneur-0.12.0-orig/lib/notify/popup.c xneur-0.12.0/lib/notify/popup.c
--- xneur-0.12.0-orig/lib/notify/popup.c	2011-01-27 05:25:42.000000000 -0500
+++ xneur-0.12.0/lib/notify/popup.c	2011-03-12 02:21:20.411001315 -0500
@@ -27,6 +27,10 @@
 
 #include <libnotify/notify.h>
 
+#ifndef NOTIFY_CHECK_VERSION /* macro did not exist before libnotify-0.5.2 */
+#  define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
+
 #include <pthread.h>
 #include <stdio.h>
 #include <string.h>
@@ -73,8 +77,12 @@
 		popup_body->header = popup_body->content;
 		popup_body->content = NULL;
 	}
-	
+
+#if NOTIFY_CHECK_VERSION(0,7,0)
+	NotifyNotification *notify = notify_notification_new(popup_body->header, popup_body->content, icon);
+#else
 	NotifyNotification *notify = notify_notification_new(popup_body->header, popup_body->content, icon, NULL);
+#endif
 
 	notify_notification_set_category(notify, type);
 	notify_notification_set_urgency(notify, urgency);
