From d3605d93627f0d756c2c570e7ef0ad6e751d8eb3 Mon Sep 17 00:00:00 2001
From: Aisha Tammy <aisha@bsd.ac>
Date: Sun, 7 Apr 2024 20:32:01 -0400
Subject: [PATCH 1/2] feat: add pyproject.toml

---
 pyproject.toml | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 pyproject.toml

diff --git a/pyproject.toml b/pyproject.toml
new file mode 100644
index 0000000..fec8478
--- /dev/null
+++ b/pyproject.toml
@@ -0,0 +1,27 @@
+[build-system]
+requires = ["setuptools >= 61.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "azote"
+version = "1.12.6"
+description = "Wallpaper manager for sway and some other WMs"
+readme = "README.md"
+license = {text = "GPL3"}
+requires-python = ">= 3.8.0"
+authors = [
+  {name = "Piotr Miller", email = "nwg.piotr@gmail.com"},
+]
+dependencies = [
+  "pillow",
+  "PyGObject",
+  "PyYAML",
+  "Send2Trash",
+]
+
+[project.urls]
+Repository = "https://github.com/nwg-piotr/azote"
+License = "https://github.com/nwg-piotr/azote/blob/master/LICENSE"
+
+[project.scripts]
+azote = "azote.main:main"

From 769d027dfa40c218075827ca50241a2395240bea Mon Sep 17 00:00:00 2001
From: Aisha Tammy <aisha@bsd.ac>
Date: Sun, 7 Apr 2024 20:33:21 -0400
Subject: [PATCH 2/2] scratch

---
 dist/azote |  5 -----
 setup.py   | 24 ------------------------
 2 files changed, 29 deletions(-)
 delete mode 100755 dist/azote
 delete mode 100644 setup.py

diff --git a/dist/azote b/dist/azote
deleted file mode 100755
index 7684782..0000000
--- a/dist/azote
+++ /dev/null
@@ -1,5 +0,0 @@
-#!/bin/sh
-
-LIB=$(python3 -Ic "from sysconfig import get_path; print(get_path('purelib'))")
-cd $LIB/azote
-exec /usr/bin/python3 main.py "$@"
diff --git a/setup.py b/setup.py
deleted file mode 100644
index 86cfe6d..0000000
--- a/setup.py
+++ /dev/null
@@ -1,24 +0,0 @@
-import os
-from setuptools import setup, find_packages
-
-
-def read(f_name):
-    return open(os.path.join(os.path.dirname(__file__), f_name)).read()
-
-
-setup(
-    name='azote',
-    version='1.12.6',
-    description='Wallpaper manager for sway and some other WMs',
-    packages=find_packages(),
-    include_package_data=True,
-    package_data={
-        "": ["images/*", "langs/*"]
-    },
-    url='https://github.com/nwg-piotr/azote',
-    license='GPL3',
-    author='Piotr Miller',
-    author_email='nwg.piotr@gmail.com',
-    python_requires='>=3.8.0',
-    install_requires=[],
-)
