diff -Naur a/src/aux_widgets/InitialRepoConfig.ui b/src/aux_widgets/InitialRepoConfig.ui
--- a/src/aux_widgets/InitialRepoConfig.ui
+++ b/src/aux_widgets/InitialRepoConfig.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>600</width>
-    <height>400</height>
+    <height>500</height>
    </rect>
   </property>
   <property name="minimumSize">
@@ -184,6 +184,12 @@
      <property name="title">
       <string>Credentials configuration</string>
      </property>
+     <property name="autoFillBackground">
+      <bool>false</bool>
+     </property>
+     <property name="styleSheet">
+      <string notr="true">background: transparent;</string>
+     </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
diff -Naur a/src/big_widgets/ConfigWidget.ui b/src/big_widgets/ConfigWidget.ui
--- a/src/big_widgets/ConfigWidget.ui
+++ b/src/big_widgets/ConfigWidget.ui
@@ -678,7 +678,7 @@
                   <number>10</number>
                  </property>
                  <item row="0" column="0" colspan="2">
-                  <widget class="QCheckBox" name="chbCredentials">
+                  <widget class="CheckBox" name="chbCredentials">
                    <property name="text">
                     <string>Store credentials using GitQlient (Current session or in Storage)</string>
                    </property>
@@ -1018,6 +1018,12 @@
          <property name="title">
           <string>Installed plugins</string>
          </property>
+         <property name="autoFillBackground">
+          <bool>false</bool>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">background: transparent;</string>
+         </property>
          <layout class="QGridLayout" name="pluginsLayout">
           <property name="leftMargin">
            <number>10</number>
@@ -1129,6 +1135,12 @@
          <property name="title">
           <string>Official GitQlient plugins</string>
          </property>
+         <property name="autoFillBackground">
+          <bool>false</bool>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">background: transparent;</string>
+         </property>
          <layout class="QGridLayout" name="availablePluginsLayout">
           <property name="leftMargin">
            <number>10</number>
@@ -1213,6 +1225,12 @@
          <property name="title">
           <string>Branches panel</string>
          </property>
+         <property name="autoFillBackground">
+          <bool>false</bool>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">background: transparent;</string>
+         </property>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="1">
            <widget class="QLabel" name="label_62">
@@ -1250,6 +1268,12 @@
          <property name="title">
           <string>GitQlient repo operations</string>
          </property>
+         <property name="autoFillBackground">
+          <bool>false</bool>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">background: transparent;</string>
+         </property>
          <layout class="QGridLayout" name="gridLayout_9">
           <item row="3" column="1">
            <widget class="QLabel" name="label_45">
@@ -1315,6 +1339,12 @@
          <property name="title">
           <string>GitQlient general</string>
          </property>
+         <property name="autoFillBackground">
+          <bool>false</bool>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">background: transparent;</string>
+         </property>
          <layout class="QGridLayout" name="gridLayout_10">
           <item row="3" column="1">
            <widget class="QLabel" name="label_54">
