--- a/setup.py
+++ b/setup.py
@@ -3,7 +3,6 @@
 from distutils.core import setup
 import distutils.core
 from distutils.command import build, install_data
-import gzip
 import os.path
 import sys
 
@@ -29,9 +28,9 @@
     def build_man_page(self):
         """
         """
-        man_path = os.path.join(self.build_base, 'osc.1.gz')
+        man_path = os.path.join(self.build_base, 'osc.1')
         distutils.log.info('generating %s' % man_path)
-        outfile = gzip.open(man_path, 'wt')
+        outfile = open(man_path, 'w')
         osccli = commandline.Osc(stdout=outfile)
         # FIXME: we cannot call the main method because osc expects an ~/.oscrc
         # file (this would break builds in environments like the obs)
@@ -100,7 +99,7 @@
 
 data_files = []
 if sys.platform[:3] != 'win':
-    data_files.append((os.path.join('share', 'man', 'man1'), ['osc.1.gz']))
+    data_files.append((os.path.join('share', 'man', 'man1'), ['osc.1']))
 
 with open("README") as fh:
     long_description = fh.read()
