--- a/Makefile.in
+++ b/Makefile.in
@@ -77,7 +77,7 @@
 $(DATABASE) $(DATAINDEX): $(EXES) data/*.txt
 	./format -p < data/places.txt > tmp
 	./format -c < data/counties.txt >> tmp
-#	./format -m < data/mcds.txt >> tmp
+	./format -m < data/mcds.txt >> tmp
 	./format -z < data/zips.txt >> tmp
 	sort tmp | ./format -f > $(DATABASE)
 	-rm -f tmp
@@ -165,7 +165,7 @@
 #		 fi \
 #	 done
 	-rm -f config.h config.cache config.status stamp-h.in stamp-h
-        -rm -f Makefile
+	-rm -f Makefile
 #	-rm -f configure Makefile # We need to keep configure
 	-rm -f $(DATABASE) $(DATABASE).dz $(DATAINDEX)
 
