--- a/Makefile.am
+++ b/Makefile.am
@@ -65,40 +65,40 @@
 	@echo '+----------------------------------------------------------------------------+'
 	@echo '|'
 	@echo '|  PnetCDF has been successfully installed under'
-	@echo '|          $(DESTDIR)$(prefix)'
+	@echo '|          $(prefix)'
 	@echo '|'
 	@echo '|  * PnetCDF header files have been installed in'
-	@echo '|          $(DESTDIR)$(prefix)/include'
+	@echo '|          $(prefix)/include'
 	@echo '|  * PnetCDF library files have been installed in'
-	@echo '|          $(DESTDIR)$(exec_prefix)/lib'
+	@echo '|          $(exec_prefix)/lib'
 	@echo '|  * PnetCDF utility programs have been installed in'
-	@echo '|          $(DESTDIR)$(exec_prefix)/bin'
+	@echo '|          $(exec_prefix)/bin'
 	@echo '|  * PnetCDF man pages have been installed in'
-	@echo '|          $(DESTDIR)$(prefix)/share/man'
+	@echo '|          $(prefix)/share/man'
 	@echo '|'
 	@echo '|  To compile your PnetCDF programs, please add the following to the command'
 	@echo '|  line, so the compiler can find the PnetCDF header files:'
-	@echo '|      -I$(DESTDIR)$(prefix)/include'
+	@echo '|      -I$(prefix)/include'
 	@echo '|'
 	@if test "x$(has_fortran)" = xyes -a 'x$(FC_MODINC)' != 'x-I' ; then \
 	 echo '|  Add the following line to compile your Fortran programs' ; \
-	 echo '|      $(FC_MODINC)$(DESTDIR)$(prefix)/include' ; \
+	 echo '|      $(FC_MODINC)$(prefix)/include' ; \
 	 echo '|' ; \
 	fi
 	@echo '|  Add the following line to link your program to PnetCDF library:'
-	@echo '|      -L$(DESTDIR)$(exec_prefix)/lib -lpnetcdf'
+	@echo '|      -L$(exec_prefix)/lib -lpnetcdf'
 	@echo '|'
 	@if test "x$(enable_shared)" = xyes ; then \
 	 echo '|  Add the following to your run-time environment variable LD_LIBRARY_PATH,' ; \
 	 echo '|  when linking your executable with the PnetCDF shared libraries.' ; \
          if test "x$(enable_netcdf4)" = xyes -a "x$(enable_adios)" = xyes ; then \
-         echo '|      $(DESTDIR)$(exec_prefix)/lib:$(netcdf4_libdir):$(adios_libdir)' ; \
+         echo '|      $(exec_prefix)/lib:$(netcdf4_libdir):$(adios_libdir)' ; \
          elif test "x$(enable_netcdf4)" = xyes ; then \
-         echo '|      $(DESTDIR)$(exec_prefix)/lib:$(netcdf4_libdir)' ; \
+         echo '|      $(exec_prefix)/lib:$(netcdf4_libdir)' ; \
          elif test "x$(enable_adios)" = xyes ; then \
-         echo '|      $(DESTDIR)$(exec_prefix)/lib:$(adios_libdir)' ; \
+         echo '|      $(exec_prefix)/lib:$(adios_libdir)' ; \
          else \
-	 echo '|      $(DESTDIR)$(exec_prefix)/lib' ; \
+	 echo '|      $(exec_prefix)/lib' ; \
 	 fi ; \
 	 echo '|' ; \
 	fi
