From b0ef14bb5f73e0a1f5345f0ddd3b3992205daebf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Philip=20M=C3=BCller?= <philm@manjaro.org>
Date: Tue, 4 May 2021 08:53:30 +0000
Subject: [PATCH] fbd-feedback-manager: fix device-specific themes path

---
 src/fbd-feedback-manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fbd-feedback-manager.c b/src/fbd-feedback-manager.c
index 39c27eb..ac6cdfa 100644
--- a/src/fbd-feedback-manager.c
+++ b/src/fbd-feedback-manager.c
@@ -455,7 +455,7 @@ find_themefile (void)
       for (i = 0; i < g_strv_length (xdg_data_dirs); i++) {
         g_autofree gchar *config_path = NULL;
 
-        config_path = g_strconcat (xdg_data_dirs[i], "feedbackd/themes/", comp, ".json", NULL);
+        config_path = g_strconcat (xdg_data_dirs[i], "/feedbackd/themes/", comp, ".json", NULL);
         g_debug ("Searching for device specific themefile in %s", config_path);
 
         // Check if file exist
-- 
GitLab

