diff --git a/README.md b/README.md
index cc6df4b..0595368 100644
--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@ OCaml implementation of the Qemu Monitor Protocol (QMP)
 Install the dependencies using OPAM:
 
 ```sh
-opam install yojson cmdliner dune ounit
+opam install yojson cmdliner dune ounit2
 ```
 
 Build the library, tests, and CLI:
diff --git a/lib_test/dune b/lib_test/dune
index acc68b9..a587594 100644
--- a/lib_test/dune
+++ b/lib_test/dune
@@ -6,5 +6,5 @@
 
 (executable
   (name messages)
-  (libraries libqmp oUnit)
+  (libraries libqmp ounit2)
 )
diff --git a/qmp.opam b/qmp.opam
index 2e21b12..571b58f 100644
--- a/qmp.opam
+++ b/qmp.opam
@@ -17,7 +17,7 @@ depends: [
   "dune" {build}
   "yojson"
   "cmdliner"
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 dev-repo: "git://github.com/xapi-project/ocaml-qmp"
 synopsis: "OCaml implementation of a Qemu Message Protocol (QMP) client"
