From a3ebbbd2760e4ba3774529a6975ac868baff091c Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:03:29 +0200
Subject: [PATCH 1/3] use ounit2

---
 lib_test/dune | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib_test/dune b/lib_test/dune
index acc68b9..a587594 100644
--- a/lib_test/dune
+++ b/lib_test/dune
@@ -6,5 +6,5 @@
 
 (executable
   (name messages)
-  (libraries libqmp oUnit)
+  (libraries libqmp ounit2)
 )

From d09f6df5f07eed691d77136cca4ea36a98e630c3 Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:05:34 +0200
Subject: [PATCH 2/3] install ounit2

---
 README.md | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README.md b/README.md
index cc6df4b..0595368 100644
--- a/README.md
+++ b/README.md
@@ -8,7 +8,7 @@ OCaml implementation of the Qemu Monitor Protocol (QMP)
 Install the dependencies using OPAM:
 
 ```sh
-opam install yojson cmdliner dune ounit
+opam install yojson cmdliner dune ounit2
 ```
 
 Build the library, tests, and CLI:

From dcd85772065d646d0272160be14ebf9e28bb6bde Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Sat, 16 Apr 2022 15:06:05 +0200
Subject: [PATCH 3/3] depend on ounit2

---
 qmp.opam | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qmp.opam b/qmp.opam
index 242b3c8..e36fb3f 100644
--- a/qmp.opam
+++ b/qmp.opam
@@ -18,7 +18,7 @@ depends: [
   "dune" {build}
   "yojson"
   "cmdliner"
-  "ounit" {with-test}
+  "ounit2" {with-test}
 ]
 dev-repo: "git://github.com/xapi-project/ocaml-qmp"
 synopsis: "OCaml implementation of a Qemu Message Protocol (QMP) client"
