diff --git a/tools/idf_tools.py b/tools/idf_tools.py
index 8a17ed224d..2103e9d60a 100755
--- a/tools/idf_tools.py
+++ b/tools/idf_tools.py
@@ -2113,6 +2113,7 @@ def action_install_python_env(args):  # type: ignore
 
 
 def action_check_python_dependencies(args):  # type: ignore
+    return
     use_constraints = not args.no_constraints
     req_paths = get_requirements('')  # no new features -> just detect the existing ones
 
