From: Pavel Sobolev <contact@paveloom.dev>
Subject: [PATCH] Remove the AppData file validation test.

Signed-off-by: Pavel Sobolev <contact@paveloom.dev>

--- a/data/meson.build
+++ b/data/meson.build
@@ -28,13 +28,6 @@ appstream_file = i18n.merge_file(
   install_dir: join_paths(get_option('datadir'), 'appdata')
 )
 
-appstream_util = find_program('appstream-util', required: false)
-if appstream_util.found()
-  test('Validate appstream file', appstream_util,
-    args: ['validate', appstream_file]
-  )
-endif
-
 install_data('com.github.tenderowl.frog.gschema.xml',
   install_dir: join_paths(get_option('datadir'), 'glib-2.0/schemas')
 )

