From 1a651a0bb19ad8c0752b497eed125a901d7f83da Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Wed, 25 Aug 2021 09:13:37 +0200
Subject: [PATCH] fix == bashism

---
 config/x_ac_handshake.m4 | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

--- a/config/x_ac_handshake.m4
+++ b/config/x_ac_handshake.m4
@@ -50,7 +50,7 @@ AC_DEFUN([X_AC_HANDSHAKE],[
 
         #Check for munge availability
         PKG_CHECK_MODULES(MUNGE, munge, , [AC_MSG_NOTICE([pkg-config could not find munge])])
-        if test "x$MUNGE_LIBS" == "x"; then
+        if test "x$MUNGE_LIBS" = "x"; then
            MUNGE_LIBS=-lmunge
         fi
         CFLAGS="$CFLAGS $MUNGE_CFLAGS"
@@ -66,26 +66,26 @@ AC_DEFUN([X_AC_HANDSHAKE],[
 
         #If the user did not specify security options, then make some choices based on what we have
         if test "x$EXPLICIT_SEC" != "xtrue"; then
-           if test "x$HAVE_MUNGE" == "xtrue"; then
+           if test "x$HAVE_MUNGE" = "xtrue"; then
              MUNGE="true"
            else
              ENABLE_NULL_ENCRYPTION="true"
            fi
         fi
 
-        if test "x$WANT_MUNGE" == "xtrue"; then
-           if test "x$HAVE_MUNGE" == "xtrue"; then
+        if test "x$WANT_MUNGE" = "xtrue"; then
+           if test "x$HAVE_MUNGE" = "xtrue"; then
               MUNGE="true"
            else
               AC_MSG_ERROR([Could not find munge])
            fi
         fi
 
-        if test "x$WANT_KEYFILE" == "xtrue"; then
+        if test "x$WANT_KEYFILE" = "xtrue"; then
            KEYFILE="true"
         fi
 
-        if test "x$WANT_NOSEC" == "xtrue"; then
+        if test "x$WANT_NOSEC" = "xtrue"; then
            ENABLE_NULL_ENCRYPTION="true"
         fi
 
@@ -100,7 +100,7 @@ AC_DEFUN([X_AC_HANDSHAKE],[
         #if test "x$KEYFILE" != "x"; then 
         #  AC_MSG_ERROR([Keyfile security is not supported at this time])
         #fi
-        #if test "x$ENABLE_NULL_ENCRYPTION x$MUNGE" == "xtrue xtrue"; then
+        #if test "x$ENABLE_NULL_ENCRYPTION x$MUNGE" = "xtrue xtrue"; then
         #  AC_MSG_ERROR([Cannot support multiple security models at this time])
         #fi
         #Remove above when expanding to more than munge
@@ -108,15 +108,15 @@ AC_DEFUN([X_AC_HANDSHAKE],[
         AC_DEFINE_UNQUOTED([SEC_KEYDIR], "[$SEC_KEYDIR]",[Directory to store key files in])
         AC_SUBST(MUNGE_CFLAGS)
         AC_SUBST(MUNGE_LIBS)
-        if test "x$MUNGE" == "xtrue"; then
+        if test "x$MUNGE" = "xtrue"; then
            AC_MSG_NOTICE([Enabling munge for security authentication])        
            AC_DEFINE([MUNGE], [1], [Use munge for authentication])
         fi
-        if test "x$KEYFILE" == "xtrue"; then
+        if test "x$KEYFILE" = "xtrue"; then
            AC_MSG_NOTICE([Enabling keyfile for security authentication])
            AC_DEFINE([KEYFILE], [1], [Use keyfile for authentication])
         fi
-        if test "x$ENABLE_NULL_ENCRYPTION" == "xtrue"; then
+        if test "x$ENABLE_NULL_ENCRYPTION" = "xtrue"; then
            AC_MSG_NOTICE([WARNING: No secure handshake will be used for the COBO layer!!!]);
            AC_DEFINE([ENABLE_NULL_ENCRYPTION], [1], [Allow NULL encryption])
         fi])
