diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 361b4aa..1f61281 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -375,7 +375,6 @@ target_compile_definitions(corectrl_lib PRIVATE ${APP_COMPILE_DEFINITIONS})
 target_link_libraries(corectrl_lib PRIVATE
   Qt6::Core
   QuaZip::QuaZip
-  stdc++fs
   pthread
   spdlog::spdlog
   pugixml::pugixml
diff --git a/src/helper/CMakeLists.txt b/src/helper/CMakeLists.txt
index 38deb12..b8ed17e 100644
--- a/src/helper/CMakeLists.txt
+++ b/src/helper/CMakeLists.txt
@@ -106,7 +106,6 @@ target_compile_definitions(corectrl_helper PRIVATE ${HELPER_COMPILE_DEFINITIONS}
 target_link_libraries(corectrl_helper PRIVATE
   Qt6::Core
   Qt6::DBus
-  stdc++fs
   pthread
   spdlog::spdlog
   ${Botan_LIBRARIES}
