diff --git a/src/core/components/controls/controlgroupxmlparser.cpp b/src/core/components/controls/controlgroupxmlparser.cpp
index c83a187..c580fc3 100644
--- a/src/core/components/controls/controlgroupxmlparser.cpp
+++ b/src/core/components/controls/controlgroupxmlparser.cpp
@@ -5,6 +5,7 @@
 
 #include <pugixml.hpp>
 #include <utility>
+#include <algorithm>
 
 class ControlGroupXMLParser::Factory final
 : public ProfilePartXMLParser::Factory
