--- a/cmake/FixupCompilerFlags.cmake
+++ b/cmake/FixupCompilerFlags.cmake
@@ -45,18 +45,6 @@
     SET(CMAKE_CXX_FLAGS "-Wall ${CMAKE_CXX_FLAGS}")
     #SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wshorten-64-to-32")
 
-    # for GCC or Clang, we want both maximum performance and debug info.
-    IF (CMAKE_BUILD_TYPE STREQUAL Debug)
-      SET(CMAKE_C_FLAGS_DEBUG "-O3 ${CMAKE_C_FLAGS_DEBUG}")
-      SET(CMAKE_CXX_FLAGS_DEBUG "-O3 ${CMAKE_CXX_FLAGS_DEBUG}")
-    ELSEIF(CMAKE_BUILD_TYPE STREQUAL Release)
-      SET(CMAKE_C_FLAGS_RELEASE "-g ${CMAKE_C_FLAGS_RELEASE}")
-      SET(CMAKE_CXX_FLAGS_RELEASE "-g ${CMAKE_CXX_FLAGS_RELEASE}")
-    ELSE()
-      SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O3")
-      SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O3")
-    ENDIF()
-
     # for some reason static library doesn't have this
     # on x86_64 and is required.
     # Solaris GCC (32 bit) doesn't have this either. so
