--- iio-sensor-proxy-${PV}/meson.build
+++ iio-sensor-proxy-${PV}/meson.build
@@ -34,11 +34,7 @@
     udev_dep = dependency('udev')
     udev_rules_dir = udev_dep.get_pkgconfig_variable('udevdir') + '/rules.d'
 endif
-systemd_system_unit_dir = get_option('systemdsystemunitdir')
-if systemd_system_unit_dir == 'auto'
-    systemd_dep = dependency('systemd')
-    systemd_system_unit_dir = systemd_dep.get_pkgconfig_variable('systemdsystemunitdir')
-endif
+systemd_system_unit_dir = '/lib/systemd/system'
 if get_option('gtk-tests')
     gtk_dep = dependency('gtk+-3.0', required: false)
 endif

--- iio-sensor-proxy-${PV}/data/meson.build
+++ iio-sensor-proxy-${PV}/data/meson.build
@@ -8,13 +8,6 @@
 data_conf.set('geoclue_user', get_option('geoclue-user'))
 
 configure_file(
-  input: 'iio-sensor-proxy.service.in',
-  output: 'iio-sensor-proxy.service',
-  configuration: data_conf,
-  install_dir: systemd_system_unit_dir,
-)
-
-configure_file(
   input: 'net.hadess.SensorProxy.conf.in',
   output: 'net.hadess.SensorProxy.conf',
   configuration: data_conf,
