--- Makefile	2022-12-05 08:07:54.867207190 -0500
+++ Makefile-tmp	2022-12-05 08:14:19.101530405 -0500
@@ -2,7 +2,7 @@
 BIN_FILES = uwufetch.c
 LIB_FILES = fetch.c
 UWUFETCH_VERSION = $(shell git describe --tags)
-CFLAGS = -O3 -DUWUFETCH_VERSION=\"$(UWUFETCH_VERSION)\"
+CFLAGS := $(CFLAGS) -O3 -DUWUFETCH_VERSION=\"$(UWUFETCH_VERSION)\"
 CFLAGS_DEBUG = -Wall -Wextra -g -pthread -DUWUFETCH_VERSION=\"$(UWUFETCH_VERSION)\"
 CC ?= cc
 AR = ar
@@ -64,7 +64,7 @@
 endif
 
 build: $(BIN_FILES) lib
-	$(CC) $(CFLAGS) -o $(NAME) $(BIN_FILES) lib$(LIB_FILES:.c=.a)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(NAME) $(BIN_FILES) lib$(LIB_FILES:.c=.a)
 
 lib: $(LIB_FILES)
 	$(CC) $(CFLAGS) -fPIC -c -o $(LIB_FILES:.c=.o) $(LIB_FILES)
