--- Makefile	2022-09-30 13:04:19.913850961 -0400
+++ Makefile	2022-09-30 13:05:14.877102396 -0400
@@ -91,12 +91,12 @@
 	./$(NAME) $(ARGS)
 
 install: build
-	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) $(ETC_DIR)/$(NAME)
+	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME)
 	cp $(NAME) $(DESTDIR)/$(PREFIX)
 	cp lib$(LIB_FILES:.c=.so) $(DESTDIR)/$(LIBDIR)
 	cp $(LIB_FILES:.c=.h) $(DESTDIR)/include
 	cp -r res/* $(DESTDIR)/$(LIBDIR)/$(NAME)
-	cp default.config $(ETC_DIR)/$(NAME)/config
+	cp default.config ${DESTDIR}/$(ETC_DIR)/$(NAME)/config
 	cp ./$(NAME).1.gz $(DESTDIR)/$(MANDIR)
 
 uninstall:
