--- Makefile	2022-10-04 12:18:03.076649484 -0400
+++ Makefile-tmp	2022-10-04 12:23:48.782933214 -0400
@@ -91,11 +91,11 @@
 	./$(NAME) $(ARGS)
 
 install: build
-	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME) ${DESTDIR}/include
+	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME) ${DESTDIR}/include ${DESTDIR}/lib/uwufetch
 	cp $(NAME) $(DESTDIR)/$(PREFIX)
 	cp lib$(LIB_FILES:.c=.so) $(DESTDIR)/$(LIBDIR)
 	cp $(LIB_FILES:.c=.h) $(DESTDIR)/include/
-	cp -r res/* $(DESTDIR)/$(LIBDIR)/$(NAME)
+	cp -r res/* $(DESTDIR)/lib/$(NAME)
 	cp default.config ${DESTDIR}/$(ETC_DIR)/$(NAME)/config
 	cp ./$(NAME).1.gz $(DESTDIR)/$(MANDIR)
 
