--- Makefile	2022-12-05 08:30:04.452071661 -0500
+++ Makefile	2022-12-05 08:35:10.611197290 -0500
@@ -69,7 +69,7 @@
 lib: $(LIB_FILES)
 	$(CC) $(CFLAGS) -fPIC -c -o $(LIB_FILES:.c=.o) $(LIB_FILES)
 	$(AR) rcs lib$(LIB_FILES:.c=.a) $(LIB_FILES:.c=.o)
-	$(CC) $(CFLAGS) -shared -o lib$(LIB_FILES:.c=.so) $(LIB_FILES:.c=.o)
+	$(CC) $(CFLAGS) -shared -Wl,-soname,lib$(LIB_FILES:.c=.so).$(UWUFETCH_VERSION) -o lib$(LIB_FILES:.c=.so) $(LIB_FILES:.c=.o)
 
 release: build
 	mkdir -pv $(NAME)_$(UWUFETCH_VERSION)-$(PLATFORM_ABBR)
