--- Makefile	2022-10-04 12:15:17.893515471 -0400
+++ Makefile-tmp	2022-10-04 12:17:03.220062320 -0400
@@ -91,10 +91,10 @@
 	./$(NAME) $(ARGS)
 
 install: build
-	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME)
+	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME) ${DESTDIR}/include
 	cp $(NAME) $(DESTDIR)/$(PREFIX)
 	cp lib$(LIB_FILES:.c=.so) $(DESTDIR)/$(LIBDIR)
-	cp $(LIB_FILES:.c=.h) $(DESTDIR)/include
+	cp $(LIB_FILES:.c=.h) $(DESTDIR)/include/
 	cp -r res/* $(DESTDIR)/$(LIBDIR)/$(NAME)
 	cp default.config ${DESTDIR}/$(ETC_DIR)/$(NAME)/config
 	cp ./$(NAME).1.gz $(DESTDIR)/$(MANDIR)
