--- Makefile	2023-09-14 13:22:17.487810992 -0400
+++ Makefile-tmp	2023-09-14 13:33:18.990358700 -0400
@@ -106,7 +106,7 @@
 	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME)
 	cp $(NAME) $(DESTDIR)/$(PREFIX)
 	cp lib$(LIB_FILES:.c=.so) $(DESTDIR)/$(LIBDIR)
-	cp $(LIB_FILES:.c=.h) $(DESTDIR)/$(INCDIR)
+	cp $(LIB_FILES:.c=.h) $(DESTDIR)/$(INCDIR)/
 	cp -r res/* $(DESTDIR)/$(LIBDIR)/$(NAME)
 	cp default.config ${DESTDIR}/$(ETC_DIR)/$(NAME)/config
 	cp ./$(NAME).1.gz $(DESTDIR)/$(MANDIR)

