--- Makefile	2022-12-05 08:42:56.586585774 -0500
+++ Makefile-tmp	2022-12-05 08:43:24.063183118 -0500
@@ -92,8 +92,7 @@
 
 install: build
 	mkdir -pv $(DESTDIR)/$(PREFIX) $(DESTDIR)/$(LIBDIR)/$(NAME) $(DESTDIR)/$(MANDIR) ${DESTDIR}/$(ETC_DIR)/$(NAME) ${DESTDIR}/include ${DESTDIR}/lib/uwufetch
-	cp $(NAME) $(DESTDIR)/$(PREFIX)
-	cp lib$(LIB_FILES:.c=.so) $(DESTDIR)/$(LIBDIR)
+	cp $(NAME) $(DESTDIR)/$(PREFIX)	
 	cp $(LIB_FILES:.c=.h) $(DESTDIR)/include/
 	cp -r res/* $(DESTDIR)/lib/$(NAME)
 	cp default.config ${DESTDIR}/$(ETC_DIR)/$(NAME)/config
