diff --git a/src/Gtk/BackupBox.vala b/src/Gtk/BackupBox.vala
old mode 100644
new mode 100755
index 3b0a855..4926503
--- a/src/Gtk/BackupBox.vala
+++ b/src/Gtk/BackupBox.vala
@@ -64,7 +64,7 @@ class BackupBox : Gtk.Box{
 		log_debug("BackupBox: BackupBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 		
diff --git a/src/Gtk/BackupDeviceBox.vala b/src/Gtk/BackupDeviceBox.vala
index 1bdde5a..ead9371 100644
--- a/src/Gtk/BackupDeviceBox.vala
+++ b/src/Gtk/BackupDeviceBox.vala
@@ -46,7 +46,7 @@ class BackupDeviceBox : Gtk.Box{
 		log_debug("BackupDeviceBox: BackupDeviceBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/BackupFinishBox.vala b/src/Gtk/BackupFinishBox.vala
index 1b57759..0343be5 100644
--- a/src/Gtk/BackupFinishBox.vala
+++ b/src/Gtk/BackupFinishBox.vala
@@ -42,7 +42,7 @@ class BackupFinishBox : Gtk.Box{
 		log_debug("BackupFinishBox: BackupFinishBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/BootOptionsBox.vala b/src/Gtk/BootOptionsBox.vala
index 5c96443..6e66372 100644
--- a/src/Gtk/BootOptionsBox.vala
+++ b/src/Gtk/BootOptionsBox.vala
@@ -47,7 +47,7 @@ class BootOptionsBox : Gtk.Box{
 		log_debug("BootOptionsBox: BootOptionsBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/DeleteBox.vala b/src/Gtk/DeleteBox.vala
index 7cbc7d4..c824bbd 100644
--- a/src/Gtk/DeleteBox.vala
+++ b/src/Gtk/DeleteBox.vala
@@ -50,7 +50,7 @@ class DeleteBox : Gtk.Box{
 		log_debug("DeleteBox: DeleteBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 		
diff --git a/src/Gtk/DeleteFinishBox.vala b/src/Gtk/DeleteFinishBox.vala
index ab9a6d0..db99034 100644
--- a/src/Gtk/DeleteFinishBox.vala
+++ b/src/Gtk/DeleteFinishBox.vala
@@ -42,7 +42,7 @@ class DeleteFinishBox : Gtk.Box{
 		log_debug("DeleteFinishBox: DeleteFinishBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/EstimateBox.vala b/src/Gtk/EstimateBox.vala
index 858545b..efdfaf7 100644
--- a/src/Gtk/EstimateBox.vala
+++ b/src/Gtk/EstimateBox.vala
@@ -48,7 +48,7 @@ class EstimateBox : Gtk.Box{
 		log_debug("EstimateBox: EstimateBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 		
diff --git a/src/Gtk/ExcludeAppsBox.vala b/src/Gtk/ExcludeAppsBox.vala
index 6d96fcb..2d26f74 100644
--- a/src/Gtk/ExcludeAppsBox.vala
+++ b/src/Gtk/ExcludeAppsBox.vala
@@ -41,7 +41,7 @@ class ExcludeAppsBox : Gtk.Box{
 		log_debug("ExcludeAppsBox: ExcludeAppsBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/ExcludeBox.vala b/src/Gtk/ExcludeBox.vala
index ac70bcf..0cdddbf 100644
--- a/src/Gtk/ExcludeBox.vala
+++ b/src/Gtk/ExcludeBox.vala
@@ -44,7 +44,7 @@ class ExcludeBox : Gtk.Box{
 		log_debug("ExcludeBox: ExcludeBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/FinishBox.vala b/src/Gtk/FinishBox.vala
index be10612..10a2d5e 100644
--- a/src/Gtk/FinishBox.vala
+++ b/src/Gtk/FinishBox.vala
@@ -45,7 +45,7 @@ class FinishBox : Gtk.Box{
 		log_debug("FinishBox: FinishBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 		show_notes = _show_notes;
diff --git a/src/Gtk/MiscBox.vala b/src/Gtk/MiscBox.vala
index 91295ca..4f48f83 100644
--- a/src/Gtk/MiscBox.vala
+++ b/src/Gtk/MiscBox.vala
@@ -46,7 +46,7 @@ class MiscBox : Gtk.Box{
 		log_debug("MiscBox: MiscBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RestoreBox.vala b/src/Gtk/RestoreBox.vala
index 3d9289d..4d483b0 100644
--- a/src/Gtk/RestoreBox.vala
+++ b/src/Gtk/RestoreBox.vala
@@ -65,7 +65,7 @@ class RestoreBox : Gtk.Box{
 		log_debug("RestoreBox: RestoreBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RestoreDeviceBox.vala b/src/Gtk/RestoreDeviceBox.vala
index 3a21930..431cb07 100644
--- a/src/Gtk/RestoreDeviceBox.vala
+++ b/src/Gtk/RestoreDeviceBox.vala
@@ -51,7 +51,7 @@ class RestoreDeviceBox : Gtk.Box{
 		log_debug("RestoreDeviceBox: RestoreDeviceBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RestoreExcludeBox.vala b/src/Gtk/RestoreExcludeBox.vala
index 67e781a..65cc655 100644
--- a/src/Gtk/RestoreExcludeBox.vala
+++ b/src/Gtk/RestoreExcludeBox.vala
@@ -46,7 +46,7 @@ class RestoreExcludeBox : Gtk.Box{
 		log_debug("RestoreExcludeBox: RestoreExcludeBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RestoreFinishBox.vala b/src/Gtk/RestoreFinishBox.vala
index 30d5690..1bb0bac 100644
--- a/src/Gtk/RestoreFinishBox.vala
+++ b/src/Gtk/RestoreFinishBox.vala
@@ -42,7 +42,7 @@ class RestoreFinishBox : Gtk.Box{
 		log_debug("RestoreFinishBox: RestoreFinishBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RestoreSummaryBox.vala b/src/Gtk/RestoreSummaryBox.vala
index 6cf87c7..ca25b3c 100644
--- a/src/Gtk/RestoreSummaryBox.vala
+++ b/src/Gtk/RestoreSummaryBox.vala
@@ -44,7 +44,7 @@ class RestoreSummaryBox : Gtk.Box{
 		log_debug("RestoreSummaryBox: RestoreSummaryBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/RsyncLogBox.vala b/src/Gtk/RsyncLogBox.vala
index 2d1394e..4006a04 100644
--- a/src/Gtk/RsyncLogBox.vala
+++ b/src/Gtk/RsyncLogBox.vala
@@ -69,7 +69,7 @@ public class RsyncLogBox : Gtk.Box {
 
 	public RsyncLogBox(Gtk.Window _window) {
 		
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 
 		this.margin = 6;
 		
diff --git a/src/Gtk/ScheduleBox.vala b/src/Gtk/ScheduleBox.vala
index b59cfbb..39f33fc 100644
--- a/src/Gtk/ScheduleBox.vala
+++ b/src/Gtk/ScheduleBox.vala
@@ -50,7 +50,7 @@ class ScheduleBox : Gtk.Box{
 		log_debug("ScheduleBox: ScheduleBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 		
diff --git a/src/Gtk/SnapshotBackendBox.vala b/src/Gtk/SnapshotBackendBox.vala
index c725ad7..09bbd0f 100755
--- a/src/Gtk/SnapshotBackendBox.vala
+++ b/src/Gtk/SnapshotBackendBox.vala
@@ -46,7 +46,7 @@ class SnapshotBackendBox : Gtk.Box{
 		log_debug("SnapshotBackendBox: SnapshotBackendBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
diff --git a/src/Gtk/SnapshotListBox.vala b/src/Gtk/SnapshotListBox.vala
index 4d6e9f1..4cef400 100644
--- a/src/Gtk/SnapshotListBox.vala
+++ b/src/Gtk/SnapshotListBox.vala
@@ -63,7 +63,7 @@ class SnapshotListBox : Gtk.Box{
 		log_debug("SnapshotListBox: SnapshotListBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 6;
 
diff --git a/src/Gtk/UsersBox.vala b/src/Gtk/UsersBox.vala
index 2d2a63e..dbcc719 100644
--- a/src/Gtk/UsersBox.vala
+++ b/src/Gtk/UsersBox.vala
@@ -49,7 +49,7 @@ class UsersBox : Gtk.Box{
 		log_debug("UsersBox: UsersBox()");
 		
 		//base(Gtk.Orientation.VERTICAL, 6); // issue with vala
-		Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
+		GLib.Object(orientation: Gtk.Orientation.VERTICAL, spacing: 6); // work-around
 		parent_window = _parent_window;
 		margin = 12;
 
