--- a/src/makefile
+++ b/src/makefile
@@ -1,6 +1,6 @@
 SHELL=/bin/bash
-CFLAGS=--std=c99
-EXECUTABLES = find msgmerge msgfmt install rm mkdir cp chmod valac
+CFLAGS += --std=c99
+EXECUTABLES = find msgmerge msgfmt install rm mkdir cp chmod $(VALAC)
 CHECKEXECS := $(foreach exec,$(EXECUTABLES), $(if $(shell which $(exec)),,$(error No $(exec) found, install it)))
 
 prefix=/usr
@@ -50,8 +50,8 @@ all: app-gtk app-console
 app-gtk:
 
 	#timeshift-gtk
-	valac -X -D'GETTEXT_PACKAGE="${app_name}"' \
-		--Xcc="-lm" --Xcc="-O3" ${symbols} \
+	$(VALAC) -v -X -D'GETTEXT_PACKAGE="${app_name}"' \
+		--cc=$(CC) $(foreach flag, $(CFLAGS) $(LDFLAGS), -X $(flag)) --Xcc="-lm" ${symbols} \
 		Core/*.vala Gtk/*.vala Utility/*.vala Utility/Gtk/*.vala \
 		-o ${app_name}-gtk \
 		--pkg glib-2.0 --pkg gio-unix-2.0 --pkg posix \
@@ -61,8 +61,8 @@ app-gtk:
 app-console:
 
 	#timeshift
-	valac -X -D'GETTEXT_PACKAGE="${app_name}"' \
-		--Xcc="-lm" --Xcc="-O3" ${symbols} \
+	$(VALAC) -v -X -D'GETTEXT_PACKAGE="${app_name}"' \
+		--cc="$(CC)" $(foreach flag, $(CFLAGS) $(LDFLAGS), -X $(flag)) --Xcc="-lm" ${symbols} \
 		Core/*.vala Utility/*.vala Utility/Gtk/*.vala Console/*.vala \
 		-o ${app_name} \
 		--pkg glib-2.0 --pkg gio-unix-2.0 --pkg posix \
@@ -71,7 +71,6 @@ app-console:
 
 manpage:
 	./${app_name} --help > ../man/${app_name}.1
-	gzip -f ../man/${app_name}.1
 
 clean:
 	rm -rfv ../release/{source,i386,amd64,armel,armhf}
@@ -115,7 +114,7 @@ install:
 	install -m 0644 ../files/${app_name}.json "$(DESTDIR)$(appconfdir)/default.json"
 
 	# man page
-	install -m 0644 ../man/${app_name}.1.gz "$(DESTDIR)$(man1dir)/${app_name}.1.gz"
+	install -m 0644 ../man/${app_name}.1 "$(DESTDIR)$(man1dir)/${app_name}.1"
 
 	# app icons
 	cp -dpr --no-preserve=ownership -t "$(DESTDIR)$(sharedir)/icons" ./share/icons/*
-- 
2.35.1

