diff --color -Naur heirloom-mailx-12.5/Makefile heirloom-mailx-12.5-patch/Makefile
--- heirloom-mailx-12.5/Makefile	2011-04-26 23:23:22.000000000 +0200
+++ heirloom-mailx-12.5-patch/Makefile	2021-08-08 21:58:46.965043853 +0200
@@ -25,7 +25,7 @@
 # displayed correctly.
 #CFLAGS		=
 #CPPFLAGS	=
-#LDFLAGS		=
+LDFLAGS		=
 #WARN		= -Wall -Wno-parentheses -Werror
 
 # Some RedHat versions need INCLUDES = -I/usr/kerberos/include to compile
