diff --git a/protocols/meson.build b/protocols/meson.build
index 9d47e0a..c082845 100644
--- a/protocols/meson.build
+++ b/protocols/meson.build
@@ -29,7 +29,7 @@ protocols = [
   ['pointer-constraints-unstable-v1.xml'],
   ['tablet-unstable-v2.xml'],
   ['idle.xml'],
-  [hl_protocol_dir, 'protocols/hyprland-toplevel-export-v1.xml']
+  ['hyprland-toplevel-export-v1.xml']
 ]
 wl_protos_src = []
 wl_protos_headers = []
