--- a/Makefile
+++ b/Makefile
@@ -22,8 +22,7 @@
   MAKEFILE_STDLIB  = libstdc++.a
 endif
 
-CXXFLAGS += -g -O0 \
-	-Wall -W -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings -fno-common \
+CXXFLAGS += -Wall -W -Wshadow -Wpointer-arith -Wcast-qual -Wcast-align -Wwrite-strings \
     -I$(BOINC_DIR) \
     -I$(BOINC_LIB_DIR) \
     -I$(BOINC_API_DIR) \
@@ -56,5 +55,5 @@
 	regexp_memory.o \
 	regexp_report.o
 
-wrapper: wrapper.o $(MAKEFILE_STDLIB) $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(REGEXP_OBJS)
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) $(MAKEFILE_LDFLAGS) -lboinc_api -lboinc -lboinc_zip $(STDCPPTC)
+wrapper: wrapper.o $(REGEXP_OBJS)
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) -lboinc_api -lboinc -lboinc_zip $(STDCPPTC)
