https://bugs.gentoo.org/887865
Fix Wincompatible-pointer-types.
--- a/src/media-backend.c
+++ b/src/media-backend.c
@@ -531,7 +531,7 @@ haze_media_backend_ready (TpSvcMediaSessionHandler *iface,
 
 static void
 haze_media_backend_error (TpSvcMediaSessionHandler *iface,
-                          guint errno,
+                          guint err,
                           const gchar *message,
                           DBusGMethodInvocation *context)
 {
@@ -561,7 +561,7 @@ haze_media_backend_error (TpSvcMediaSessionHandler *iface,
     }
 
   DEBUG ("Media.SessionHandler::Error called, error %u (%s) -- "
-      "emitting error on each stream", errno, message);
+      "emitting error on each stream", err, message);
 
   purple_media_end (priv->media, NULL, NULL);
 
@@ -579,7 +579,7 @@ haze_media_backend_error (TpSvcMediaSessionHandler *iface,
     {
       HazeMediaStream *stream = g_ptr_array_index (tmp, i);
 
-      haze_media_stream_error (stream, errno, message, NULL);
+      haze_media_stream_error (stream, err, message, NULL);
     }
 
   g_ptr_array_free (tmp, TRUE);
--- a/src/media-stream.c
+++ b/src/media-stream.c
@@ -849,14 +849,14 @@ haze_media_stream_error (HazeMediaStream *self,
  */
 static void
 haze_media_stream_error_async (TpSvcMediaStreamHandler *iface,
-                               guint errno,
+                               guint err,
                                const gchar *message,
                                DBusGMethodInvocation *context)
 {
   HazeMediaStream *self = HAZE_MEDIA_STREAM (iface);
   GError *error = NULL;
 
-  if (haze_media_stream_error (self, errno, message, &error))
+  if (haze_media_stream_error (self, err, message, &error))
     {
       tp_svc_media_stream_handler_return_from_error (context);
     }
