diff --git c/pyproject.toml w/pyproject.toml
index 5f530a5e..8daf38ba 100644
--- c/pyproject.toml
+++ w/pyproject.toml
@@ -87,10 +87,10 @@
 
 	# Linux + macOS + Windows (MINGW64)
 	ext-modules = [
-		{name = "phazor", sources = ["src/phazor/kissfft/kiss_fftr.c", "src/phazor/kissfft/kiss_fft.c", "src/phazor/phazor.c"], include-dirs = ["/usr/include/opus", "/opt/homebrew/include/opus", "/opt/homebrew/include", "C:/msys64/mingw64/include/opus"], libraries = ["samplerate", "wavpack", "opusfile", "vorbisfile", "mpg123", "FLAC", "openmpt", "gme"], library-dirs = ["/opt/homebrew/lib"], extra-compile-args = ["-D MINI"] },
+		{name = "phazor", sources = ["src/phazor/phazor.c"], include-dirs = ["/usr/include/opus", "/opt/homebrew/include/opus", "/opt/homebrew/include", "C:/msys64/mingw64/include/opus"], libraries = ["kissfft-float-openmp", "samplerate", "wavpack", "opusfile", "vorbisfile", "mpg123", "FLAC", "openmpt", "gme"], library-dirs = ["/opt/homebrew/lib"], extra-compile-args = ["-D MINI"] },
 		# Set as optional to allow soft-failure, it's expected not to build on Windows/macOS, but we don't want to fail the entire build on it
 		# I have not found a better way to solve this, ideally we would somehow tag this as Linux-exclusive
-		{name = "phazor-pw", sources = ["src/phazor/kissfft/kiss_fftr.c", "src/phazor/kissfft/kiss_fft.c", "src/phazor/phazor.c"], include-dirs = ["/usr/include/opus", "/usr/include/pipewire-0.3", "/usr/include/spa-0.2"], libraries = ["samplerate", "wavpack", "opusfile", "vorbisfile", "mpg123", "FLAC", "openmpt", "gme", "pipewire-0.3"], extra-compile-args = ["-D PIPE"], optional = true },
+		{name = "phazor-pw", sources = ["src/phazor/phazor.c"], include-dirs = ["/usr/include/opus", "/usr/include/pipewire-0.3", "/usr/include/spa-0.2"], libraries = ["kissfft-float-openmp", "samplerate", "wavpack", "opusfile", "vorbisfile", "mpg123", "FLAC", "openmpt", "gme", "pipewire-0.3"], extra-compile-args = ["-D PIPE"] },
 	]
 
 	package-dir = {"" = "src"}
