diff --git i/src/tauon/t_modules/t_main.py w/src/tauon/t_modules/t_main.py
index 4da572e5..acde4380 100644
--- i/src/tauon/t_modules/t_main.py
+++ w/src/tauon/t_modules/t_main.py
@@ -39337,8 +39337,8 @@ def main(holder: Holder) -> None:
 	locale_directory = install_directory / "locale"
 	if flatpak_mode:
 		locale_directory = Path("/app/share/locale")
-	#elif str(install_directory).startswith(("/opt/", "/usr/")):
-	#	locale_directory = Path("/usr/share/locale")
+	elif str(install_directory).startswith(("/opt/", "/usr/")):
+		locale_directory = Path("/usr/share/locale")
 
 	dirs = Directories(
 		install_directory=install_directory,
