--- a/meson.build
+++ b/meson.build
@@ -79,8 +79,10 @@
 
 cpp_compiler = meson.get_compiler('cpp')
 
-extended_encoders_available = (cpp_compiler.has_header('emmintrin.h') and
-    host_machine.cpu_family() == 'x86' and host_machine.cpu_family() == 'x86_64')
+extended_encoders_available = (
+    cpp_compiler.has_header('emmintrin.h') and
+    (host_machine.cpu_family() == 'x86' or host_machine.cpu_family() == 'x86_64')
+)
 
 if not extended_encoders_available
   add_project_arguments('-DBUNDLE_NO_SSE', language: 'cpp')
