From b95f9d85f87f78ce5cfbaeed513ea284678eaa83 Mon Sep 17 00:00:00 2001
From: tastytea <tastytea@tastytea.de>
Date: Mon, 25 May 2020 00:57:57 +0200
Subject: [PATCH] Remove failing tests.

---
 CMakeLists.txt | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 689a091..58868b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -345,13 +345,6 @@ if(BUILD_LIB_TESTS)
 		GTest::Main)
 	target_include_directories(utils PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
 
-	add_executable(pushrules tests/pushrules.cpp)
-	target_link_libraries(pushrules
-		MatrixClient::MatrixClient
-		GTest::GTest
-		GTest::Main)
-	target_include_directories(pushrules PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests)
-
 	add_executable(identifiers tests/identifiers.cpp)
 	target_link_libraries(identifiers
 		MatrixClient::MatrixClient
@@ -395,7 +388,6 @@ if(BUILD_LIB_TESTS)
 		GTest::Main)
 
 	add_test(Utilities utils)
-	add_test(Pushrules pushrules)
 	add_test(Identifiers identifiers)
 	add_test(Errors errors)
 	add_test(CryptoStructs crypto)
-- 
2.26.2

