--- src/Makefile
+++ src/Makefile
@@ -82,11 +82,6 @@
   LDLIBS += -pthread
 endif
 
-# Check for gnuplot existance
-ifneq (, $(shell which gnuplot))
-  CFLAGS += -DGNUPLOT
-endif
-
 # OpenBSD does not implement wordexp()
 ifeq ($(shell uname -s),OpenBSD)
   CFLAGS += -DNO_WORDEXP
@@ -97,77 +92,6 @@
   LDLIBS += -ldl
 endif
 
-ifneq (, $(shell which pkg-config))
-  # Any system with pkg-config
-
-  # NOTE: ncursesw (required)
-  ifeq ($(shell uname -s),Darwin)
-    # macOS' ncurses is built with wide-char support
-    LDFLAGS += -lncurses
-  else ifneq ($(shell pkg-config --exists ncursesw || echo 'no'),no)
-    CFLAGS += $(shell pkg-config --cflags ncursesw)
-    LDLIBS += $(shell pkg-config --libs ncursesw)
-  else ifneq ($(shell pkg-config --exists ncurses || echo 'no'),no)
-    # hopefully this includes wide character support then
-    CFLAGS += $(shell pkg-config --cflags ncurses)
-    LDLIBS += $(shell pkg-config --libs ncurses)
-  else
-    LDLIBS += -lncursesw
-  endif
-
-  # NOTE: libxlsreader (libxls) is required for xls file reading support
-  ifneq ($(shell pkg-config --exists libxls || echo 'no'),no)
-    CFLAGS += -DXLS $(shell pkg-config --cflags libxls)
-    LDLIBS += $(shell pkg-config --libs libxls)
-  endif
-
-  # NOTE: libxml and libzip are required for xlsx file import support
-  ifneq ($(shell pkg-config --exists libzip libxml-2.0 || echo 'no'),no)
-    CFLAGS += -DODS -DXLSX $(shell pkg-config --cflags libxml-2.0 libzip)
-    LDLIBS += $(shell pkg-config --libs libxml-2.0 libzip)
-  endif
-
-  # NOTE: libxlsxwriter is required for xlsx file export support
-  ifneq ($(shell pkg-config --exists xlsxwriter || echo 'no'),no)
-    CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter)
-    LDLIBS += $(shell pkg-config --libs xlsxwriter)
-  endif
-
-  # NOTE: lua support
-  ifneq ($(shell pkg-config --exists lua || echo 'no'),no) # Check for user's default lua
-    CFLAGS += -DXLUA $(shell pkg-config --cflags lua)
-    ifneq ($(shell uname -s),Darwin)
-      LDLIBS += $(shell pkg-config --libs lua) -Wl,--export-dynamic
-    else
-      LDLIBS += $(shell pkg-config --libs lua) -rdynamic
-    endif
-  else ifneq ($(shell pkg-config --exists luajit || echo 'no'),no) # If not found, check for luajit
-    CFLAGS += -DXLUA $(shell pkg-config --cflags luajit)
-    ifneq ($(shell uname -s),Darwin)
-      LDLIBS += $(shell pkg-config --libs luajit) -Wl,--export-dynamic
-    else
-      LDLIBS += $(shell pkg-config --libs luajit) -rdynamic
-    endif
-  endif
-else ifeq ($(shell uname -s),Darwin)
-  # macOS without pkg-config
-
-  # macOS' ncurses is built with wide-char support
-  LDFLAGS += -lncurses
-else ifeq ($(shell uname -s),NetBSD)
-  # NetBSD without pkg-config
-
-  CFLAGS  += -I/usr/pkg/include
-  CFLAGS  += -I/usr/pkg/include/ncursesw
-
-  LDFLAGS += -L/usr/pkg/lib
-  LDFLAGS += -Wl,-R/usr/pkg/lib
-
-  LDLIBS += -lncursesw
-else
-  LDFLAGS += -lncursesw
-endif
-
 OBJS = $(patsubst %.c, %.o, $(wildcard *.c) $(wildcard utils/*.c)) gram.o
 
 .PHONY : all clean install docs man_install man_uninstall
