--- src/Makefile
+++ src/Makefile
@@ -83,7 +83,7 @@
 endif
 
 # Check for gnuplot existance
-ifneq (, $(shell which gnuplot))
+ifeq ($(PLOTS),yes)
   CFLAGS += -DGNUPLOT
 endif
 
@@ -116,38 +116,31 @@
   endif
 
   # NOTE: libxlsreader (libxls) is required for xls file reading support
-  ifneq ($(shell pkg-config --exists libxls || echo 'no'),no)
+  ifeq ($(XLS),yes)
     CFLAGS += -DXLS $(shell pkg-config --cflags libxls)
     LDLIBS += $(shell pkg-config --libs libxls)
   endif
 
   # NOTE: libxml and libzip are required for xlsx file import support
-  ifneq ($(shell pkg-config --exists libzip libxml-2.0 || echo 'no'),no)
+  ifeq ($(XML_ZIP),yes)
     CFLAGS += -DODS -DXLSX $(shell pkg-config --cflags libxml-2.0 libzip)
     LDLIBS += $(shell pkg-config --libs libxml-2.0 libzip)
   endif
 
   # NOTE: libxlsxwriter is required for xlsx file export support
-  ifneq ($(shell pkg-config --exists xlsxwriter || echo 'no'),no)
+  ifeq ($(XLSX),yes)
     CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter)
     LDLIBS += $(shell pkg-config --libs xlsxwriter)
   endif
 
   # NOTE: lua support
-  ifneq ($(shell pkg-config --exists lua || echo 'no'),no) # Check for user's default lua
+  ifeq ($(LUA),yes)
     CFLAGS += -DXLUA $(shell pkg-config --cflags lua)
     ifneq ($(shell uname -s),Darwin)
       LDLIBS += $(shell pkg-config --libs lua) -Wl,--export-dynamic
     else
       LDLIBS += $(shell pkg-config --libs lua) -rdynamic
     endif
-  else ifneq ($(shell pkg-config --exists luajit || echo 'no'),no) # If not found, check for luajit
-    CFLAGS += -DXLUA $(shell pkg-config --cflags luajit)
-    ifneq ($(shell uname -s),Darwin)
-      LDLIBS += $(shell pkg-config --libs luajit) -Wl,--export-dynamic
-    else
-      LDLIBS += $(shell pkg-config --libs luajit) -rdynamic
-    endif
   endif
 else ifeq ($(shell uname -s),Darwin)
   # macOS without pkg-config
