diff -Naur arianna-v24.02.0/CMakeLists.txt arianna-patched/CMakeLists.txt
--- arianna-v24.02.0/CMakeLists.txt	2024-02-15 23:50:18.000000000 +0000
+++ arianna-patched/CMakeLists.txt	2024-03-08 18:31:10.980327465 +0000
@@ -41,7 +41,8 @@
     VERSION_HEADER ${CMAKE_CURRENT_BINARY_DIR}/arianna-version.h
 )
 
-find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS Widgets Core Quick Gui QuickControls2 WebSockets WebChannel Test Svg Sql Xml HttpServer)
+find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS WebChannel Test)
+find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Quick Qml Gui HttpServer Network Core WebSockets QuickControls2 Svg Sql Xml Widgets)
 set_package_properties(Qt6 PROPERTIES
     TYPE REQUIRED
     PURPOSE "Basic application components"
@@ -61,7 +62,7 @@
 find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS WebEngineQuick)
 
 if (NOT WIN32)
-    find_package(KF6 ${KF_MIN_VERSION} COMPONENTS Baloo)
+    find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS Baloo)
 endif()
 
 ecm_find_qmlmodule(QtWebChannel)
