--- a/driver/Makefile
+++ b/driver/Makefile
@@ -1,8 +1,7 @@
 include ./Make.inc
 
 psort: psort_driver.cc
-	$(CXX) $(CXXFLAGS) -I../src psort_driver.cc -o psort $(LIBS)
-	strip psort
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -fPIE -I../src psort_driver.cc -o psort $(LIBS)
 
 clean:
 	rm -f *.o psort results
--- a/driver/Make.inc
+++ b/driver/Make.inc
@@ -1,4 +1,4 @@
 CXX=mpic++
-CXXFLAGS=-O2 -DUSE_FUNNEL -Wall 
+CXXFLAGS+=-DUSE_FUNNEL -Wall
 LIBS=-lm
 
