--- a/Makefile.am
+++ b/Makefile.am
@@ -726,11 +726,11 @@
 clean-local: clean-rpm clean-deb clean-doc
 
 install-data-hook:
-	chmod +x $(mpiompexamplesdir)/run.sh
-	chmod +x $(mpiompomptexamplesdir)/run.sh
-	chmod +x $(mpiompssexamplesdir)/run.sh
-	chmod +x $(omptexamplesdir)/run.sh
-	chmod +x $(statsexamplesdir)/run.sh
+	chmod +x ${DESTDIR}/$(mpiompexamplesdir)/run.sh
+	chmod +x ${DESTDIR}/$(mpiompomptexamplesdir)/run.sh
+	chmod +x ${DESTDIR}/$(mpiompssexamplesdir)/run.sh
+	chmod +x ${DESTDIR}/$(omptexamplesdir)/run.sh
+	chmod +x ${DESTDIR}/$(statsexamplesdir)/run.sh
 
 dist-hook:
 	if [ -x "$(GIT)" ]; \
