--- a/bin/hollywood
+++ b/bin/hollywood
@@ -18,7 +18,7 @@
 # limitations under the License.
 
 PKG="hollywood"
-trap "pkill -f -9 lib/hollywood/ >/dev/null 2>&1; exit 0" INT
+trap "pkill -f -9 libexec/hollywood/ >/dev/null 2>&1; exit 0" INT
 
 spin_up() {
 	local w=
@@ -43,7 +43,7 @@
 	done
 }
 
-WIDGET_DIR="$(dirname $0)/../lib/$PKG"
+WIDGET_DIR="$(dirname $0)/../libexec/$PKG"
 SPLITS=$(ls "$WIDGET_DIR" | wc -l)
 DELAY=10
 while [ ! -z "$1" ]; do
--- a/bin/wallstreet
+++ b/bin/wallstreet
@@ -19,10 +19,10 @@
 
 PKG="wallstreet"
 [ -n "$TMDIR" ] || export TMPDIR=$(mktemp -d /dev/shm/$PKG.XXXXXXXXX)
-trap "rm -rf $TMPDIR; pkill -f -9 lib/$PKG/ >/dev/null 2>&1; exit 0" EXIT HUP INT QUIT TERM
+trap "rm -rf $TMPDIR; pkill -f -9 libexec/$PKG/ >/dev/null 2>&1; exit 0" EXIT HUP INT QUIT TERM
 
 dir="-v"
-widget_dir="$(dirname $0)/../lib/$PKG"
+widget_dir="$(dirname $0)/../libexec/$PKG"
 widget1=$(ls "$widget_dir/" | sort -R | head -n1)
 
 if [ -n "$1" ]; then
