diff '--color=auto' -aur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-11-28 02:48:17.057740689 -0600
+++ b/CMakeLists.txt	2024-11-28 02:51:56.721795566 -0600
@@ -495,7 +495,7 @@
 	modules/benchmark/md5.c
 	modules/benchmark/sha1.c
 	PROPERTIES
-	COMPILE_FLAGS "-O0 -Wall -Wextra -Werror=implicit-function-declaration"
+	COMPILE_FLAGS "-Wall -Wextra -Werror=implicit-function-declaration"
 )
 
 set_source_files_properties(
