--- a/test/bugs.bats
+++ b/test/bugs.bats
@@ -1,7 +1,7 @@
 #!/usr/bin/env bats
 
-load 'test_helper/bats-support/load'
-load 'test_helper/bats-assert/load'
+load '/usr/share/bats-support/load.bash'
+load '/usr/share/bats-assert/load.bash'
 load 'test_helper/util'
 
 # https://github.com/paulirish/dotfiles/commit/6743b907ff586c28cd36e08d1e1c634e2968893e#commitcomment-13459061
--- a/test/diff-so-fancy.bats
+++ b/test/diff-so-fancy.bats
@@ -1,7 +1,7 @@
 #!/usr/bin/env bats
 
-load 'test_helper/bats-support/load'
-load 'test_helper/bats-assert/load'
+load '/usr/share/bats-support/load.bash'
+load '/usr/share/bats-assert/load.bash'
 load 'test_helper/util'
 
 set_env
