--- a/diff-so-fancy
+++ b/diff-so-fancy
@@ -9,7 +9,7 @@
 use File::Basename;                                         # For dirname
 use Encode;                                                 # For handling UTF8 stuff
 use Cwd qw(abs_path);                                       # For realpath()
-use lib dirname(abs_path(File::Spec->catdir($0))) . "/lib"; # Add the local lib/ to @INC
+use lib "/usr/share/diff-so-fancy";
 use DiffHighlight;
 
 use strict;
