--- a/include/CombBLAS/SpParHelper.cpp
+++ b/include/CombBLAS/SpParHelper.cpp
@@ -121,7 +121,7 @@
 		{
             long * dist_in = new long[nprocs];
             for(int i=0; i< nprocs; ++i)    dist_in[i] = (long) dist[i];
-            vpsort::parallel_sort (array, array+length,  dist_in, comm);
+            psort::parallel_sort (array, array+length,  dist_in, comm);
             delete [] dist_in;
 		}
 		else
@@ -157,7 +157,7 @@
 			MPI_Comm_create(comm, real_group, &real_comm);
 			if(!excluded)
 			{
-				vpsort::parallel_sort (array, array+length,  dist_in, real_comm);
+				psort::parallel_sort (array, array+length,  dist_in, real_comm);
 				MPI_Comm_free(&real_comm);
 			}
 			MPI_Group_free(&real_group);
Solo in /var/tmp/portage/sci-libs/CombBLAS-0_pre20220331/work/CombBLAS-426f6be0b29831025cdcacc1f8f69e3520bfb0ff: psort-1.0
Solo in /var/tmp/portage/sci-libs/CombBLAS-0_pre20220331/work/CombBLAS-426f6be0b29831025cdcacc1f8f69e3520bfb0ff: usort
