--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,7 +19,7 @@
 
 # set include directories
 target_include_directories(CombBLAS PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
-target_include_directories(CombBLAS PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/psort-1.0/include> $<INSTALL_INTERFACE:include>)
+target_include_directories(CombBLAS PUBLIC /usr/include/psort $<INSTALL_INTERFACE:include>)
 target_include_directories(CombBLAS PRIVATE include/CombBLAS)
 
 # MPI and OpenMP dependencies
@@ -43,11 +43,9 @@
   target_link_libraries(CombBLAS PUBLIC "${OpenMP_CXX_FLAGS}")
 endif()
 
-add_subdirectory(usort)
-target_link_libraries(CombBLAS PUBLIC Usortlib)
+target_link_libraries(CombBLAS PUBLIC -lUsortlib)
 
-add_subdirectory(graph500-1.2/generator)
-target_link_libraries(CombBLAS PUBLIC GraphGenlib)
+target_link_libraries(CombBLAS PUBLIC -lGraphGenlib)
 
 # Set up exported configuration
 # This allows CombBLAS to be installed in two ways:
@@ -65,7 +63,6 @@
 
 # installation
 install(DIRECTORY include/ DESTINATION include)
-install(DIRECTORY psort-1.0/include/ DESTINATION include)
 install(TARGETS CombBLAS EXPORT CombBLASTargets
         LIBRARY DESTINATION lib64
         ARCHIVE DESTINATION lib64
