--- a/Matlab/startup.m
+++ b/Matlab/startup.m
@@ -1,12 +1,8 @@
-disp('This is C:\Users\John\Documents\Gblas\Matlab\startup.m')
+disp('This is /usr/share/octave/site/m/CombBLAS/startup.m')
 
 format compact
 
-addpath('C:\Users\John\Documents\LincolnLabs\matlab')
-addpath('C:\Users\John\Documents\MATLAB\meshpart')
-addpath('C:\Users\John\Documents\MATLAB\UFget')
-addpath('C:\Users\John\Documents\MATLAB\ncm')
-addpath('C:\Users\John\Documents\Gblas\Matlab\utils')
-addpath('C:\Users\John\Documents\Gblas\Matlab\spindex')
-addpath('C:\Users\John\Documents\Gblas\Matlab\apsp')
-addpath('C:\Users\John\Documents\Gblas\Matlab\stuff')
\ Manca newline alla fine del file
+addpath('/usr/share/octave/site/m/CombBLAS/utils')
+addpath('/usr/share/octave/site/m/CombBLAS/spindex')
+addpath('/usr/share/octave/site/m/CombBLAS/apsp')
+addpath('/usr/share/octave/site/m/CombBLAS/stuff')
