--- a/include/CombBLAS/dcsc.cpp
+++ b/include/CombBLAS/dcsc.cpp
@@ -1223,7 +1223,7 @@
 template<class VT>	
 void Dcsc<IT,NT>::FillColInds(const VT * colnums, IT nind, std::vector< std::pair<IT,IT> > & colinds, IT * aux, IT csize) const
 {
-	if ( aux == NULL || (nzc / nind) < THRESHOLD)   	// use scanning indexing
+	if ( aux == NULL || (nzc / nind) < COMBBLAS_THRESHOLD)   	// use scanning indexing
 	{
 		IT mink = std::min(nzc, nind);
 		std::pair<IT,IT> * isect = new std::pair<IT,IT>[mink];
--- a/include/CombBLAS/SpDefs.h
+++ b/include/CombBLAS/SpDefs.h
@@ -122,8 +122,8 @@
 #define ALIGN 8
 #endif
 
-#ifndef THRESHOLD
-#define THRESHOLD 4	// if range1.size() / range2.size() < threshold, use scanning based indexing
+#ifndef COMBBLAS_THRESHOLD
+#define COMBBLAS_THRESHOLD 4	// if range1.size() / range2.size() < threshold, use scanning based indexing
 #endif
 
 #ifndef MEMORYINBYTES
