--- a/makefile	2019-08-31 06:10:16.000000000 +0200
+++ b/makefile	2021-10-18 11:35:15.173731696 +0200
@@ -4,17 +4,14 @@
 LIBS = -lm
 
 ###
-CFLAGS  = -std=c99
-CFLAGS += -g
+CFLAGS += -std=c99
 CFLAGS += -Wall
 CFLAGS += -Wextra
 CFLAGS += -pedantic
-CFLAGS += -Werror
 CFLAGS += -Wmissing-declarations
 CFLAGS += -DUNITY_SUPPORT_64
 
 BUILDFLAGS = -Wall
-BUILDFLAGS += -g
 BUILDFLAGS += -fPIC
 BUILDFLAGS += -shared
 
@@ -38,8 +35,8 @@
 
 tests.out: test/*.c src/*.c src/*.h
 	@echo Compiling $@
-	@$(CC) $(CFLAGS) src/*.c test/vendor/unity.c test/*.c -o tests.out $(LIBS)
+	@$(CC) $(CFLAGS) -fPIE $(LDFLAGS) src/*.c test/vendor/unity.c test/*.c -o tests.out $(LIBS)
 
 build: src/*.c src/*.h
 	@echo Compiling $@
-	@$(CC) $(BUILDFLAGS) src/*.c -o libpreloadvaccine.so $(BUILDLIBS)
\ Manca newline alla fine del file
+	@$(CC) $(BUILDFLAGS) $(CFLAGS) $(LDFLAGS) -Wl,-soname,libpreloadvaccine.so src/*.c -o libpreloadvaccine.so $(BUILDLIBS)
