From 9b55c68551fccf7bdb7514af35045a592b9da2db Mon Sep 17 00:00:00 2001
From: Alessandro Barbieri <lssndrbarbieri@gmail.com>
Date: Thu, 21 Oct 2021 07:15:26 +0200
Subject: [PATCH] fix typo

--- a/test/test_libpreloadvaccine.c
+++ b/test/test_libpreloadvaccine.c
@@ -33,7 +33,7 @@ static void test_file_contains_object(void)
 static void test_file_contains_object_bad_file(void)
 {
     char *working_dir = getenv("PWD");
-    char *test_path = strcat(working_dir,"/test/test_data/ld.so.prload");
+    char *test_path = strcat(working_dir,"/test/test_data/ld.so.preload");
     TEST_ASSERT_FALSE(file_contains_object("/tmp/libpreload.so", test_path));
 }
 
@@ -62,4 +62,4 @@ int main(void)
    RUN_TEST(test_file_contains_object_bad_file);
 
    return UnityEnd();
-}
\ No newline at end of file
+}
